{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "4c332526-fdb2-48b2-ab26-db1720feb57d",
   "metadata": {},
   "source": [
    "# Make ASV taxa table (custom)\n",
    "\n",
    "- From custom blast search xml file\n",
    "- Analyze PR2 results, get top hits"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "179b0dae-f73c-4e88-aa02-98b1c3f6658f",
   "metadata": {},
   "source": [
    "### Import libraries and set directory location"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "851b9142-f70a-4fde-8b28-9f7204cfe5a2",
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import glob\n",
    "from Bio.Blast import NCBIXML   #to parse XML files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "1cce6749-f2db-4cbf-84ed-a1db60df820e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/kpitz/github/GLOMICON/intercomparison/Merged_Datasets/PR2/GLOMICON_PR2_version_5.1.0_blastn_041025.xml\n",
      "/Users/kpitz/github/GLOMICON/intercomparison/Merged_Datasets/data/GLOMICON_taxa_blastnr.csv\n",
      "/Users/kpitz/github/GLOMICON/intercomparison/Merged_Datasets/data/GLOMICON_seq_merged.csv\n"
     ]
    }
   ],
   "source": [
    "directory = '/Users/kpitz/github/GLOMICON/intercomparison/Merged_Datasets/'\n",
    "prefix = 'GLOMICON'\n",
    "marker = '18S'\n",
    "blast_file = '/Users/kpitz/github/GLOMICON/intercomparison/Merged_Datasets/PR2/GLOMICON_PR2_version_5.1.0_blastn_041025.xml'\n",
    "print(blast_file)\n",
    "taxa_table = '/Users/kpitz/github/GLOMICON/intercomparison/Merged_Datasets/data/GLOMICON_taxa_blastnr.csv'\n",
    "print(taxa_table)\n",
    "seq_table = '/Users/kpitz/github/GLOMICON/intercomparison/Merged_Datasets/data/GLOMICON_seq_merged.csv'\n",
    "print(seq_table)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "42352bb3-98c7-4f97-b33d-76caac1f1e9e",
   "metadata": {},
   "source": [
    "## Import Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "041cfe77-b0c4-438d-94ce-5e69bca33de1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Domain</th>\n",
       "      <th>Kingdom</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ASV_1950</th>\n",
       "      <td>Archaea</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>archaeon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_9581</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_63</th>\n",
       "      <td>Archaea</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>archaeon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_3182</th>\n",
       "      <td>Archaea</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>archaeon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_603</th>\n",
       "      <td>Archaea</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>archaeon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_12927</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_12880</th>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bacillariophyta</td>\n",
       "      <td>Bacillariophyceae</td>\n",
       "      <td>Mastogloiales</td>\n",
       "      <td>Achnanthaceae</td>\n",
       "      <td>Achnanthes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_14103</th>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>uncultured marine eukaryote</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_13385</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_5807</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>14547 rows Ã— 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              Domain Kingdom           Phylum              Class  \\\n",
       "ASV                                                                \n",
       "ASV_1950     Archaea     NaN          unknown            unknown   \n",
       "ASV_9581         NaN     NaN              NaN                NaN   \n",
       "ASV_63       Archaea     NaN          unknown            unknown   \n",
       "ASV_3182     Archaea     NaN          unknown            unknown   \n",
       "ASV_603      Archaea     NaN          unknown            unknown   \n",
       "...              ...     ...              ...                ...   \n",
       "ASV_12927        NaN     NaN              NaN                NaN   \n",
       "ASV_12880  Eukaryota     NaN  Bacillariophyta  Bacillariophyceae   \n",
       "ASV_14103  Eukaryota     NaN          unknown            unknown   \n",
       "ASV_13385        NaN     NaN              NaN                NaN   \n",
       "ASV_5807         NaN     NaN              NaN                NaN   \n",
       "\n",
       "                   Order         Family       Genus  \\\n",
       "ASV                                                   \n",
       "ASV_1950         unknown        unknown     unknown   \n",
       "ASV_9581             NaN            NaN         NaN   \n",
       "ASV_63           unknown        unknown     unknown   \n",
       "ASV_3182         unknown        unknown     unknown   \n",
       "ASV_603          unknown        unknown     unknown   \n",
       "...                  ...            ...         ...   \n",
       "ASV_12927            NaN            NaN         NaN   \n",
       "ASV_12880  Mastogloiales  Achnanthaceae  Achnanthes   \n",
       "ASV_14103        unknown        unknown     unknown   \n",
       "ASV_13385            NaN            NaN         NaN   \n",
       "ASV_5807             NaN            NaN         NaN   \n",
       "\n",
       "                               Species  \n",
       "ASV                                     \n",
       "ASV_1950                      archaeon  \n",
       "ASV_9581                           NaN  \n",
       "ASV_63                        archaeon  \n",
       "ASV_3182                      archaeon  \n",
       "ASV_603                       archaeon  \n",
       "...                                ...  \n",
       "ASV_12927                          NaN  \n",
       "ASV_12880                          NaN  \n",
       "ASV_14103  uncultured marine eukaryote  \n",
       "ASV_13385                          NaN  \n",
       "ASV_5807                           NaN  \n",
       "\n",
       "[14547 rows x 8 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#sequence table\n",
    "df = pd.read_csv(seq_table)\n",
    "df = df.rename(columns={'Unnamed: 0':'ASV'})\n",
    "df.set_index('ASV', inplace=True)\n",
    "seq = df.copy()\n",
    "seq.head()\n",
    "\n",
    "# Dada2, MEGAN, Blastn results\n",
    "df = pd.read_csv(taxa_table)\n",
    "df = df.rename(columns={'Unnamed: 0':'ASV'})\n",
    "df.set_index('ASV', inplace=True)\n",
    "taxa = df.copy()\n",
    "taxa.head()\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3e0d5ee7-dbed-4185-8c16-ec75712e5e6c",
   "metadata": {},
   "source": [
    "## Parse XML file to get top hits per ASV and percent ID and bitscore values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "3a082438-00f6-425b-a8a7-041a102d6ad0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Blast XML file:  /Users/kpitz/github/GLOMICON/intercomparison/Merged_Datasets/PR2/GLOMICON_PR2_version_5.1.0_blastn_041025.xml\n",
      "Done parsing  /Users/kpitz/github/GLOMICON/intercomparison/Merged_Datasets/PR2/GLOMICON_PR2_version_5.1.0_blastn_041025.xml\n"
     ]
    }
   ],
   "source": [
    "# Only save top 2% of bitscore hits to query\n",
    "print('Blast XML file: ', blast_file)\n",
    "hit_dict ={}  #Dictionary to store hits\n",
    "result_handle =open(blast_file)\n",
    "#parse blast records\n",
    "blast_records= NCBIXML.parse(result_handle)\n",
    "for blast_record in blast_records:\n",
    "    query = blast_record.query\n",
    "    hit_ID=0 #hit counter\n",
    "    for alignment in blast_record.alignments:\n",
    "        hit_ID+=1\n",
    "        key = (query, hit_ID)\n",
    "        hsp = alignment.hsps[0] #only look at top hsp per alignment to genbank sequence\n",
    "        #set limits on evalue and bitscore and % Identity\n",
    "        per_iden = hsp.identities/float(hsp.align_length)\n",
    "        per_iden = (per_iden *100)\n",
    "        per_iden = round(per_iden, 2)\n",
    "        evalue = hsp.expect\n",
    "        bitscore = int(hsp.score)\n",
    "        # set bitscore limit for hits, within top 2%\n",
    "        if hit_ID ==1:\n",
    "            top_bitscore = bitscore\n",
    "            min_bitscore = top_bitscore - (top_bitscore*.02)\n",
    "        if bitscore < min_bitscore:\n",
    "            #continue to next alignment if bitscore too low\n",
    "            continue\n",
    "        #alignment length\n",
    "        align_len = hsp.align_length\n",
    "        # save results to dictionary\n",
    "        value = (evalue, bitscore, per_iden, alignment.hit_def, align_len)\n",
    "        hit_dict[key]=value\n",
    "print ('Done parsing ', blast_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "a93b717e-af8d-49e6-bb35-9a5fb118c382",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "KC488454.1.1667_U|18S_rRNA|nucleus|clone_HL5aSF04.03|Eukaryota|Haptista|Haptophyta|Haptophyta_X|Prymnesiophyceae|Phaeocystales|Phaeocystaceae|Phaeocystis|Phaeocystis_pouchetii\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hit_number</th>\n",
       "      <th>eval</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>%ID</th>\n",
       "      <th>hit_def</th>\n",
       "      <th>align_len</th>\n",
       "      <th>PR2_ID</th>\n",
       "      <th>PR2_marker</th>\n",
       "      <th>location</th>\n",
       "      <th>nuc</th>\n",
       "      <th>Domain</th>\n",
       "      <th>Supergroup</th>\n",
       "      <th>Division</th>\n",
       "      <th>Subdivision</th>\n",
       "      <th>Class</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ASV_1</th>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>350</td>\n",
       "      <td>100.0</td>\n",
       "      <td>KC488454.1.1667_U|18S_rRNA|nucleus|clone_HL5aS...</td>\n",
       "      <td>350</td>\n",
       "      <td>KC488454.1.1667_U</td>\n",
       "      <td>18S_rRNA</td>\n",
       "      <td>nucleus</td>\n",
       "      <td>clone_HL5aSF04.03</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Haptista</td>\n",
       "      <td>Haptophyta</td>\n",
       "      <td>Haptophyta_X</td>\n",
       "      <td>Prymnesiophyceae</td>\n",
       "      <td>Phaeocystales</td>\n",
       "      <td>Phaeocystaceae</td>\n",
       "      <td>Phaeocystis</td>\n",
       "      <td>Phaeocystis_pouchetii</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_1</th>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>350</td>\n",
       "      <td>100.0</td>\n",
       "      <td>HM561166.1.964_U|18S_rRNA|nucleus|clone_CFL133...</td>\n",
       "      <td>350</td>\n",
       "      <td>HM561166.1.964_U</td>\n",
       "      <td>18S_rRNA</td>\n",
       "      <td>nucleus</td>\n",
       "      <td>clone_CFL133R09</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Haptista</td>\n",
       "      <td>Haptophyta</td>\n",
       "      <td>Haptophyta_X</td>\n",
       "      <td>Prymnesiophyceae</td>\n",
       "      <td>Phaeocystales</td>\n",
       "      <td>Phaeocystaceae</td>\n",
       "      <td>Phaeocystis</td>\n",
       "      <td>Phaeocystis_pouchetii</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_1</th>\n",
       "      <td>3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>350</td>\n",
       "      <td>100.0</td>\n",
       "      <td>HM561167.1.839_U|18S_rRNA|nucleus|clone_CFL133...</td>\n",
       "      <td>350</td>\n",
       "      <td>HM561167.1.839_U</td>\n",
       "      <td>18S_rRNA</td>\n",
       "      <td>nucleus</td>\n",
       "      <td>clone_CFL133R10</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Haptista</td>\n",
       "      <td>Haptophyta</td>\n",
       "      <td>Haptophyta_X</td>\n",
       "      <td>Prymnesiophyceae</td>\n",
       "      <td>Phaeocystales</td>\n",
       "      <td>Phaeocystaceae</td>\n",
       "      <td>Phaeocystis</td>\n",
       "      <td>Phaeocystis_pouchetii</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_1</th>\n",
       "      <td>4</td>\n",
       "      <td>0.0</td>\n",
       "      <td>350</td>\n",
       "      <td>100.0</td>\n",
       "      <td>HM561164.1.955_U|18S_rRNA|nucleus|clone_CFL133...</td>\n",
       "      <td>350</td>\n",
       "      <td>HM561164.1.955_U</td>\n",
       "      <td>18S_rRNA</td>\n",
       "      <td>nucleus</td>\n",
       "      <td>clone_CFL133DB11</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Haptista</td>\n",
       "      <td>Haptophyta</td>\n",
       "      <td>Haptophyta_X</td>\n",
       "      <td>Prymnesiophyceae</td>\n",
       "      <td>Phaeocystales</td>\n",
       "      <td>Phaeocystaceae</td>\n",
       "      <td>Phaeocystis</td>\n",
       "      <td>Phaeocystis_pouchetii</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_1</th>\n",
       "      <td>5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>350</td>\n",
       "      <td>100.0</td>\n",
       "      <td>HM561163.1.945_U|18S_rRNA|nucleus|clone_CFL133...</td>\n",
       "      <td>350</td>\n",
       "      <td>HM561163.1.945_U</td>\n",
       "      <td>18S_rRNA</td>\n",
       "      <td>nucleus</td>\n",
       "      <td>clone_CFL133DB10</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Haptista</td>\n",
       "      <td>Haptophyta</td>\n",
       "      <td>Haptophyta_X</td>\n",
       "      <td>Prymnesiophyceae</td>\n",
       "      <td>Phaeocystales</td>\n",
       "      <td>Phaeocystaceae</td>\n",
       "      <td>Phaeocystis</td>\n",
       "      <td>Phaeocystis_pouchetii</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Hit_number eval bitscore    %ID  \\\n",
       "ASV                                      \n",
       "ASV_1           1  0.0      350  100.0   \n",
       "ASV_1           2  0.0      350  100.0   \n",
       "ASV_1           3  0.0      350  100.0   \n",
       "ASV_1           4  0.0      350  100.0   \n",
       "ASV_1           5  0.0      350  100.0   \n",
       "\n",
       "                                                 hit_def align_len  \\\n",
       "ASV                                                                  \n",
       "ASV_1  KC488454.1.1667_U|18S_rRNA|nucleus|clone_HL5aS...       350   \n",
       "ASV_1  HM561166.1.964_U|18S_rRNA|nucleus|clone_CFL133...       350   \n",
       "ASV_1  HM561167.1.839_U|18S_rRNA|nucleus|clone_CFL133...       350   \n",
       "ASV_1  HM561164.1.955_U|18S_rRNA|nucleus|clone_CFL133...       350   \n",
       "ASV_1  HM561163.1.945_U|18S_rRNA|nucleus|clone_CFL133...       350   \n",
       "\n",
       "                  PR2_ID PR2_marker location                nuc     Domain  \\\n",
       "ASV                                                                          \n",
       "ASV_1  KC488454.1.1667_U   18S_rRNA  nucleus  clone_HL5aSF04.03  Eukaryota   \n",
       "ASV_1   HM561166.1.964_U   18S_rRNA  nucleus    clone_CFL133R09  Eukaryota   \n",
       "ASV_1   HM561167.1.839_U   18S_rRNA  nucleus    clone_CFL133R10  Eukaryota   \n",
       "ASV_1   HM561164.1.955_U   18S_rRNA  nucleus   clone_CFL133DB11  Eukaryota   \n",
       "ASV_1   HM561163.1.945_U   18S_rRNA  nucleus   clone_CFL133DB10  Eukaryota   \n",
       "\n",
       "      Supergroup    Division   Subdivision             Class          Order  \\\n",
       "ASV                                                                           \n",
       "ASV_1   Haptista  Haptophyta  Haptophyta_X  Prymnesiophyceae  Phaeocystales   \n",
       "ASV_1   Haptista  Haptophyta  Haptophyta_X  Prymnesiophyceae  Phaeocystales   \n",
       "ASV_1   Haptista  Haptophyta  Haptophyta_X  Prymnesiophyceae  Phaeocystales   \n",
       "ASV_1   Haptista  Haptophyta  Haptophyta_X  Prymnesiophyceae  Phaeocystales   \n",
       "ASV_1   Haptista  Haptophyta  Haptophyta_X  Prymnesiophyceae  Phaeocystales   \n",
       "\n",
       "               Family        Genus                Species  \n",
       "ASV                                                        \n",
       "ASV_1  Phaeocystaceae  Phaeocystis  Phaeocystis_pouchetii  \n",
       "ASV_1  Phaeocystaceae  Phaeocystis  Phaeocystis_pouchetii  \n",
       "ASV_1  Phaeocystaceae  Phaeocystis  Phaeocystis_pouchetii  \n",
       "ASV_1  Phaeocystaceae  Phaeocystis  Phaeocystis_pouchetii  \n",
       "ASV_1  Phaeocystaceae  Phaeocystis  Phaeocystis_pouchetii  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Save Blast Results to file\n",
    "df = pd.DataFrame(hit_dict)\n",
    "df=df.T\n",
    "df.columns = ['eval','bitscore', '%ID', 'hit_def', 'align_len']\n",
    "df.reset_index(inplace=True)\n",
    "df = df.rename(columns={'level_0': 'ASV', 'level_1': 'Hit_number'})\n",
    "# parse out PR2 information from 'hit_def'\n",
    "print(df['hit_def'].iloc[0])\n",
    "# domain, supergroup, division, subdivision, class, order, family, genus, species\n",
    "col1 = ['PR2_ID', 'PR2_marker', 'location','nuc',\n",
    "        'Domain','Supergroup','Division','Subdivision','Class', 'Order',\n",
    "        'Family','Genus', 'Species']\n",
    "#col2 = ['Kingdom', 'Phylum', 'Class', 'Order', 'Family','subfamily','Genus', 'Species']\n",
    "# split PR2 columns\n",
    "for i in range(len(col1)):\n",
    "    df[col1[i]] = df['hit_def'].str.split('|').str[i]\n",
    "# split taxonomy levels\n",
    "# for i in range(len(col2)):\n",
    "#     df[col2[i]] = df['PR2_taxonomy'].str.split(',').str[i]\n",
    "df.set_index('ASV', inplace=True)\n",
    "blast_results = df.copy()\n",
    "blast_results.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2f594041-4ff1-43b6-bee6-a1de683153f6",
   "metadata": {},
   "source": [
    "## Export blast results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "3a9dacc3-34ad-497b-bfca-4e7cb90f5a6f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hit_number</th>\n",
       "      <th>eval</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>%ID</th>\n",
       "      <th>hit_def</th>\n",
       "      <th>align_len</th>\n",
       "      <th>PR2_ID</th>\n",
       "      <th>PR2_marker</th>\n",
       "      <th>location</th>\n",
       "      <th>nuc</th>\n",
       "      <th>Domain</th>\n",
       "      <th>Supergroup</th>\n",
       "      <th>Division</th>\n",
       "      <th>Subdivision</th>\n",
       "      <th>Class</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ASV_1</th>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>350</td>\n",
       "      <td>100.0</td>\n",
       "      <td>KC488454.1.1667_U|18S_rRNA|nucleus|clone_HL5aS...</td>\n",
       "      <td>350</td>\n",
       "      <td>KC488454.1.1667_U</td>\n",
       "      <td>18S_rRNA</td>\n",
       "      <td>nucleus</td>\n",
       "      <td>clone_HL5aSF04.03</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Haptista</td>\n",
       "      <td>Haptophyta</td>\n",
       "      <td>Haptophyta_X</td>\n",
       "      <td>Prymnesiophyceae</td>\n",
       "      <td>Phaeocystales</td>\n",
       "      <td>Phaeocystaceae</td>\n",
       "      <td>Phaeocystis</td>\n",
       "      <td>Phaeocystis_pouchetii</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_2</th>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>127</td>\n",
       "      <td>100.0</td>\n",
       "      <td>AJ535170.1.1763_U|18S_rRNA|nucleus|clone_p928|...</td>\n",
       "      <td>127</td>\n",
       "      <td>AJ535170.1.1763_U</td>\n",
       "      <td>18S_rRNA</td>\n",
       "      <td>nucleus</td>\n",
       "      <td>clone_p928</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>TSAR</td>\n",
       "      <td>Stramenopiles</td>\n",
       "      <td>Gyrista</td>\n",
       "      <td>Mediophyceae</td>\n",
       "      <td>Thalassiosirales</td>\n",
       "      <td>Thalassiosiraceae</td>\n",
       "      <td>Conticribra</td>\n",
       "      <td>Conticribra_weissflogii</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_3</th>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>118</td>\n",
       "      <td>97.64</td>\n",
       "      <td>KJ961704.1.1788_U|18S_rRNA|nucleus|strain_52|E...</td>\n",
       "      <td>127</td>\n",
       "      <td>KJ961704.1.1788_U</td>\n",
       "      <td>18S_rRNA</td>\n",
       "      <td>nucleus</td>\n",
       "      <td>strain_52</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>TSAR</td>\n",
       "      <td>Stramenopiles</td>\n",
       "      <td>Gyrista</td>\n",
       "      <td>Coscinodiscophyceae</td>\n",
       "      <td>Melosirales</td>\n",
       "      <td>Melosiraceae</td>\n",
       "      <td>Melosira</td>\n",
       "      <td>Melosira_sp.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_4</th>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>127</td>\n",
       "      <td>100.0</td>\n",
       "      <td>GU373969.1.1789_U|18S_rRNA|nucleus|strain_CCMP...</td>\n",
       "      <td>127</td>\n",
       "      <td>GU373969.1.1789_U</td>\n",
       "      <td>18S_rRNA</td>\n",
       "      <td>nucleus</td>\n",
       "      <td>strain_CCMP1440</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>TSAR</td>\n",
       "      <td>Stramenopiles</td>\n",
       "      <td>Gyrista</td>\n",
       "      <td>Bacillariophyceae</td>\n",
       "      <td>Bacillariales</td>\n",
       "      <td>Bacillariaceae</td>\n",
       "      <td>Pseudo-nitzschia</td>\n",
       "      <td>Pseudo-nitzschia_seriata</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_5</th>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>126</td>\n",
       "      <td>100.0</td>\n",
       "      <td>MN965778.1.1765_U|18S_rRNA|nucleus||Eukaryota|...</td>\n",
       "      <td>126</td>\n",
       "      <td>MN965778.1.1765_U</td>\n",
       "      <td>18S_rRNA</td>\n",
       "      <td>nucleus</td>\n",
       "      <td></td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>TSAR</td>\n",
       "      <td>Alveolata</td>\n",
       "      <td>Dinoflagellata</td>\n",
       "      <td>Dinophyceae</td>\n",
       "      <td>Gymnodiniales</td>\n",
       "      <td>Kareniaceae</td>\n",
       "      <td>Shimiella</td>\n",
       "      <td>Shimiella_gracilenta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_14543</th>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>357</td>\n",
       "      <td>97.9</td>\n",
       "      <td>KY979976.1.1705_U|18S_rRNA|nucleus||Eukaryota|...</td>\n",
       "      <td>381</td>\n",
       "      <td>KY979976.1.1705_U</td>\n",
       "      <td>18S_rRNA</td>\n",
       "      <td>nucleus</td>\n",
       "      <td></td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>TSAR</td>\n",
       "      <td>Stramenopiles</td>\n",
       "      <td>Gyrista</td>\n",
       "      <td>Pelagophyceae</td>\n",
       "      <td>Pelagomonadales</td>\n",
       "      <td>Pelagomonadales_clade_A</td>\n",
       "      <td>Pelagomonadales_clade_A_X</td>\n",
       "      <td>Pelagomonadales_clade_A_X_sp.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_14544</th>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>361</td>\n",
       "      <td>99.73</td>\n",
       "      <td>KR064016.1.709_U|18S_rRNA|nucleus|clone_St4_DC...</td>\n",
       "      <td>364</td>\n",
       "      <td>KR064016.1.709_U</td>\n",
       "      <td>18S_rRNA</td>\n",
       "      <td>nucleus</td>\n",
       "      <td>clone_St4_DCM_31-71_502f</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>TSAR</td>\n",
       "      <td>Alveolata</td>\n",
       "      <td>Ciliophora</td>\n",
       "      <td>Oligohymenophorea</td>\n",
       "      <td>OLIGO5</td>\n",
       "      <td>OLIGO5_X</td>\n",
       "      <td>OLIGO5_XX</td>\n",
       "      <td>OLIGO5_XX_sp.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_14545</th>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>368</td>\n",
       "      <td>100.0</td>\n",
       "      <td>KJ762992.1.1771_U|18S_rRNA|nucleus|clone_SGUH1...</td>\n",
       "      <td>368</td>\n",
       "      <td>KJ762992.1.1771_U</td>\n",
       "      <td>18S_rRNA</td>\n",
       "      <td>nucleus</td>\n",
       "      <td>clone_SGUH1368</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>TSAR</td>\n",
       "      <td>Alveolata</td>\n",
       "      <td>Ciliophora</td>\n",
       "      <td>Spirotrichea</td>\n",
       "      <td>Oligotrichida</td>\n",
       "      <td>Strombidiidae</td>\n",
       "      <td>Strombidiidae_X</td>\n",
       "      <td>Strombidiidae_X_sp.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_14546</th>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>365</td>\n",
       "      <td>100.0</td>\n",
       "      <td>KJ763536.1.1777_U|18S_rRNA|nucleus|clone_SGYP1...</td>\n",
       "      <td>365</td>\n",
       "      <td>KJ763536.1.1777_U</td>\n",
       "      <td>18S_rRNA</td>\n",
       "      <td>nucleus</td>\n",
       "      <td>clone_SGYP1304</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>TSAR</td>\n",
       "      <td>Rhizaria</td>\n",
       "      <td>Radiolaria</td>\n",
       "      <td>Acantharea</td>\n",
       "      <td>Acantharea_4</td>\n",
       "      <td>Acantharea_4d</td>\n",
       "      <td>Acantharea_4d_X</td>\n",
       "      <td>Acantharea_4d_X_sp.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_14547</th>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>369</td>\n",
       "      <td>100.0</td>\n",
       "      <td>PQ554980.1.536_U|18S_rRNA|nucleus||Eukaryota|A...</td>\n",
       "      <td>369</td>\n",
       "      <td>PQ554980.1.536_U</td>\n",
       "      <td>18S_rRNA</td>\n",
       "      <td>nucleus</td>\n",
       "      <td></td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Archaeplastida</td>\n",
       "      <td>Chlorophyta</td>\n",
       "      <td>Chlorophyta_X</td>\n",
       "      <td>Mamiellophyceae</td>\n",
       "      <td>Mamiellales</td>\n",
       "      <td>Mamiellaceae</td>\n",
       "      <td>Micromonas</td>\n",
       "      <td>Micromonas_polaris</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>13408 rows Ã— 19 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           Hit_number eval bitscore    %ID  \\\n",
       "ASV                                          \n",
       "ASV_1               1  0.0      350  100.0   \n",
       "ASV_2               1  0.0      127  100.0   \n",
       "ASV_3               1  0.0      118  97.64   \n",
       "ASV_4               1  0.0      127  100.0   \n",
       "ASV_5               1  0.0      126  100.0   \n",
       "...               ...  ...      ...    ...   \n",
       "ASV_14543           1  0.0      357   97.9   \n",
       "ASV_14544           1  0.0      361  99.73   \n",
       "ASV_14545           1  0.0      368  100.0   \n",
       "ASV_14546           1  0.0      365  100.0   \n",
       "ASV_14547           1  0.0      369  100.0   \n",
       "\n",
       "                                                     hit_def align_len  \\\n",
       "ASV                                                                      \n",
       "ASV_1      KC488454.1.1667_U|18S_rRNA|nucleus|clone_HL5aS...       350   \n",
       "ASV_2      AJ535170.1.1763_U|18S_rRNA|nucleus|clone_p928|...       127   \n",
       "ASV_3      KJ961704.1.1788_U|18S_rRNA|nucleus|strain_52|E...       127   \n",
       "ASV_4      GU373969.1.1789_U|18S_rRNA|nucleus|strain_CCMP...       127   \n",
       "ASV_5      MN965778.1.1765_U|18S_rRNA|nucleus||Eukaryota|...       126   \n",
       "...                                                      ...       ...   \n",
       "ASV_14543  KY979976.1.1705_U|18S_rRNA|nucleus||Eukaryota|...       381   \n",
       "ASV_14544  KR064016.1.709_U|18S_rRNA|nucleus|clone_St4_DC...       364   \n",
       "ASV_14545  KJ762992.1.1771_U|18S_rRNA|nucleus|clone_SGUH1...       368   \n",
       "ASV_14546  KJ763536.1.1777_U|18S_rRNA|nucleus|clone_SGYP1...       365   \n",
       "ASV_14547  PQ554980.1.536_U|18S_rRNA|nucleus||Eukaryota|A...       369   \n",
       "\n",
       "                      PR2_ID PR2_marker location                       nuc  \\\n",
       "ASV                                                                          \n",
       "ASV_1      KC488454.1.1667_U   18S_rRNA  nucleus         clone_HL5aSF04.03   \n",
       "ASV_2      AJ535170.1.1763_U   18S_rRNA  nucleus                clone_p928   \n",
       "ASV_3      KJ961704.1.1788_U   18S_rRNA  nucleus                 strain_52   \n",
       "ASV_4      GU373969.1.1789_U   18S_rRNA  nucleus           strain_CCMP1440   \n",
       "ASV_5      MN965778.1.1765_U   18S_rRNA  nucleus                             \n",
       "...                      ...        ...      ...                       ...   \n",
       "ASV_14543  KY979976.1.1705_U   18S_rRNA  nucleus                             \n",
       "ASV_14544   KR064016.1.709_U   18S_rRNA  nucleus  clone_St4_DCM_31-71_502f   \n",
       "ASV_14545  KJ762992.1.1771_U   18S_rRNA  nucleus            clone_SGUH1368   \n",
       "ASV_14546  KJ763536.1.1777_U   18S_rRNA  nucleus            clone_SGYP1304   \n",
       "ASV_14547   PQ554980.1.536_U   18S_rRNA  nucleus                             \n",
       "\n",
       "              Domain      Supergroup       Division     Subdivision  \\\n",
       "ASV                                                                   \n",
       "ASV_1      Eukaryota        Haptista     Haptophyta    Haptophyta_X   \n",
       "ASV_2      Eukaryota            TSAR  Stramenopiles         Gyrista   \n",
       "ASV_3      Eukaryota            TSAR  Stramenopiles         Gyrista   \n",
       "ASV_4      Eukaryota            TSAR  Stramenopiles         Gyrista   \n",
       "ASV_5      Eukaryota            TSAR      Alveolata  Dinoflagellata   \n",
       "...              ...             ...            ...             ...   \n",
       "ASV_14543  Eukaryota            TSAR  Stramenopiles         Gyrista   \n",
       "ASV_14544  Eukaryota            TSAR      Alveolata      Ciliophora   \n",
       "ASV_14545  Eukaryota            TSAR      Alveolata      Ciliophora   \n",
       "ASV_14546  Eukaryota            TSAR       Rhizaria      Radiolaria   \n",
       "ASV_14547  Eukaryota  Archaeplastida    Chlorophyta   Chlorophyta_X   \n",
       "\n",
       "                         Class             Order                   Family  \\\n",
       "ASV                                                                         \n",
       "ASV_1         Prymnesiophyceae     Phaeocystales           Phaeocystaceae   \n",
       "ASV_2             Mediophyceae  Thalassiosirales        Thalassiosiraceae   \n",
       "ASV_3      Coscinodiscophyceae       Melosirales             Melosiraceae   \n",
       "ASV_4        Bacillariophyceae     Bacillariales           Bacillariaceae   \n",
       "ASV_5              Dinophyceae     Gymnodiniales              Kareniaceae   \n",
       "...                        ...               ...                      ...   \n",
       "ASV_14543        Pelagophyceae   Pelagomonadales  Pelagomonadales_clade_A   \n",
       "ASV_14544    Oligohymenophorea            OLIGO5                 OLIGO5_X   \n",
       "ASV_14545         Spirotrichea     Oligotrichida            Strombidiidae   \n",
       "ASV_14546           Acantharea      Acantharea_4            Acantharea_4d   \n",
       "ASV_14547      Mamiellophyceae       Mamiellales             Mamiellaceae   \n",
       "\n",
       "                               Genus                        Species  \n",
       "ASV                                                                  \n",
       "ASV_1                    Phaeocystis          Phaeocystis_pouchetii  \n",
       "ASV_2                    Conticribra        Conticribra_weissflogii  \n",
       "ASV_3                       Melosira                   Melosira_sp.  \n",
       "ASV_4               Pseudo-nitzschia       Pseudo-nitzschia_seriata  \n",
       "ASV_5                      Shimiella           Shimiella_gracilenta  \n",
       "...                              ...                            ...  \n",
       "ASV_14543  Pelagomonadales_clade_A_X  Pelagomonadales_clade_A_X_sp.  \n",
       "ASV_14544                  OLIGO5_XX                  OLIGO5_XX_sp.  \n",
       "ASV_14545            Strombidiidae_X            Strombidiidae_X_sp.  \n",
       "ASV_14546            Acantharea_4d_X            Acantharea_4d_X_sp.  \n",
       "ASV_14547                 Micromonas             Micromonas_polaris  \n",
       "\n",
       "[13408 rows x 19 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# just top hits\n",
    "df = blast_results.copy()\n",
    "df = df.reset_index()\n",
    "df = df.drop_duplicates('ASV')\n",
    "df = df.set_index('ASV')\n",
    "df.to_csv(directory+'PR2/PR2_blast_results_tophits.csv')\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "35a71824-e726-47cd-aa79-7cbb2b15811f",
   "metadata": {},
   "source": [
    "## Compare to blastn MEGAN results (nt)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "fedd2e7c-3403-4705-be01-f51bd36cf5a1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/kpitz/github/GLOMICON/intercomparison/Merged_Datasets/PR2/test_diatom.csv\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr:last-of-type th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th colspan=\"10\" halign=\"left\">PR2</th>\n",
       "      <th>...</th>\n",
       "      <th colspan=\"5\" halign=\"left\">MEGAN</th>\n",
       "      <th>Class_match</th>\n",
       "      <th>Order_match</th>\n",
       "      <th>Family_match</th>\n",
       "      <th>Genus_match</th>\n",
       "      <th>Species_match</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>Hit_number</th>\n",
       "      <th>eval</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>%ID</th>\n",
       "      <th>hit_def</th>\n",
       "      <th>align_len</th>\n",
       "      <th>PR2_ID</th>\n",
       "      <th>PR2_marker</th>\n",
       "      <th>location</th>\n",
       "      <th>nuc</th>\n",
       "      <th>...</th>\n",
       "      <th>Class</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ASV_4114</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>97</td>\n",
       "      <td>92.19</td>\n",
       "      <td>AJ535150.1.1787_U|18S_rRNA|nucleus|clone_p387|...</td>\n",
       "      <td>128</td>\n",
       "      <td>AJ535150.1.1787_U</td>\n",
       "      <td>18S_rRNA</td>\n",
       "      <td>nucleus</td>\n",
       "      <td>clone_p387</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_12880</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>319</td>\n",
       "      <td>95.2</td>\n",
       "      <td>AJ535150.1.1787_U|18S_rRNA|nucleus|clone_p387|...</td>\n",
       "      <td>375</td>\n",
       "      <td>AJ535150.1.1787_U</td>\n",
       "      <td>18S_rRNA</td>\n",
       "      <td>nucleus</td>\n",
       "      <td>clone_p387</td>\n",
       "      <td>...</td>\n",
       "      <td>Bacillariophyceae</td>\n",
       "      <td>Mastogloiales</td>\n",
       "      <td>Achnanthaceae</td>\n",
       "      <td>Achnanthes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_4469</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>94</td>\n",
       "      <td>92.56</td>\n",
       "      <td>HQ655888.1.1730_U|18S_rRNA|nucleus||Eukaryota|...</td>\n",
       "      <td>121</td>\n",
       "      <td>HQ655888.1.1730_U</td>\n",
       "      <td>18S_rRNA</td>\n",
       "      <td>nucleus</td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td>Bacillariophyceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_972</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>127</td>\n",
       "      <td>100.0</td>\n",
       "      <td>OR286793.1.1783_U|18S_rRNA|nucleus||Eukaryota|...</td>\n",
       "      <td>127</td>\n",
       "      <td>OR286793.1.1783_U</td>\n",
       "      <td>18S_rRNA</td>\n",
       "      <td>nucleus</td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td>Bacillariophyceae</td>\n",
       "      <td>Bacillariales</td>\n",
       "      <td>Bacillariaceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_10097</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>124</td>\n",
       "      <td>99.21</td>\n",
       "      <td>OR286793.1.1783_U|18S_rRNA|nucleus||Eukaryota|...</td>\n",
       "      <td>127</td>\n",
       "      <td>OR286793.1.1783_U</td>\n",
       "      <td>18S_rRNA</td>\n",
       "      <td>nucleus</td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td>Bacillariophyceae</td>\n",
       "      <td>Bacillariales</td>\n",
       "      <td>Bacillariaceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_431</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Mediophyceae</td>\n",
       "      <td>Cymatosirales</td>\n",
       "      <td>Cymatosiraceae</td>\n",
       "      <td>Brockmanniella</td>\n",
       "      <td>Brockmanniella brockmannii</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_1577</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Coscinodiscophyceae</td>\n",
       "      <td>Stephanodiscales</td>\n",
       "      <td>Stephanodiscaceae</td>\n",
       "      <td>Cyclotella</td>\n",
       "      <td>Cyclotella choctawhatcheeana</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_3465</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Coscinodiscophyceae</td>\n",
       "      <td>Paraliales</td>\n",
       "      <td>Paraliaceae</td>\n",
       "      <td>Paralia</td>\n",
       "      <td>Paralia sulcata</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_9410</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Coscinodiscophyceae</td>\n",
       "      <td>Chaetocerotales</td>\n",
       "      <td>Chaetocerotaceae</td>\n",
       "      <td>Chaetoceros</td>\n",
       "      <td>Chaetoceros costatus</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_2514</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Coscinodiscophyceae</td>\n",
       "      <td>Melosirales</td>\n",
       "      <td>Melosiraceae</td>\n",
       "      <td>Melosira</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1327 rows Ã— 32 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 PR2                       \\\n",
       "          Hit_number eval bitscore    %ID   \n",
       "ASV                                         \n",
       "ASV_4114         1.0  0.0       97  92.19   \n",
       "ASV_12880        1.0  0.0      319   95.2   \n",
       "ASV_4469         1.0  0.0       94  92.56   \n",
       "ASV_972          1.0  0.0      127  100.0   \n",
       "ASV_10097        1.0  0.0      124  99.21   \n",
       "...              ...  ...      ...    ...   \n",
       "ASV_431          NaN  NaN      NaN    NaN   \n",
       "ASV_1577         NaN  NaN      NaN    NaN   \n",
       "ASV_3465         NaN  NaN      NaN    NaN   \n",
       "ASV_9410         NaN  NaN      NaN    NaN   \n",
       "ASV_2514         NaN  NaN      NaN    NaN   \n",
       "\n",
       "                                                                        \\\n",
       "                                                     hit_def align_len   \n",
       "ASV                                                                      \n",
       "ASV_4114   AJ535150.1.1787_U|18S_rRNA|nucleus|clone_p387|...       128   \n",
       "ASV_12880  AJ535150.1.1787_U|18S_rRNA|nucleus|clone_p387|...       375   \n",
       "ASV_4469   HQ655888.1.1730_U|18S_rRNA|nucleus||Eukaryota|...       121   \n",
       "ASV_972    OR286793.1.1783_U|18S_rRNA|nucleus||Eukaryota|...       127   \n",
       "ASV_10097  OR286793.1.1783_U|18S_rRNA|nucleus||Eukaryota|...       127   \n",
       "...                                                      ...       ...   \n",
       "ASV_431                                                  NaN       NaN   \n",
       "ASV_1577                                                 NaN       NaN   \n",
       "ASV_3465                                                 NaN       NaN   \n",
       "ASV_9410                                                 NaN       NaN   \n",
       "ASV_2514                                                 NaN       NaN   \n",
       "\n",
       "                                                              ...  \\\n",
       "                      PR2_ID PR2_marker location         nuc  ...   \n",
       "ASV                                                           ...   \n",
       "ASV_4114   AJ535150.1.1787_U   18S_rRNA  nucleus  clone_p387  ...   \n",
       "ASV_12880  AJ535150.1.1787_U   18S_rRNA  nucleus  clone_p387  ...   \n",
       "ASV_4469   HQ655888.1.1730_U   18S_rRNA  nucleus              ...   \n",
       "ASV_972    OR286793.1.1783_U   18S_rRNA  nucleus              ...   \n",
       "ASV_10097  OR286793.1.1783_U   18S_rRNA  nucleus              ...   \n",
       "...                      ...        ...      ...         ...  ...   \n",
       "ASV_431                  NaN        NaN      NaN         NaN  ...   \n",
       "ASV_1577                 NaN        NaN      NaN         NaN  ...   \n",
       "ASV_3465                 NaN        NaN      NaN         NaN  ...   \n",
       "ASV_9410                 NaN        NaN      NaN         NaN  ...   \n",
       "ASV_2514                 NaN        NaN      NaN         NaN  ...   \n",
       "\n",
       "                         MEGAN                                       \\\n",
       "                         Class             Order             Family   \n",
       "ASV                                                                   \n",
       "ASV_4114                   NaN               NaN                NaN   \n",
       "ASV_12880    Bacillariophyceae     Mastogloiales      Achnanthaceae   \n",
       "ASV_4469     Bacillariophyceae               NaN                NaN   \n",
       "ASV_972      Bacillariophyceae     Bacillariales     Bacillariaceae   \n",
       "ASV_10097    Bacillariophyceae     Bacillariales     Bacillariaceae   \n",
       "...                        ...               ...                ...   \n",
       "ASV_431           Mediophyceae     Cymatosirales     Cymatosiraceae   \n",
       "ASV_1577   Coscinodiscophyceae  Stephanodiscales  Stephanodiscaceae   \n",
       "ASV_3465   Coscinodiscophyceae        Paraliales        Paraliaceae   \n",
       "ASV_9410   Coscinodiscophyceae   Chaetocerotales   Chaetocerotaceae   \n",
       "ASV_2514   Coscinodiscophyceae       Melosirales       Melosiraceae   \n",
       "\n",
       "                                                        Class_match  \\\n",
       "                    Genus                       Species               \n",
       "ASV                                                                   \n",
       "ASV_4114              NaN                           NaN           0   \n",
       "ASV_12880      Achnanthes                           NaN           1   \n",
       "ASV_4469              NaN                           NaN           1   \n",
       "ASV_972               NaN                           NaN           1   \n",
       "ASV_10097             NaN                           NaN           1   \n",
       "...                   ...                           ...         ...   \n",
       "ASV_431    Brockmanniella    Brockmanniella brockmannii           0   \n",
       "ASV_1577       Cyclotella  Cyclotella choctawhatcheeana           0   \n",
       "ASV_3465          Paralia               Paralia sulcata           0   \n",
       "ASV_9410      Chaetoceros          Chaetoceros costatus           0   \n",
       "ASV_2514         Melosira                           NaN           0   \n",
       "\n",
       "          Order_match Family_match Genus_match Species_match  \n",
       "                                                              \n",
       "ASV                                                           \n",
       "ASV_4114            0            0           0             0  \n",
       "ASV_12880           0            1           1             0  \n",
       "ASV_4469            0            0           0             0  \n",
       "ASV_972             1            1           0             0  \n",
       "ASV_10097           1            1           0             0  \n",
       "...               ...          ...         ...           ...  \n",
       "ASV_431             0            0           0             0  \n",
       "ASV_1577            0            0           0             0  \n",
       "ASV_3465            0            0           0             0  \n",
       "ASV_9410            0            0           0             0  \n",
       "ASV_2514            0            0           0             0  \n",
       "\n",
       "[1327 rows x 32 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = blast_results.copy()\n",
    "df = df.reset_index()\n",
    "df = df.drop_duplicates('ASV')\n",
    "df = df.set_index('ASV')\n",
    "df = pd.concat([df, taxa], axis=1, keys=['PR2','MEGAN'])\n",
    "\n",
    "levels = ['Class', 'Order', 'Family','Genus', 'Species']\n",
    "for tax in levels:\n",
    "    df[tax+'_match'] = 0\n",
    "    df.loc[df['PR2', tax]==df['MEGAN', tax], tax+'_match'] = 1\n",
    "    # 26245\n",
    "\n",
    "cols = ['Class_match', 'Order_match', 'Family_match','Genus_match', 'Species_match']\n",
    "df = df.sort_values(cols)\n",
    "\n",
    "'''#Genus\n",
    "df['genus_match'] = 0\n",
    "df.loc[df['PR2', 'Genus']==df['MEGAN', 'Genus'], 'genus_match'] = 1\n",
    "# 26245\n",
    "df = df.loc[df['genus_match']==1]\n",
    "#3953 genus\n",
    "df = df.sort_values('genus_match')'''\n",
    "\n",
    "#print(directory+'test.csv')\n",
    "df.to_csv(directory+'test.csv')\n",
    "df = df.loc[df['MEGAN','Phylum']=='Bacillariophyta']\n",
    "df = df.sort_values([('PR2','Class'), ('PR2','Order'), ('PR2','Family'),('PR2','Genus'), ('PR2','Species')])\n",
    "#df = df.drop_duplicates([('PR2','Species'), ('MEGAN', 'Species')])\n",
    "#df['count'] = 1\n",
    "#df = df.groupby([('PR2','Species'), ('MEGAN', 'Species')]).sum()\n",
    "#df = df[['count']]\n",
    "df.to_csv(directory+'PR2/test_diatom.csv')\n",
    "print(directory+'PR2/test_diatom.csv')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "f653ae6d-720e-4238-99c0-b5ffb192f076",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/kpitz/github/GLOMICON/intercomparison/Merged_Datasets/test.csv\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr:last-of-type th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>(PR2, Family)</th>\n",
       "      <th>(MEGAN, Family)</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Achnanthaceae</th>\n",
       "      <th>Achnanthaceae</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Anaulaceae</th>\n",
       "      <th>Anaulaceae</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Asterionellopsidaceae</th>\n",
       "      <th>Fragilariaceae</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Asterolampraceae</th>\n",
       "      <th>Asterolampraceae</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Attheyaceae</th>\n",
       "      <th>Attheyaceae</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">Thalassiosiraceae</th>\n",
       "      <th>Cymatosiraceae</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pleurosigmataceae</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Stephanodiscaceae</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Thalassiosiraceae</th>\n",
       "      <td>285</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Triceratiaceae</th>\n",
       "      <th>Triceratiaceae</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>93 rows Ã— 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        count\n",
       "                                             \n",
       "(PR2, Family)         (MEGAN, Family)        \n",
       "Achnanthaceae         Achnanthaceae         1\n",
       "Anaulaceae            Anaulaceae            2\n",
       "Asterionellopsidaceae Fragilariaceae        2\n",
       "Asterolampraceae      Asterolampraceae      4\n",
       "Attheyaceae           Attheyaceae           4\n",
       "...                                       ...\n",
       "Thalassiosiraceae     Cymatosiraceae        3\n",
       "                      Pleurosigmataceae     1\n",
       "                      Stephanodiscaceae     1\n",
       "                      Thalassiosiraceae   285\n",
       "Triceratiaceae        Triceratiaceae        3\n",
       "\n",
       "[93 rows x 1 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = blast_results.copy()\n",
    "df = df.reset_index()\n",
    "df = df.drop_duplicates('ASV')\n",
    "df = df.set_index('ASV')\n",
    "df = pd.concat([df, taxa], axis=1, keys=['PR2','MEGAN'])\n",
    "\n",
    "levels = ['Class', 'Order', 'Family','Genus', 'Species']\n",
    "for tax in levels:\n",
    "    df[tax+'_match'] = 0\n",
    "    df.loc[df['PR2', tax]==df['MEGAN', tax], tax+'_match'] = 1\n",
    "    # 26245\n",
    "\n",
    "cols = ['Class_match', 'Order_match', 'Family_match','Genus_match', 'Species_match']\n",
    "df = df.sort_values(cols)\n",
    "\n",
    "'''#Genus\n",
    "df['genus_match'] = 0\n",
    "df.loc[df['PR2', 'Genus']==df['MEGAN', 'Genus'], 'genus_match'] = 1\n",
    "# 26245\n",
    "df = df.loc[df['genus_match']==1]\n",
    "#3953 genus\n",
    "df = df.sort_values('genus_match')'''\n",
    "\n",
    "print(directory+'PR2/test.csv')\n",
    "df.to_csv(directory+'PR2/test.csv')\n",
    "#df = df.loc[df['PR2','Class']=='Dinophyceae']\n",
    "df = df.loc[df['MEGAN','Phylum']=='Bacillariophyta']\n",
    "df = df.sort_values([('PR2','Class'), ('PR2','Order'), ('PR2','Family'),('PR2','Genus'), ('PR2','Species')])\n",
    "#df = df.drop_duplicates([('PR2','Species'), ('MEGAN', 'Species')])\n",
    "df['count'] = 1\n",
    "df = df.groupby([('PR2','Family'), ('MEGAN', 'Family')]).sum()\n",
    "df = df[['count']]\n",
    "df.to_csv(directory+'PR2/test_Class.csv')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "22616115-b572-4397-bc93-dee5fbc4d96d",
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "('PR2', 'Kingdom')",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "File \u001b[0;32m~/anaconda3/envs/scipy_test/lib/python3.12/site-packages/pandas/core/indexes/multi.py:3053\u001b[0m, in \u001b[0;36mMultiIndex.get_loc\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   3052\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m-> 3053\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_engine\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_loc\u001b[49m\u001b[43m(\u001b[49m\u001b[43mkey\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   3054\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m err:\n",
      "File \u001b[0;32mindex.pyx:776\u001b[0m, in \u001b[0;36mpandas._libs.index.BaseMultiIndexCodesEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32mindex.pyx:167\u001b[0m, in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32mindex.pyx:196\u001b[0m, in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32mpandas/_libs/hashtable_class_helper.pxi:2152\u001b[0m, in \u001b[0;36mpandas._libs.hashtable.UInt64HashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32mpandas/_libs/hashtable_class_helper.pxi:2176\u001b[0m, in \u001b[0;36mpandas._libs.hashtable.UInt64HashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 73",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[17], line 11\u001b[0m\n\u001b[1;32m      9\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m tax \u001b[38;5;129;01min\u001b[39;00m levels:\n\u001b[1;32m     10\u001b[0m     df[tax\u001b[38;5;241m+\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m_match\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m=\u001b[39m \u001b[38;5;241m0\u001b[39m\n\u001b[0;32m---> 11\u001b[0m     df\u001b[38;5;241m.\u001b[39mloc[\u001b[43mdf\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mPR2\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtax\u001b[49m\u001b[43m]\u001b[49m\u001b[38;5;241m==\u001b[39mdf[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mMEGAN\u001b[39m\u001b[38;5;124m'\u001b[39m, tax], tax\u001b[38;5;241m+\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m_match\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m=\u001b[39m \u001b[38;5;241m1\u001b[39m\n\u001b[1;32m     12\u001b[0m     \u001b[38;5;66;03m# 26245\u001b[39;00m\n\u001b[1;32m     14\u001b[0m cols \u001b[38;5;241m=\u001b[39m [\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mKingdom_match\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mPhylum_match\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mClass_match\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mOrder_match\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mFamily_match\u001b[39m\u001b[38;5;124m'\u001b[39m,\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mGenus_match\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mSpecies_match\u001b[39m\u001b[38;5;124m'\u001b[39m]\n",
      "File \u001b[0;32m~/anaconda3/envs/scipy_test/lib/python3.12/site-packages/pandas/core/frame.py:4089\u001b[0m, in \u001b[0;36mDataFrame.__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   4087\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m is_single_key:\n\u001b[1;32m   4088\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mcolumns\u001b[38;5;241m.\u001b[39mnlevels \u001b[38;5;241m>\u001b[39m \u001b[38;5;241m1\u001b[39m:\n\u001b[0;32m-> 4089\u001b[0m         \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_getitem_multilevel\u001b[49m\u001b[43m(\u001b[49m\u001b[43mkey\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   4090\u001b[0m     indexer \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mcolumns\u001b[38;5;241m.\u001b[39mget_loc(key)\n\u001b[1;32m   4091\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m is_integer(indexer):\n",
      "File \u001b[0;32m~/anaconda3/envs/scipy_test/lib/python3.12/site-packages/pandas/core/frame.py:4147\u001b[0m, in \u001b[0;36mDataFrame._getitem_multilevel\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   4145\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21m_getitem_multilevel\u001b[39m(\u001b[38;5;28mself\u001b[39m, key):\n\u001b[1;32m   4146\u001b[0m     \u001b[38;5;66;03m# self.columns is a MultiIndex\u001b[39;00m\n\u001b[0;32m-> 4147\u001b[0m     loc \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcolumns\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_loc\u001b[49m\u001b[43m(\u001b[49m\u001b[43mkey\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   4148\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(loc, (\u001b[38;5;28mslice\u001b[39m, np\u001b[38;5;241m.\u001b[39mndarray)):\n\u001b[1;32m   4149\u001b[0m         new_columns \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mcolumns[loc]\n",
      "File \u001b[0;32m~/anaconda3/envs/scipy_test/lib/python3.12/site-packages/pandas/core/indexes/multi.py:3055\u001b[0m, in \u001b[0;36mMultiIndex.get_loc\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   3053\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_engine\u001b[38;5;241m.\u001b[39mget_loc(key)\n\u001b[1;32m   3054\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m err:\n\u001b[0;32m-> 3055\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m(key) \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01merr\u001b[39;00m\n\u001b[1;32m   3056\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mTypeError\u001b[39;00m:\n\u001b[1;32m   3057\u001b[0m     \u001b[38;5;66;03m# e.g. test_partial_slicing_with_multiindex partial string slicing\u001b[39;00m\n\u001b[1;32m   3058\u001b[0m     loc, _ \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mget_loc_level(key, \u001b[38;5;28mlist\u001b[39m(\u001b[38;5;28mrange\u001b[39m(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mnlevels)))\n",
      "\u001b[0;31mKeyError\u001b[0m: ('PR2', 'Kingdom')"
     ]
    }
   ],
   "source": [
    "df = blast_results.copy()\n",
    "df = df.reset_index()\n",
    "df = df.drop_duplicates('ASV')\n",
    "df = df.set_index('ASV')\n",
    "df = pd.concat([df, taxa], axis=1, keys=['PR2','MEGAN'])\n",
    "df = df.loc[df['PR2','Genus']=='Pseudocalanus']\n",
    "\n",
    "levels = ['Kingdom', 'Phylum', 'Class', 'Order', 'Family','Genus', 'Species']\n",
    "for tax in levels:\n",
    "    df[tax+'_match'] = 0\n",
    "    df.loc[df['PR2', tax]==df['MEGAN', tax], tax+'_match'] = 1\n",
    "    # 26245\n",
    "\n",
    "cols = ['Kingdom_match', 'Phylum_match', 'Class_match', 'Order_match', 'Family_match','Genus_match', 'Species_match']\n",
    "df = df.sort_values(cols)\n",
    "\n",
    "'''#Genus\n",
    "df['genus_match'] = 0\n",
    "df.loc[df['PR2', 'Genus']==df['MEGAN', 'Genus'], 'genus_match'] = 1\n",
    "# 26245\n",
    "df = df.loc[df['genus_match']==1]\n",
    "#3953 genus\n",
    "df = df.sort_values('genus_match')'''\n",
    "\n",
    "'''print(directory+'test.csv')\n",
    "df.to_csv(directory+'test.csv')\n",
    "df = df.loc[df['PR2','Class']=='Copepoda']\n",
    "df = df.sort_values([('PR2','Class'), ('PR2','Order'), ('PR2','Family'),('PR2','Genus'), ('PR2','Species')])\n",
    "#df = df.drop_duplicates([('PR2','Species'), ('MEGAN', 'Species')])\n",
    "df['count'] = 1\n",
    "df = df.groupby([('PR2','Species'), ('MEGAN', 'Species')]).sum()\n",
    "df = df[['count']]\n",
    "df.to_csv(directory+'test_copepoda.csv')'''\n",
    "df = df.sort_values([('PR2','Class'), ('PR2','Order'), ('PR2','Family'),('PR2','Genus'), ('PR2','Species')])\n",
    "df = df.loc[(df[('PR2', 'Species')]=='Pseudocalanus mimus')&(df[('MEGAN', 'Species')]!='Pseudocalanus mimus')]\n",
    "df.to_csv(directory+'PR2/test_copepoda.csv')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4bad96bd-fa46-4ec9-90a3-2a4299bc2239",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e92b8592-a1d7-4769-b4f3-afe828364e04",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "55e8ea2b-72f6-462b-89fc-1fd812f5a18a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a8161aac-5686-4e42-8944-eac775225299",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "27fb0d3c-3e9d-4899-95e2-6fafd91c70cd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ASV</th>\n",
       "      <th>Hit_number</th>\n",
       "      <th>eval</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>%ID</th>\n",
       "      <th>hit_def</th>\n",
       "      <th>align_len</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>ASV_3</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>110</td>\n",
       "      <td>79.73</td>\n",
       "      <td>BHNHM212-24|COI-5P|Afghanistan|Animalia,Arthro...</td>\n",
       "      <td>291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>ASV_5</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>113</td>\n",
       "      <td>79.52</td>\n",
       "      <td>GMQQT008-18|COI-5P|Australia|Animalia,Arthropo...</td>\n",
       "      <td>293</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>161</th>\n",
       "      <td>ASV_20</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>113</td>\n",
       "      <td>79.52</td>\n",
       "      <td>GMQQT008-18|COI-5P|Australia|Animalia,Arthropo...</td>\n",
       "      <td>293</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>163</th>\n",
       "      <td>ASV_23</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>105</td>\n",
       "      <td>78.69</td>\n",
       "      <td>CRBOF22983-24|COI-5P|Costa Rica|Animalia,Arthr...</td>\n",
       "      <td>291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>367</th>\n",
       "      <td>ASV_27</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>122</td>\n",
       "      <td>79.81</td>\n",
       "      <td>GBAAY62677-24|COI-5P|Unrecoverable|Mixture,Uns...</td>\n",
       "      <td>312</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>272978</th>\n",
       "      <td>ASV_269846</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>101</td>\n",
       "      <td>79.21</td>\n",
       "      <td>NBINS004-14|COI-5P|United States|Animalia,Arth...</td>\n",
       "      <td>279</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>274072</th>\n",
       "      <td>ASV_270224</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>97</td>\n",
       "      <td>77.59</td>\n",
       "      <td>POMS7650-24|COI-5P|United Kingdom|Animalia,Art...</td>\n",
       "      <td>299</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>277208</th>\n",
       "      <td>ASV_272913</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>102</td>\n",
       "      <td>78.35</td>\n",
       "      <td>CRALC29584-21|COI-5P|Costa Rica|Animalia,Arthr...</td>\n",
       "      <td>291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>277375</th>\n",
       "      <td>ASV_273151</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>98</td>\n",
       "      <td>78.16</td>\n",
       "      <td>GBAAY56519-24|COI-5P|Unrecoverable|Protista,Ha...</td>\n",
       "      <td>293</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>278590</th>\n",
       "      <td>ASV_274120</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>107</td>\n",
       "      <td>78.91</td>\n",
       "      <td>GMSVD8417-23|COI-5P|United States|Animalia,Art...</td>\n",
       "      <td>294</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3624 rows Ã— 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               ASV  Hit_number eval bitscore    %ID  \\\n",
       "38           ASV_3           1  0.0      110  79.73   \n",
       "49           ASV_5           1  0.0      113  79.52   \n",
       "161         ASV_20           1  0.0      113  79.52   \n",
       "163         ASV_23           1  0.0      105  78.69   \n",
       "367         ASV_27           1  0.0      122  79.81   \n",
       "...            ...         ...  ...      ...    ...   \n",
       "272978  ASV_269846           1  0.0      101  79.21   \n",
       "274072  ASV_270224           1  0.0       97  77.59   \n",
       "277208  ASV_272913           1  0.0      102  78.35   \n",
       "277375  ASV_273151           1  0.0       98  78.16   \n",
       "278590  ASV_274120           1  0.0      107  78.91   \n",
       "\n",
       "                                                  hit_def align_len  \n",
       "38      BHNHM212-24|COI-5P|Afghanistan|Animalia,Arthro...       291  \n",
       "49      GMQQT008-18|COI-5P|Australia|Animalia,Arthropo...       293  \n",
       "161     GMQQT008-18|COI-5P|Australia|Animalia,Arthropo...       293  \n",
       "163     CRBOF22983-24|COI-5P|Costa Rica|Animalia,Arthr...       291  \n",
       "367     GBAAY62677-24|COI-5P|Unrecoverable|Mixture,Uns...       312  \n",
       "...                                                   ...       ...  \n",
       "272978  NBINS004-14|COI-5P|United States|Animalia,Arth...       279  \n",
       "274072  POMS7650-24|COI-5P|United Kingdom|Animalia,Art...       299  \n",
       "277208  CRALC29584-21|COI-5P|Costa Rica|Animalia,Arthr...       291  \n",
       "277375  GBAAY56519-24|COI-5P|Unrecoverable|Protista,Ha...       293  \n",
       "278590  GMSVD8417-23|COI-5P|United States|Animalia,Art...       294  \n",
       "\n",
       "[3624 rows x 7 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = blast_results.copy()\n",
    "#df = df.loc[df['ASV']=='ASV_5']\n",
    "df = df.drop_duplicates('ASV')\n",
    "df = df.loc[df['%ID']<80]\n",
    "#df= df.loc[df['bitscore']>(308 - (308*.02))]\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "9c62fdfd-c55f-43c4-8b8b-328e9ce105e3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BHNHM212-24|COI-5P|Afghanistan|Animalia,Arthropoda,Insecta,Lepidoptera,Noctuidae,Noctuinae,Caradrina,Caradrina fergana,None\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ASV</th>\n",
       "      <th>Hit_number</th>\n",
       "      <th>eval</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>%ID</th>\n",
       "      <th>hit_def</th>\n",
       "      <th>align_len</th>\n",
       "      <th>BOLD_ID</th>\n",
       "      <th>BOLD_marker</th>\n",
       "      <th>BOLD_location</th>\n",
       "      <th>BOLD_taxonomy</th>\n",
       "      <th>Kingdom</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>subfamily</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>ASV_3</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>110</td>\n",
       "      <td>79.73</td>\n",
       "      <td>BHNHM212-24|COI-5P|Afghanistan|Animalia,Arthro...</td>\n",
       "      <td>291</td>\n",
       "      <td>BHNHM212-24</td>\n",
       "      <td>COI-5P</td>\n",
       "      <td>Afghanistan</td>\n",
       "      <td>Animalia,Arthropoda,Insecta,Lepidoptera,Noctui...</td>\n",
       "      <td>Animalia</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Insecta</td>\n",
       "      <td>Lepidoptera</td>\n",
       "      <td>Noctuidae</td>\n",
       "      <td>Noctuinae</td>\n",
       "      <td>Caradrina</td>\n",
       "      <td>Caradrina fergana</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>ASV_5</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>113</td>\n",
       "      <td>79.52</td>\n",
       "      <td>GMQQT008-18|COI-5P|Australia|Animalia,Arthropo...</td>\n",
       "      <td>293</td>\n",
       "      <td>GMQQT008-18</td>\n",
       "      <td>COI-5P</td>\n",
       "      <td>Australia</td>\n",
       "      <td>Animalia,Arthropoda,Insecta,Diptera,Rhagionida...</td>\n",
       "      <td>Animalia</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Insecta</td>\n",
       "      <td>Diptera</td>\n",
       "      <td>Rhagionidae</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>161</th>\n",
       "      <td>ASV_20</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>113</td>\n",
       "      <td>79.52</td>\n",
       "      <td>GMQQT008-18|COI-5P|Australia|Animalia,Arthropo...</td>\n",
       "      <td>293</td>\n",
       "      <td>GMQQT008-18</td>\n",
       "      <td>COI-5P</td>\n",
       "      <td>Australia</td>\n",
       "      <td>Animalia,Arthropoda,Insecta,Diptera,Rhagionida...</td>\n",
       "      <td>Animalia</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Insecta</td>\n",
       "      <td>Diptera</td>\n",
       "      <td>Rhagionidae</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>163</th>\n",
       "      <td>ASV_23</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>105</td>\n",
       "      <td>78.69</td>\n",
       "      <td>CRBOF22983-24|COI-5P|Costa Rica|Animalia,Arthr...</td>\n",
       "      <td>291</td>\n",
       "      <td>CRBOF22983-24</td>\n",
       "      <td>COI-5P</td>\n",
       "      <td>Costa Rica</td>\n",
       "      <td>Animalia,Arthropoda,Insecta,Diptera,Ceratopogo...</td>\n",
       "      <td>Animalia</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Insecta</td>\n",
       "      <td>Diptera</td>\n",
       "      <td>Ceratopogonidae</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>367</th>\n",
       "      <td>ASV_27</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>122</td>\n",
       "      <td>79.81</td>\n",
       "      <td>GBAAY62677-24|COI-5P|Unrecoverable|Mixture,Uns...</td>\n",
       "      <td>312</td>\n",
       "      <td>GBAAY62677-24</td>\n",
       "      <td>COI-5P</td>\n",
       "      <td>Unrecoverable</td>\n",
       "      <td>Mixture,Unspecified,None,None,None,None,None,N...</td>\n",
       "      <td>Mixture</td>\n",
       "      <td>Unspecified</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>272978</th>\n",
       "      <td>ASV_269846</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>101</td>\n",
       "      <td>79.21</td>\n",
       "      <td>NBINS004-14|COI-5P|United States|Animalia,Arth...</td>\n",
       "      <td>279</td>\n",
       "      <td>NBINS004-14</td>\n",
       "      <td>COI-5P</td>\n",
       "      <td>United States</td>\n",
       "      <td>Animalia,Arthropoda,Insecta,Coleoptera,Tenebri...</td>\n",
       "      <td>Animalia</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Insecta</td>\n",
       "      <td>Coleoptera</td>\n",
       "      <td>Tenebrionidae</td>\n",
       "      <td>Tenebrioninae</td>\n",
       "      <td>Cerenopus</td>\n",
       "      <td>Cerenopus concolor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>274072</th>\n",
       "      <td>ASV_270224</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>97</td>\n",
       "      <td>77.59</td>\n",
       "      <td>POMS7650-24|COI-5P|United Kingdom|Animalia,Art...</td>\n",
       "      <td>299</td>\n",
       "      <td>POMS7650-24</td>\n",
       "      <td>COI-5P</td>\n",
       "      <td>United Kingdom</td>\n",
       "      <td>Animalia,Arthropoda,Insecta,Diptera,Tachinidae...</td>\n",
       "      <td>Animalia</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Insecta</td>\n",
       "      <td>Diptera</td>\n",
       "      <td>Tachinidae</td>\n",
       "      <td>Dexiinae</td>\n",
       "      <td>Eriothrix</td>\n",
       "      <td>Eriothrix rufomaculata</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>277208</th>\n",
       "      <td>ASV_272913</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>102</td>\n",
       "      <td>78.35</td>\n",
       "      <td>CRALC29584-21|COI-5P|Costa Rica|Animalia,Arthr...</td>\n",
       "      <td>291</td>\n",
       "      <td>CRALC29584-21</td>\n",
       "      <td>COI-5P</td>\n",
       "      <td>Costa Rica</td>\n",
       "      <td>Animalia,Arthropoda,Insecta,Diptera,Ceratopogo...</td>\n",
       "      <td>Animalia</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Insecta</td>\n",
       "      <td>Diptera</td>\n",
       "      <td>Ceratopogonidae</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>277375</th>\n",
       "      <td>ASV_273151</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>98</td>\n",
       "      <td>78.16</td>\n",
       "      <td>GBAAY56519-24|COI-5P|Unrecoverable|Protista,Ha...</td>\n",
       "      <td>293</td>\n",
       "      <td>GBAAY56519-24</td>\n",
       "      <td>COI-5P</td>\n",
       "      <td>Unrecoverable</td>\n",
       "      <td>Protista,Haptophyta,Prymnesiophyceae,Isochrysi...</td>\n",
       "      <td>Protista</td>\n",
       "      <td>Haptophyta</td>\n",
       "      <td>Prymnesiophyceae</td>\n",
       "      <td>Isochrysidales</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>278590</th>\n",
       "      <td>ASV_274120</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>107</td>\n",
       "      <td>78.91</td>\n",
       "      <td>GMSVD8417-23|COI-5P|United States|Animalia,Art...</td>\n",
       "      <td>294</td>\n",
       "      <td>GMSVD8417-23</td>\n",
       "      <td>COI-5P</td>\n",
       "      <td>United States</td>\n",
       "      <td>Animalia,Arthropoda,Insecta,Diptera,Heleomyzid...</td>\n",
       "      <td>Animalia</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Insecta</td>\n",
       "      <td>Diptera</td>\n",
       "      <td>Heleomyzidae</td>\n",
       "      <td>Trixoscelidinae</td>\n",
       "      <td>Trixoscelis</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3624 rows Ã— 19 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               ASV  Hit_number eval bitscore    %ID  \\\n",
       "38           ASV_3           1  0.0      110  79.73   \n",
       "49           ASV_5           1  0.0      113  79.52   \n",
       "161         ASV_20           1  0.0      113  79.52   \n",
       "163         ASV_23           1  0.0      105  78.69   \n",
       "367         ASV_27           1  0.0      122  79.81   \n",
       "...            ...         ...  ...      ...    ...   \n",
       "272978  ASV_269846           1  0.0      101  79.21   \n",
       "274072  ASV_270224           1  0.0       97  77.59   \n",
       "277208  ASV_272913           1  0.0      102  78.35   \n",
       "277375  ASV_273151           1  0.0       98  78.16   \n",
       "278590  ASV_274120           1  0.0      107  78.91   \n",
       "\n",
       "                                                  hit_def align_len  \\\n",
       "38      BHNHM212-24|COI-5P|Afghanistan|Animalia,Arthro...       291   \n",
       "49      GMQQT008-18|COI-5P|Australia|Animalia,Arthropo...       293   \n",
       "161     GMQQT008-18|COI-5P|Australia|Animalia,Arthropo...       293   \n",
       "163     CRBOF22983-24|COI-5P|Costa Rica|Animalia,Arthr...       291   \n",
       "367     GBAAY62677-24|COI-5P|Unrecoverable|Mixture,Uns...       312   \n",
       "...                                                   ...       ...   \n",
       "272978  NBINS004-14|COI-5P|United States|Animalia,Arth...       279   \n",
       "274072  POMS7650-24|COI-5P|United Kingdom|Animalia,Art...       299   \n",
       "277208  CRALC29584-21|COI-5P|Costa Rica|Animalia,Arthr...       291   \n",
       "277375  GBAAY56519-24|COI-5P|Unrecoverable|Protista,Ha...       293   \n",
       "278590  GMSVD8417-23|COI-5P|United States|Animalia,Art...       294   \n",
       "\n",
       "              BOLD_ID BOLD_marker   BOLD_location  \\\n",
       "38        BHNHM212-24      COI-5P     Afghanistan   \n",
       "49        GMQQT008-18      COI-5P       Australia   \n",
       "161       GMQQT008-18      COI-5P       Australia   \n",
       "163     CRBOF22983-24      COI-5P      Costa Rica   \n",
       "367     GBAAY62677-24      COI-5P   Unrecoverable   \n",
       "...               ...         ...             ...   \n",
       "272978    NBINS004-14      COI-5P   United States   \n",
       "274072    POMS7650-24      COI-5P  United Kingdom   \n",
       "277208  CRALC29584-21      COI-5P      Costa Rica   \n",
       "277375  GBAAY56519-24      COI-5P   Unrecoverable   \n",
       "278590   GMSVD8417-23      COI-5P   United States   \n",
       "\n",
       "                                            BOLD_taxonomy   Kingdom  \\\n",
       "38      Animalia,Arthropoda,Insecta,Lepidoptera,Noctui...  Animalia   \n",
       "49      Animalia,Arthropoda,Insecta,Diptera,Rhagionida...  Animalia   \n",
       "161     Animalia,Arthropoda,Insecta,Diptera,Rhagionida...  Animalia   \n",
       "163     Animalia,Arthropoda,Insecta,Diptera,Ceratopogo...  Animalia   \n",
       "367     Mixture,Unspecified,None,None,None,None,None,N...   Mixture   \n",
       "...                                                   ...       ...   \n",
       "272978  Animalia,Arthropoda,Insecta,Coleoptera,Tenebri...  Animalia   \n",
       "274072  Animalia,Arthropoda,Insecta,Diptera,Tachinidae...  Animalia   \n",
       "277208  Animalia,Arthropoda,Insecta,Diptera,Ceratopogo...  Animalia   \n",
       "277375  Protista,Haptophyta,Prymnesiophyceae,Isochrysi...  Protista   \n",
       "278590  Animalia,Arthropoda,Insecta,Diptera,Heleomyzid...  Animalia   \n",
       "\n",
       "             Phylum             Class           Order           Family  \\\n",
       "38       Arthropoda           Insecta     Lepidoptera        Noctuidae   \n",
       "49       Arthropoda           Insecta         Diptera      Rhagionidae   \n",
       "161      Arthropoda           Insecta         Diptera      Rhagionidae   \n",
       "163      Arthropoda           Insecta         Diptera  Ceratopogonidae   \n",
       "367     Unspecified              None            None             None   \n",
       "...             ...               ...             ...              ...   \n",
       "272978   Arthropoda           Insecta      Coleoptera    Tenebrionidae   \n",
       "274072   Arthropoda           Insecta         Diptera       Tachinidae   \n",
       "277208   Arthropoda           Insecta         Diptera  Ceratopogonidae   \n",
       "277375   Haptophyta  Prymnesiophyceae  Isochrysidales             None   \n",
       "278590   Arthropoda           Insecta         Diptera     Heleomyzidae   \n",
       "\n",
       "              subfamily        Genus                 Species  \n",
       "38            Noctuinae    Caradrina       Caradrina fergana  \n",
       "49                 None         None                    None  \n",
       "161                None         None                    None  \n",
       "163                None         None                    None  \n",
       "367                None         None                    None  \n",
       "...                 ...          ...                     ...  \n",
       "272978    Tenebrioninae    Cerenopus      Cerenopus concolor  \n",
       "274072         Dexiinae    Eriothrix  Eriothrix rufomaculata  \n",
       "277208             None         None                    None  \n",
       "277375             None         None                    None  \n",
       "278590  Trixoscelidinae  Trixoscelis                    None  \n",
       "\n",
       "[3624 rows x 19 columns]"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = blast_results.copy()\n",
    "#df = df.loc[df['ASV']=='ASV_5']\n",
    "df = df.drop_duplicates('ASV')\n",
    "df = df.loc[df['%ID']<80]\n",
    "print(df['hit_def'].iloc[0])\n",
    "col1 = ['PR2_ID', 'PR2_marker', 'PR2_location', 'PR2_taxonomy']\n",
    "col2 = ['Kingdom', 'Phylum', 'Class', 'Order', 'Family','subfamily','Genus', 'Species']\n",
    "for i in range(len(col1)):\n",
    "    df[col1[i]] = df['hit_def'].str.split('|').str[i]\n",
    "\n",
    "for i in range(len(col2)):\n",
    "    df[col2[i]] = df['PR2_taxonomy'].str.split(',').str[i]\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "473bfdea-f182-424e-ae98-9550ec679745",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Save Blast Results to file\n",
    "df = pd.DataFrame(hit_dict)\n",
    "df=df.T\n",
    "df.columns = ['eval','bitscore', '%ID', 'hit_def', 'align_len']\n",
    "df.reset_index(inplace=True)\n",
    "df = df.rename(columns={'level_0': 'ASV', 'level_1': 'Hit_number'})\n",
    "\n",
    "\n",
    "#Keep the best hit of each category of hit\n",
    "df = df.drop_duplicates(subset=['genus', 'species', 'OTU'], keep='first')\n",
    "#Sort by OTU number\n",
    "df['OTU_Number']= df['OTU'].str.split('_').str[1]\n",
    "df['OTU_Number'] = df['OTU_Number'].astype(int)\n",
    "df.sort_values(['OTU_Number','Hit_number'], inplace=True)\n",
    "df.drop('OTU_Number', axis=1, inplace=True)\n",
    "df.set_index(['OTU', 'Hit_number'], inplace = True)\n",
    "outfile = filename.replace('.xml','_TopBLASThits.csv')\n",
    "print(outfile)\n",
    "df.to_csv(outfile)\n",
    "xml_dfs.append(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "173291f9-80e4-4980-acd6-562454304252",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "\n",
    "\n",
    "#Save Blast Results to file\n",
    "df = pd.DataFrame(hit_dict)\n",
    "df=df.T\n",
    "df.columns = ['eval','bitscore', '%ID', 'genus', 'species', 'hit_def', 'align_len']\n",
    "df.reset_index(inplace=True)\n",
    "df = df.rename(columns={'level_0': 'OTU', 'level_1': 'Hit_number'})\n",
    "#Keep the best hit of each category of hit\n",
    "df = df.drop_duplicates(subset=['genus', 'species', 'OTU'], keep='first')\n",
    "#Sort by OTU number\n",
    "df['OTU_Number']= df['OTU'].str.split('_').str[1]\n",
    "df['OTU_Number'] = df['OTU_Number'].astype(int)\n",
    "df.sort_values(['OTU_Number','Hit_number'], inplace=True)\n",
    "df.drop('OTU_Number', axis=1, inplace=True)\n",
    "df.set_index(['OTU', 'Hit_number'], inplace = True)\n",
    "outfile = filename.replace('.xml','_TopBLASThits.csv')\n",
    "print(outfile)\n",
    "df.to_csv(outfile)\n",
    "xml_dfs.append(df)\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "    #parse filename and directory location\n",
    "    print('Blast XML file: ', filename)\n",
    "    #directory = filename.split('/')[:-2]\n",
    "    #string_s= '/'\n",
    "    #directory=string_s.join(directory)\n",
    "    #directory = directory +'/'\n",
    "    #print('directory', directory)\n",
    "    genus=[]\n",
    "    species=[]\n",
    "    query=[]\n",
    "    hit_dict ={}  #Dictionary to store hits\n",
    "    result_handle =open(filename)\n",
    "    #Phix names to filter out - added for diversity in sequencing process\n",
    "    taxon = ['PhiX', 'phiX']\n",
    "    #parse blast records\n",
    "    blast_records= NCBIXML.parse(result_handle)\n",
    "    #Locations to store PhiX hits\n",
    "    taxon_A =[]\n",
    "    taxon_B =[]\n",
    "    query_A =[]\n",
    "    query_B =[]\n",
    "    for blast_record in blast_records:\n",
    "        query = blast_record.query\n",
    "        genus = str(taxa_dict[query][-2]) #look up genus assignment\n",
    "        species = str(taxa_dict[query][-1]) #look up species assignment\n",
    "        hit_ID=0 #hit counter\n",
    "        for alignment in blast_record.alignments:\n",
    "            hit_ID+=1\n",
    "            key = (query, hit_ID)\n",
    "            hsp = alignment.hsps[0] #only look at top hsp per alignment to genbank sequence\n",
    "            #set limits on evalue and bitscore and % Identity\n",
    "            per_iden = hsp.identities/float(hsp.align_length)\n",
    "            per_iden = (per_iden *100)\n",
    "            per_iden = round(per_iden, 2)\n",
    "            evalue = hsp.expect\n",
    "            bitscore = int(hsp.score)\n",
    "            #alignment length\n",
    "            align_len = hsp.align_length\n",
    "            #Check if MEGAN assigned ASV at species level;look for species in hit ID if yes\n",
    "            if species in ['s_', 'no_hit', 'unassigned']:\n",
    "                species_h = 'species_unassigned'\n",
    "            else:\n",
    "                if species in alignment.hit_def:\n",
    "                    species_h = species\n",
    "                else:\n",
    "                    species_h ='Not_Found'\n",
    "            #Check if MEGAN assigned ASV at genus level;look for species in hit ID if yes\n",
    "            if genus in ['g_', 'no_hit', 'unassigned']:\n",
    "                genus_h = 'genus_unassigned'\n",
    "            else:\n",
    "                if genus in alignment.hit_def:\n",
    "                    genus_h = genus\n",
    "                else:\n",
    "                    genus_h ='Not_Found'\n",
    "            value = (evalue, bitscore, per_iden, genus_h, species_h, alignment.hit_def, align_len)\n",
    "            hit_dict[key]=value\n",
    "            genus_h='' #reset variables\n",
    "            species_h=''#reset variables\n",
    "            #Look for PhiX hits\n",
    "            if taxon[0] in alignment.title:\n",
    "                #print (alignment.title)\n",
    "                taxon_A.append(blast_record.query)\n",
    "                query_A.append(alignment.title)\n",
    "            if taxon[1] in alignment.title:\n",
    "                #print (alignment.title)\n",
    "                taxon_B.append(blast_record.query)\n",
    "                query_B.append(alignment.title)\n",
    "        #break\n",
    "    print ('Done parsing ', filename)\n",
    "\n",
    "    #Save Blast Results to file\n",
    "    df = pd.DataFrame(hit_dict)\n",
    "    df=df.T\n",
    "    df.columns = ['eval','bitscore', '%ID', 'genus', 'species', 'hit_def', 'align_len']\n",
    "    df.reset_index(inplace=True)\n",
    "    df = df.rename(columns={'level_0': 'OTU', 'level_1': 'Hit_number'})\n",
    "    #Keep the best hit of each category of hit\n",
    "    df = df.drop_duplicates(subset=['genus', 'species', 'OTU'], keep='first')\n",
    "    #Sort by OTU number\n",
    "    df['OTU_Number']= df['OTU'].str.split('_').str[1]\n",
    "    df['OTU_Number'] = df['OTU_Number'].astype(int)\n",
    "    df.sort_values(['OTU_Number','Hit_number'], inplace=True)\n",
    "    df.drop('OTU_Number', axis=1, inplace=True)\n",
    "    df.set_index(['OTU', 'Hit_number'], inplace = True)\n",
    "    outfile = filename.replace('.xml','_TopBLASThits.csv')\n",
    "    print(outfile)\n",
    "    df.to_csv(outfile)\n",
    "    xml_dfs.append(df)\n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8a96ceb6-f4c6-4dc1-a334-159da0e35abc",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "4c5ed0f3-f3bc-41eb-82ba-c61c5c71ea74",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Species bitscore limit:  400 , Percent Identity Limit:  97\n",
      "Genus bitscore limit:  350 , Percent Identity Limit:  95\n",
      "Blast XML file:  /Users/kpitz/Projects/custom_blast/Blast_VM_output/mbon-worker3__.xml\n",
      "Done parsing  /Users/kpitz/Projects/custom_blast/Blast_VM_output/mbon-worker3__.xml\n",
      "/Users/kpitz/Projects/custom_blast/Blast_VM_output/mbon-worker3___TopBLASThits.csv\n",
      "Blast XML file:  /Users/kpitz/Projects/custom_blast/Blast_VM_output/mbon-worker1__.xml\n",
      "Done parsing  /Users/kpitz/Projects/custom_blast/Blast_VM_output/mbon-worker1__.xml\n",
      "/Users/kpitz/Projects/custom_blast/Blast_VM_output/mbon-worker1___TopBLASThits.csv\n",
      "Blast XML file:  /Users/kpitz/Projects/custom_blast/Blast_VM_output/mbon-worker5__.xml\n",
      "Done parsing  /Users/kpitz/Projects/custom_blast/Blast_VM_output/mbon-worker5__.xml\n",
      "/Users/kpitz/Projects/custom_blast/Blast_VM_output/mbon-worker5___TopBLASThits.csv\n",
      "Blast XML file:  /Users/kpitz/Projects/custom_blast/Blast_VM_output/mbon-worker2__.xml\n",
      "Done parsing  /Users/kpitz/Projects/custom_blast/Blast_VM_output/mbon-worker2__.xml\n",
      "/Users/kpitz/Projects/custom_blast/Blast_VM_output/mbon-worker2___TopBLASThits.csv\n",
      "Blast XML file:  /Users/kpitz/Projects/custom_blast/Blast_VM_output/mbon-worker6__.xml\n",
      "Done parsing  /Users/kpitz/Projects/custom_blast/Blast_VM_output/mbon-worker6__.xml\n",
      "/Users/kpitz/Projects/custom_blast/Blast_VM_output/mbon-worker6___TopBLASThits.csv\n",
      "Blast XML file:  /Users/kpitz/Projects/custom_blast/Blast_VM_output/mbon-worker4__.xml\n",
      "Done parsing  /Users/kpitz/Projects/custom_blast/Blast_VM_output/mbon-worker4__.xml\n",
      "/Users/kpitz/Projects/custom_blast/Blast_VM_output/mbon-worker4___TopBLASThits.csv\n"
     ]
    }
   ],
   "source": [
    "# Parse genbank xml files (from mbon-master workers, 6 total)\n",
    "\n",
    "'''# COI limits:\n",
    "#Species limits\n",
    "#bitscore_sp = int(sys.argv[1])\n",
    "#per_ID_sp = int(sys.argv[2])\n",
    "bitscore_sp=400\n",
    "per_ID_sp=97\n",
    "print('Species bitscore limit: ', bitscore_sp, ', Percent Identity Limit: ', per_ID_sp)\n",
    "#Genus limits\n",
    "#bitscore_gn = int(sys.argv[3])\n",
    "#per_ID_gn = int(sys.argv[4])\n",
    "bitscore_gn=350\n",
    "per_ID_gn=95\n",
    "print('Genus bitscore limit: ', bitscore_gn, ', Percent Identity Limit: ', per_ID_gn)'''\n",
    "\n",
    "#Taxa Table\n",
    "taxa_tab = pd.concat([megan, seq], axis=1)\n",
    "taxa_tab = taxa_tab[levels]\n",
    "\n",
    "#Create dictionary of ASV ID to taxonomy\n",
    "taxa_dict = taxa_tab.T.to_dict('list')\n",
    "\n",
    "\n",
    "#print(directory)\n",
    "file_loc = '/Users/kpitz/Projects/custom_blast/Blast_VM_output/*xml'\n",
    "\n",
    "files = glob.glob(file_loc)\n",
    "# Parse through XML files\n",
    "xml_dfs = []  #will store results from each xml file\n",
    "for filename in files:\n",
    "    #parse filename and directory location\n",
    "    print('Blast XML file: ', filename)\n",
    "    #directory = filename.split('/')[:-2]\n",
    "    #string_s= '/'\n",
    "    #directory=string_s.join(directory)\n",
    "    #directory = directory +'/'\n",
    "    #print('directory', directory)\n",
    "    genus=[]\n",
    "    species=[]\n",
    "    query=[]\n",
    "    hit_dict ={}  #Dictionary to store hits\n",
    "    result_handle =open(filename)\n",
    "    #Phix names to filter out - added for diversity in sequencing process\n",
    "    taxon = ['PhiX', 'phiX']\n",
    "    #parse blast records\n",
    "    blast_records= NCBIXML.parse(result_handle)\n",
    "    #Locations to store PhiX hits\n",
    "    taxon_A =[]\n",
    "    taxon_B =[]\n",
    "    query_A =[]\n",
    "    query_B =[]\n",
    "    for blast_record in blast_records:\n",
    "        query = blast_record.query\n",
    "        genus = str(taxa_dict[query][-2]) #look up genus assignment\n",
    "        species = str(taxa_dict[query][-1]) #look up species assignment\n",
    "        hit_ID=0 #hit counter\n",
    "        for alignment in blast_record.alignments:\n",
    "            hit_ID+=1\n",
    "            key = (query, hit_ID)\n",
    "            hsp = alignment.hsps[0] #only look at top hsp per alignment to genbank sequence\n",
    "            #set limits on evalue and bitscore and % Identity\n",
    "            per_iden = hsp.identities/float(hsp.align_length)\n",
    "            per_iden = (per_iden *100)\n",
    "            per_iden = round(per_iden, 2)\n",
    "            evalue = hsp.expect\n",
    "            bitscore = int(hsp.score)\n",
    "            #alignment length\n",
    "            align_len = hsp.align_length\n",
    "            #Check if MEGAN assigned ASV at species level;look for species in hit ID if yes\n",
    "            if species in ['s_', 'no_hit', 'unassigned']:\n",
    "                species_h = 'species_unassigned'\n",
    "            else:\n",
    "                if species in alignment.hit_def:\n",
    "                    species_h = species\n",
    "                else:\n",
    "                    species_h ='Not_Found'\n",
    "            #Check if MEGAN assigned ASV at genus level;look for species in hit ID if yes\n",
    "            if genus in ['g_', 'no_hit', 'unassigned']:\n",
    "                genus_h = 'genus_unassigned'\n",
    "            else:\n",
    "                if genus in alignment.hit_def:\n",
    "                    genus_h = genus\n",
    "                else:\n",
    "                    genus_h ='Not_Found'\n",
    "            value = (evalue, bitscore, per_iden, genus_h, species_h, alignment.hit_def, align_len)\n",
    "            hit_dict[key]=value\n",
    "            genus_h='' #reset variables\n",
    "            species_h=''#reset variables\n",
    "            #Look for PhiX hits\n",
    "            if taxon[0] in alignment.title:\n",
    "                #print (alignment.title)\n",
    "                taxon_A.append(blast_record.query)\n",
    "                query_A.append(alignment.title)\n",
    "            if taxon[1] in alignment.title:\n",
    "                #print (alignment.title)\n",
    "                taxon_B.append(blast_record.query)\n",
    "                query_B.append(alignment.title)\n",
    "        #break\n",
    "    print ('Done parsing ', filename)\n",
    "\n",
    "    #Save Blast Results to file\n",
    "    df = pd.DataFrame(hit_dict)\n",
    "    df=df.T\n",
    "    df.columns = ['eval','bitscore', '%ID', 'genus', 'species', 'hit_def', 'align_len']\n",
    "    df.reset_index(inplace=True)\n",
    "    df = df.rename(columns={'level_0': 'OTU', 'level_1': 'Hit_number'})\n",
    "    #Keep the best hit of each category of hit\n",
    "    df = df.drop_duplicates(subset=['genus', 'species', 'OTU'], keep='first')\n",
    "    #Sort by OTU number\n",
    "    df['OTU_Number']= df['OTU'].str.split('_').str[1]\n",
    "    df['OTU_Number'] = df['OTU_Number'].astype(int)\n",
    "    df.sort_values(['OTU_Number','Hit_number'], inplace=True)\n",
    "    df.drop('OTU_Number', axis=1, inplace=True)\n",
    "    df.set_index(['OTU', 'Hit_number'], inplace = True)\n",
    "    outfile = filename.replace('.xml','_TopBLASThits.csv')\n",
    "    print(outfile)\n",
    "    df.to_csv(outfile)\n",
    "    xml_dfs.append(df)\n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bbb2229d-2ee0-4d4d-a723-0c85c5a77ceb",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d2d098f8-4043-43f5-ba17-ff98a75a443b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c4477fea-c7e5-48d9-adcc-b0fd838a1346",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "56cf8c30-f841-4a7e-8c55-cec355cbf565",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "78655ead-ad58-4468-8157-80c12c0af7cd",
   "metadata": {},
   "source": [
    "## Get df of sequence data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "ccdb2edc-bdfc-4252-b006-dd156f4df856",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of Ids: 15713\n",
      "Number of Seqs: 15713\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ASV_1</th>\n",
       "      <td>TCTAGCAGGGATTCAAGCTCATTCAGGAGGTTCTGTTGATTTAGCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_2</th>\n",
       "      <td>TCTAGCAGGGATTCAAGCTCATTCAGGAGGTTCTGTTGATTTAGCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_3</th>\n",
       "      <td>TTAAGAATAAATATCGCCCATTCAGGCCCATCTGTCGATTTTGCTA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_4</th>\n",
       "      <td>TCTAGCAGGGATTCAAGCTCATTCAGGAGGTTCTGTTGATTTAGCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_5</th>\n",
       "      <td>TCTAGCAGGGATTCAAGCTCATTCAGGAGGTTCTGTTGATTTAGCA...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                sequence\n",
       "ASV_1  TCTAGCAGGGATTCAAGCTCATTCAGGAGGTTCTGTTGATTTAGCA...\n",
       "ASV_2  TCTAGCAGGGATTCAAGCTCATTCAGGAGGTTCTGTTGATTTAGCA...\n",
       "ASV_3  TTAAGAATAAATATCGCCCATTCAGGCCCATCTGTCGATTTTGCTA...\n",
       "ASV_4  TCTAGCAGGGATTCAAGCTCATTCAGGAGGTTCTGTTGATTTAGCA...\n",
       "ASV_5  TCTAGCAGGGATTCAAGCTCATTCAGGAGGTTCTGTTGATTTAGCA..."
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "input_file = directory + 'Flyer2018_COI_seq_merged.fasta'\n",
    "# fasta file to dataframe:\n",
    "\n",
    "#From fasta file create pandas df of ASV and sequence\n",
    "def from_fasta_to_df(file):\n",
    "    #print(file)\n",
    "    with open(file) as f:\n",
    "        Ids=[]\n",
    "        seqs =[]\n",
    "        for strline in f:\n",
    "            if strline[0]=='>':\n",
    "                Ids.append(strline[1:].strip())\n",
    "            else:\n",
    "                seqs.append(strline.strip())\n",
    "    print('Number of Ids:',len(Ids))\n",
    "    print('Number of Seqs:',len(seqs))\n",
    "    seq_dict = dict(zip(Ids, seqs))\n",
    "    #make pandas df\n",
    "    df= pd.DataFrame.from_dict(seq_dict,orient='index', columns=['sequence'])\n",
    "    return df\n",
    "\n",
    "df = from_fasta_to_df(input_file)\n",
    "seq = df.copy()\n",
    "seq.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2f3fd9a2-73ae-4464-aa2e-d6521385cb71",
   "metadata": {},
   "source": [
    "## Parse MEGAN results from genbank blastn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "49cf358e-33ca-46cb-a1b1-6017f2eec1ec",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[D] Eukaryota; [K] Metazoa; [P] Arthropoda; [C] Insecta; [O] Diptera;\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>taxonomy</th>\n",
       "      <th>Domain</th>\n",
       "      <th>Kingdom</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ASV_20618</th>\n",
       "      <td>[D] Archaea; [P] Thaumarchaeota;</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Thaumarchaeota</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_73595</th>\n",
       "      <td>[D] Bacteria; [P] Bacteroidetes; [C] Flavobact...</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>Flavobacteriia</td>\n",
       "      <td>Flavobacteriales</td>\n",
       "      <td>Flavobacteriaceae</td>\n",
       "      <td>Cellulophaga</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_2609</th>\n",
       "      <td>[D] Bacteria; [P] Bacteroidetes; [C] Flavobact...</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>Flavobacteriia</td>\n",
       "      <td>Flavobacteriales</td>\n",
       "      <td>Flavobacteriaceae</td>\n",
       "      <td>Cellulophaga</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_10045</th>\n",
       "      <td>[D] Bacteria; [P] Bacteroidetes; [C] Flavobact...</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>Flavobacteriia</td>\n",
       "      <td>Flavobacteriales</td>\n",
       "      <td>Flavobacteriaceae</td>\n",
       "      <td>Cellulophaga</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_21215</th>\n",
       "      <td>[D] Bacteria; [P] Bacteroidetes; [C] Flavobact...</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>Flavobacteriia</td>\n",
       "      <td>Flavobacteriales</td>\n",
       "      <td>Flavobacteriaceae</td>\n",
       "      <td>Flavobacterium</td>\n",
       "      <td>Flavobacterium sp. F-29</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    taxonomy    Domain  \\\n",
       "ASV                                                                      \n",
       "ASV_20618                   [D] Archaea; [P] Thaumarchaeota;   Archaea   \n",
       "ASV_73595  [D] Bacteria; [P] Bacteroidetes; [C] Flavobact...  Bacteria   \n",
       "ASV_2609   [D] Bacteria; [P] Bacteroidetes; [C] Flavobact...  Bacteria   \n",
       "ASV_10045  [D] Bacteria; [P] Bacteroidetes; [C] Flavobact...  Bacteria   \n",
       "ASV_21215  [D] Bacteria; [P] Bacteroidetes; [C] Flavobact...  Bacteria   \n",
       "\n",
       "          Kingdom          Phylum           Class             Order  \\\n",
       "ASV                                                                   \n",
       "ASV_20618     NaN  Thaumarchaeota             NaN               NaN   \n",
       "ASV_73595     NaN   Bacteroidetes  Flavobacteriia  Flavobacteriales   \n",
       "ASV_2609      NaN   Bacteroidetes  Flavobacteriia  Flavobacteriales   \n",
       "ASV_10045     NaN   Bacteroidetes  Flavobacteriia  Flavobacteriales   \n",
       "ASV_21215     NaN   Bacteroidetes  Flavobacteriia  Flavobacteriales   \n",
       "\n",
       "                      Family           Genus                  Species  \n",
       "ASV                                                                    \n",
       "ASV_20618                NaN             NaN                      NaN  \n",
       "ASV_73595  Flavobacteriaceae    Cellulophaga                      NaN  \n",
       "ASV_2609   Flavobacteriaceae    Cellulophaga                      NaN  \n",
       "ASV_10045  Flavobacteriaceae    Cellulophaga                      NaN  \n",
       "ASV_21215  Flavobacteriaceae  Flavobacterium  Flavobacterium sp. F-29  "
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#K metazoa Kingdom\n",
    "levels = ['Domain', 'Kingdom', 'Phylum', 'Class', 'Order', 'Family', 'Genus', 'Species']\n",
    "df = pd.read_csv(taxa_table_file, sep=\"\\t\", header=None, names =['ASV', 'taxonomy'])\n",
    "\n",
    "#print(df['taxonomy'][405])\n",
    "# extract all levels present in the taxonomy:\n",
    "df.set_index('ASV', inplace=True)\n",
    "#df['test'] = df['taxonomy'].str.split(';')\n",
    "#df['len'] = df['test'].str.len()\n",
    "df['Domain'] = df['taxonomy'].str.extract(r'\\[D\\] ([^;]*);.*')\n",
    "df['Kingdom'] = df['taxonomy'].str.extract(r'\\[K\\] ([^;]*);.*')\n",
    "df['Phylum'] = df['taxonomy'].str.extract(r'\\[P\\] ([^;]*);.*')\n",
    "df['Class'] = df['taxonomy'].str.extract(r'\\[C\\] ([^;]*);.*')\n",
    "df['Order'] = df['taxonomy'].str.extract(r'\\[O\\] ([^;]*);.*')\n",
    "df['Family'] = df['taxonomy'].str.extract(r'\\[F\\] ([^;]*);.*')\n",
    "df['Genus'] = df['taxonomy'].str.extract(r'\\[G\\] ([^;]*);.*')\n",
    "df['Species'] = df['taxonomy'].str.extract(r'\\[S\\] ([^;]*);.*')\n",
    "\n",
    "df = df.sort_values(levels)\n",
    "\n",
    "# MEGAN already exports 'unknown' term but is inconsistent for some terms, like 'Kingdom'\n",
    "#df = df.fillna('not assigned')\n",
    "# if no taxonomy:\n",
    "for level in levels:\n",
    "    df.loc[df['taxonomy'].isna(), level] = 'not assigned'\n",
    "\n",
    "#df = df.dropna(how='all')\n",
    "#df.to_csv('/Users/kpitz/Documents/test.csv')\n",
    "megan = df.copy()\n",
    "megan.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e634bf53-7f2f-42e0-aebb-d54c7f9edbef",
   "metadata": {},
   "source": [
    "## Parse genbank XML files to get percent ID and bitscore values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "2df6f37b-7fe1-4d8a-8645-fc026c7be60f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Species bitscore limit:  400 , Percent Identity Limit:  97\n",
      "Genus bitscore limit:  350 , Percent Identity Limit:  95\n",
      "Blast XML file:  /Users/kpitz/Projects/custom_blast/Blast_VM_output/mbon-worker3__.xml\n",
      "Done parsing  /Users/kpitz/Projects/custom_blast/Blast_VM_output/mbon-worker3__.xml\n",
      "/Users/kpitz/Projects/custom_blast/Blast_VM_output/mbon-worker3___TopBLASThits.csv\n",
      "Blast XML file:  /Users/kpitz/Projects/custom_blast/Blast_VM_output/mbon-worker1__.xml\n",
      "Done parsing  /Users/kpitz/Projects/custom_blast/Blast_VM_output/mbon-worker1__.xml\n",
      "/Users/kpitz/Projects/custom_blast/Blast_VM_output/mbon-worker1___TopBLASThits.csv\n",
      "Blast XML file:  /Users/kpitz/Projects/custom_blast/Blast_VM_output/mbon-worker5__.xml\n",
      "Done parsing  /Users/kpitz/Projects/custom_blast/Blast_VM_output/mbon-worker5__.xml\n",
      "/Users/kpitz/Projects/custom_blast/Blast_VM_output/mbon-worker5___TopBLASThits.csv\n",
      "Blast XML file:  /Users/kpitz/Projects/custom_blast/Blast_VM_output/mbon-worker2__.xml\n",
      "Done parsing  /Users/kpitz/Projects/custom_blast/Blast_VM_output/mbon-worker2__.xml\n",
      "/Users/kpitz/Projects/custom_blast/Blast_VM_output/mbon-worker2___TopBLASThits.csv\n",
      "Blast XML file:  /Users/kpitz/Projects/custom_blast/Blast_VM_output/mbon-worker6__.xml\n",
      "Done parsing  /Users/kpitz/Projects/custom_blast/Blast_VM_output/mbon-worker6__.xml\n",
      "/Users/kpitz/Projects/custom_blast/Blast_VM_output/mbon-worker6___TopBLASThits.csv\n",
      "Blast XML file:  /Users/kpitz/Projects/custom_blast/Blast_VM_output/mbon-worker4__.xml\n",
      "Done parsing  /Users/kpitz/Projects/custom_blast/Blast_VM_output/mbon-worker4__.xml\n",
      "/Users/kpitz/Projects/custom_blast/Blast_VM_output/mbon-worker4___TopBLASThits.csv\n"
     ]
    }
   ],
   "source": [
    "# Parse genbank xml files (from mbon-master workers, 6 total)\n",
    "\n",
    "# COI limits:\n",
    "#Species limits\n",
    "#bitscore_sp = int(sys.argv[1])\n",
    "#per_ID_sp = int(sys.argv[2])\n",
    "bitscore_sp=400\n",
    "per_ID_sp=97\n",
    "print('Species bitscore limit: ', bitscore_sp, ', Percent Identity Limit: ', per_ID_sp)\n",
    "#Genus limits\n",
    "#bitscore_gn = int(sys.argv[3])\n",
    "#per_ID_gn = int(sys.argv[4])\n",
    "bitscore_gn=350\n",
    "per_ID_gn=95\n",
    "print('Genus bitscore limit: ', bitscore_gn, ', Percent Identity Limit: ', per_ID_gn)\n",
    "\n",
    "#Taxa Table\n",
    "taxa_tab = pd.concat([megan, seq], axis=1)\n",
    "taxa_tab = taxa_tab[levels]\n",
    "\n",
    "#Create dictionary of ASV ID to taxonomy\n",
    "taxa_dict = taxa_tab.T.to_dict('list')\n",
    "\n",
    "\n",
    "#print(directory)\n",
    "file_loc = '/Users/kpitz/Projects/custom_blast/Blast_VM_output/*xml'\n",
    "\n",
    "files = glob.glob(file_loc)\n",
    "# Parse through XML files\n",
    "xml_dfs = []  #will store results from each xml file\n",
    "for filename in files:\n",
    "    #parse filename and directory location\n",
    "    print('Blast XML file: ', filename)\n",
    "    #directory = filename.split('/')[:-2]\n",
    "    #string_s= '/'\n",
    "    #directory=string_s.join(directory)\n",
    "    #directory = directory +'/'\n",
    "    #print('directory', directory)\n",
    "    genus=[]\n",
    "    species=[]\n",
    "    query=[]\n",
    "    hit_dict ={}  #Dictionary to store hits\n",
    "    result_handle =open(filename)\n",
    "    #Phix names to filter out - added for diversity in sequencing process\n",
    "    taxon = ['PhiX', 'phiX']\n",
    "    #parse blast records\n",
    "    blast_records= NCBIXML.parse(result_handle)\n",
    "    #Locations to store PhiX hits\n",
    "    taxon_A =[]\n",
    "    taxon_B =[]\n",
    "    query_A =[]\n",
    "    query_B =[]\n",
    "    for blast_record in blast_records:\n",
    "        query = blast_record.query\n",
    "        genus = str(taxa_dict[query][-2]) #look up genus assignment\n",
    "        species = str(taxa_dict[query][-1]) #look up species assignment\n",
    "        hit_ID=0 #hit counter\n",
    "        for alignment in blast_record.alignments:\n",
    "            hit_ID+=1\n",
    "            key = (query, hit_ID)\n",
    "            hsp = alignment.hsps[0] #only look at top hsp per alignment to genbank sequence\n",
    "            #set limits on evalue and bitscore and % Identity\n",
    "            per_iden = hsp.identities/float(hsp.align_length)\n",
    "            per_iden = (per_iden *100)\n",
    "            per_iden = round(per_iden, 2)\n",
    "            evalue = hsp.expect\n",
    "            bitscore = int(hsp.score)\n",
    "            #alignment length\n",
    "            align_len = hsp.align_length\n",
    "            #Check if MEGAN assigned ASV at species level;look for species in hit ID if yes\n",
    "            if species in ['s_', 'no_hit', 'unassigned']:\n",
    "                species_h = 'species_unassigned'\n",
    "            else:\n",
    "                if species in alignment.hit_def:\n",
    "                    species_h = species\n",
    "                else:\n",
    "                    species_h ='Not_Found'\n",
    "            #Check if MEGAN assigned ASV at genus level;look for species in hit ID if yes\n",
    "            if genus in ['g_', 'no_hit', 'unassigned']:\n",
    "                genus_h = 'genus_unassigned'\n",
    "            else:\n",
    "                if genus in alignment.hit_def:\n",
    "                    genus_h = genus\n",
    "                else:\n",
    "                    genus_h ='Not_Found'\n",
    "            value = (evalue, bitscore, per_iden, genus_h, species_h, alignment.hit_def, align_len)\n",
    "            hit_dict[key]=value\n",
    "            genus_h='' #reset variables\n",
    "            species_h=''#reset variables\n",
    "            #Look for PhiX hits\n",
    "            if taxon[0] in alignment.title:\n",
    "                #print (alignment.title)\n",
    "                taxon_A.append(blast_record.query)\n",
    "                query_A.append(alignment.title)\n",
    "            if taxon[1] in alignment.title:\n",
    "                #print (alignment.title)\n",
    "                taxon_B.append(blast_record.query)\n",
    "                query_B.append(alignment.title)\n",
    "        #break\n",
    "    print ('Done parsing ', filename)\n",
    "\n",
    "    #Save Blast Results to file\n",
    "    df = pd.DataFrame(hit_dict)\n",
    "    df=df.T\n",
    "    df.columns = ['eval','bitscore', '%ID', 'genus', 'species', 'hit_def', 'align_len']\n",
    "    df.reset_index(inplace=True)\n",
    "    df = df.rename(columns={'level_0': 'OTU', 'level_1': 'Hit_number'})\n",
    "    #Keep the best hit of each category of hit\n",
    "    df = df.drop_duplicates(subset=['genus', 'species', 'OTU'], keep='first')\n",
    "    #Sort by OTU number\n",
    "    df['OTU_Number']= df['OTU'].str.split('_').str[1]\n",
    "    df['OTU_Number'] = df['OTU_Number'].astype(int)\n",
    "    df.sort_values(['OTU_Number','Hit_number'], inplace=True)\n",
    "    df.drop('OTU_Number', axis=1, inplace=True)\n",
    "    df.set_index(['OTU', 'Hit_number'], inplace = True)\n",
    "    outfile = filename.replace('.xml','_TopBLASThits.csv')\n",
    "    print(outfile)\n",
    "    df.to_csv(outfile)\n",
    "    xml_dfs.append(df)\n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "id": "45a48525-f18f-4039-9da6-0f42cc84d608",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Domain</th>\n",
       "      <th>Kingdom</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ASV_1</th>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Haptophyta</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Isochrysidales</td>\n",
       "      <td>Noelaerhabdaceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_2</th>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Haptophyta</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Isochrysidales</td>\n",
       "      <td>Noelaerhabdaceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_3</th>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Metazoa</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g_</td>\n",
       "      <td>s_</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_4</th>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Haptophyta</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Isochrysidales</td>\n",
       "      <td>Noelaerhabdaceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_5</th>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Haptophyta</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Isochrysidales</td>\n",
       "      <td>Noelaerhabdaceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_138295</th>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Phaeophyceae</td>\n",
       "      <td>Ectocarpales</td>\n",
       "      <td>Chordariaceae</td>\n",
       "      <td>g_</td>\n",
       "      <td>s_</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_138351</th>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Oomycota</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Saprolegniales</td>\n",
       "      <td>Saprolegniaceae</td>\n",
       "      <td>g_</td>\n",
       "      <td>s_</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_138768</th>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g_</td>\n",
       "      <td>s_</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_138814</th>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g_</td>\n",
       "      <td>s_</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_138825</th>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Haptophyta</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Phaeocystales</td>\n",
       "      <td>Phaeocystaceae</td>\n",
       "      <td>Phaeocystis</td>\n",
       "      <td>s_</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9828 rows Ã— 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               Domain  Kingdom      Phylum         Class           Order  \\\n",
       "ASV                                                                        \n",
       "ASV_1       Eukaryota      NaN  Haptophyta       unknown  Isochrysidales   \n",
       "ASV_2       Eukaryota      NaN  Haptophyta       unknown  Isochrysidales   \n",
       "ASV_3       Eukaryota  Metazoa  Arthropoda           NaN             NaN   \n",
       "ASV_4       Eukaryota      NaN  Haptophyta       unknown  Isochrysidales   \n",
       "ASV_5       Eukaryota      NaN  Haptophyta       unknown  Isochrysidales   \n",
       "...               ...      ...         ...           ...             ...   \n",
       "ASV_138295  Eukaryota      NaN     unknown  Phaeophyceae    Ectocarpales   \n",
       "ASV_138351  Eukaryota      NaN    Oomycota       unknown  Saprolegniales   \n",
       "ASV_138768  Eukaryota      NaN         NaN           NaN             NaN   \n",
       "ASV_138814  Eukaryota      NaN         NaN           NaN             NaN   \n",
       "ASV_138825  Eukaryota      NaN  Haptophyta       unknown   Phaeocystales   \n",
       "\n",
       "                      Family        Genus Species  \n",
       "ASV                                                \n",
       "ASV_1       Noelaerhabdaceae          NaN     NaN  \n",
       "ASV_2       Noelaerhabdaceae          NaN     NaN  \n",
       "ASV_3                    NaN           g_      s_  \n",
       "ASV_4       Noelaerhabdaceae          NaN     NaN  \n",
       "ASV_5       Noelaerhabdaceae          NaN     NaN  \n",
       "...                      ...          ...     ...  \n",
       "ASV_138295     Chordariaceae           g_      s_  \n",
       "ASV_138351   Saprolegniaceae           g_      s_  \n",
       "ASV_138768               NaN           g_      s_  \n",
       "ASV_138814               NaN           g_      s_  \n",
       "ASV_138825    Phaeocystaceae  Phaeocystis      s_  \n",
       "\n",
       "[9828 rows x 8 columns]"
      ]
     },
     "execution_count": 98,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Look at all XML Files together\n",
    "\n",
    "df= pd.concat(xml_dfs, axis=0)\n",
    "\n",
    "# Begin filtering Taxonomic Annotations by Species and Genus limits\n",
    "df[['eval', 'bitscore', '%ID', 'align_len']] = df[['eval', 'bitscore', '%ID', 'align_len']].apply(pd.to_numeric)\n",
    "#Get top hits for taxa annotation assigned by MEGAN\n",
    "df = df.reset_index().set_index('OTU')\n",
    "df = df.join(taxa_tab, how='left')   #Join blast results with taxa table; anything without a blast hit won't join\n",
    "df.reset_index(inplace=True)\n",
    "df=df.sort_values(['index', 'Hit_number'])\n",
    "df.set_index(['index', 'Hit_number'], inplace=True)\n",
    "\n",
    "#Need to limit by top hit that includes genus and species, or just genus if that's all that's available\n",
    "#Or if neither of them are available, set g_ and s_\n",
    "#need to sort df by these levels; higher number, less specific assignment\n",
    "df['G_lev']= 1 # 1 = assigned at that level; start out with everything =1\n",
    "df['S_lev']= 1 # 1 = assigned at that level; start out with everything =1\n",
    "# Now find unassigned ASVs:\n",
    "df.loc[df['genus'] == 'genus_unassigned' , 'G_lev'] = 3\n",
    "df.loc[df['genus'] == 'Not_Found' , 'G_lev'] = 2\n",
    "df.loc[df['species'] == 'species_unassigned' , 'S_lev'] = 3\n",
    "df.loc[df['species'] == 'Not_Found' , 'S_lev'] = 2\n",
    "\n",
    "df = df.sort_values(['S_lev', 'G_lev'])\n",
    "#Now just want to take the top hit for each OTU.\n",
    "df=df.reset_index()\n",
    "df = df.drop_duplicates('index', keep='first')\n",
    "df.drop(['G_lev', 'S_lev'], axis=1, inplace=True)\n",
    "\n",
    "#check no duplicates, one entry per OTU\n",
    "dups = df.reset_index().duplicated(subset=['index'])\n",
    "if dups.unique() != [False]:\n",
    "    print('Error!! Duplicate OTUs in Table. Something went wrong.')\n",
    "\n",
    "\n",
    "#Limit by percent ID\n",
    "df.loc[df['%ID'] < per_ID_gn , 'Genus'] = \"g_\"\n",
    "df.loc[df['%ID'] < per_ID_sp , 'Species'] = \"s_\"\n",
    "#Limit by bitscore\n",
    "df.loc[df['bitscore'] < bitscore_gn , 'Genus'] = \"g_\"\n",
    "df.loc[df['bitscore'] < bitscore_sp , 'Species'] = \"s_\"\n",
    "\n",
    "#sort by ASV number and hit number\n",
    "df.reset_index(inplace=True)\n",
    "df['OTU_Number']= df['index'].str.split('_').str[1]\n",
    "df['OTU_Number'] = df['OTU_Number'].astype(int)\n",
    "df.sort_values(['OTU_Number','Hit_number'], inplace=True)\n",
    "df.set_index(['index','Hit_number'], inplace=True)\n",
    "\n",
    "#Create New Taxa Table from selection\n",
    "taxa_lim = df[levels]  #limit columns to taxonomy\n",
    "taxa_lim = taxa_lim.reset_index().drop('Hit_number', axis=1)\n",
    "taxa_lim = taxa_lim.rename(columns = {'index':'ASV'})\n",
    "taxa_lim.set_index('ASV', inplace=True)\n",
    "taxa_lim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "id": "53a59a63-2125-4250-9513-c60f282fd43b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/kpitz/Projects/custom_blast/Filtered_ASV_taxa_table.csv\n",
      "XXXX  Check Stats of hits that have been filtered: XXXX\n",
      "Mean %ID of ASVs with g_: 84.05301163050773\n",
      "Mean bitscore of ASVs with g_: 350.2860657571013\n",
      "Mean %ID of ASVs with s_: 96.00779761904762\n",
      "Mean bitscore of ASVs with s_: 538.202380952381\n",
      "/Users/kpitz/Projects/custom_blast/ASV_tophit_Ftaxonomy.csv\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>eval</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>%ID</th>\n",
       "      <th>genus</th>\n",
       "      <th>species</th>\n",
       "      <th>hit_def</th>\n",
       "      <th>align_len</th>\n",
       "      <th>Domain</th>\n",
       "      <th>Kingdom</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>index</th>\n",
       "      <th>Hit_number</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ASV_1</th>\n",
       "      <th>1</th>\n",
       "      <td>7.533310e-154</td>\n",
       "      <td>616</td>\n",
       "      <td>100.00</td>\n",
       "      <td>Not_Found</td>\n",
       "      <td>Not_Found</td>\n",
       "      <td>Gephyrocapsa muellerae strain RCC3370 mitochon...</td>\n",
       "      <td>308</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Haptophyta</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Isochrysidales</td>\n",
       "      <td>Noelaerhabdaceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_2</th>\n",
       "      <th>1</th>\n",
       "      <td>7.533310e-154</td>\n",
       "      <td>616</td>\n",
       "      <td>100.00</td>\n",
       "      <td>Not_Found</td>\n",
       "      <td>Not_Found</td>\n",
       "      <td>Emiliania huxleyi strain RCC4002 mitochondrion...</td>\n",
       "      <td>308</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Haptophyta</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Isochrysidales</td>\n",
       "      <td>Noelaerhabdaceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_3</th>\n",
       "      <th>1</th>\n",
       "      <td>2.999550e-70</td>\n",
       "      <td>309</td>\n",
       "      <td>80.13</td>\n",
       "      <td>Not_Found</td>\n",
       "      <td>Not_Found</td>\n",
       "      <td>Oncaea mediterranea voucher P0134 cytochrome o...</td>\n",
       "      <td>307</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Metazoa</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g_</td>\n",
       "      <td>s_</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_4</th>\n",
       "      <th>1</th>\n",
       "      <td>7.533310e-154</td>\n",
       "      <td>616</td>\n",
       "      <td>100.00</td>\n",
       "      <td>Not_Found</td>\n",
       "      <td>Not_Found</td>\n",
       "      <td>Gephyrocapsa parvula strain RCC4034 mitochondr...</td>\n",
       "      <td>308</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Haptophyta</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Isochrysidales</td>\n",
       "      <td>Noelaerhabdaceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_5</th>\n",
       "      <th>1</th>\n",
       "      <td>3.203250e-152</td>\n",
       "      <td>611</td>\n",
       "      <td>99.68</td>\n",
       "      <td>Not_Found</td>\n",
       "      <td>Not_Found</td>\n",
       "      <td>Gephyrocapsa parvula strain RCC4034 mitochondr...</td>\n",
       "      <td>308</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Haptophyta</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Isochrysidales</td>\n",
       "      <td>Noelaerhabdaceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_138295</th>\n",
       "      <th>1</th>\n",
       "      <td>1.050850e-69</td>\n",
       "      <td>307</td>\n",
       "      <td>81.10</td>\n",
       "      <td>Not_Found</td>\n",
       "      <td>Not_Found</td>\n",
       "      <td>Hecatonema maculans mitochondrial partial COI ...</td>\n",
       "      <td>291</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Phaeophyceae</td>\n",
       "      <td>Ectocarpales</td>\n",
       "      <td>Chordariaceae</td>\n",
       "      <td>g_</td>\n",
       "      <td>s_</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_138351</th>\n",
       "      <th>1</th>\n",
       "      <td>3.274330e-76</td>\n",
       "      <td>331</td>\n",
       "      <td>82.99</td>\n",
       "      <td>Aphanomyces</td>\n",
       "      <td>Aphanomyces iridis</td>\n",
       "      <td>Aphanomyces iridis voucher CBS52487 cytochrome...</td>\n",
       "      <td>288</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Oomycota</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Saprolegniales</td>\n",
       "      <td>Saprolegniaceae</td>\n",
       "      <td>g_</td>\n",
       "      <td>s_</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_138768</th>\n",
       "      <th>1</th>\n",
       "      <td>1.105320e-75</td>\n",
       "      <td>329</td>\n",
       "      <td>82.27</td>\n",
       "      <td>Not_Found</td>\n",
       "      <td>Not_Found</td>\n",
       "      <td>Laurencia sp. ARS-2010 voucher ARS03166 cytoch...</td>\n",
       "      <td>299</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g_</td>\n",
       "      <td>s_</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_138814</th>\n",
       "      <th>1</th>\n",
       "      <td>2.550410e-77</td>\n",
       "      <td>334</td>\n",
       "      <td>82.88</td>\n",
       "      <td>Not_Found</td>\n",
       "      <td>Not_Found</td>\n",
       "      <td>Ovalopodium rosalinum clone 2 cytochrome C oxi...</td>\n",
       "      <td>292</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g_</td>\n",
       "      <td>s_</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_138825</th>\n",
       "      <th>1</th>\n",
       "      <td>1.013010e-126</td>\n",
       "      <td>516</td>\n",
       "      <td>95.83</td>\n",
       "      <td>Phaeocystis</td>\n",
       "      <td>Phaeocystis globosa</td>\n",
       "      <td>Phaeocystis globosa mitochondrion, partial genome</td>\n",
       "      <td>288</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Haptophyta</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Phaeocystales</td>\n",
       "      <td>Phaeocystaceae</td>\n",
       "      <td>Phaeocystis</td>\n",
       "      <td>s_</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9828 rows Ã— 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                eval  bitscore     %ID        genus  \\\n",
       "index      Hit_number                                                 \n",
       "ASV_1      1           7.533310e-154       616  100.00    Not_Found   \n",
       "ASV_2      1           7.533310e-154       616  100.00    Not_Found   \n",
       "ASV_3      1            2.999550e-70       309   80.13    Not_Found   \n",
       "ASV_4      1           7.533310e-154       616  100.00    Not_Found   \n",
       "ASV_5      1           3.203250e-152       611   99.68    Not_Found   \n",
       "...                              ...       ...     ...          ...   \n",
       "ASV_138295 1            1.050850e-69       307   81.10    Not_Found   \n",
       "ASV_138351 1            3.274330e-76       331   82.99  Aphanomyces   \n",
       "ASV_138768 1            1.105320e-75       329   82.27    Not_Found   \n",
       "ASV_138814 1            2.550410e-77       334   82.88    Not_Found   \n",
       "ASV_138825 1           1.013010e-126       516   95.83  Phaeocystis   \n",
       "\n",
       "                                   species  \\\n",
       "index      Hit_number                        \n",
       "ASV_1      1                     Not_Found   \n",
       "ASV_2      1                     Not_Found   \n",
       "ASV_3      1                     Not_Found   \n",
       "ASV_4      1                     Not_Found   \n",
       "ASV_5      1                     Not_Found   \n",
       "...                                    ...   \n",
       "ASV_138295 1                     Not_Found   \n",
       "ASV_138351 1            Aphanomyces iridis   \n",
       "ASV_138768 1                     Not_Found   \n",
       "ASV_138814 1                     Not_Found   \n",
       "ASV_138825 1           Phaeocystis globosa   \n",
       "\n",
       "                                                                 hit_def  \\\n",
       "index      Hit_number                                                      \n",
       "ASV_1      1           Gephyrocapsa muellerae strain RCC3370 mitochon...   \n",
       "ASV_2      1           Emiliania huxleyi strain RCC4002 mitochondrion...   \n",
       "ASV_3      1           Oncaea mediterranea voucher P0134 cytochrome o...   \n",
       "ASV_4      1           Gephyrocapsa parvula strain RCC4034 mitochondr...   \n",
       "ASV_5      1           Gephyrocapsa parvula strain RCC4034 mitochondr...   \n",
       "...                                                                  ...   \n",
       "ASV_138295 1           Hecatonema maculans mitochondrial partial COI ...   \n",
       "ASV_138351 1           Aphanomyces iridis voucher CBS52487 cytochrome...   \n",
       "ASV_138768 1           Laurencia sp. ARS-2010 voucher ARS03166 cytoch...   \n",
       "ASV_138814 1           Ovalopodium rosalinum clone 2 cytochrome C oxi...   \n",
       "ASV_138825 1           Phaeocystis globosa mitochondrion, partial genome   \n",
       "\n",
       "                       align_len     Domain  Kingdom      Phylum  \\\n",
       "index      Hit_number                                              \n",
       "ASV_1      1                 308  Eukaryota      NaN  Haptophyta   \n",
       "ASV_2      1                 308  Eukaryota      NaN  Haptophyta   \n",
       "ASV_3      1                 307  Eukaryota  Metazoa  Arthropoda   \n",
       "ASV_4      1                 308  Eukaryota      NaN  Haptophyta   \n",
       "ASV_5      1                 308  Eukaryota      NaN  Haptophyta   \n",
       "...                          ...        ...      ...         ...   \n",
       "ASV_138295 1                 291  Eukaryota      NaN     unknown   \n",
       "ASV_138351 1                 288  Eukaryota      NaN    Oomycota   \n",
       "ASV_138768 1                 299  Eukaryota      NaN         NaN   \n",
       "ASV_138814 1                 292  Eukaryota      NaN         NaN   \n",
       "ASV_138825 1                 288  Eukaryota      NaN  Haptophyta   \n",
       "\n",
       "                              Class           Order            Family  \\\n",
       "index      Hit_number                                                   \n",
       "ASV_1      1                unknown  Isochrysidales  Noelaerhabdaceae   \n",
       "ASV_2      1                unknown  Isochrysidales  Noelaerhabdaceae   \n",
       "ASV_3      1                    NaN             NaN               NaN   \n",
       "ASV_4      1                unknown  Isochrysidales  Noelaerhabdaceae   \n",
       "ASV_5      1                unknown  Isochrysidales  Noelaerhabdaceae   \n",
       "...                             ...             ...               ...   \n",
       "ASV_138295 1           Phaeophyceae    Ectocarpales     Chordariaceae   \n",
       "ASV_138351 1                unknown  Saprolegniales   Saprolegniaceae   \n",
       "ASV_138768 1                    NaN             NaN               NaN   \n",
       "ASV_138814 1                    NaN             NaN               NaN   \n",
       "ASV_138825 1                unknown   Phaeocystales    Phaeocystaceae   \n",
       "\n",
       "                             Genus Species  \n",
       "index      Hit_number                       \n",
       "ASV_1      1                   NaN     NaN  \n",
       "ASV_2      1                   NaN     NaN  \n",
       "ASV_3      1                    g_      s_  \n",
       "ASV_4      1                   NaN     NaN  \n",
       "ASV_5      1                   NaN     NaN  \n",
       "...                            ...     ...  \n",
       "ASV_138295 1                    g_      s_  \n",
       "ASV_138351 1                    g_      s_  \n",
       "ASV_138768 1                    g_      s_  \n",
       "ASV_138814 1                    g_      s_  \n",
       "ASV_138825 1           Phaeocystis      s_  \n",
       "\n",
       "[9828 rows x 15 columns]"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "directory = '/Users/kpitz/Projects/custom_blast/'\n",
    "outname = directory + 'Filtered_ASV_taxa_table.csv'\n",
    "print(outname)\n",
    "taxa_lim.to_csv(outname)\n",
    "\n",
    "#Get average %ID of genus and species corrected ASVs:\n",
    "print('XXXX  Check Stats of hits that have been filtered: XXXX')\n",
    "df_g = df.loc[df['Genus']=='g_']\n",
    "print('Mean %ID of ASVs with g_:',df_g['%ID'].mean())\n",
    "print('Mean bitscore of ASVs with g_:',df_g['bitscore'].mean())\n",
    "df_s = df.loc[df['Species']=='s_']\n",
    "df_s = df_s.loc[df_s['Genus']!='g_']\n",
    "print('Mean %ID of ASVs with s_:',df_s['%ID'].mean())\n",
    "print('Mean bitscore of ASVs with s_:',df_s['bitscore'].mean())\n",
    "\n",
    "df.drop('OTU_Number', axis=1, inplace=True)\n",
    "df.drop('level_0', axis=1, inplace=True)\n",
    "outname = directory + 'ASV_tophit_Ftaxonomy.csv'\n",
    "print(outname)\n",
    "df.to_csv(outname)\n",
    "tophit_Ftaxonomy = df.copy()\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "id": "a8fb7c12-8e1c-4933-b299-aacf33e31d99",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'\\n\\n#Remove PhiX OTUs\\nlist_DUP = (taxon_A, taxon_B)\\n#Number of OTUs without duplicates\\nfor i in range(len(list_DUP)):\\n    y=list(set(list_DUP[i]))   #Remove duplicate entries from each list\\n    print(taxon[i], \\' number of OTUs: \\', len(y))    #Get number of OTUs which hit each taxon\\n\\n#Get total number of otus and reads with PhiX and phiX; list of unique ASVs with PhiX hits\\ntaxons = taxon_A + taxon_B\\ntaxons= list(set(taxons))\\nprint(\\'Total Number OTUs with PhiX hits:\\',len(taxons))   #Number of OTUS\\n\\nif len(taxons)== 0 :\\n    print(\\'No Phix OTUs removed\\')\\nelse:\\n    #remove these ASVs from the OTU table\\n    df = otu_tab.copy()\\n    df.reset_index(inplace=True)\\n    df=df.loc[df[\\'ASV\\'].isin(taxons)==False]\\n    otu_tab = df.copy()\\n\\n#ASV taxa table taken from Blast XML files will only contain ASVs with blast hits\\n#Ideally want to assign taxonomy from old table for those hits that weren\\'t changed by this filtering\\n#They should all be \\'no_hit\\'\\n\\n#Join  new taxonomy and old taxonomy table together; check \\'no_hit\\' for all missing taxa\\ndf= taxa_lim.copy()\\ndf[\\'New\\']=1\\ndf=pd.concat([df, taxa_tab], axis=1, keys=[\\'new\\',\\'old\\'],sort=False)\\ndf=df.sort_values((\\'new\\', \\'New\\'))\\ndf=df.loc[df[(\\'new\\', \\'New\\')]!=1]\\nprint(\\'Unique Kingdom IDs from taxa without a Blast hit: \\')\\nprint(\"Should just be \\'no_hit\\':\")\\nprint(df[(\\'old\\', \\'Kingdom\\')].unique())\\n\\n#Since they\\'re all no_hit, assign any missing values as \\'no_hit\\'\\ndf=pd.concat([taxa_lim, otu_tab], axis=1,sort=False)\\ndf=df.fillna(\\'no_hit\\')\\ndf.index.name =\\'ASV\\'\\n\\n#export \\'OTU_taxa_table_all.csv\\'\\noutname = directory + \\'Filtered_ASV_taxa_table_all.csv\\'\\nprint(outname)\\ndf.to_csv(outname)\\n\\nprint(\\'Finished Filtering Taxonomy by BLAST hits\\')'"
      ]
     },
     "execution_count": 88,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'''\n",
    "\n",
    "#Remove PhiX OTUs\n",
    "list_DUP = (taxon_A, taxon_B)\n",
    "#Number of OTUs without duplicates\n",
    "for i in range(len(list_DUP)):\n",
    "    y=list(set(list_DUP[i]))   #Remove duplicate entries from each list\n",
    "    print(taxon[i], ' number of OTUs: ', len(y))    #Get number of OTUs which hit each taxon\n",
    "\n",
    "#Get total number of otus and reads with PhiX and phiX; list of unique ASVs with PhiX hits\n",
    "taxons = taxon_A + taxon_B\n",
    "taxons= list(set(taxons))\n",
    "print('Total Number OTUs with PhiX hits:',len(taxons))   #Number of OTUS\n",
    "\n",
    "if len(taxons)== 0 :\n",
    "    print('No Phix OTUs removed')\n",
    "else:\n",
    "    #remove these ASVs from the OTU table\n",
    "    df = otu_tab.copy()\n",
    "    df.reset_index(inplace=True)\n",
    "    df=df.loc[df['ASV'].isin(taxons)==False]\n",
    "    otu_tab = df.copy()\n",
    "\n",
    "#ASV taxa table taken from Blast XML files will only contain ASVs with blast hits\n",
    "#Ideally want to assign taxonomy from old table for those hits that weren't changed by this filtering\n",
    "#They should all be 'no_hit'\n",
    "\n",
    "#Join  new taxonomy and old taxonomy table together; check 'no_hit' for all missing taxa\n",
    "df= taxa_lim.copy()\n",
    "df['New']=1\n",
    "df=pd.concat([df, taxa_tab], axis=1, keys=['new','old'],sort=False)\n",
    "df=df.sort_values(('new', 'New'))\n",
    "df=df.loc[df[('new', 'New')]!=1]\n",
    "print('Unique Kingdom IDs from taxa without a Blast hit: ')\n",
    "print(\"Should just be 'no_hit':\")\n",
    "print(df[('old', 'Kingdom')].unique())\n",
    "\n",
    "#Since they're all no_hit, assign any missing values as 'no_hit'\n",
    "df=pd.concat([taxa_lim, otu_tab], axis=1,sort=False)\n",
    "df=df.fillna('no_hit')\n",
    "df.index.name ='ASV'\n",
    "\n",
    "#export 'OTU_taxa_table_all.csv'\n",
    "outname = directory + 'Filtered_ASV_taxa_table_all.csv'\n",
    "print(outname)\n",
    "df.to_csv(outname)\n",
    "\n",
    "print('Finished Filtering Taxonomy by BLAST hits')'''"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "12c36dc3-0643-4ce7-8f07-17c10fed1d33",
   "metadata": {},
   "source": [
    "## Parse results from MZG blastn search"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "530fdd52-1a7b-4efa-b1e5-9d2e860472d5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Blast XML file:  /Users/kpitz/Projects/custom_blast/Flyer2018_COI_MZG_COI_blastn_061223.xml\n",
      "Done parsing  /Users/kpitz/Projects/custom_blast/Flyer2018_COI_MZG_COI_blastn_061223.xml\n"
     ]
    }
   ],
   "source": [
    "# from Bio.Blast import NCBIXML\n",
    "\n",
    "# Parse MZGB xml file\n",
    "files = ['/Users/kpitz/Projects/custom_blast/Flyer2018_COI_MZG_COI_blastn_061223.xml']\n",
    "\n",
    "# just keep top 4 hits for each query\n",
    "\n",
    "# Parse through XML files\n",
    "xml_dfs = []  #will store results from each xml file\n",
    "for filename in files:\n",
    "    print('Blast XML file: ', filename)\n",
    "    query=[]\n",
    "    hit_dict ={}  #Dictionary to store hits\n",
    "    result_handle =open(filename)\n",
    "    #Phix names to filter out - added for diversity in sequencing process\n",
    "    taxon = ['PhiX', 'phiX']\n",
    "    #parse blast records\n",
    "    blast_records= NCBIXML.parse(result_handle)\n",
    "    for blast_record in blast_records:\n",
    "        query = blast_record.query\n",
    "        hit_ID=0 #hit counter\n",
    "        for alignment in blast_record.alignments:\n",
    "            hit_ID+=1\n",
    "            if hit_ID >4:\n",
    "                continue\n",
    "            key = (query, hit_ID)\n",
    "            hsp = alignment.hsps[0] #only look at top hsp per alignment to genbank sequence\n",
    "            #set limits on evalue and bitscore and % Identity\n",
    "            per_iden = hsp.identities/float(hsp.align_length)\n",
    "            per_iden = (per_iden *100)\n",
    "            per_iden = round(per_iden, 2)\n",
    "            evalue = hsp.expect\n",
    "            bitscore = int(hsp.score)\n",
    "            #alignment length\n",
    "            align_len = hsp.align_length\n",
    "            value = (evalue, bitscore, per_iden, alignment.hit_def, align_len)\n",
    "            hit_dict[key]=value\n",
    "    print ('Done parsing ', filename)\n",
    "    #Save Blast Results to file\n",
    "    df = pd.DataFrame(hit_dict)\n",
    "    df=df.T\n",
    "    df.columns = ['eval','bitscore', '%ID', 'hit_def', 'align_len']\n",
    "    df.reset_index(inplace=True)\n",
    "    df = df.rename(columns={'level_0': 'ASV', 'level_1': 'Hit_number'})\n",
    "    xml_dfs.append(df)\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "01dbde8d-0ec4-403d-ab8b-eddc3a88add2",
   "metadata": {},
   "source": [
    "## Limit by percent ID and just keep one hit per query"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "0e215a2d-b21b-4d31-906e-b06c1feffafc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hit_number</th>\n",
       "      <th>eval</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>%ID</th>\n",
       "      <th>hit_def</th>\n",
       "      <th>align_len</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ASV_1</th>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>226</td>\n",
       "      <td>98.72</td>\n",
       "      <td>AB649198__Emiliania_huxleyi   MZGdb-v2023-m04-...</td>\n",
       "      <td>235</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_2</th>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>228</td>\n",
       "      <td>98.73</td>\n",
       "      <td>AB649198__Emiliania_huxleyi   MZGdb-v2023-m04-...</td>\n",
       "      <td>237</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_3</th>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>98</td>\n",
       "      <td>86.23</td>\n",
       "      <td>MG669420__Thysanopoda_orientalis   MZGdb-v2023...</td>\n",
       "      <td>167</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_4</th>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>231</td>\n",
       "      <td>99.16</td>\n",
       "      <td>AB649198__Emiliania_huxleyi   MZGdb-v2023-m04-...</td>\n",
       "      <td>237</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_5</th>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>234</td>\n",
       "      <td>99.58</td>\n",
       "      <td>AB649198__Emiliania_huxleyi   MZGdb-v2023-m04-...</td>\n",
       "      <td>237</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_138196</th>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>234</td>\n",
       "      <td>99.58</td>\n",
       "      <td>AB649198__Emiliania_huxleyi   MZGdb-v2023-m04-...</td>\n",
       "      <td>237</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_138288</th>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>120</td>\n",
       "      <td>80.62</td>\n",
       "      <td>KJ960337__Apoglossum_ruscifolium   MZGdb-v2023...</td>\n",
       "      <td>289</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_138768</th>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>135</td>\n",
       "      <td>81.88</td>\n",
       "      <td>MT760752__Phaeocystis_globosa   MZGdb-v2023-m0...</td>\n",
       "      <td>298</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_138814</th>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>125</td>\n",
       "      <td>81.1</td>\n",
       "      <td>AB441225__Stylocoeniella_guentheri   MZGdb-v20...</td>\n",
       "      <td>291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_138825</th>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>252</td>\n",
       "      <td>95.83</td>\n",
       "      <td>KC967226__Phaeocystis_globosa   MZGdb-v2023-m0...</td>\n",
       "      <td>288</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7237 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            Hit_number eval bitscore    %ID  \\\n",
       "ASV                                           \n",
       "ASV_1                1  0.0      226  98.72   \n",
       "ASV_2                1  0.0      228  98.73   \n",
       "ASV_3                2  0.0       98  86.23   \n",
       "ASV_4                1  0.0      231  99.16   \n",
       "ASV_5                1  0.0      234  99.58   \n",
       "...                ...  ...      ...    ...   \n",
       "ASV_138196           1  0.0      234  99.58   \n",
       "ASV_138288           1  0.0      120  80.62   \n",
       "ASV_138768           1  0.0      135  81.88   \n",
       "ASV_138814           1  0.0      125   81.1   \n",
       "ASV_138825           1  0.0      252  95.83   \n",
       "\n",
       "                                                      hit_def align_len  \n",
       "ASV                                                                      \n",
       "ASV_1       AB649198__Emiliania_huxleyi   MZGdb-v2023-m04-...       235  \n",
       "ASV_2       AB649198__Emiliania_huxleyi   MZGdb-v2023-m04-...       237  \n",
       "ASV_3       MG669420__Thysanopoda_orientalis   MZGdb-v2023...       167  \n",
       "ASV_4       AB649198__Emiliania_huxleyi   MZGdb-v2023-m04-...       237  \n",
       "ASV_5       AB649198__Emiliania_huxleyi   MZGdb-v2023-m04-...       237  \n",
       "...                                                       ...       ...  \n",
       "ASV_138196  AB649198__Emiliania_huxleyi   MZGdb-v2023-m04-...       237  \n",
       "ASV_138288  KJ960337__Apoglossum_ruscifolium   MZGdb-v2023...       289  \n",
       "ASV_138768  MT760752__Phaeocystis_globosa   MZGdb-v2023-m0...       298  \n",
       "ASV_138814  AB441225__Stylocoeniella_guentheri   MZGdb-v20...       291  \n",
       "ASV_138825  KC967226__Phaeocystis_globosa   MZGdb-v2023-m0...       288  \n",
       "\n",
       "[7237 rows x 6 columns]"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.concat(xml_dfs, axis=1)\n",
    "df = df.loc[df['%ID']>=80]\n",
    "df = df.drop_duplicates('ASV', keep='first')\n",
    "#df = df.loc[df['ASV']=='ASV_3']\n",
    "df.to_csv('/Users/kpitz/Documents/test.csv')\n",
    "df = df.set_index('ASV')\n",
    "MZgb = df.copy()\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a4f011e8-1acf-4492-9894-6809d48f2cb3",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Join MZG results with genbank results\n",
    "- Make columns to show when genera or species results match between methods"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "id": "7a82d338-3b67-4c5a-8d79-84e4ea887b34",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr:last-of-type th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th colspan=\"6\" halign=\"left\">MZGdb</th>\n",
       "      <th colspan=\"13\" halign=\"left\">genbank</th>\n",
       "      <th>genus_match</th>\n",
       "      <th>sp_match</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>Hit_number</th>\n",
       "      <th>eval</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>%ID</th>\n",
       "      <th>hit_def</th>\n",
       "      <th>align_len</th>\n",
       "      <th>Hit_number</th>\n",
       "      <th>eval</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>%ID</th>\n",
       "      <th>...</th>\n",
       "      <th>Domain</th>\n",
       "      <th>Kingdom</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ASV_1623</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>316</td>\n",
       "      <td>100.0</td>\n",
       "      <td>CNIDC447-16__Muggiaea_atlantica   MZGdb_v2023-...</td>\n",
       "      <td>316</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.951310e-123</td>\n",
       "      <td>505.0</td>\n",
       "      <td>92.06</td>\n",
       "      <td>...</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Metazoa</td>\n",
       "      <td>Cnidaria</td>\n",
       "      <td>Hydrozoa</td>\n",
       "      <td>Siphonophorae</td>\n",
       "      <td>Diphyidae</td>\n",
       "      <td>g_</td>\n",
       "      <td>s_</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_2672</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>313</td>\n",
       "      <td>100.0</td>\n",
       "      <td>NC_013935__Pycnococcus_provasolii   MZGdb-v202...</td>\n",
       "      <td>313</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.481300e-156</td>\n",
       "      <td>626.0</td>\n",
       "      <td>100.00</td>\n",
       "      <td>...</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Viridiplantae</td>\n",
       "      <td>Chlorophyta</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Pycnococcaceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_537</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>313</td>\n",
       "      <td>100.0</td>\n",
       "      <td>JN050306__Pseudonitzschia_cuspidata   MZGdb-v2...</td>\n",
       "      <td>313</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.481300e-156</td>\n",
       "      <td>626.0</td>\n",
       "      <td>100.00</td>\n",
       "      <td>...</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bacillariophyta</td>\n",
       "      <td>Bacillariophyceae</td>\n",
       "      <td>Bacillariales</td>\n",
       "      <td>Bacillariaceae</td>\n",
       "      <td>Pseudo-nitzschia</td>\n",
       "      <td>Pseudo-nitzschia cuspidata</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_14705</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>313</td>\n",
       "      <td>100.0</td>\n",
       "      <td>HM473956__Strongylocentrotus_fragilis   MZGdb-...</td>\n",
       "      <td>313</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.481300e-156</td>\n",
       "      <td>626.0</td>\n",
       "      <td>100.00</td>\n",
       "      <td>...</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Metazoa</td>\n",
       "      <td>Echinodermata</td>\n",
       "      <td>Echinoidea</td>\n",
       "      <td>Camarodonta</td>\n",
       "      <td>Strongylocentrotidae</td>\n",
       "      <td>Allocentrotus</td>\n",
       "      <td>Allocentrotus fragilis</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_23324</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>313</td>\n",
       "      <td>100.0</td>\n",
       "      <td>MW446643__Stenella_coeruleoalba   MZGdb-v2023-...</td>\n",
       "      <td>313</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.481300e-156</td>\n",
       "      <td>626.0</td>\n",
       "      <td>100.00</td>\n",
       "      <td>...</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Metazoa</td>\n",
       "      <td>Chordata</td>\n",
       "      <td>Mammalia</td>\n",
       "      <td>Artiodactyla</td>\n",
       "      <td>Delphinidae</td>\n",
       "      <td>Delphinus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_16621</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>8.357320e-06</td>\n",
       "      <td>70.0</td>\n",
       "      <td>91.11</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g_</td>\n",
       "      <td>s_</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_26530</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>8.312630e-06</td>\n",
       "      <td>70.0</td>\n",
       "      <td>95.00</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g_</td>\n",
       "      <td>s_</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_25986</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>8.312630e-06</td>\n",
       "      <td>70.0</td>\n",
       "      <td>91.11</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g_</td>\n",
       "      <td>s_</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_44269</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>8.848930e-06</td>\n",
       "      <td>70.0</td>\n",
       "      <td>95.00</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g_</td>\n",
       "      <td>s_</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_65214</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>8.357320e-06</td>\n",
       "      <td>70.0</td>\n",
       "      <td>91.11</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g_</td>\n",
       "      <td>s_</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9749 rows Ã— 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               MZGdb                       \\\n",
       "          Hit_number eval bitscore    %ID   \n",
       "ASV                                         \n",
       "ASV_1623         1.0  0.0      316  100.0   \n",
       "ASV_2672         1.0  0.0      313  100.0   \n",
       "ASV_537          1.0  0.0      313  100.0   \n",
       "ASV_14705        1.0  0.0      313  100.0   \n",
       "ASV_23324        1.0  0.0      313  100.0   \n",
       "...              ...  ...      ...    ...   \n",
       "ASV_16621        NaN  NaN      NaN    NaN   \n",
       "ASV_26530        NaN  NaN      NaN    NaN   \n",
       "ASV_25986        NaN  NaN      NaN    NaN   \n",
       "ASV_44269        NaN  NaN      NaN    NaN   \n",
       "ASV_65214        NaN  NaN      NaN    NaN   \n",
       "\n",
       "                                                                        \\\n",
       "                                                     hit_def align_len   \n",
       "ASV                                                                      \n",
       "ASV_1623   CNIDC447-16__Muggiaea_atlantica   MZGdb_v2023-...       316   \n",
       "ASV_2672   NC_013935__Pycnococcus_provasolii   MZGdb-v202...       313   \n",
       "ASV_537    JN050306__Pseudonitzschia_cuspidata   MZGdb-v2...       313   \n",
       "ASV_14705  HM473956__Strongylocentrotus_fragilis   MZGdb-...       313   \n",
       "ASV_23324  MW446643__Stenella_coeruleoalba   MZGdb-v2023-...       313   \n",
       "...                                                      ...       ...   \n",
       "ASV_16621                                                NaN       NaN   \n",
       "ASV_26530                                                NaN       NaN   \n",
       "ASV_25986                                                NaN       NaN   \n",
       "ASV_44269                                                NaN       NaN   \n",
       "ASV_65214                                                NaN       NaN   \n",
       "\n",
       "             genbank                                  ...             \\\n",
       "          Hit_number           eval bitscore     %ID  ...     Domain   \n",
       "ASV                                                   ...              \n",
       "ASV_1623         2.0  1.951310e-123    505.0   92.06  ...  Eukaryota   \n",
       "ASV_2672         1.0  1.481300e-156    626.0  100.00  ...  Eukaryota   \n",
       "ASV_537          1.0  1.481300e-156    626.0  100.00  ...  Eukaryota   \n",
       "ASV_14705        1.0  1.481300e-156    626.0  100.00  ...  Eukaryota   \n",
       "ASV_23324        1.0  1.481300e-156    626.0  100.00  ...  Eukaryota   \n",
       "...              ...            ...      ...     ...  ...        ...   \n",
       "ASV_16621        1.0   8.357320e-06     70.0   91.11  ...        NaN   \n",
       "ASV_26530        1.0   8.312630e-06     70.0   95.00  ...        NaN   \n",
       "ASV_25986        1.0   8.312630e-06     70.0   91.11  ...        NaN   \n",
       "ASV_44269        1.0   8.848930e-06     70.0   95.00  ...        NaN   \n",
       "ASV_65214        1.0   8.357320e-06     70.0   91.11  ...        NaN   \n",
       "\n",
       "                                                                             \\\n",
       "                 Kingdom           Phylum              Class          Order   \n",
       "ASV                                                                           \n",
       "ASV_1623         Metazoa         Cnidaria           Hydrozoa  Siphonophorae   \n",
       "ASV_2672   Viridiplantae      Chlorophyta            unknown        unknown   \n",
       "ASV_537              NaN  Bacillariophyta  Bacillariophyceae  Bacillariales   \n",
       "ASV_14705        Metazoa    Echinodermata         Echinoidea    Camarodonta   \n",
       "ASV_23324        Metazoa         Chordata           Mammalia   Artiodactyla   \n",
       "...                  ...              ...                ...            ...   \n",
       "ASV_16621            NaN              NaN                NaN            NaN   \n",
       "ASV_26530            NaN              NaN                NaN            NaN   \n",
       "ASV_25986            NaN              NaN                NaN            NaN   \n",
       "ASV_44269            NaN              NaN                NaN            NaN   \n",
       "ASV_65214            NaN              NaN                NaN            NaN   \n",
       "\n",
       "                                                                               \\\n",
       "                         Family             Genus                     Species   \n",
       "ASV                                                                             \n",
       "ASV_1623              Diphyidae                g_                          s_   \n",
       "ASV_2672         Pycnococcaceae               NaN                         NaN   \n",
       "ASV_537          Bacillariaceae  Pseudo-nitzschia  Pseudo-nitzschia cuspidata   \n",
       "ASV_14705  Strongylocentrotidae     Allocentrotus      Allocentrotus fragilis   \n",
       "ASV_23324           Delphinidae         Delphinus                         NaN   \n",
       "...                         ...               ...                         ...   \n",
       "ASV_16621                   NaN                g_                          s_   \n",
       "ASV_26530                   NaN                g_                          s_   \n",
       "ASV_25986                   NaN                g_                          s_   \n",
       "ASV_44269                   NaN                g_                          s_   \n",
       "ASV_65214                   NaN                g_                          s_   \n",
       "\n",
       "          genus_match sp_match  \n",
       "                                \n",
       "ASV                             \n",
       "ASV_1623        False    False  \n",
       "ASV_2672        False    False  \n",
       "ASV_537         False    False  \n",
       "ASV_14705       False    False  \n",
       "ASV_23324       False    False  \n",
       "...               ...      ...  \n",
       "ASV_16621       False    False  \n",
       "ASV_26530       False    False  \n",
       "ASV_25986       False    False  \n",
       "ASV_44269       False    False  \n",
       "ASV_65214       False    False  \n",
       "\n",
       "[9749 rows x 24 columns]"
      ]
     },
     "execution_count": 110,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# join with MEGAN:\n",
    "df = tophit_Ftaxonomy.reset_index()\n",
    "df = df.rename(columns={'index':'ASV'})\n",
    "df = df.set_index('ASV')\n",
    "df = pd.concat([MZgb, df], axis=1, keys=['MZGdb', 'genbank'])\n",
    "df.to_csv(directory + 'MZGdb_Genbank_megan_comparison.csv')\n",
    "# Now compare\n",
    "#Genus\n",
    "df['genus_match'] = df.apply(lambda x: str(x['genbank'].Genus) in str(x['MZGdb'].hit_def), axis=1)\n",
    "df.loc[df['genbank']['Genus'].isna(), 'genus_match'] = False\n",
    "df = df.sort_values('genus_match')\n",
    "# Species\n",
    "df['hit_sp_def'] = df['MZGdb']['hit_def'].str.replace('_',' ')\n",
    "df['sp_match'] = df.apply(lambda x: str(x['genbank'].Species) in str(x.hit_sp_def), axis=1)\n",
    "df.loc[df['genbank']['Species'].isna(), 'sp_match'] = False\n",
    "df = df.sort_values('sp_match')\n",
    "\n",
    "# drop unecessary columns\n",
    "df = df.drop(['hit_sp_def'], axis=1)\n",
    "df = df.loc[(df['sp_match']==False) & (df['genus_match']==False)]\n",
    "df = df.sort_values([('MZGdb', 'bitscore'), ('genbank', 'bitscore')], ascending=False)\n",
    "df.to_csv(directory + 'MZGdb_Genbank_megan_diffs.csv')\n",
    "\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a1ccabe4-31a7-426b-8c72-156a40da0c6a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "06aab2ac-aa69-4f03-a560-2f9036736089",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "id": "119354ad-0544-466c-b174-8a082904aca0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr:last-of-type th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th colspan=\"6\" halign=\"left\">MZGdb</th>\n",
       "      <th colspan=\"13\" halign=\"left\">genbank</th>\n",
       "      <th>genus_match</th>\n",
       "      <th>sp_match</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>Hit_number</th>\n",
       "      <th>eval</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>%ID</th>\n",
       "      <th>hit_def</th>\n",
       "      <th>align_len</th>\n",
       "      <th>Hit_number</th>\n",
       "      <th>eval</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>%ID</th>\n",
       "      <th>...</th>\n",
       "      <th>Domain</th>\n",
       "      <th>Kingdom</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ASV_1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>226</td>\n",
       "      <td>98.72</td>\n",
       "      <td>AB649198__Emiliania_huxleyi   MZGdb-v2023-m04-...</td>\n",
       "      <td>235</td>\n",
       "      <td>1.0</td>\n",
       "      <td>7.533310e-154</td>\n",
       "      <td>616.0</td>\n",
       "      <td>100.00</td>\n",
       "      <td>...</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Haptophyta</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Isochrysidales</td>\n",
       "      <td>Noelaerhabdaceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_125642</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>149</td>\n",
       "      <td>83.79</td>\n",
       "      <td>AB948162__Skeletonema_menzelii   MZGdb-v2023-m...</td>\n",
       "      <td>290</td>\n",
       "      <td>6.0</td>\n",
       "      <td>5.438750e-86</td>\n",
       "      <td>366.0</td>\n",
       "      <td>87.31</td>\n",
       "      <td>...</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bacillariophyta</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g_</td>\n",
       "      <td>s_</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_125681</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>114</td>\n",
       "      <td>98.33</td>\n",
       "      <td>KHBC002-13__Pandea_rubra   MZGdb_v2023-m04-01 ...</td>\n",
       "      <td>120</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.739310e-50</td>\n",
       "      <td>235.0</td>\n",
       "      <td>99.17</td>\n",
       "      <td>...</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Metazoa</td>\n",
       "      <td>Chordata</td>\n",
       "      <td>Mammalia</td>\n",
       "      <td>Primates</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>g_</td>\n",
       "      <td>s_</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_125687</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>161</td>\n",
       "      <td>85.37</td>\n",
       "      <td>LR880960__Eurypon_clavigerum   MZGdb-v2023-m04...</td>\n",
       "      <td>287</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.062680e-110</td>\n",
       "      <td>457.0</td>\n",
       "      <td>91.41</td>\n",
       "      <td>...</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Metazoa</td>\n",
       "      <td>Porifera</td>\n",
       "      <td>Demospongiae</td>\n",
       "      <td>Axinellida</td>\n",
       "      <td>Raspailiidae</td>\n",
       "      <td>g_</td>\n",
       "      <td>s_</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_130778</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>210</td>\n",
       "      <td>96.2</td>\n",
       "      <td>AB649198__Emiliania_huxleyi   MZGdb-v2023-m04-...</td>\n",
       "      <td>237</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.426170e-120</td>\n",
       "      <td>492.0</td>\n",
       "      <td>92.69</td>\n",
       "      <td>...</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Haptophyta</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Isochrysidales</td>\n",
       "      <td>Noelaerhabdaceae</td>\n",
       "      <td>g_</td>\n",
       "      <td>s_</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_907</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>302</td>\n",
       "      <td>99.35</td>\n",
       "      <td>MG136783__Leuckartiara_longicalcar   MZGdb-v20...</td>\n",
       "      <td>308</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.902350e-151</td>\n",
       "      <td>606.0</td>\n",
       "      <td>99.35</td>\n",
       "      <td>...</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Metazoa</td>\n",
       "      <td>Cnidaria</td>\n",
       "      <td>Hydrozoa</td>\n",
       "      <td>Anthoathecata</td>\n",
       "      <td>Pandeidae</td>\n",
       "      <td>Leuckartiara</td>\n",
       "      <td>Leuckartiara longicalcar</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_16879</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>305</td>\n",
       "      <td>99.68</td>\n",
       "      <td>MN745808__Ctenocalanus_vanus   MZGdb-v2023-m04...</td>\n",
       "      <td>308</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.203250e-152</td>\n",
       "      <td>611.0</td>\n",
       "      <td>99.68</td>\n",
       "      <td>...</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Metazoa</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Hexanauplia</td>\n",
       "      <td>Calanoida</td>\n",
       "      <td>Calanidae</td>\n",
       "      <td>Ctenocalanus</td>\n",
       "      <td>Ctenocalanus vanus</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_3893</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>305</td>\n",
       "      <td>99.68</td>\n",
       "      <td>KC287601__Clausocalanus_paululus   MZGdb-v2023...</td>\n",
       "      <td>308</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.203250e-152</td>\n",
       "      <td>611.0</td>\n",
       "      <td>99.68</td>\n",
       "      <td>...</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Metazoa</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Hexanauplia</td>\n",
       "      <td>Calanoida</td>\n",
       "      <td>Clausocalanidae</td>\n",
       "      <td>Clausocalanus</td>\n",
       "      <td>Clausocalanus paululus</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_24535</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>287</td>\n",
       "      <td>97.73</td>\n",
       "      <td>KX650375__Oncaea_scottodicarloi   MZGdb-v2023-...</td>\n",
       "      <td>308</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.452580e-144</td>\n",
       "      <td>581.0</td>\n",
       "      <td>97.73</td>\n",
       "      <td>...</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Metazoa</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Hexanauplia</td>\n",
       "      <td>Poecilostomatoida</td>\n",
       "      <td>Oncaeidae</td>\n",
       "      <td>Oncaea</td>\n",
       "      <td>Oncaea scottodicarloi</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_5228</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>302</td>\n",
       "      <td>99.35</td>\n",
       "      <td>MG136783__Leuckartiara_longicalcar   MZGdb-v20...</td>\n",
       "      <td>308</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.902350e-151</td>\n",
       "      <td>606.0</td>\n",
       "      <td>99.35</td>\n",
       "      <td>...</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Metazoa</td>\n",
       "      <td>Cnidaria</td>\n",
       "      <td>Hydrozoa</td>\n",
       "      <td>Anthoathecata</td>\n",
       "      <td>Pandeidae</td>\n",
       "      <td>Leuckartiara</td>\n",
       "      <td>Leuckartiara longicalcar</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10305 rows Ã— 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                MZGdb                       \\\n",
       "           Hit_number eval bitscore    %ID   \n",
       "ASV                                          \n",
       "ASV_1             1.0  0.0      226  98.72   \n",
       "ASV_125642        1.0  0.0      149  83.79   \n",
       "ASV_125681        1.0  0.0      114  98.33   \n",
       "ASV_125687        1.0  0.0      161  85.37   \n",
       "ASV_130778        1.0  0.0      210   96.2   \n",
       "...               ...  ...      ...    ...   \n",
       "ASV_907           1.0  0.0      302  99.35   \n",
       "ASV_16879         1.0  0.0      305  99.68   \n",
       "ASV_3893          1.0  0.0      305  99.68   \n",
       "ASV_24535         1.0  0.0      287  97.73   \n",
       "ASV_5228          1.0  0.0      302  99.35   \n",
       "\n",
       "                                                                         \\\n",
       "                                                      hit_def align_len   \n",
       "ASV                                                                       \n",
       "ASV_1       AB649198__Emiliania_huxleyi   MZGdb-v2023-m04-...       235   \n",
       "ASV_125642  AB948162__Skeletonema_menzelii   MZGdb-v2023-m...       290   \n",
       "ASV_125681  KHBC002-13__Pandea_rubra   MZGdb_v2023-m04-01 ...       120   \n",
       "ASV_125687  LR880960__Eurypon_clavigerum   MZGdb-v2023-m04...       287   \n",
       "ASV_130778  AB649198__Emiliania_huxleyi   MZGdb-v2023-m04-...       237   \n",
       "...                                                       ...       ...   \n",
       "ASV_907     MG136783__Leuckartiara_longicalcar   MZGdb-v20...       308   \n",
       "ASV_16879   MN745808__Ctenocalanus_vanus   MZGdb-v2023-m04...       308   \n",
       "ASV_3893    KC287601__Clausocalanus_paululus   MZGdb-v2023...       308   \n",
       "ASV_24535   KX650375__Oncaea_scottodicarloi   MZGdb-v2023-...       308   \n",
       "ASV_5228    MG136783__Leuckartiara_longicalcar   MZGdb-v20...       308   \n",
       "\n",
       "              genbank                                  ...             \\\n",
       "           Hit_number           eval bitscore     %ID  ...     Domain   \n",
       "ASV                                                    ...              \n",
       "ASV_1             1.0  7.533310e-154    616.0  100.00  ...  Eukaryota   \n",
       "ASV_125642        6.0   5.438750e-86    366.0   87.31  ...  Eukaryota   \n",
       "ASV_125681        1.0   3.739310e-50    235.0   99.17  ...  Eukaryota   \n",
       "ASV_125687        1.0  2.062680e-110    457.0   91.41  ...  Eukaryota   \n",
       "ASV_130778        1.0  3.426170e-120    492.0   92.69  ...  Eukaryota   \n",
       "...               ...            ...      ...     ...  ...        ...   \n",
       "ASV_907           1.0  3.902350e-151    606.0   99.35  ...  Eukaryota   \n",
       "ASV_16879         1.0  3.203250e-152    611.0   99.68  ...  Eukaryota   \n",
       "ASV_3893          1.0  3.203250e-152    611.0   99.68  ...  Eukaryota   \n",
       "ASV_24535         1.0  4.452580e-144    581.0   97.73  ...  Eukaryota   \n",
       "ASV_5228          1.0  3.902350e-151    606.0   99.35  ...  Eukaryota   \n",
       "\n",
       "                                                                       \\\n",
       "            Kingdom           Phylum         Class              Order   \n",
       "ASV                                                                     \n",
       "ASV_1           NaN       Haptophyta       unknown     Isochrysidales   \n",
       "ASV_125642      NaN  Bacillariophyta           NaN                NaN   \n",
       "ASV_125681  Metazoa         Chordata      Mammalia           Primates   \n",
       "ASV_125687  Metazoa         Porifera  Demospongiae         Axinellida   \n",
       "ASV_130778      NaN       Haptophyta       unknown     Isochrysidales   \n",
       "...             ...              ...           ...                ...   \n",
       "ASV_907     Metazoa         Cnidaria      Hydrozoa      Anthoathecata   \n",
       "ASV_16879   Metazoa       Arthropoda   Hexanauplia          Calanoida   \n",
       "ASV_3893    Metazoa       Arthropoda   Hexanauplia          Calanoida   \n",
       "ASV_24535   Metazoa       Arthropoda   Hexanauplia  Poecilostomatoida   \n",
       "ASV_5228    Metazoa         Cnidaria      Hydrozoa      Anthoathecata   \n",
       "\n",
       "                                                                       \\\n",
       "                      Family          Genus                   Species   \n",
       "ASV                                                                     \n",
       "ASV_1       Noelaerhabdaceae            NaN                       NaN   \n",
       "ASV_125642               NaN             g_                        s_   \n",
       "ASV_125681         Hominidae             g_                        s_   \n",
       "ASV_125687      Raspailiidae             g_                        s_   \n",
       "ASV_130778  Noelaerhabdaceae             g_                        s_   \n",
       "...                      ...            ...                       ...   \n",
       "ASV_907            Pandeidae   Leuckartiara  Leuckartiara longicalcar   \n",
       "ASV_16879          Calanidae   Ctenocalanus        Ctenocalanus vanus   \n",
       "ASV_3893     Clausocalanidae  Clausocalanus    Clausocalanus paululus   \n",
       "ASV_24535          Oncaeidae         Oncaea     Oncaea scottodicarloi   \n",
       "ASV_5228           Pandeidae   Leuckartiara  Leuckartiara longicalcar   \n",
       "\n",
       "           genus_match sp_match  \n",
       "                                 \n",
       "ASV                              \n",
       "ASV_1            False    False  \n",
       "ASV_125642       False    False  \n",
       "ASV_125681       False    False  \n",
       "ASV_125687       False    False  \n",
       "ASV_130778       False    False  \n",
       "...                ...      ...  \n",
       "ASV_907           True     True  \n",
       "ASV_16879         True     True  \n",
       "ASV_3893          True     True  \n",
       "ASV_24535         True     True  \n",
       "ASV_5228          True     True  \n",
       "\n",
       "[10305 rows x 24 columns]"
      ]
     },
     "execution_count": 106,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## compare results\n",
    "#Genus\n",
    "df['genus_match'] = df.apply(lambda x: str(x['genbank'].Genus) in str(x['MZGdb'].hit_def), axis=1)\n",
    "df.loc[df['genbank']['Genus'].isna(), 'genus_match'] = False\n",
    "df = df.sort_values('genus_match')\n",
    "# Species\n",
    "df['hit_sp_def'] = df['MZGdb']['hit_def'].str.replace('_',' ')\n",
    "df['sp_match'] = df.apply(lambda x: str(x['genbank'].Species) in str(x.hit_sp_def), axis=1)\n",
    "df.loc[df['genbank']['Species'].isna(), 'sp_match'] = False\n",
    "df = df.sort_values('sp_match')\n",
    "\n",
    "# drop unecessary columns\n",
    "df = df.drop(['hit_sp_def'], axis=1)\n",
    "df.to_csv('/Users/kpitz/Documents/test.csv')\n",
    "df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5b9636be-2e69-471a-a5fe-9b437f6510f2",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2bb9a7c9-0067-4c8a-92d3-e1ddab1af07a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2f8c1fe9-3989-41da-a1a4-741349d96d68",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "89f22b8d-817b-46e7-bfc9-26d7f34f9b10",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "377f23cf-945d-4bb0-a28a-6eff0d19215f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "a2c2159d-e14a-4006-8863-83a1dd221085",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hit_number</th>\n",
       "      <th>eval</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>%ID</th>\n",
       "      <th>hit_def</th>\n",
       "      <th>align_len</th>\n",
       "      <th>Domain</th>\n",
       "      <th>Kingdom</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "      <th>genus_match</th>\n",
       "      <th>sp_match</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ASV_1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>226</td>\n",
       "      <td>98.72</td>\n",
       "      <td>AB649198__Emiliania_huxleyi   MZGdb-v2023-m04-...</td>\n",
       "      <td>235</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Haptophyta</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Isochrysidales</td>\n",
       "      <td>Noelaerhabdaceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_2745</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>47</td>\n",
       "      <td>84.27</td>\n",
       "      <td>EU982725__Cadlina_rumia   MZGdb-v2023-m04-01 (...</td>\n",
       "      <td>89</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_3142</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>296</td>\n",
       "      <td>98.08</td>\n",
       "      <td>MW691182__Mesonerilla_roscovita   MZGdb-v2023-...</td>\n",
       "      <td>312</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Metazoa</td>\n",
       "      <td>Chordata</td>\n",
       "      <td>Mammalia</td>\n",
       "      <td>Primates</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Homo</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_3136</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>124</td>\n",
       "      <td>81.03</td>\n",
       "      <td>CFAD137-11__Pagurus_villosus   MZGdb_v2023-m04...</td>\n",
       "      <td>290</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Fungi</td>\n",
       "      <td>Basidiomycota</td>\n",
       "      <td>Agaricomycetes</td>\n",
       "      <td>Agaricales</td>\n",
       "      <td>Pleurotaceae</td>\n",
       "      <td>Pleurotus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_2734</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>155</td>\n",
       "      <td>98.76</td>\n",
       "      <td>HQ646565__Octactis_octonaria   MZGdb-v2023-m04...</td>\n",
       "      <td>161</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Dictyochophyceae</td>\n",
       "      <td>Dictyochales</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Dictyocha</td>\n",
       "      <td>Dictyocha speculum</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_98427</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>133</td>\n",
       "      <td>81.29</td>\n",
       "      <td>MT760752__Phaeocystis_globosa   MZGdb-v2023-m0...</td>\n",
       "      <td>310</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Haptophyta</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Phaeocystales</td>\n",
       "      <td>Phaeocystaceae</td>\n",
       "      <td>Phaeocystis</td>\n",
       "      <td>Phaeocystis globosa</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_27159</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>308</td>\n",
       "      <td>100.0</td>\n",
       "      <td>MN745887__Ctenocalanus_vanus   MZGdb-v2023-m04...</td>\n",
       "      <td>308</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Metazoa</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Hexanauplia</td>\n",
       "      <td>Calanoida</td>\n",
       "      <td>Calanidae</td>\n",
       "      <td>Ctenocalanus</td>\n",
       "      <td>Ctenocalanus vanus</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_98421</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>254</td>\n",
       "      <td>94.16</td>\n",
       "      <td>MT760738__Phaeocystis_globosa   MZGdb-v2023-m0...</td>\n",
       "      <td>308</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Haptophyta</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Phaeocystales</td>\n",
       "      <td>Phaeocystaceae</td>\n",
       "      <td>Phaeocystis</td>\n",
       "      <td>Phaeocystis globosa</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_64966</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>240</td>\n",
       "      <td>94.16</td>\n",
       "      <td>KC967226__Phaeocystis_globosa   MZGdb-v2023-m0...</td>\n",
       "      <td>291</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Haptophyta</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Phaeocystales</td>\n",
       "      <td>Phaeocystaceae</td>\n",
       "      <td>Phaeocystis</td>\n",
       "      <td>Phaeocystis globosa</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_62934</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>244</td>\n",
       "      <td>93.18</td>\n",
       "      <td>KC967226__Phaeocystis_globosa   MZGdb-v2023-m0...</td>\n",
       "      <td>308</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Haptophyta</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Phaeocystales</td>\n",
       "      <td>Phaeocystaceae</td>\n",
       "      <td>Phaeocystis</td>\n",
       "      <td>Phaeocystis globosa</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10098 rows Ã— 16 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           Hit_number eval bitscore    %ID  \\\n",
       "ASV                                          \n",
       "ASV_1             1.0  0.0      226  98.72   \n",
       "ASV_2745          1.0  0.0       47  84.27   \n",
       "ASV_3142          1.0  0.0      296  98.08   \n",
       "ASV_3136          1.0  0.0      124  81.03   \n",
       "ASV_2734          1.0  0.0      155  98.76   \n",
       "...               ...  ...      ...    ...   \n",
       "ASV_98427         1.0  0.0      133  81.29   \n",
       "ASV_27159         1.0  0.0      308  100.0   \n",
       "ASV_98421         1.0  0.0      254  94.16   \n",
       "ASV_64966         1.0  0.0      240  94.16   \n",
       "ASV_62934         1.0  0.0      244  93.18   \n",
       "\n",
       "                                                     hit_def align_len  \\\n",
       "ASV                                                                      \n",
       "ASV_1      AB649198__Emiliania_huxleyi   MZGdb-v2023-m04-...       235   \n",
       "ASV_2745   EU982725__Cadlina_rumia   MZGdb-v2023-m04-01 (...        89   \n",
       "ASV_3142   MW691182__Mesonerilla_roscovita   MZGdb-v2023-...       312   \n",
       "ASV_3136   CFAD137-11__Pagurus_villosus   MZGdb_v2023-m04...       290   \n",
       "ASV_2734   HQ646565__Octactis_octonaria   MZGdb-v2023-m04...       161   \n",
       "...                                                      ...       ...   \n",
       "ASV_98427  MT760752__Phaeocystis_globosa   MZGdb-v2023-m0...       310   \n",
       "ASV_27159  MN745887__Ctenocalanus_vanus   MZGdb-v2023-m04...       308   \n",
       "ASV_98421  MT760738__Phaeocystis_globosa   MZGdb-v2023-m0...       308   \n",
       "ASV_64966  KC967226__Phaeocystis_globosa   MZGdb-v2023-m0...       291   \n",
       "ASV_62934  KC967226__Phaeocystis_globosa   MZGdb-v2023-m0...       308   \n",
       "\n",
       "              Domain  Kingdom         Phylum             Class  \\\n",
       "ASV                                                              \n",
       "ASV_1      Eukaryota      NaN     Haptophyta           unknown   \n",
       "ASV_2745         NaN      NaN            NaN               NaN   \n",
       "ASV_3142   Eukaryota  Metazoa       Chordata          Mammalia   \n",
       "ASV_3136   Eukaryota    Fungi  Basidiomycota    Agaricomycetes   \n",
       "ASV_2734   Eukaryota      NaN        unknown  Dictyochophyceae   \n",
       "...              ...      ...            ...               ...   \n",
       "ASV_98427  Eukaryota      NaN     Haptophyta           unknown   \n",
       "ASV_27159  Eukaryota  Metazoa     Arthropoda       Hexanauplia   \n",
       "ASV_98421  Eukaryota      NaN     Haptophyta           unknown   \n",
       "ASV_64966  Eukaryota      NaN     Haptophyta           unknown   \n",
       "ASV_62934  Eukaryota      NaN     Haptophyta           unknown   \n",
       "\n",
       "                    Order            Family         Genus  \\\n",
       "ASV                                                         \n",
       "ASV_1      Isochrysidales  Noelaerhabdaceae           NaN   \n",
       "ASV_2745              NaN               NaN           NaN   \n",
       "ASV_3142         Primates         Hominidae          Homo   \n",
       "ASV_3136       Agaricales      Pleurotaceae     Pleurotus   \n",
       "ASV_2734     Dictyochales           unknown     Dictyocha   \n",
       "...                   ...               ...           ...   \n",
       "ASV_98427   Phaeocystales    Phaeocystaceae   Phaeocystis   \n",
       "ASV_27159       Calanoida         Calanidae  Ctenocalanus   \n",
       "ASV_98421   Phaeocystales    Phaeocystaceae   Phaeocystis   \n",
       "ASV_64966   Phaeocystales    Phaeocystaceae   Phaeocystis   \n",
       "ASV_62934   Phaeocystales    Phaeocystaceae   Phaeocystis   \n",
       "\n",
       "                       Species  genus_match  sp_match  \n",
       "ASV                                                    \n",
       "ASV_1                      NaN        False     False  \n",
       "ASV_2745                   NaN        False     False  \n",
       "ASV_3142          Homo sapiens        False     False  \n",
       "ASV_3136                   NaN        False     False  \n",
       "ASV_2734    Dictyocha speculum        False     False  \n",
       "...                        ...          ...       ...  \n",
       "ASV_98427  Phaeocystis globosa         True      True  \n",
       "ASV_27159   Ctenocalanus vanus         True      True  \n",
       "ASV_98421  Phaeocystis globosa         True      True  \n",
       "ASV_64966  Phaeocystis globosa         True      True  \n",
       "ASV_62934  Phaeocystis globosa         True      True  \n",
       "\n",
       "[10098 rows x 16 columns]"
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# join with MEGAN:\n",
    "df = pd.concat([MZgb, megan], axis=1)\n",
    "#df['genus_match'] = ''\n",
    "df['genus_match'] = df.apply(lambda x: str(x.Genus) in str(x.hit_def), axis=1)\n",
    "df.loc[df['Genus'].isna(), 'genus_match'] = False\n",
    "df = df.sort_values('genus_match')\n",
    "# Species\n",
    "df['hit_sp_def'] = df['hit_def'].str.replace('_',' ')\n",
    "df['sp_match'] = df.apply(lambda x: str(x.Species) in str(x.hit_sp_def), axis=1)\n",
    "df.loc[df['Species'].isna(), 'sp_match'] = False\n",
    "df = df.sort_values('sp_match')\n",
    "\n",
    "# drop unecessary columns\n",
    "df = df.drop(['taxonomy', 'hit_sp_def'], axis=1)\n",
    "df.to_csv('/Users/kpitz/Documents/test.csv')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "58c2a6e7-d2b3-4bc6-b170-7a50bd22abd5",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0f1b7cd9-b12e-430b-8ace-d59471368d4c",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Export Files\n",
    "# df = pd.concat([meta[['sample_name']], otu_table.T], axis=1, sort=False)\n",
    "#rename columns as sample_names\n",
    "# df.set_index('sample_name', inplace = True)\n",
    "#Join with taxonomy\n",
    "samples = list(df.T)\n",
    "print('Number of Samples in OTU table:', len(samples))\n",
    "df = pd.concat([taxa_tab, df.T], axis=1, sort=False)\n",
    "df.fillna('no_hit', inplace=True)\n",
    "#sort by ASV number - relates to abundance of ASV\n",
    "df['OTU_num'] = df.index.str.split('_').str[-1].astype(int)\n",
    "df=df.sort_values('OTU_num')\n",
    "df.drop('OTU_num', inplace=True, axis=1)\n",
    "#Make index names\n",
    "df.index.name='ASV'\n",
    "#export 'OTU_taxa_table_all.csv'\n",
    "df.to_csv(directory+'ASV_taxa_table_all.csv')\n",
    "#export taxa table\n",
    "taxa_tab_all = df[levels].copy()\n",
    "taxa_tab_all.index.name='#ASV'  #Biom conversion needs # in index header\n",
    "taxa_tab_all.to_csv(directory+'Taxa_table.tsv', sep='\\t')\n",
    "#export otu table\n",
    "otu_tab_all = df[samples].copy()\n",
    "otu_tab_all.to_csv(directory+'ASV_table.tsv', sep='\\t')\n",
    "#Print some basic stats of total reads and ASVs\n",
    "df['Total_Reads'] = df[samples].sum(axis=1)\n",
    "df['Total_ASVs']=1\n",
    "df=df.groupby(['Kingdom','Phylum']).sum()\n",
    "print('XXXXXX Summary of Total Reads and ASVs by Kingdom and Phylum')\n",
    "print(df[['Total_Reads','Total_ASVs']])\n",
    "print('XXXXXX End Summary')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python scipy_test",
   "language": "python",
   "name": "scipy_test"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
