{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "0ffb7cd6-d661-422b-8742-6a7a4c7f96ad",
   "metadata": {},
   "source": [
    "# Merge Metadata\n",
    "\n",
    "- pull from downloaded google sheets of FAIRe terms"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ad8556e4-6d01-4e2b-a7c8-49ac3474fb21",
   "metadata": {},
   "source": [
    "### Import"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "d5caa7f6-1c19-4acb-b19a-6f3487c18e2d",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import glob"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8e8a2643-c755-4e19-bec0-311dcae46a4d",
   "metadata": {},
   "source": [
    "### Functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "ac3a5f40-ea2f-443c-aab8-7f731b7b916d",
   "metadata": {},
   "outputs": [],
   "source": [
    "def water_amplicon_merge(water_df, amplicon_df):\n",
    "    df = water_df.set_index('sample_name')\n",
    "    df = df.drop(['sheet'], axis=1)\n",
    "    # add amplicon metadata to water sample metadata (some water samples don't have amplicon data if not sequenced in institute)\n",
    "    df = df.join(amplicon_df.set_index('sample_name'), rsuffix='_amplicon_sheet')\n",
    "    df = df.drop(['sheet'], axis=1)\n",
    "    return df\n",
    "def study_data_merge(study_meta, bioinf_meta, institute_name):\n",
    "    df = pd.concat([study_meta, bioinf_meta], axis=1)\n",
    "    df = df.drop(['sheet', 'date_modified'], axis=1)\n",
    "    df = df.T\n",
    "    df = df.rename(columns={0:institute_name})\n",
    "    return df\n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8d9f11d4-f7aa-42e7-a2cf-dec8d8b98c2c",
   "metadata": {},
   "source": [
    "### Metadata Directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "31aa8c3c-1550-42f6-88d7-563114560fd9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['/Users/kpitz/github/GLOMICON/intercomparison/metadata/SBR_NOAA_MIMARKS.survey.water.6.0_sharing.xlsx', '/Users/kpitz/github/GLOMICON/intercomparison/metadata/~$AWI_NOAA_MIMARKS.survey.water.6.0_sharing.xlsx', '/Users/kpitz/github/GLOMICON/intercomparison/metadata/MBARI_NOAA_MIMARKS.survey.water.6.0_sharing.xlsx', '/Users/kpitz/github/GLOMICON/intercomparison/metadata/AWI_NOAA_MIMARKS.survey.water.6.0_sharing.xlsx', '/Users/kpitz/github/GLOMICON/intercomparison/metadata/NOAA_NOAA_MIMARKS.survey.water.6.0_sharing.xlsx']\n"
     ]
    }
   ],
   "source": [
    "meta_dir = '/Users/kpitz/github/GLOMICON/intercomparison/metadata/'\n",
    "files = glob.glob(meta_dir + '*.xlsx')\n",
    "print(files)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ea5d2c10-f45d-40dc-ad88-11e94c9c7712",
   "metadata": {},
   "source": [
    "## Import and Merge Metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "29a23a9c-b7ad-42d6-9b3b-89324169713b",
   "metadata": {},
   "outputs": [],
   "source": [
    "study_dfs = []\n",
    "samp_dfs = []\n",
    "institutes = []"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e0b5155d-b97e-4451-a3be-e7781f126b0e",
   "metadata": {},
   "source": [
    "### MBARI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 262,
   "id": "d7f98e04-8ccc-4231-b4d8-d0866774a513",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/kpitz/github/GLOMICON/intercomparison/metadata/MBARI_NOAA_MIMARKS.survey.water.6.0_sharing.xlsx\n",
      "Sheets in excel file: ['README', 'study_data', 'water_sample_data', 'amplicon_prep_data', 'metag_prep_data', 'analysis_data', 'SRA_template', 'measurement_metadata', 'SRA Terms', 'Version log']\n",
      "study_data 5\n",
      "water_sample_data 8\n",
      "amplicon_prep_data 6\n",
      "analysis_data 6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/kpitz/anaconda3/envs/scipy_test/lib/python3.12/site-packages/openpyxl/worksheet/_read_only.py:85: UserWarning: Data Validation extension is not supported and will be removed\n",
      "  for idx, row in parser.parse():\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>amplicon_sequenced</th>\n",
       "      <th>trim_method</th>\n",
       "      <th>cluster_method</th>\n",
       "      <th>pid_clustering</th>\n",
       "      <th>taxa_class_method</th>\n",
       "      <th>taxa_ref_db</th>\n",
       "      <th>code_repo</th>\n",
       "      <th>date_modified</th>\n",
       "      <th>sheet</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>18S V9</td>\n",
       "      <td>atropos</td>\n",
       "      <td>dada2</td>\n",
       "      <td>ASV</td>\n",
       "      <td>BLAST; MEGAN LCA</td>\n",
       "      <td>genbank NR</td>\n",
       "      <td>https://github.com/MBARI-BOG/BOG-Banzai-Dada2-...</td>\n",
       "      <td>2024-08-15 08:44:22.928</td>\n",
       "      <td>analysis_data</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  amplicon_sequenced trim_method cluster_method pid_clustering  \\\n",
       "0             18S V9     atropos          dada2            ASV   \n",
       "\n",
       "  taxa_class_method taxa_ref_db  \\\n",
       "0  BLAST; MEGAN LCA  genbank NR   \n",
       "\n",
       "                                           code_repo           date_modified  \\\n",
       "0  https://github.com/MBARI-BOG/BOG-Banzai-Dada2-... 2024-08-15 08:44:22.928   \n",
       "\n",
       "           sheet  \n",
       "0  analysis_data  "
      ]
     },
     "execution_count": 262,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# import\n",
    "#file = files[1]\n",
    "file = '/Users/kpitz/github/GLOMICON/intercomparison/metadata/MBARI_NOAA_MIMARKS.survey.water.6.0_sharing.xlsx'\n",
    "print(file)\n",
    "inst_name = 'MBARI'\n",
    "institutes.append(inst_name)\n",
    "xl = pd.ExcelFile(file)\n",
    "print('Sheets in excel file:',xl.sheet_names)  # see all sheet names\n",
    "#sheets = xl.sheet_names\n",
    "sheets = ['study_data', 'water_sample_data', 'amplicon_prep_data', 'analysis_data']\n",
    "dfs= []\n",
    "skiprow_num = [5,8,6,6]\n",
    "for sheet, sr in zip(sheets, skiprow_num):\n",
    "    print(sheet,sr)\n",
    "    df = xl.parse(sheet,skiprows=sr)\n",
    "    df = df.dropna(how='all', axis=1)\n",
    "    df = df.dropna(how='all', axis=0)\n",
    "    df['sheet'] = sheet\n",
    "    dfs.append(df)\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 263,
   "id": "07dfef98-3ad2-4a86-a80f-89635a43554d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# get columns in water sample metadata\n",
    "water_samp_cols = list(dfs[1])\n",
    "amplicon_samp_cols = list(dfs[2])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "edc0096d-78a0-48ac-81d1-7136017b9fe2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MBARI</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>project_id</th>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>project_name</th>\n",
       "      <td>Intercomparison of filtered seawater samples a...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>project_id_external</th>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>project_contact</th>\n",
       "      <td>Kathleen Pitz, kpitz@mbari.org, https://orcid....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>project_description</th>\n",
       "      <td>Replicate filters, originating from surface wa...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                 MBARI\n",
       "project_id                                          GLOMICON_INTERCOMP\n",
       "project_name         Intercomparison of filtered seawater samples a...\n",
       "project_id_external                                 GLOMICON_INTERCOMP\n",
       "project_contact      Kathleen Pitz, kpitz@mbari.org, https://orcid....\n",
       "project_description  Replicate filters, originating from surface wa..."
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = study_data_merge(dfs[0], dfs[3], inst_name)\n",
    "study_dfs.append(df)\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "38323198-029f-447d-b2fe-a4d9450affdb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Collecting Institute</th>\n",
       "      <th>Analyzing Institute</th>\n",
       "      <th>cruise_id</th>\n",
       "      <th>station</th>\n",
       "      <th>ctd_bottle_no</th>\n",
       "      <th>sample_replicate</th>\n",
       "      <th>source_material_id</th>\n",
       "      <th>extract_number</th>\n",
       "      <th>sample_title</th>\n",
       "      <th>project_id</th>\n",
       "      <th>...</th>\n",
       "      <th>target_gene</th>\n",
       "      <th>target_subfragment</th>\n",
       "      <th>pcr_primer_forward</th>\n",
       "      <th>pcr_primer_reverse</th>\n",
       "      <th>pcr_primer_name_forward</th>\n",
       "      <th>pcr_primer_name_reverse</th>\n",
       "      <th>pcr_primer_reference</th>\n",
       "      <th>pcr_cond</th>\n",
       "      <th>date_modified_amplicon_sheet</th>\n",
       "      <th>modified_by_amplicon_sheet</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sample_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CN18Sc37_12_Rep_Stdy5_AO</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy5</td>\n",
       "      <td>AO9</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>GLOMICON</td>\n",
       "      <td>...</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V9</td>\n",
       "      <td>GTACACACCGCCCGTC</td>\n",
       "      <td>TGATCCTTCTGCAGGTTCACCTAC</td>\n",
       "      <td>Euk_1391f</td>\n",
       "      <td>EukBr</td>\n",
       "      <td>Amaral-Zettler et al. 2009, Stoeck et al. 2010...</td>\n",
       "      <td>dx.doi.org/10.17504/protocols.io.36wgq3d8olk5/v2</td>\n",
       "      <td>2024-08-15 08:42:28.469</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN18Sc37_12_Rep_Stdy10_AO</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy10</td>\n",
       "      <td>AO10</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>GLOMICON</td>\n",
       "      <td>...</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V9</td>\n",
       "      <td>GTACACACCGCCCGTC</td>\n",
       "      <td>TGATCCTTCTGCAGGTTCACCTAC</td>\n",
       "      <td>Euk_1391f</td>\n",
       "      <td>EukBr</td>\n",
       "      <td>Amaral-Zettler et al. 2009, Stoeck et al. 2010...</td>\n",
       "      <td>dx.doi.org/10.17504/protocols.io.36wgq3d8olk5/v2</td>\n",
       "      <td>2024-06-18 19:31:14.358</td>\n",
       "      <td>kpitz@mbari.org</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN18Sc37_12_Rep_Stdy14_AO</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy14</td>\n",
       "      <td>AO11</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>GLOMICON</td>\n",
       "      <td>...</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V9</td>\n",
       "      <td>GTACACACCGCCCGTC</td>\n",
       "      <td>TGATCCTTCTGCAGGTTCACCTAC</td>\n",
       "      <td>Euk_1391f</td>\n",
       "      <td>EukBr</td>\n",
       "      <td>Amaral-Zettler et al. 2009, Stoeck et al. 2010...</td>\n",
       "      <td>dx.doi.org/10.17504/protocols.io.36wgq3d8olk5/v2</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN18Sc37_12_Rep_Stdy18_AO</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy18</td>\n",
       "      <td>AO12</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>GLOMICON</td>\n",
       "      <td>...</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V9</td>\n",
       "      <td>GTACACACCGCCCGTC</td>\n",
       "      <td>TGATCCTTCTGCAGGTTCACCTAC</td>\n",
       "      <td>Euk_1391f</td>\n",
       "      <td>EukBr</td>\n",
       "      <td>Amaral-Zettler et al. 2009, Stoeck et al. 2010...</td>\n",
       "      <td>dx.doi.org/10.17504/protocols.io.36wgq3d8olk5/v2</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN18Sc37_12_Rep_Stdy22_AO</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy22</td>\n",
       "      <td>AO13</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>GLOMICON</td>\n",
       "      <td>...</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V9</td>\n",
       "      <td>GTACACACCGCCCGTC</td>\n",
       "      <td>TGATCCTTCTGCAGGTTCACCTAC</td>\n",
       "      <td>Euk_1391f</td>\n",
       "      <td>EukBr</td>\n",
       "      <td>Amaral-Zettler et al. 2009, Stoeck et al. 2010...</td>\n",
       "      <td>dx.doi.org/10.17504/protocols.io.36wgq3d8olk5/v2</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 62 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                          Collecting Institute Analyzing Institute cruise_id  \\\n",
       "sample_name                                                                    \n",
       "CN18Sc37_12_Rep_Stdy5_AO                 MBARI               MBARI     CN18S   \n",
       "CN18Sc37_12_Rep_Stdy10_AO                MBARI               MBARI     CN18S   \n",
       "CN18Sc37_12_Rep_Stdy14_AO                MBARI               MBARI     CN18S   \n",
       "CN18Sc37_12_Rep_Stdy18_AO                MBARI               MBARI     CN18S   \n",
       "CN18Sc37_12_Rep_Stdy22_AO                MBARI               MBARI     CN18S   \n",
       "\n",
       "                          station  ctd_bottle_no  sample_replicate  \\\n",
       "sample_name                                                          \n",
       "CN18Sc37_12_Rep_Stdy5_AO       TS           12.0               5.0   \n",
       "CN18Sc37_12_Rep_Stdy10_AO      TS           12.0              10.0   \n",
       "CN18Sc37_12_Rep_Stdy14_AO      TS           12.0              14.0   \n",
       "CN18Sc37_12_Rep_Stdy18_AO      TS           12.0              18.0   \n",
       "CN18Sc37_12_Rep_Stdy22_AO      TS           12.0              22.0   \n",
       "\n",
       "                               source_material_id extract_number  \\\n",
       "sample_name                                                        \n",
       "CN18Sc37_12_Rep_Stdy5_AO    CN18Sc37_12_Rep_Stdy5            AO9   \n",
       "CN18Sc37_12_Rep_Stdy10_AO  CN18Sc37_12_Rep_Stdy10           AO10   \n",
       "CN18Sc37_12_Rep_Stdy14_AO  CN18Sc37_12_Rep_Stdy14           AO11   \n",
       "CN18Sc37_12_Rep_Stdy18_AO  CN18Sc37_12_Rep_Stdy18           AO12   \n",
       "CN18Sc37_12_Rep_Stdy22_AO  CN18Sc37_12_Rep_Stdy22           AO13   \n",
       "\n",
       "                                                      sample_title project_id  \\\n",
       "sample_name                                                                     \n",
       "CN18Sc37_12_Rep_Stdy5_AO   Seawater sample Monterey Bay (GLOMICON)   GLOMICON   \n",
       "CN18Sc37_12_Rep_Stdy10_AO  Seawater sample Monterey Bay (GLOMICON)   GLOMICON   \n",
       "CN18Sc37_12_Rep_Stdy14_AO  Seawater sample Monterey Bay (GLOMICON)   GLOMICON   \n",
       "CN18Sc37_12_Rep_Stdy18_AO  Seawater sample Monterey Bay (GLOMICON)   GLOMICON   \n",
       "CN18Sc37_12_Rep_Stdy22_AO  Seawater sample Monterey Bay (GLOMICON)   GLOMICON   \n",
       "\n",
       "                           ... target_gene target_subfragment  \\\n",
       "sample_name                ...                                  \n",
       "CN18Sc37_12_Rep_Stdy5_AO   ...    18S rRNA                 V9   \n",
       "CN18Sc37_12_Rep_Stdy10_AO  ...    18S rRNA                 V9   \n",
       "CN18Sc37_12_Rep_Stdy14_AO  ...    18S rRNA                 V9   \n",
       "CN18Sc37_12_Rep_Stdy18_AO  ...    18S rRNA                 V9   \n",
       "CN18Sc37_12_Rep_Stdy22_AO  ...    18S rRNA                 V9   \n",
       "\n",
       "                          pcr_primer_forward        pcr_primer_reverse  \\\n",
       "sample_name                                                              \n",
       "CN18Sc37_12_Rep_Stdy5_AO    GTACACACCGCCCGTC  TGATCCTTCTGCAGGTTCACCTAC   \n",
       "CN18Sc37_12_Rep_Stdy10_AO   GTACACACCGCCCGTC  TGATCCTTCTGCAGGTTCACCTAC   \n",
       "CN18Sc37_12_Rep_Stdy14_AO   GTACACACCGCCCGTC  TGATCCTTCTGCAGGTTCACCTAC   \n",
       "CN18Sc37_12_Rep_Stdy18_AO   GTACACACCGCCCGTC  TGATCCTTCTGCAGGTTCACCTAC   \n",
       "CN18Sc37_12_Rep_Stdy22_AO   GTACACACCGCCCGTC  TGATCCTTCTGCAGGTTCACCTAC   \n",
       "\n",
       "                          pcr_primer_name_forward pcr_primer_name_reverse  \\\n",
       "sample_name                                                                 \n",
       "CN18Sc37_12_Rep_Stdy5_AO                Euk_1391f                   EukBr   \n",
       "CN18Sc37_12_Rep_Stdy10_AO               Euk_1391f                   EukBr   \n",
       "CN18Sc37_12_Rep_Stdy14_AO               Euk_1391f                   EukBr   \n",
       "CN18Sc37_12_Rep_Stdy18_AO               Euk_1391f                   EukBr   \n",
       "CN18Sc37_12_Rep_Stdy22_AO               Euk_1391f                   EukBr   \n",
       "\n",
       "                                                        pcr_primer_reference  \\\n",
       "sample_name                                                                    \n",
       "CN18Sc37_12_Rep_Stdy5_AO   Amaral-Zettler et al. 2009, Stoeck et al. 2010...   \n",
       "CN18Sc37_12_Rep_Stdy10_AO  Amaral-Zettler et al. 2009, Stoeck et al. 2010...   \n",
       "CN18Sc37_12_Rep_Stdy14_AO  Amaral-Zettler et al. 2009, Stoeck et al. 2010...   \n",
       "CN18Sc37_12_Rep_Stdy18_AO  Amaral-Zettler et al. 2009, Stoeck et al. 2010...   \n",
       "CN18Sc37_12_Rep_Stdy22_AO  Amaral-Zettler et al. 2009, Stoeck et al. 2010...   \n",
       "\n",
       "                                                                   pcr_cond  \\\n",
       "sample_name                                                                   \n",
       "CN18Sc37_12_Rep_Stdy5_AO   dx.doi.org/10.17504/protocols.io.36wgq3d8olk5/v2   \n",
       "CN18Sc37_12_Rep_Stdy10_AO  dx.doi.org/10.17504/protocols.io.36wgq3d8olk5/v2   \n",
       "CN18Sc37_12_Rep_Stdy14_AO  dx.doi.org/10.17504/protocols.io.36wgq3d8olk5/v2   \n",
       "CN18Sc37_12_Rep_Stdy18_AO  dx.doi.org/10.17504/protocols.io.36wgq3d8olk5/v2   \n",
       "CN18Sc37_12_Rep_Stdy22_AO  dx.doi.org/10.17504/protocols.io.36wgq3d8olk5/v2   \n",
       "\n",
       "                          date_modified_amplicon_sheet  \\\n",
       "sample_name                                              \n",
       "CN18Sc37_12_Rep_Stdy5_AO       2024-08-15 08:42:28.469   \n",
       "CN18Sc37_12_Rep_Stdy10_AO      2024-06-18 19:31:14.358   \n",
       "CN18Sc37_12_Rep_Stdy14_AO                          NaT   \n",
       "CN18Sc37_12_Rep_Stdy18_AO                          NaT   \n",
       "CN18Sc37_12_Rep_Stdy22_AO                          NaT   \n",
       "\n",
       "                          modified_by_amplicon_sheet  \n",
       "sample_name                                           \n",
       "CN18Sc37_12_Rep_Stdy5_AO                         NaN  \n",
       "CN18Sc37_12_Rep_Stdy10_AO            kpitz@mbari.org  \n",
       "CN18Sc37_12_Rep_Stdy14_AO                        NaN  \n",
       "CN18Sc37_12_Rep_Stdy18_AO                        NaN  \n",
       "CN18Sc37_12_Rep_Stdy22_AO                        NaN  \n",
       "\n",
       "[5 rows x 62 columns]"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = water_amplicon_merge(dfs[1], dfs[2])\n",
    "samp_dfs.append(df)\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "20ca2aab-b625-4f7a-8204-1146ab291e1f",
   "metadata": {},
   "source": [
    "### AOML"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "291f0d48-d05e-4450-9be2-cae2a5aac9c2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/kpitz/github/GLOMICON/intercomparison/metadata/NOAA_NOAA_MIMARKS.survey.water.6.0_sharing.xlsx\n",
      "Sheets in excel file: ['README', 'study_data', 'water_sample_data', 'amplicon_prep_data', 'metag_prep_data', 'analysis_data', 'SRA_template', 'measurement_metadata', 'SRA Terms', 'Version log']\n",
      "study_data 5\n",
      "water_sample_data 8\n",
      "amplicon_prep_data 6\n",
      "analysis_data 6\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>amplicon_sequenced</th>\n",
       "      <th>ampliconSize</th>\n",
       "      <th>trim_method</th>\n",
       "      <th>cluster_method</th>\n",
       "      <th>pid_clustering</th>\n",
       "      <th>taxa_class_method</th>\n",
       "      <th>taxa_ref_db</th>\n",
       "      <th>code_repo</th>\n",
       "      <th>identificationReferences</th>\n",
       "      <th>controls_used</th>\n",
       "      <th>date_modified</th>\n",
       "      <th>modified_by</th>\n",
       "      <th>sheet</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>16S V4-V5</td>\n",
       "      <td>411</td>\n",
       "      <td>cutadapt</td>\n",
       "      <td>Tourmaline; qiime2-2021.2; dada2</td>\n",
       "      <td>ASV</td>\n",
       "      <td>Tourmaline; qiime2-2021.2; naive-bayes classifier</td>\n",
       "      <td>Silva SSU Ref NR 99 v138.1; 515f-926r region; ...</td>\n",
       "      <td>https://github.com/aomlomics/gomecc</td>\n",
       "      <td>10.5281/zenodo.8392695 | https://github.com/ao...</td>\n",
       "      <td>12 distilled water blanks | 2 PCR no-template ...</td>\n",
       "      <td>2024-04-11 07:12:09.354</td>\n",
       "      <td>aomlomics@gmail.com</td>\n",
       "      <td>analysis_data</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>18S V9</td>\n",
       "      <td>260</td>\n",
       "      <td>cutadapt</td>\n",
       "      <td>Tourmaline; qiime2-2021.2; dada2</td>\n",
       "      <td>ASV</td>\n",
       "      <td>Tourmaline; qiime2-2021.2; naive-bayes classifier</td>\n",
       "      <td>PR2 v5.0.1; V9 1391f-1510r region; 10.5281/zen...</td>\n",
       "      <td>https://github.com/aomlomics/gomecc</td>\n",
       "      <td>10.5281/zenodo.8392706 | https://pr2-database....</td>\n",
       "      <td>12 distilled water blanks | 2 PCR no-template ...</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>analysis_data</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  amplicon_sequenced  ampliconSize trim_method  \\\n",
       "0          16S V4-V5           411    cutadapt   \n",
       "1             18S V9           260    cutadapt   \n",
       "\n",
       "                     cluster_method pid_clustering  \\\n",
       "0  Tourmaline; qiime2-2021.2; dada2            ASV   \n",
       "1  Tourmaline; qiime2-2021.2; dada2            ASV   \n",
       "\n",
       "                                   taxa_class_method  \\\n",
       "0  Tourmaline; qiime2-2021.2; naive-bayes classifier   \n",
       "1  Tourmaline; qiime2-2021.2; naive-bayes classifier   \n",
       "\n",
       "                                         taxa_ref_db  \\\n",
       "0  Silva SSU Ref NR 99 v138.1; 515f-926r region; ...   \n",
       "1  PR2 v5.0.1; V9 1391f-1510r region; 10.5281/zen...   \n",
       "\n",
       "                             code_repo  \\\n",
       "0  https://github.com/aomlomics/gomecc   \n",
       "1  https://github.com/aomlomics/gomecc   \n",
       "\n",
       "                            identificationReferences  \\\n",
       "0  10.5281/zenodo.8392695 | https://github.com/ao...   \n",
       "1  10.5281/zenodo.8392706 | https://pr2-database....   \n",
       "\n",
       "                                       controls_used           date_modified  \\\n",
       "0  12 distilled water blanks | 2 PCR no-template ... 2024-04-11 07:12:09.354   \n",
       "1  12 distilled water blanks | 2 PCR no-template ...                     NaT   \n",
       "\n",
       "           modified_by          sheet  \n",
       "0  aomlomics@gmail.com  analysis_data  \n",
       "1                  NaN  analysis_data  "
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# import\n",
    "file = '/Users/kpitz/github/GLOMICON/intercomparison/metadata/NOAA_NOAA_MIMARKS.survey.water.6.0_sharing.xlsx'\n",
    "print(file)\n",
    "inst_name = 'AOML'\n",
    "institutes.append(inst_name)\n",
    "xl = pd.ExcelFile(file)\n",
    "print('Sheets in excel file:',xl.sheet_names)  # see all sheet names\n",
    "#sheets = xl.sheet_names\n",
    "sheets = ['study_data', 'water_sample_data', 'amplicon_prep_data', 'analysis_data']\n",
    "dfs= []\n",
    "skiprow_num = [5,8,6,6]\n",
    "for sheet, sr in zip(sheets, skiprow_num):\n",
    "    print(sheet,sr)\n",
    "    df = xl.parse(sheet,skiprows=sr)\n",
    "    df = df.dropna(how='all', axis=1)\n",
    "    df = df.dropna(how='all', axis=0)\n",
    "    df['sheet'] = sheet\n",
    "    dfs.append(df)\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "40e6e233-5743-4e65-910b-992222500a96",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AOML</th>\n",
       "      <th>AOML_2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>project_id</th>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>project_name</th>\n",
       "      <td>Intercomparison of filtered seawater samples a...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>project_id_external</th>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>project_contact</th>\n",
       "      <td>Kathleen Pitz, kpitz@mbari.org, https://orcid....</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>type</th>\n",
       "      <td>Occurrence</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>license</th>\n",
       "      <td>CC-BY 4.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>external_links</th>\n",
       "      <td>https://github.com/GLOMICON</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>amplicon_sequenced</th>\n",
       "      <td>16S V4-V5</td>\n",
       "      <td>18S V9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ampliconSize</th>\n",
       "      <td>411</td>\n",
       "      <td>260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>trim_method</th>\n",
       "      <td>cutadapt</td>\n",
       "      <td>cutadapt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cluster_method</th>\n",
       "      <td>Tourmaline; qiime2-2021.2; dada2</td>\n",
       "      <td>Tourmaline; qiime2-2021.2; dada2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pid_clustering</th>\n",
       "      <td>ASV</td>\n",
       "      <td>ASV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taxa_class_method</th>\n",
       "      <td>Tourmaline; qiime2-2021.2; naive-bayes classifier</td>\n",
       "      <td>Tourmaline; qiime2-2021.2; naive-bayes classifier</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taxa_ref_db</th>\n",
       "      <td>Silva SSU Ref NR 99 v138.1; 515f-926r region; ...</td>\n",
       "      <td>PR2 v5.0.1; V9 1391f-1510r region; 10.5281/zen...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>code_repo</th>\n",
       "      <td>https://github.com/aomlomics/gomecc</td>\n",
       "      <td>https://github.com/aomlomics/gomecc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>identificationReferences</th>\n",
       "      <td>10.5281/zenodo.8392695 | https://github.com/ao...</td>\n",
       "      <td>10.5281/zenodo.8392706 | https://pr2-database....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>controls_used</th>\n",
       "      <td>12 distilled water blanks | 2 PCR no-template ...</td>\n",
       "      <td>12 distilled water blanks | 2 PCR no-template ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>modified_by</th>\n",
       "      <td>aomlomics@gmail.com</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                       AOML  \\\n",
       "project_id                                               GLOMICON_INTERCOMP   \n",
       "project_name              Intercomparison of filtered seawater samples a...   \n",
       "project_id_external                                      GLOMICON_INTERCOMP   \n",
       "project_contact           Kathleen Pitz, kpitz@mbari.org, https://orcid....   \n",
       "type                                                             Occurrence   \n",
       "license                                                           CC-BY 4.0   \n",
       "external_links                                  https://github.com/GLOMICON   \n",
       "amplicon_sequenced                                                16S V4-V5   \n",
       "ampliconSize                                                            411   \n",
       "trim_method                                                        cutadapt   \n",
       "cluster_method                             Tourmaline; qiime2-2021.2; dada2   \n",
       "pid_clustering                                                          ASV   \n",
       "taxa_class_method         Tourmaline; qiime2-2021.2; naive-bayes classifier   \n",
       "taxa_ref_db               Silva SSU Ref NR 99 v138.1; 515f-926r region; ...   \n",
       "code_repo                               https://github.com/aomlomics/gomecc   \n",
       "identificationReferences  10.5281/zenodo.8392695 | https://github.com/ao...   \n",
       "controls_used             12 distilled water blanks | 2 PCR no-template ...   \n",
       "modified_by                                             aomlomics@gmail.com   \n",
       "\n",
       "                                                                     AOML_2  \n",
       "project_id                                                              NaN  \n",
       "project_name                                                            NaN  \n",
       "project_id_external                                                     NaN  \n",
       "project_contact                                                         NaN  \n",
       "type                                                                    NaN  \n",
       "license                                                                 NaN  \n",
       "external_links                                                          NaN  \n",
       "amplicon_sequenced                                                   18S V9  \n",
       "ampliconSize                                                            260  \n",
       "trim_method                                                        cutadapt  \n",
       "cluster_method                             Tourmaline; qiime2-2021.2; dada2  \n",
       "pid_clustering                                                          ASV  \n",
       "taxa_class_method         Tourmaline; qiime2-2021.2; naive-bayes classifier  \n",
       "taxa_ref_db               PR2 v5.0.1; V9 1391f-1510r region; 10.5281/zen...  \n",
       "code_repo                               https://github.com/aomlomics/gomecc  \n",
       "identificationReferences  10.5281/zenodo.8392706 | https://pr2-database....  \n",
       "controls_used             12 distilled water blanks | 2 PCR no-template ...  \n",
       "modified_by                                                             NaN  "
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = study_data_merge(dfs[0], dfs[3], inst_name)\n",
    "df = df.rename(columns={1:'AOML_2'})\n",
    "study_dfs.append(df)\n",
    "df.head(n=50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "bab3a57a-7cca-4881-821c-aa58badc2c84",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Collecting Institute</th>\n",
       "      <th>Analyzing Institute</th>\n",
       "      <th>description</th>\n",
       "      <th>date_modified</th>\n",
       "      <th>collecting_institute</th>\n",
       "      <th>analyzing_Institute</th>\n",
       "      <th>library_id</th>\n",
       "      <th>library_strategy</th>\n",
       "      <th>library_source</th>\n",
       "      <th>library_selection</th>\n",
       "      <th>lib_layout</th>\n",
       "      <th>platform</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>filetype</th>\n",
       "      <th>filename</th>\n",
       "      <th>filename2</th>\n",
       "      <th>sequencing_run_folder</th>\n",
       "      <th>pcr_primer_name_reverse</th>\n",
       "      <th>date_modified_amplicon_sheet</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sample_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GLOMICON_AWI_12</th>\n",
       "      <td>AWI</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>2024-09-26 09:18:08.071</td>\n",
       "      <td>2024-09-26 09:18:37.306</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>GLOMICON_AWI_12</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>paired</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>fastq</td>\n",
       "      <td>GLOMICON_AWI_12_S69_L001_R1_001.fastq.gz</td>\n",
       "      <td>GLOMICON_AWI_12_S69_L001_R2_001.fastq.gz</td>\n",
       "      <td>20240401_18S-Amplicon_PE150</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2024-09-12 20:02:53.389</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GLOMICON_AWI_16</th>\n",
       "      <td>AWI</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaT</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>GLOMICON_AWI_16</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>paired</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>fastq</td>\n",
       "      <td>GLOMICON_AWI_16_S68_L001_R1_001.fastq.gz</td>\n",
       "      <td>GLOMICON_AWI_16_S68_L001_R2_001.fastq.gz</td>\n",
       "      <td>20240401_18S-Amplicon_PE150</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GLOMICON_AWI_20</th>\n",
       "      <td>AWI</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaT</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>GLOMICON_AWI_20</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>paired</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>fastq</td>\n",
       "      <td>GLOMICON_AWI_20_S67_L001_R1_001.fastq.gz</td>\n",
       "      <td>GLOMICON_AWI_20_S67_L001_R2_001.fastq.gz</td>\n",
       "      <td>20240401_18S-Amplicon_PE150</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GLOMICON_AWI_4</th>\n",
       "      <td>AWI</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaT</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>GLOMICON_AWI_4</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>paired</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>fastq</td>\n",
       "      <td>GLOMICON_AWI_4_S71_L001_R1_001.fastq.gz</td>\n",
       "      <td>GLOMICON_AWI_4_S71_L001_R2_001.fastq.gz</td>\n",
       "      <td>20240401_18S-Amplicon_PE150</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GLOMICON_AWI_8</th>\n",
       "      <td>AWI</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaT</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>GLOMICON_AWI_8</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>paired</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>fastq</td>\n",
       "      <td>GLOMICON_AWI_8_S70_L001_R1_001.fastq.gz</td>\n",
       "      <td>GLOMICON_AWI_8_S70_L001_R2_001.fastq.gz</td>\n",
       "      <td>20240401_18S-Amplicon_PE150</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GLOMICON_BloomMock_1</th>\n",
       "      <td>MOCKBLOOM</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaT</td>\n",
       "      <td>MOCKBLOOM</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>GLOMICON_BloomMock_1</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>paired</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>fastq</td>\n",
       "      <td>GLOMICON_BloomMock_1_S77_L001_R1_001.fastq.gz</td>\n",
       "      <td>GLOMICON_BloomMock_1_S77_L001_R2_001.fastq.gz</td>\n",
       "      <td>20240401_18S-Amplicon_PE150</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GLOMICON_BloomMock_10</th>\n",
       "      <td>MOCKBLOOM</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaT</td>\n",
       "      <td>MOCKBLOOM</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>GLOMICON_BloomMock_10</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>paired</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>fastq</td>\n",
       "      <td>GLOMICON_BloomMock_10_S80_L001_R1_001.fastq.gz</td>\n",
       "      <td>GLOMICON_BloomMock_10_S80_L001_R2_001.fastq.gz</td>\n",
       "      <td>20240401_18S-Amplicon_PE150</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GLOMICON_BloomMock_30</th>\n",
       "      <td>MOCKBLOOM</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaT</td>\n",
       "      <td>MOCKBLOOM</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>GLOMICON_BloomMock_30</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>paired</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>fastq</td>\n",
       "      <td>GLOMICON_BloomMock_30_S81_L001_R1_001.fastq.gz</td>\n",
       "      <td>GLOMICON_BloomMock_30_S81_L001_R2_001.fastq.gz</td>\n",
       "      <td>20240401_18S-Amplicon_PE150</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GLOMICON_BloomMock_4</th>\n",
       "      <td>MOCKBLOOM</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaT</td>\n",
       "      <td>MOCKBLOOM</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>GLOMICON_BloomMock_4</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>paired</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>fastq</td>\n",
       "      <td>GLOMICON_BloomMock_4_S78_L001_R1_001.fastq.gz</td>\n",
       "      <td>GLOMICON_BloomMock_4_S78_L001_R2_001.fastq.gz</td>\n",
       "      <td>20240401_18S-Amplicon_PE150</td>\n",
       "      <td>x</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GLOMICON_BloomMock_6</th>\n",
       "      <td>MOCKBLOOM</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaT</td>\n",
       "      <td>MOCKBLOOM</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>GLOMICON_BloomMock_6</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>paired</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>fastq</td>\n",
       "      <td>GLOMICON_BloomMock_6_S79_L001_R1_001.fastq.gz</td>\n",
       "      <td>GLOMICON_BloomMock_6_S79_L001_R2_001.fastq.gz</td>\n",
       "      <td>20240401_18S-Amplicon_PE150</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GLOMICON_EvenMock_1</th>\n",
       "      <td>MOCKEVEN</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaT</td>\n",
       "      <td>MOCKEVEN</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>GLOMICON_EvenMock_1</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>paired</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>fastq</td>\n",
       "      <td>GLOMICON_EvenMock_1_S82_L001_R1_001.fastq.gz</td>\n",
       "      <td>GLOMICON_EvenMock_1_S82_L001_R2_001.fastq.gz</td>\n",
       "      <td>20240401_18S-Amplicon_PE150</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GLOMICON_EvenMock_4</th>\n",
       "      <td>MOCKEVEN</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaT</td>\n",
       "      <td>MOCKEVEN</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>GLOMICON_EvenMock_4</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>paired</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>fastq</td>\n",
       "      <td>GLOMICON_EvenMock_4_S83_L001_R1_001.fastq.gz</td>\n",
       "      <td>GLOMICON_EvenMock_4_S83_L001_R2_001.fastq.gz</td>\n",
       "      <td>20240401_18S-Amplicon_PE150</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GLOMICON_EvenMock_5</th>\n",
       "      <td>MOCKEVEN</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaT</td>\n",
       "      <td>MOCKEVEN</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>GLOMICON_EvenMock_5</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>paired</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>fastq</td>\n",
       "      <td>GLOMICON_EvenMock_5_S84_L001_R1_001.fastq.gz</td>\n",
       "      <td>GLOMICON_EvenMock_5_S84_L001_R2_001.fastq.gz</td>\n",
       "      <td>20240401_18S-Amplicon_PE150</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GLOMICON_EvenMock_A</th>\n",
       "      <td>MOCKEVEN</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaT</td>\n",
       "      <td>MOCKEVEN</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>GLOMICON_EvenMock_A</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>paired</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>fastq</td>\n",
       "      <td>GLOMICON_EvenMock_A_S85_L001_R1_001.fastq.gz</td>\n",
       "      <td>GLOMICON_EvenMock_A_S85_L001_R2_001.fastq.gz</td>\n",
       "      <td>20240401_18S-Amplicon_PE150</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GLOMICON_EvenMock_B</th>\n",
       "      <td>MOCKEVEN</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaT</td>\n",
       "      <td>MOCKEVEN</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>GLOMICON_EvenMock_B</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>paired</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>fastq</td>\n",
       "      <td>GLOMICON_EvenMock_B_S86_L001_R1_001.fastq.gz</td>\n",
       "      <td>GLOMICON_EvenMock_B_S86_L001_R2_001.fastq.gz</td>\n",
       "      <td>20240401_18S-Amplicon_PE150</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GLOMICON_NOC_10</th>\n",
       "      <td>NOC</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NOC</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>GLOMICON_NOC_10</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>paired</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>fastq</td>\n",
       "      <td>GLOMICON_NOC_10_S75_L001_R1_001.fastq.gz</td>\n",
       "      <td>GLOMICON_NOC_10_S75_L001_R2_001.fastq.gz</td>\n",
       "      <td>20240401_18S-Amplicon_PE150</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GLOMICON_NOC_16</th>\n",
       "      <td>NOC</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NOC</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>GLOMICON_NOC_16</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>paired</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>fastq</td>\n",
       "      <td>GLOMICON_NOC_16_S74_L001_R1_001.fastq.gz</td>\n",
       "      <td>GLOMICON_NOC_16_S74_L001_R2_001.fastq.gz</td>\n",
       "      <td>20240401_18S-Amplicon_PE150</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GLOMICON_NOC_22</th>\n",
       "      <td>NOC</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NOC</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>GLOMICON_NOC_22</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>paired</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>fastq</td>\n",
       "      <td>GLOMICON_NOC_22_S73_L001_R1_001.fastq.gz</td>\n",
       "      <td>GLOMICON_NOC_22_S73_L001_R2_001.fastq.gz</td>\n",
       "      <td>20240401_18S-Amplicon_PE150</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GLOMICON_NOC_3</th>\n",
       "      <td>NOC</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NOC</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>GLOMICON_NOC_3</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>paired</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>fastq</td>\n",
       "      <td>GLOMICON_NOC_3_S76_L001_R1_001.fastq.gz</td>\n",
       "      <td>GLOMICON_NOC_3_S76_L001_R2_001.fastq.gz</td>\n",
       "      <td>20240401_18S-Amplicon_PE150</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GLOMICON_NOC_32</th>\n",
       "      <td>NOC</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NOC</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>GLOMICON_NOC_32</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>paired</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>fastq</td>\n",
       "      <td>GLOMICON_NOC_32_S72_L001_R1_001.fastq.gz</td>\n",
       "      <td>GLOMICON_NOC_32_S72_L001_R2_001.fastq.gz</td>\n",
       "      <td>20240401_18S-Amplicon_PE150</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GLOMICON_Roscoff_11</th>\n",
       "      <td>SBR</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaT</td>\n",
       "      <td>SBR</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>GLOMICON_Roscoff_11</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>paired</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>fastq</td>\n",
       "      <td>GLOMICON_Roscoff_11_S64_L001_R1_001.fastq.gz</td>\n",
       "      <td>GLOMICON_Roscoff_11_S64_L001_R2_001.fastq.gz</td>\n",
       "      <td>20240401_18S-Amplicon_PE150</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GLOMICON_Roscoff_15</th>\n",
       "      <td>SBR</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaT</td>\n",
       "      <td>SBR</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>GLOMICON_Roscoff_15</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>paired</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>fastq</td>\n",
       "      <td>GLOMICON_Roscoff_15_S63_L001_R1_001.fastq.gz</td>\n",
       "      <td>GLOMICON_Roscoff_15_S63_L001_R2_001.fastq.gz</td>\n",
       "      <td>20240401_18S-Amplicon_PE150</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GLOMICON_Roscoff_19</th>\n",
       "      <td>SBR</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>2024-09-12 19:50:18.026</td>\n",
       "      <td>NaT</td>\n",
       "      <td>SBR</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>GLOMICON_Roscoff_19</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>paired</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>fastq</td>\n",
       "      <td>GLOMICON_Roscoff_19_S62_L001_R1_001.fastq.gz</td>\n",
       "      <td>GLOMICON_Roscoff_19_S62_L001_R2_001.fastq.gz</td>\n",
       "      <td>20240401_18S-Amplicon_PE150</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GLOMICON_Roscoff_3</th>\n",
       "      <td>SBR</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaT</td>\n",
       "      <td>SBR</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>GLOMICON_Roscoff_3</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>paired</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>fastq</td>\n",
       "      <td>GLOMICON_Roscoff_3_S66_L001_R1_001.fastq.gz</td>\n",
       "      <td>GLOMICON_Roscoff_3_S66_L001_R2_001.fastq.gz</td>\n",
       "      <td>20240401_18S-Amplicon_PE150</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GLOMICON_Roscoff_7</th>\n",
       "      <td>SBR</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaT</td>\n",
       "      <td>SBR</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>GLOMICON_Roscoff_7</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>paired</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>fastq</td>\n",
       "      <td>GLOMICON_Roscoff_7_S65_L001_R1_001.fastq.gz</td>\n",
       "      <td>GLOMICON_Roscoff_7_S65_L001_R2_001.fastq.gz</td>\n",
       "      <td>20240401_18S-Amplicon_PE150</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      Collecting Institute Analyzing Institute  \\\n",
       "sample_name                                                      \n",
       "GLOMICON_AWI_12                        AWI                NOAA   \n",
       "GLOMICON_AWI_16                        AWI                NOAA   \n",
       "GLOMICON_AWI_20                        AWI                NOAA   \n",
       "GLOMICON_AWI_4                         AWI                NOAA   \n",
       "GLOMICON_AWI_8                         AWI                NOAA   \n",
       "GLOMICON_BloomMock_1             MOCKBLOOM                NOAA   \n",
       "GLOMICON_BloomMock_10            MOCKBLOOM                NOAA   \n",
       "GLOMICON_BloomMock_30            MOCKBLOOM                NOAA   \n",
       "GLOMICON_BloomMock_4             MOCKBLOOM                NOAA   \n",
       "GLOMICON_BloomMock_6             MOCKBLOOM                NOAA   \n",
       "GLOMICON_EvenMock_1               MOCKEVEN                NOAA   \n",
       "GLOMICON_EvenMock_4               MOCKEVEN                NOAA   \n",
       "GLOMICON_EvenMock_5               MOCKEVEN                NOAA   \n",
       "GLOMICON_EvenMock_A               MOCKEVEN                NOAA   \n",
       "GLOMICON_EvenMock_B               MOCKEVEN                NOAA   \n",
       "GLOMICON_NOC_10                        NOC                NOAA   \n",
       "GLOMICON_NOC_16                        NOC                NOAA   \n",
       "GLOMICON_NOC_22                        NOC                NOAA   \n",
       "GLOMICON_NOC_3                         NOC                NOAA   \n",
       "GLOMICON_NOC_32                        NOC                NOAA   \n",
       "GLOMICON_Roscoff_11                    SBR                NOAA   \n",
       "GLOMICON_Roscoff_15                    SBR                NOAA   \n",
       "GLOMICON_Roscoff_19                    SBR                NOAA   \n",
       "GLOMICON_Roscoff_3                     SBR                NOAA   \n",
       "GLOMICON_Roscoff_7                     SBR                NOAA   \n",
       "\n",
       "                                  description           date_modified  \\\n",
       "sample_name                                                             \n",
       "GLOMICON_AWI_12       2024-09-26 09:18:08.071 2024-09-26 09:18:37.306   \n",
       "GLOMICON_AWI_16                           NaT                     NaT   \n",
       "GLOMICON_AWI_20                           NaT                     NaT   \n",
       "GLOMICON_AWI_4                            NaT                     NaT   \n",
       "GLOMICON_AWI_8                            NaT                     NaT   \n",
       "GLOMICON_BloomMock_1                      NaT                     NaT   \n",
       "GLOMICON_BloomMock_10                     NaT                     NaT   \n",
       "GLOMICON_BloomMock_30                     NaT                     NaT   \n",
       "GLOMICON_BloomMock_4                      NaT                     NaT   \n",
       "GLOMICON_BloomMock_6                      NaT                     NaT   \n",
       "GLOMICON_EvenMock_1                       NaT                     NaT   \n",
       "GLOMICON_EvenMock_4                       NaT                     NaT   \n",
       "GLOMICON_EvenMock_5                       NaT                     NaT   \n",
       "GLOMICON_EvenMock_A                       NaT                     NaT   \n",
       "GLOMICON_EvenMock_B                       NaT                     NaT   \n",
       "GLOMICON_NOC_10                           NaT                     NaT   \n",
       "GLOMICON_NOC_16                           NaT                     NaT   \n",
       "GLOMICON_NOC_22                           NaT                     NaT   \n",
       "GLOMICON_NOC_3                            NaT                     NaT   \n",
       "GLOMICON_NOC_32                           NaT                     NaT   \n",
       "GLOMICON_Roscoff_11                       NaT                     NaT   \n",
       "GLOMICON_Roscoff_15                       NaT                     NaT   \n",
       "GLOMICON_Roscoff_19   2024-09-12 19:50:18.026                     NaT   \n",
       "GLOMICON_Roscoff_3                        NaT                     NaT   \n",
       "GLOMICON_Roscoff_7                        NaT                     NaT   \n",
       "\n",
       "                      collecting_institute analyzing_Institute  \\\n",
       "sample_name                                                      \n",
       "GLOMICON_AWI_12                        AWI                NOAA   \n",
       "GLOMICON_AWI_16                        AWI                NOAA   \n",
       "GLOMICON_AWI_20                        AWI                NOAA   \n",
       "GLOMICON_AWI_4                         AWI                NOAA   \n",
       "GLOMICON_AWI_8                         AWI                NOAA   \n",
       "GLOMICON_BloomMock_1             MOCKBLOOM                NOAA   \n",
       "GLOMICON_BloomMock_10            MOCKBLOOM                NOAA   \n",
       "GLOMICON_BloomMock_30            MOCKBLOOM                NOAA   \n",
       "GLOMICON_BloomMock_4             MOCKBLOOM                NOAA   \n",
       "GLOMICON_BloomMock_6             MOCKBLOOM                NOAA   \n",
       "GLOMICON_EvenMock_1               MOCKEVEN                NOAA   \n",
       "GLOMICON_EvenMock_4               MOCKEVEN                NOAA   \n",
       "GLOMICON_EvenMock_5               MOCKEVEN                NOAA   \n",
       "GLOMICON_EvenMock_A               MOCKEVEN                NOAA   \n",
       "GLOMICON_EvenMock_B               MOCKEVEN                NOAA   \n",
       "GLOMICON_NOC_10                        NOC                NOAA   \n",
       "GLOMICON_NOC_16                        NOC                NOAA   \n",
       "GLOMICON_NOC_22                        NOC                NOAA   \n",
       "GLOMICON_NOC_3                         NOC                NOAA   \n",
       "GLOMICON_NOC_32                        NOC                NOAA   \n",
       "GLOMICON_Roscoff_11                    SBR                NOAA   \n",
       "GLOMICON_Roscoff_15                    SBR                NOAA   \n",
       "GLOMICON_Roscoff_19                    SBR                NOAA   \n",
       "GLOMICON_Roscoff_3                     SBR                NOAA   \n",
       "GLOMICON_Roscoff_7                     SBR                NOAA   \n",
       "\n",
       "                                  library_id library_strategy library_source  \\\n",
       "sample_name                                                                    \n",
       "GLOMICON_AWI_12              GLOMICON_AWI_12         AMPLICON    METAGENOMIC   \n",
       "GLOMICON_AWI_16              GLOMICON_AWI_16         AMPLICON    METAGENOMIC   \n",
       "GLOMICON_AWI_20              GLOMICON_AWI_20         AMPLICON    METAGENOMIC   \n",
       "GLOMICON_AWI_4                GLOMICON_AWI_4         AMPLICON    METAGENOMIC   \n",
       "GLOMICON_AWI_8                GLOMICON_AWI_8         AMPLICON    METAGENOMIC   \n",
       "GLOMICON_BloomMock_1    GLOMICON_BloomMock_1         AMPLICON    METAGENOMIC   \n",
       "GLOMICON_BloomMock_10  GLOMICON_BloomMock_10         AMPLICON    METAGENOMIC   \n",
       "GLOMICON_BloomMock_30  GLOMICON_BloomMock_30         AMPLICON    METAGENOMIC   \n",
       "GLOMICON_BloomMock_4    GLOMICON_BloomMock_4         AMPLICON    METAGENOMIC   \n",
       "GLOMICON_BloomMock_6    GLOMICON_BloomMock_6         AMPLICON    METAGENOMIC   \n",
       "GLOMICON_EvenMock_1      GLOMICON_EvenMock_1         AMPLICON    METAGENOMIC   \n",
       "GLOMICON_EvenMock_4      GLOMICON_EvenMock_4         AMPLICON    METAGENOMIC   \n",
       "GLOMICON_EvenMock_5      GLOMICON_EvenMock_5         AMPLICON    METAGENOMIC   \n",
       "GLOMICON_EvenMock_A      GLOMICON_EvenMock_A         AMPLICON    METAGENOMIC   \n",
       "GLOMICON_EvenMock_B      GLOMICON_EvenMock_B         AMPLICON    METAGENOMIC   \n",
       "GLOMICON_NOC_10              GLOMICON_NOC_10         AMPLICON    METAGENOMIC   \n",
       "GLOMICON_NOC_16              GLOMICON_NOC_16         AMPLICON    METAGENOMIC   \n",
       "GLOMICON_NOC_22              GLOMICON_NOC_22         AMPLICON    METAGENOMIC   \n",
       "GLOMICON_NOC_3                GLOMICON_NOC_3         AMPLICON    METAGENOMIC   \n",
       "GLOMICON_NOC_32              GLOMICON_NOC_32         AMPLICON    METAGENOMIC   \n",
       "GLOMICON_Roscoff_11      GLOMICON_Roscoff_11         AMPLICON    METAGENOMIC   \n",
       "GLOMICON_Roscoff_15      GLOMICON_Roscoff_15         AMPLICON    METAGENOMIC   \n",
       "GLOMICON_Roscoff_19      GLOMICON_Roscoff_19         AMPLICON    METAGENOMIC   \n",
       "GLOMICON_Roscoff_3        GLOMICON_Roscoff_3         AMPLICON    METAGENOMIC   \n",
       "GLOMICON_Roscoff_7        GLOMICON_Roscoff_7         AMPLICON    METAGENOMIC   \n",
       "\n",
       "                      library_selection lib_layout  platform instrument_model  \\\n",
       "sample_name                                                                     \n",
       "GLOMICON_AWI_12                     PCR     paired  ILLUMINA   Illumina MiSeq   \n",
       "GLOMICON_AWI_16                     PCR     paired  ILLUMINA   Illumina MiSeq   \n",
       "GLOMICON_AWI_20                     PCR     paired  ILLUMINA   Illumina MiSeq   \n",
       "GLOMICON_AWI_4                      PCR     paired  ILLUMINA   Illumina MiSeq   \n",
       "GLOMICON_AWI_8                      PCR     paired  ILLUMINA   Illumina MiSeq   \n",
       "GLOMICON_BloomMock_1                PCR     paired  ILLUMINA   Illumina MiSeq   \n",
       "GLOMICON_BloomMock_10               PCR     paired  ILLUMINA   Illumina MiSeq   \n",
       "GLOMICON_BloomMock_30               PCR     paired  ILLUMINA   Illumina MiSeq   \n",
       "GLOMICON_BloomMock_4                PCR     paired  ILLUMINA   Illumina MiSeq   \n",
       "GLOMICON_BloomMock_6                PCR     paired  ILLUMINA   Illumina MiSeq   \n",
       "GLOMICON_EvenMock_1                 PCR     paired  ILLUMINA   Illumina MiSeq   \n",
       "GLOMICON_EvenMock_4                 PCR     paired  ILLUMINA   Illumina MiSeq   \n",
       "GLOMICON_EvenMock_5                 PCR     paired  ILLUMINA   Illumina MiSeq   \n",
       "GLOMICON_EvenMock_A                 PCR     paired  ILLUMINA   Illumina MiSeq   \n",
       "GLOMICON_EvenMock_B                 PCR     paired  ILLUMINA   Illumina MiSeq   \n",
       "GLOMICON_NOC_10                     PCR     paired  ILLUMINA   Illumina MiSeq   \n",
       "GLOMICON_NOC_16                     PCR     paired  ILLUMINA   Illumina MiSeq   \n",
       "GLOMICON_NOC_22                     PCR     paired  ILLUMINA   Illumina MiSeq   \n",
       "GLOMICON_NOC_3                      PCR     paired  ILLUMINA   Illumina MiSeq   \n",
       "GLOMICON_NOC_32                     PCR     paired  ILLUMINA   Illumina MiSeq   \n",
       "GLOMICON_Roscoff_11                 PCR     paired  ILLUMINA   Illumina MiSeq   \n",
       "GLOMICON_Roscoff_15                 PCR     paired  ILLUMINA   Illumina MiSeq   \n",
       "GLOMICON_Roscoff_19                 PCR     paired  ILLUMINA   Illumina MiSeq   \n",
       "GLOMICON_Roscoff_3                  PCR     paired  ILLUMINA   Illumina MiSeq   \n",
       "GLOMICON_Roscoff_7                  PCR     paired  ILLUMINA   Illumina MiSeq   \n",
       "\n",
       "                      filetype  \\\n",
       "sample_name                      \n",
       "GLOMICON_AWI_12          fastq   \n",
       "GLOMICON_AWI_16          fastq   \n",
       "GLOMICON_AWI_20          fastq   \n",
       "GLOMICON_AWI_4           fastq   \n",
       "GLOMICON_AWI_8           fastq   \n",
       "GLOMICON_BloomMock_1     fastq   \n",
       "GLOMICON_BloomMock_10    fastq   \n",
       "GLOMICON_BloomMock_30    fastq   \n",
       "GLOMICON_BloomMock_4     fastq   \n",
       "GLOMICON_BloomMock_6     fastq   \n",
       "GLOMICON_EvenMock_1      fastq   \n",
       "GLOMICON_EvenMock_4      fastq   \n",
       "GLOMICON_EvenMock_5      fastq   \n",
       "GLOMICON_EvenMock_A      fastq   \n",
       "GLOMICON_EvenMock_B      fastq   \n",
       "GLOMICON_NOC_10          fastq   \n",
       "GLOMICON_NOC_16          fastq   \n",
       "GLOMICON_NOC_22          fastq   \n",
       "GLOMICON_NOC_3           fastq   \n",
       "GLOMICON_NOC_32          fastq   \n",
       "GLOMICON_Roscoff_11      fastq   \n",
       "GLOMICON_Roscoff_15      fastq   \n",
       "GLOMICON_Roscoff_19      fastq   \n",
       "GLOMICON_Roscoff_3       fastq   \n",
       "GLOMICON_Roscoff_7       fastq   \n",
       "\n",
       "                                                             filename  \\\n",
       "sample_name                                                             \n",
       "GLOMICON_AWI_12              GLOMICON_AWI_12_S69_L001_R1_001.fastq.gz   \n",
       "GLOMICON_AWI_16              GLOMICON_AWI_16_S68_L001_R1_001.fastq.gz   \n",
       "GLOMICON_AWI_20              GLOMICON_AWI_20_S67_L001_R1_001.fastq.gz   \n",
       "GLOMICON_AWI_4                GLOMICON_AWI_4_S71_L001_R1_001.fastq.gz   \n",
       "GLOMICON_AWI_8                GLOMICON_AWI_8_S70_L001_R1_001.fastq.gz   \n",
       "GLOMICON_BloomMock_1    GLOMICON_BloomMock_1_S77_L001_R1_001.fastq.gz   \n",
       "GLOMICON_BloomMock_10  GLOMICON_BloomMock_10_S80_L001_R1_001.fastq.gz   \n",
       "GLOMICON_BloomMock_30  GLOMICON_BloomMock_30_S81_L001_R1_001.fastq.gz   \n",
       "GLOMICON_BloomMock_4    GLOMICON_BloomMock_4_S78_L001_R1_001.fastq.gz   \n",
       "GLOMICON_BloomMock_6    GLOMICON_BloomMock_6_S79_L001_R1_001.fastq.gz   \n",
       "GLOMICON_EvenMock_1      GLOMICON_EvenMock_1_S82_L001_R1_001.fastq.gz   \n",
       "GLOMICON_EvenMock_4      GLOMICON_EvenMock_4_S83_L001_R1_001.fastq.gz   \n",
       "GLOMICON_EvenMock_5      GLOMICON_EvenMock_5_S84_L001_R1_001.fastq.gz   \n",
       "GLOMICON_EvenMock_A      GLOMICON_EvenMock_A_S85_L001_R1_001.fastq.gz   \n",
       "GLOMICON_EvenMock_B      GLOMICON_EvenMock_B_S86_L001_R1_001.fastq.gz   \n",
       "GLOMICON_NOC_10              GLOMICON_NOC_10_S75_L001_R1_001.fastq.gz   \n",
       "GLOMICON_NOC_16              GLOMICON_NOC_16_S74_L001_R1_001.fastq.gz   \n",
       "GLOMICON_NOC_22              GLOMICON_NOC_22_S73_L001_R1_001.fastq.gz   \n",
       "GLOMICON_NOC_3                GLOMICON_NOC_3_S76_L001_R1_001.fastq.gz   \n",
       "GLOMICON_NOC_32              GLOMICON_NOC_32_S72_L001_R1_001.fastq.gz   \n",
       "GLOMICON_Roscoff_11      GLOMICON_Roscoff_11_S64_L001_R1_001.fastq.gz   \n",
       "GLOMICON_Roscoff_15      GLOMICON_Roscoff_15_S63_L001_R1_001.fastq.gz   \n",
       "GLOMICON_Roscoff_19      GLOMICON_Roscoff_19_S62_L001_R1_001.fastq.gz   \n",
       "GLOMICON_Roscoff_3        GLOMICON_Roscoff_3_S66_L001_R1_001.fastq.gz   \n",
       "GLOMICON_Roscoff_7        GLOMICON_Roscoff_7_S65_L001_R1_001.fastq.gz   \n",
       "\n",
       "                                                            filename2  \\\n",
       "sample_name                                                             \n",
       "GLOMICON_AWI_12              GLOMICON_AWI_12_S69_L001_R2_001.fastq.gz   \n",
       "GLOMICON_AWI_16              GLOMICON_AWI_16_S68_L001_R2_001.fastq.gz   \n",
       "GLOMICON_AWI_20              GLOMICON_AWI_20_S67_L001_R2_001.fastq.gz   \n",
       "GLOMICON_AWI_4                GLOMICON_AWI_4_S71_L001_R2_001.fastq.gz   \n",
       "GLOMICON_AWI_8                GLOMICON_AWI_8_S70_L001_R2_001.fastq.gz   \n",
       "GLOMICON_BloomMock_1    GLOMICON_BloomMock_1_S77_L001_R2_001.fastq.gz   \n",
       "GLOMICON_BloomMock_10  GLOMICON_BloomMock_10_S80_L001_R2_001.fastq.gz   \n",
       "GLOMICON_BloomMock_30  GLOMICON_BloomMock_30_S81_L001_R2_001.fastq.gz   \n",
       "GLOMICON_BloomMock_4    GLOMICON_BloomMock_4_S78_L001_R2_001.fastq.gz   \n",
       "GLOMICON_BloomMock_6    GLOMICON_BloomMock_6_S79_L001_R2_001.fastq.gz   \n",
       "GLOMICON_EvenMock_1      GLOMICON_EvenMock_1_S82_L001_R2_001.fastq.gz   \n",
       "GLOMICON_EvenMock_4      GLOMICON_EvenMock_4_S83_L001_R2_001.fastq.gz   \n",
       "GLOMICON_EvenMock_5      GLOMICON_EvenMock_5_S84_L001_R2_001.fastq.gz   \n",
       "GLOMICON_EvenMock_A      GLOMICON_EvenMock_A_S85_L001_R2_001.fastq.gz   \n",
       "GLOMICON_EvenMock_B      GLOMICON_EvenMock_B_S86_L001_R2_001.fastq.gz   \n",
       "GLOMICON_NOC_10              GLOMICON_NOC_10_S75_L001_R2_001.fastq.gz   \n",
       "GLOMICON_NOC_16              GLOMICON_NOC_16_S74_L001_R2_001.fastq.gz   \n",
       "GLOMICON_NOC_22              GLOMICON_NOC_22_S73_L001_R2_001.fastq.gz   \n",
       "GLOMICON_NOC_3                GLOMICON_NOC_3_S76_L001_R2_001.fastq.gz   \n",
       "GLOMICON_NOC_32              GLOMICON_NOC_32_S72_L001_R2_001.fastq.gz   \n",
       "GLOMICON_Roscoff_11      GLOMICON_Roscoff_11_S64_L001_R2_001.fastq.gz   \n",
       "GLOMICON_Roscoff_15      GLOMICON_Roscoff_15_S63_L001_R2_001.fastq.gz   \n",
       "GLOMICON_Roscoff_19      GLOMICON_Roscoff_19_S62_L001_R2_001.fastq.gz   \n",
       "GLOMICON_Roscoff_3        GLOMICON_Roscoff_3_S66_L001_R2_001.fastq.gz   \n",
       "GLOMICON_Roscoff_7        GLOMICON_Roscoff_7_S65_L001_R2_001.fastq.gz   \n",
       "\n",
       "                             sequencing_run_folder pcr_primer_name_reverse  \\\n",
       "sample_name                                                                  \n",
       "GLOMICON_AWI_12        20240401_18S-Amplicon_PE150                     NaN   \n",
       "GLOMICON_AWI_16        20240401_18S-Amplicon_PE150                     NaN   \n",
       "GLOMICON_AWI_20        20240401_18S-Amplicon_PE150                     NaN   \n",
       "GLOMICON_AWI_4         20240401_18S-Amplicon_PE150                     NaN   \n",
       "GLOMICON_AWI_8         20240401_18S-Amplicon_PE150                     NaN   \n",
       "GLOMICON_BloomMock_1   20240401_18S-Amplicon_PE150                     NaN   \n",
       "GLOMICON_BloomMock_10  20240401_18S-Amplicon_PE150                     NaN   \n",
       "GLOMICON_BloomMock_30  20240401_18S-Amplicon_PE150                     NaN   \n",
       "GLOMICON_BloomMock_4   20240401_18S-Amplicon_PE150                       x   \n",
       "GLOMICON_BloomMock_6   20240401_18S-Amplicon_PE150                     NaN   \n",
       "GLOMICON_EvenMock_1    20240401_18S-Amplicon_PE150                     NaN   \n",
       "GLOMICON_EvenMock_4    20240401_18S-Amplicon_PE150                     NaN   \n",
       "GLOMICON_EvenMock_5    20240401_18S-Amplicon_PE150                     NaN   \n",
       "GLOMICON_EvenMock_A    20240401_18S-Amplicon_PE150                     NaN   \n",
       "GLOMICON_EvenMock_B    20240401_18S-Amplicon_PE150                     NaN   \n",
       "GLOMICON_NOC_10        20240401_18S-Amplicon_PE150                     NaN   \n",
       "GLOMICON_NOC_16        20240401_18S-Amplicon_PE150                     NaN   \n",
       "GLOMICON_NOC_22        20240401_18S-Amplicon_PE150                     NaN   \n",
       "GLOMICON_NOC_3         20240401_18S-Amplicon_PE150                     NaN   \n",
       "GLOMICON_NOC_32        20240401_18S-Amplicon_PE150                     NaN   \n",
       "GLOMICON_Roscoff_11    20240401_18S-Amplicon_PE150                     NaN   \n",
       "GLOMICON_Roscoff_15    20240401_18S-Amplicon_PE150                     NaN   \n",
       "GLOMICON_Roscoff_19    20240401_18S-Amplicon_PE150                     NaN   \n",
       "GLOMICON_Roscoff_3     20240401_18S-Amplicon_PE150                     NaN   \n",
       "GLOMICON_Roscoff_7     20240401_18S-Amplicon_PE150                     NaN   \n",
       "\n",
       "                      date_modified_amplicon_sheet  \n",
       "sample_name                                         \n",
       "GLOMICON_AWI_12            2024-09-12 20:02:53.389  \n",
       "GLOMICON_AWI_16                                NaT  \n",
       "GLOMICON_AWI_20                                NaT  \n",
       "GLOMICON_AWI_4                                 NaT  \n",
       "GLOMICON_AWI_8                                 NaT  \n",
       "GLOMICON_BloomMock_1                           NaT  \n",
       "GLOMICON_BloomMock_10                          NaT  \n",
       "GLOMICON_BloomMock_30                          NaT  \n",
       "GLOMICON_BloomMock_4                           NaT  \n",
       "GLOMICON_BloomMock_6                           NaT  \n",
       "GLOMICON_EvenMock_1                            NaT  \n",
       "GLOMICON_EvenMock_4                            NaT  \n",
       "GLOMICON_EvenMock_5                            NaT  \n",
       "GLOMICON_EvenMock_A                            NaT  \n",
       "GLOMICON_EvenMock_B                            NaT  \n",
       "GLOMICON_NOC_10                                NaT  \n",
       "GLOMICON_NOC_16                                NaT  \n",
       "GLOMICON_NOC_22                                NaT  \n",
       "GLOMICON_NOC_3                                 NaT  \n",
       "GLOMICON_NOC_32                                NaT  \n",
       "GLOMICON_Roscoff_11                            NaT  \n",
       "GLOMICON_Roscoff_15                            NaT  \n",
       "GLOMICON_Roscoff_19                            NaT  \n",
       "GLOMICON_Roscoff_3                             NaT  \n",
       "GLOMICON_Roscoff_7                             NaT  "
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = water_amplicon_merge(dfs[1], dfs[2])\n",
    "samp_dfs.append(df)\n",
    "df.head(n=50)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3b8c7a74-51a8-436a-9ad9-b57327f94f38",
   "metadata": {},
   "source": [
    "### AWI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "b22e293a-133f-4e38-a0e3-0dc5dce7fcf2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/kpitz/github/GLOMICON/intercomparison/metadata/AWI_NOAA_MIMARKS.survey.water.6.0_sharing.xlsx\n",
      "Sheets in excel file: ['README', 'study_data', 'water_sample_data', 'amplicon_prep_data', 'metag_prep_data', 'analysis_data', 'SRA_template', 'measurement_metadata', 'SRA Terms', 'Version log']\n",
      "study_data 5\n",
      "water_sample_data 8\n",
      "amplicon_prep_data 6\n",
      "analysis_data 8\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>amplicon_sequenced</th>\n",
       "      <th>ampliconSize</th>\n",
       "      <th>trim_method</th>\n",
       "      <th>cluster_method</th>\n",
       "      <th>pid_clustering</th>\n",
       "      <th>taxa_class_method</th>\n",
       "      <th>taxa_ref_db</th>\n",
       "      <th>code_repo</th>\n",
       "      <th>identificationReferences</th>\n",
       "      <th>controls_used</th>\n",
       "      <th>date_modified</th>\n",
       "      <th>modified_by</th>\n",
       "      <th>sheet</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>18S V4</td>\n",
       "      <td>356-394</td>\n",
       "      <td>cutadapt v2.8</td>\n",
       "      <td>dada denoising; dada2 v1.30 in R v4.3.2 (learn...</td>\n",
       "      <td>ASV</td>\n",
       "      <td>dada2 implementation of naive-bayes classifier...</td>\n",
       "      <td>PR2 v5.0.0; full set; https://github.com/pr2da...</td>\n",
       "      <td>???</td>\n",
       "      <td>https://github.com/pr2database/pr2database/rel...</td>\n",
       "      <td>???</td>\n",
       "      <td>2024-09-06 09:12:48.224</td>\n",
       "      <td>stefan.neuhaus@awi.de</td>\n",
       "      <td>analysis_data</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  amplicon_sequenced ampliconSize    trim_method  \\\n",
       "0             18S V4      356-394  cutadapt v2.8   \n",
       "\n",
       "                                      cluster_method pid_clustering  \\\n",
       "0  dada denoising; dada2 v1.30 in R v4.3.2 (learn...            ASV   \n",
       "\n",
       "                                   taxa_class_method  \\\n",
       "0  dada2 implementation of naive-bayes classifier...   \n",
       "\n",
       "                                         taxa_ref_db code_repo  \\\n",
       "0  PR2 v5.0.0; full set; https://github.com/pr2da...       ???   \n",
       "\n",
       "                            identificationReferences controls_used  \\\n",
       "0  https://github.com/pr2database/pr2database/rel...           ???   \n",
       "\n",
       "            date_modified            modified_by          sheet  \n",
       "0 2024-09-06 09:12:48.224  stefan.neuhaus@awi.de  analysis_data  "
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# import\n",
    "file = '/Users/kpitz/github/GLOMICON/intercomparison/metadata/AWI_NOAA_MIMARKS.survey.water.6.0_sharing.xlsx'\n",
    "print(file)\n",
    "inst_name = 'AWI'\n",
    "institutes.append(inst_name)\n",
    "xl = pd.ExcelFile(file)\n",
    "print('Sheets in excel file:',xl.sheet_names)  # see all sheet names\n",
    "#sheets = xl.sheet_names\n",
    "sheets = ['study_data', 'water_sample_data', 'amplicon_prep_data', 'analysis_data']\n",
    "dfs= []\n",
    "skiprow_num = [5,8,6,8]\n",
    "for sheet, sr in zip(sheets, skiprow_num):\n",
    "    print(sheet,sr)\n",
    "    df = xl.parse(sheet,skiprows=sr)\n",
    "    df = df.dropna(how='all', axis=1)\n",
    "    df = df.dropna(how='all', axis=0)\n",
    "    df['sheet'] = sheet\n",
    "    dfs.append(df)\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "3991cf1c-839d-433d-ad5a-e667ec319093",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AWI</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>project_id</th>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>project_name</th>\n",
       "      <td>Intercomparison of filtered seawater samples a...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>project_id_external</th>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>project_contact</th>\n",
       "      <td>Kathleen Pitz, kpitz@mbari.org, https://orcid....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>project_description</th>\n",
       "      <td>Replicate filters, originating from surface wa...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>type</th>\n",
       "      <td>Occurrence</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>license</th>\n",
       "      <td>CC-BY 4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>associated_parties</th>\n",
       "      <td>Felix Janssen, felix.janssen@awi.de, https://o...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>study_area_description</th>\n",
       "      <td>Surface waters from different marine time seri...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>external_links</th>\n",
       "      <td>https://github.com/GLOMICON</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>amplicon_sequenced</th>\n",
       "      <td>18S V4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ampliconSize</th>\n",
       "      <td>356-394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>trim_method</th>\n",
       "      <td>cutadapt v2.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cluster_method</th>\n",
       "      <td>dada denoising; dada2 v1.30 in R v4.3.2 (learn...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pid_clustering</th>\n",
       "      <td>ASV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taxa_class_method</th>\n",
       "      <td>dada2 implementation of naive-bayes classifier...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taxa_ref_db</th>\n",
       "      <td>PR2 v5.0.0; full set; https://github.com/pr2da...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>code_repo</th>\n",
       "      <td>???</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>identificationReferences</th>\n",
       "      <td>https://github.com/pr2database/pr2database/rel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>controls_used</th>\n",
       "      <td>???</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>modified_by</th>\n",
       "      <td>stefan.neuhaus@awi.de</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                        AWI\n",
       "project_id                                               GLOMICON_INTERCOMP\n",
       "project_name              Intercomparison of filtered seawater samples a...\n",
       "project_id_external                                      GLOMICON_INTERCOMP\n",
       "project_contact           Kathleen Pitz, kpitz@mbari.org, https://orcid....\n",
       "project_description       Replicate filters, originating from surface wa...\n",
       "type                                                             Occurrence\n",
       "license                                                           CC-BY 4.0\n",
       "associated_parties        Felix Janssen, felix.janssen@awi.de, https://o...\n",
       "study_area_description    Surface waters from different marine time seri...\n",
       "external_links                                  https://github.com/GLOMICON\n",
       "amplicon_sequenced                                                   18S V4\n",
       "ampliconSize                                                        356-394\n",
       "trim_method                                                   cutadapt v2.8\n",
       "cluster_method            dada denoising; dada2 v1.30 in R v4.3.2 (learn...\n",
       "pid_clustering                                                          ASV\n",
       "taxa_class_method         dada2 implementation of naive-bayes classifier...\n",
       "taxa_ref_db               PR2 v5.0.0; full set; https://github.com/pr2da...\n",
       "code_repo                                                               ???\n",
       "identificationReferences  https://github.com/pr2database/pr2database/rel...\n",
       "controls_used                                                           ???\n",
       "modified_by                                           stefan.neuhaus@awi.de"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = study_data_merge(dfs[0], dfs[3], inst_name)\n",
    "df = df.rename(columns={1:inst_name+'_2'})\n",
    "study_dfs.append(df)\n",
    "df.head(n=50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "id": "87fd0faa-f463-4ef0-a416-63c534838615",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Collecting Institute</th>\n",
       "      <th>Analyzing Institute</th>\n",
       "      <th>cruise_id</th>\n",
       "      <th>station</th>\n",
       "      <th>ctd_bottle_no</th>\n",
       "      <th>sample_replicate</th>\n",
       "      <th>source_material_id</th>\n",
       "      <th>sample_title</th>\n",
       "      <th>notes_sampling</th>\n",
       "      <th>project_id</th>\n",
       "      <th>...</th>\n",
       "      <th>amplicon_sequenced</th>\n",
       "      <th>target_gene</th>\n",
       "      <th>target_subfragment</th>\n",
       "      <th>pcr_primer_forward</th>\n",
       "      <th>pcr_primer_reverse</th>\n",
       "      <th>pcr_primer_name_forward</th>\n",
       "      <th>pcr_primer_name_reverse</th>\n",
       "      <th>pcr_primer_reference</th>\n",
       "      <th>pcr_cond</th>\n",
       "      <th>date_modified_amplicon_sheet</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sample_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Samples taken / prepared and analyzed by AWI (complete information)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Framstrait_01</th>\n",
       "      <td>AWI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>PS114 (2018)</td>\n",
       "      <td>PS114-36-4</td>\n",
       "      <td>pooled</td>\n",
       "      <td>01</td>\n",
       "      <td>GLOMICON_INTERCOMP_Framstrait_01</td>\n",
       "      <td>Seawater sample Fram Strait (GLOMICON_INTERCOMP)</td>\n",
       "      <td>Surface waters from the northern part of Fram ...</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V4</td>\n",
       "      <td>5'-GCG GTA ATT CCA GCT CCA A-3'</td>\n",
       "      <td>5'-AC TTT CGT TCT TGA TYR R-3'</td>\n",
       "      <td>528i F</td>\n",
       "      <td>964i R</td>\n",
       "      <td>528F: Elwood et al. (1985) doi:10.1093/oxfordj...</td>\n",
       "      <td>\\ninitial denaturation:95degC_3min;denaturatio...</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Framstrait_01</th>\n",
       "      <td>AWI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>PS114 (2018)</td>\n",
       "      <td>PS114-36-4</td>\n",
       "      <td>pooled</td>\n",
       "      <td>01</td>\n",
       "      <td>GLOMICON_INTERCOMP_Framstrait_01</td>\n",
       "      <td>Seawater sample Fram Strait (GLOMICON_INTERCOMP)</td>\n",
       "      <td>Surface waters from the northern part of Fram ...</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V4</td>\n",
       "      <td>5'-GCG GTA ATT CCA GCT CCA A-3'</td>\n",
       "      <td>5'-AC TTT CGT TCT TGA TYR R-3'</td>\n",
       "      <td>528i F</td>\n",
       "      <td>964i R</td>\n",
       "      <td>528F: Elwood et al. (1985) doi:10.1093/oxfordj...</td>\n",
       "      <td>\\ninitial denaturation:95degC_3min;denaturatio...</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Framstrait_05</th>\n",
       "      <td>AWI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>PS114 (2018)</td>\n",
       "      <td>PS114-36-4</td>\n",
       "      <td>pooled</td>\n",
       "      <td>05</td>\n",
       "      <td>GLOMICON_INTERCOMP_Framstrait_05</td>\n",
       "      <td>Seawater sample Fram Strait (GLOMICON_INTERCOMP)</td>\n",
       "      <td>Surface waters from the northern part of Fram ...</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V4</td>\n",
       "      <td>5'-GCG GTA ATT CCA GCT CCA A-3'</td>\n",
       "      <td>5'-AC TTT CGT TCT TGA TYR R-3'</td>\n",
       "      <td>528i F</td>\n",
       "      <td>964i R</td>\n",
       "      <td>528F: Elwood et al. (1985) doi:10.1093/oxfordj...</td>\n",
       "      <td>\\ninitial denaturation:95degC_3min;denaturatio...</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Framstrait_05</th>\n",
       "      <td>AWI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>PS114 (2018)</td>\n",
       "      <td>PS114-36-4</td>\n",
       "      <td>pooled</td>\n",
       "      <td>05</td>\n",
       "      <td>GLOMICON_INTERCOMP_Framstrait_05</td>\n",
       "      <td>Seawater sample Fram Strait (GLOMICON_INTERCOMP)</td>\n",
       "      <td>Surface waters from the northern part of Fram ...</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V4</td>\n",
       "      <td>5'-GCG GTA ATT CCA GCT CCA A-3'</td>\n",
       "      <td>5'-AC TTT CGT TCT TGA TYR R-3'</td>\n",
       "      <td>528i F</td>\n",
       "      <td>964i R</td>\n",
       "      <td>528F: Elwood et al. (1985) doi:10.1093/oxfordj...</td>\n",
       "      <td>\\ninitial denaturation:95degC_3min;denaturatio...</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Framstrait_09</th>\n",
       "      <td>AWI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>PS114 (2018)</td>\n",
       "      <td>PS114-36-4</td>\n",
       "      <td>pooled</td>\n",
       "      <td>09</td>\n",
       "      <td>GLOMICON_INTERCOMP_Framstrait_09</td>\n",
       "      <td>Seawater sample Fram Strait (GLOMICON_INTERCOMP)</td>\n",
       "      <td>Surface waters from the northern part of Fram ...</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V4</td>\n",
       "      <td>5'-GCG GTA ATT CCA GCT CCA A-3'</td>\n",
       "      <td>5'-AC TTT CGT TCT TGA TYR R-3'</td>\n",
       "      <td>528i F</td>\n",
       "      <td>964i R</td>\n",
       "      <td>528F: Elwood et al. (1985) doi:10.1093/oxfordj...</td>\n",
       "      <td>\\ninitial denaturation:95degC_3min;denaturatio...</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Framstrait_09</th>\n",
       "      <td>AWI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>PS114 (2018)</td>\n",
       "      <td>PS114-36-4</td>\n",
       "      <td>pooled</td>\n",
       "      <td>09</td>\n",
       "      <td>GLOMICON_INTERCOMP_Framstrait_09</td>\n",
       "      <td>Seawater sample Fram Strait (GLOMICON_INTERCOMP)</td>\n",
       "      <td>Surface waters from the northern part of Fram ...</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V4</td>\n",
       "      <td>5'-GCG GTA ATT CCA GCT CCA A-3'</td>\n",
       "      <td>5'-AC TTT CGT TCT TGA TYR R-3'</td>\n",
       "      <td>528i F</td>\n",
       "      <td>964i R</td>\n",
       "      <td>528F: Elwood et al. (1985) doi:10.1093/oxfordj...</td>\n",
       "      <td>\\ninitial denaturation:95degC_3min;denaturatio...</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Framstrait_13</th>\n",
       "      <td>AWI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>PS114 (2018)</td>\n",
       "      <td>PS114-36-4</td>\n",
       "      <td>pooled</td>\n",
       "      <td>13</td>\n",
       "      <td>GLOMICON_INTERCOMP_Framstrait_13</td>\n",
       "      <td>Seawater sample Fram Strait (GLOMICON_INTERCOMP)</td>\n",
       "      <td>Surface waters from the northern part of Fram ...</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V4</td>\n",
       "      <td>5'-GCG GTA ATT CCA GCT CCA A-3'</td>\n",
       "      <td>5'-AC TTT CGT TCT TGA TYR R-3'</td>\n",
       "      <td>528i F</td>\n",
       "      <td>964i R</td>\n",
       "      <td>528F: Elwood et al. (1985) doi:10.1093/oxfordj...</td>\n",
       "      <td>\\ninitial denaturation:95degC_3min;denaturatio...</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Framstrait_13</th>\n",
       "      <td>AWI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>PS114 (2018)</td>\n",
       "      <td>PS114-36-4</td>\n",
       "      <td>pooled</td>\n",
       "      <td>13</td>\n",
       "      <td>GLOMICON_INTERCOMP_Framstrait_13</td>\n",
       "      <td>Seawater sample Fram Strait (GLOMICON_INTERCOMP)</td>\n",
       "      <td>Surface waters from the northern part of Fram ...</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V4</td>\n",
       "      <td>5'-GCG GTA ATT CCA GCT CCA A-3'</td>\n",
       "      <td>5'-AC TTT CGT TCT TGA TYR R-3'</td>\n",
       "      <td>528i F</td>\n",
       "      <td>964i R</td>\n",
       "      <td>528F: Elwood et al. (1985) doi:10.1093/oxfordj...</td>\n",
       "      <td>\\ninitial denaturation:95degC_3min;denaturatio...</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Framstrait_17</th>\n",
       "      <td>AWI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>PS114 (2018)</td>\n",
       "      <td>PS114-36-4</td>\n",
       "      <td>pooled</td>\n",
       "      <td>17</td>\n",
       "      <td>GLOMICON_INTERCOMP_Framstrait_17</td>\n",
       "      <td>Seawater sample Fram Strait (GLOMICON_INTERCOMP)</td>\n",
       "      <td>Surface waters from the northern part of Fram ...</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V4</td>\n",
       "      <td>5'-GCG GTA ATT CCA GCT CCA A-3'</td>\n",
       "      <td>5'-AC TTT CGT TCT TGA TYR R-3'</td>\n",
       "      <td>528i F</td>\n",
       "      <td>964i R</td>\n",
       "      <td>528F: Elwood et al. (1985) doi:10.1093/oxfordj...</td>\n",
       "      <td>\\ninitial denaturation:95degC_3min;denaturatio...</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Framstrait_17</th>\n",
       "      <td>AWI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>PS114 (2018)</td>\n",
       "      <td>PS114-36-4</td>\n",
       "      <td>pooled</td>\n",
       "      <td>17</td>\n",
       "      <td>GLOMICON_INTERCOMP_Framstrait_17</td>\n",
       "      <td>Seawater sample Fram Strait (GLOMICON_INTERCOMP)</td>\n",
       "      <td>Surface waters from the northern part of Fram ...</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V4</td>\n",
       "      <td>5'-GCG GTA ATT CCA GCT CCA A-3'</td>\n",
       "      <td>5'-AC TTT CGT TCT TGA TYR R-3'</td>\n",
       "      <td>528i F</td>\n",
       "      <td>964i R</td>\n",
       "      <td>528F: Elwood et al. (1985) doi:10.1093/oxfordj...</td>\n",
       "      <td>\\ninitial denaturation:95degC_3min;denaturatio...</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evenMock_01</th>\n",
       "      <td>AWI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>01</td>\n",
       "      <td>GLOMICON_INTERCOMP_evenMock_01</td>\n",
       "      <td>Even mock community (GLOMICON_INTERCOMP)</td>\n",
       "      <td>mock community with even 18S rRNA gene copy nu...</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V4</td>\n",
       "      <td>5'-GCG GTA ATT CCA GCT CCA A-3'</td>\n",
       "      <td>5'-AC TTT CGT TCT TGA TYR R-3'</td>\n",
       "      <td>528i F</td>\n",
       "      <td>964i R</td>\n",
       "      <td>528F: Elwood et al. (1985) doi:10.1093/oxfordj...</td>\n",
       "      <td>\\ninitial denaturation:95degC_3min;denaturatio...</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evenMock_02</th>\n",
       "      <td>AWI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>02</td>\n",
       "      <td>GLOMICON_INTERCOMP_evenMock_02</td>\n",
       "      <td>Even mock community (GLOMICON_INTERCOMP)</td>\n",
       "      <td>mock community with even 18S rRNA gene copy nu...</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V4</td>\n",
       "      <td>5'-GCG GTA ATT CCA GCT CCA A-3'</td>\n",
       "      <td>5'-AC TTT CGT TCT TGA TYR R-3'</td>\n",
       "      <td>528i F</td>\n",
       "      <td>964i R</td>\n",
       "      <td>528F: Elwood et al. (1985) doi:10.1093/oxfordj...</td>\n",
       "      <td>\\ninitial denaturation:95degC_3min;denaturatio...</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evenMock_03</th>\n",
       "      <td>AWI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>03</td>\n",
       "      <td>GLOMICON_INTERCOMP_evenMock_03</td>\n",
       "      <td>Even mock community (GLOMICON_INTERCOMP)</td>\n",
       "      <td>mock community with even 18S rRNA gene copy nu...</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V4</td>\n",
       "      <td>5'-GCG GTA ATT CCA GCT CCA A-3'</td>\n",
       "      <td>5'-AC TTT CGT TCT TGA TYR R-3'</td>\n",
       "      <td>528i F</td>\n",
       "      <td>964i R</td>\n",
       "      <td>528F: Elwood et al. (1985) doi:10.1093/oxfordj...</td>\n",
       "      <td>\\ninitial denaturation:95degC_3min;denaturatio...</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evenMock_04</th>\n",
       "      <td>AWI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>04</td>\n",
       "      <td>GLOMICON_INTERCOMP_evenMock_04</td>\n",
       "      <td>Even mock community (GLOMICON_INTERCOMP)</td>\n",
       "      <td>mock community with even 18S rRNA gene copy nu...</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V4</td>\n",
       "      <td>5'-GCG GTA ATT CCA GCT CCA A-3'</td>\n",
       "      <td>5'-AC TTT CGT TCT TGA TYR R-3'</td>\n",
       "      <td>528i F</td>\n",
       "      <td>964i R</td>\n",
       "      <td>528F: Elwood et al. (1985) doi:10.1093/oxfordj...</td>\n",
       "      <td>\\ninitial denaturation:95degC_3min;denaturatio...</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evenMock_05</th>\n",
       "      <td>AWI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>05</td>\n",
       "      <td>GLOMICON_INTERCOMP_evenMock_05</td>\n",
       "      <td>Even mock community (GLOMICON_INTERCOMP)</td>\n",
       "      <td>mock community with even 18S rRNA gene copy nu...</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V4</td>\n",
       "      <td>5'-GCG GTA ATT CCA GCT CCA A-3'</td>\n",
       "      <td>5'-AC TTT CGT TCT TGA TYR R-3'</td>\n",
       "      <td>528i F</td>\n",
       "      <td>964i R</td>\n",
       "      <td>528F: Elwood et al. (1985) doi:10.1093/oxfordj...</td>\n",
       "      <td>\\ninitial denaturation:95degC_3min;denaturatio...</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>bloomMock_01</th>\n",
       "      <td>AWI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>05</td>\n",
       "      <td>GLOMICON_INTERCOMP_bloomMock_05</td>\n",
       "      <td>Bloom mock community (GLOMICON_INTERCOMP)</td>\n",
       "      <td>mock community with uneven 18S rRNA gene copy ...</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V4</td>\n",
       "      <td>5'-GCG GTA ATT CCA GCT CCA A-3'</td>\n",
       "      <td>5'-AC TTT CGT TCT TGA TYR R-3'</td>\n",
       "      <td>528i F</td>\n",
       "      <td>964i R</td>\n",
       "      <td>528F: Elwood et al. (1985) doi:10.1093/oxfordj...</td>\n",
       "      <td>\\ninitial denaturation:95degC_3min;denaturatio...</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>bloomMock_02</th>\n",
       "      <td>AWI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11</td>\n",
       "      <td>GLOMICON_INTERCOMP_bloomMock_11</td>\n",
       "      <td>Bloom mock community (GLOMICON_INTERCOMP)</td>\n",
       "      <td>mock community with uneven 18S rRNA gene copy ...</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V4</td>\n",
       "      <td>5'-GCG GTA ATT CCA GCT CCA A-3'</td>\n",
       "      <td>5'-AC TTT CGT TCT TGA TYR R-3'</td>\n",
       "      <td>528i F</td>\n",
       "      <td>964i R</td>\n",
       "      <td>528F: Elwood et al. (1985) doi:10.1093/oxfordj...</td>\n",
       "      <td>\\ninitial denaturation:95degC_3min;denaturatio...</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>bloomMock_03</th>\n",
       "      <td>AWI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12</td>\n",
       "      <td>GLOMICON_INTERCOMP_bloomMock_12</td>\n",
       "      <td>Bloom mock community (GLOMICON_INTERCOMP)</td>\n",
       "      <td>mock community with uneven 18S rRNA gene copy ...</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V4</td>\n",
       "      <td>5'-GCG GTA ATT CCA GCT CCA A-3'</td>\n",
       "      <td>5'-AC TTT CGT TCT TGA TYR R-3'</td>\n",
       "      <td>528i F</td>\n",
       "      <td>964i R</td>\n",
       "      <td>528F: Elwood et al. (1985) doi:10.1093/oxfordj...</td>\n",
       "      <td>\\ninitial denaturation:95degC_3min;denaturatio...</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>bloomMock_04</th>\n",
       "      <td>AWI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>19</td>\n",
       "      <td>GLOMICON_INTERCOMP_bloomMock_19</td>\n",
       "      <td>Bloom mock community (GLOMICON_INTERCOMP)</td>\n",
       "      <td>mock community with uneven 18S rRNA gene copy ...</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V4</td>\n",
       "      <td>5'-GCG GTA ATT CCA GCT CCA A-3'</td>\n",
       "      <td>5'-AC TTT CGT TCT TGA TYR R-3'</td>\n",
       "      <td>528i F</td>\n",
       "      <td>964i R</td>\n",
       "      <td>528F: Elwood et al. (1985) doi:10.1093/oxfordj...</td>\n",
       "      <td>\\ninitial denaturation:95degC_3min;denaturatio...</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>bloomMock_05</th>\n",
       "      <td>AWI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>23</td>\n",
       "      <td>GLOMICON_INTERCOMP_bloomMock_23</td>\n",
       "      <td>Bloom mock community (GLOMICON_INTERCOMP)</td>\n",
       "      <td>mock community with uneven 18S rRNA gene copy ...</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V4</td>\n",
       "      <td>5'-GCG GTA ATT CCA GCT CCA A-3'</td>\n",
       "      <td>5'-AC TTT CGT TCT TGA TYR R-3'</td>\n",
       "      <td>528i F</td>\n",
       "      <td>964i R</td>\n",
       "      <td>528F: Elwood et al. (1985) doi:10.1093/oxfordj...</td>\n",
       "      <td>\\ninitial denaturation:95degC_3min;denaturatio...</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>blank_01</th>\n",
       "      <td>AWI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>01</td>\n",
       "      <td>GLOMICON_INTERCOMP_blank_01</td>\n",
       "      <td>Blank PCR water (GLOMICON_INTERCOMP)</td>\n",
       "      <td>PCR water used to dilute mock community DNA ex...</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V4</td>\n",
       "      <td>5'-GCG GTA ATT CCA GCT CCA A-3'</td>\n",
       "      <td>5'-AC TTT CGT TCT TGA TYR R-3'</td>\n",
       "      <td>528i F</td>\n",
       "      <td>964i R</td>\n",
       "      <td>528F: Elwood et al. (1985) doi:10.1093/oxfordj...</td>\n",
       "      <td>\\ninitial denaturation:95degC_3min;denaturatio...</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Samples taken / prepared by GLOMICON partners and analyzed by AWI (sampling metadata information missing)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Davenport_02</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V4</td>\n",
       "      <td>5'-GCG GTA ATT CCA GCT CCA A-3'</td>\n",
       "      <td>5'-AC TTT CGT TCT TGA TYR R-3'</td>\n",
       "      <td>528i F</td>\n",
       "      <td>964i R</td>\n",
       "      <td>528F: Elwood et al. (1985) doi:10.1093/oxfordj...</td>\n",
       "      <td>\\ninitial denaturation:95degC_3min;denaturatio...</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Davenport_02</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V4</td>\n",
       "      <td>5'-GCG GTA ATT CCA GCT CCA A-3'</td>\n",
       "      <td>5'-AC TTT CGT TCT TGA TYR R-3'</td>\n",
       "      <td>528i F</td>\n",
       "      <td>964i R</td>\n",
       "      <td>528F: Elwood et al. (1985) doi:10.1093/oxfordj...</td>\n",
       "      <td>\\ninitial denaturation:95degC_3min;denaturatio...</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Davenport_06</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V4</td>\n",
       "      <td>5'-GCG GTA ATT CCA GCT CCA A-3'</td>\n",
       "      <td>5'-AC TTT CGT TCT TGA TYR R-3'</td>\n",
       "      <td>528i F</td>\n",
       "      <td>964i R</td>\n",
       "      <td>528F: Elwood et al. (1985) doi:10.1093/oxfordj...</td>\n",
       "      <td>\\ninitial denaturation:95degC_3min;denaturatio...</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Davenport_06</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V4</td>\n",
       "      <td>5'-GCG GTA ATT CCA GCT CCA A-3'</td>\n",
       "      <td>5'-AC TTT CGT TCT TGA TYR R-3'</td>\n",
       "      <td>528i F</td>\n",
       "      <td>964i R</td>\n",
       "      <td>528F: Elwood et al. (1985) doi:10.1093/oxfordj...</td>\n",
       "      <td>\\ninitial denaturation:95degC_3min;denaturatio...</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Davenport_11</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V4</td>\n",
       "      <td>5'-GCG GTA ATT CCA GCT CCA A-3'</td>\n",
       "      <td>5'-AC TTT CGT TCT TGA TYR R-3'</td>\n",
       "      <td>528i F</td>\n",
       "      <td>964i R</td>\n",
       "      <td>528F: Elwood et al. (1985) doi:10.1093/oxfordj...</td>\n",
       "      <td>\\ninitial denaturation:95degC_3min;denaturatio...</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Davenport_11</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V4</td>\n",
       "      <td>5'-GCG GTA ATT CCA GCT CCA A-3'</td>\n",
       "      <td>5'-AC TTT CGT TCT TGA TYR R-3'</td>\n",
       "      <td>528i F</td>\n",
       "      <td>964i R</td>\n",
       "      <td>528F: Elwood et al. (1985) doi:10.1093/oxfordj...</td>\n",
       "      <td>\\ninitial denaturation:95degC_3min;denaturatio...</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Davenport_15</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V4</td>\n",
       "      <td>5'-GCG GTA ATT CCA GCT CCA A-3'</td>\n",
       "      <td>5'-AC TTT CGT TCT TGA TYR R-3'</td>\n",
       "      <td>528i F</td>\n",
       "      <td>964i R</td>\n",
       "      <td>528F: Elwood et al. (1985) doi:10.1093/oxfordj...</td>\n",
       "      <td>\\ninitial denaturation:95degC_3min;denaturatio...</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Davenport_15</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V4</td>\n",
       "      <td>5'-GCG GTA ATT CCA GCT CCA A-3'</td>\n",
       "      <td>5'-AC TTT CGT TCT TGA TYR R-3'</td>\n",
       "      <td>528i F</td>\n",
       "      <td>964i R</td>\n",
       "      <td>528F: Elwood et al. (1985) doi:10.1093/oxfordj...</td>\n",
       "      <td>\\ninitial denaturation:95degC_3min;denaturatio...</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Davenport_19</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V4</td>\n",
       "      <td>5'-GCG GTA ATT CCA GCT CCA A-3'</td>\n",
       "      <td>5'-AC TTT CGT TCT TGA TYR R-3'</td>\n",
       "      <td>528i F</td>\n",
       "      <td>964i R</td>\n",
       "      <td>528F: Elwood et al. (1985) doi:10.1093/oxfordj...</td>\n",
       "      <td>\\ninitial denaturation:95degC_3min;denaturatio...</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Davenport_19</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V4</td>\n",
       "      <td>5'-GCG GTA ATT CCA GCT CCA A-3'</td>\n",
       "      <td>5'-AC TTT CGT TCT TGA TYR R-3'</td>\n",
       "      <td>528i F</td>\n",
       "      <td>964i R</td>\n",
       "      <td>528F: Elwood et al. (1985) doi:10.1093/oxfordj...</td>\n",
       "      <td>\\ninitial denaturation:95degC_3min;denaturatio...</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LaJolla_03</th>\n",
       "      <td>NOAA</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LaJolla_09</th>\n",
       "      <td>NOAA</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LaJolla_15</th>\n",
       "      <td>NOAA</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LaJolla_21</th>\n",
       "      <td>NOAA</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LaJolla_27</th>\n",
       "      <td>NOAA</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Plymouth_06</th>\n",
       "      <td>NOC</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Plymouth_13</th>\n",
       "      <td>NOC</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Plymouth_19</th>\n",
       "      <td>NOC</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Plymouth_29</th>\n",
       "      <td>NOC</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Plymouth_30</th>\n",
       "      <td>NOC</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bedford_01</th>\n",
       "      <td>UDalhousie</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bedford_07</th>\n",
       "      <td>UDalhousie</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bedford_13</th>\n",
       "      <td>UDalhousie</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bedford_19</th>\n",
       "      <td>UDalhousie</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bedford_25</th>\n",
       "      <td>UDalhousie</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Samples taken / prepared by AWI and analyzed by GLOMICON partners (almost complete information)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>[to be added from GLOMICON partner's metadata]</th>\n",
       "      <td>AWI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>PS114 (2018)</td>\n",
       "      <td>PS114-36-4</td>\n",
       "      <td>pooled</td>\n",
       "      <td>02</td>\n",
       "      <td>GLOMICON_INTERCOMP_Framstrait_02</td>\n",
       "      <td>Seawater sample Fram Strait (GLOMICON_INTERCOMP)</td>\n",
       "      <td>Surface waters from the northern part of Fram ...</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>50 rows × 69 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   Collecting Institute  \\\n",
       "sample_name                                                               \n",
       "Samples taken / prepared and analyzed by AWI (c...                  NaN   \n",
       "Framstrait_01                                                       AWI   \n",
       "Framstrait_01                                                       AWI   \n",
       "Framstrait_05                                                       AWI   \n",
       "Framstrait_05                                                       AWI   \n",
       "Framstrait_09                                                       AWI   \n",
       "Framstrait_09                                                       AWI   \n",
       "Framstrait_13                                                       AWI   \n",
       "Framstrait_13                                                       AWI   \n",
       "Framstrait_17                                                       AWI   \n",
       "Framstrait_17                                                       AWI   \n",
       "evenMock_01                                                         AWI   \n",
       "evenMock_02                                                         AWI   \n",
       "evenMock_03                                                         AWI   \n",
       "evenMock_04                                                         AWI   \n",
       "evenMock_05                                                         AWI   \n",
       "bloomMock_01                                                        AWI   \n",
       "bloomMock_02                                                        AWI   \n",
       "bloomMock_03                                                        AWI   \n",
       "bloomMock_04                                                        AWI   \n",
       "bloomMock_05                                                        AWI   \n",
       "blank_01                                                            AWI   \n",
       "Samples taken / prepared by GLOMICON partners a...                  NaN   \n",
       "Davenport_02                                                      MBARI   \n",
       "Davenport_02                                                      MBARI   \n",
       "Davenport_06                                                      MBARI   \n",
       "Davenport_06                                                      MBARI   \n",
       "Davenport_11                                                      MBARI   \n",
       "Davenport_11                                                      MBARI   \n",
       "Davenport_15                                                      MBARI   \n",
       "Davenport_15                                                      MBARI   \n",
       "Davenport_19                                                      MBARI   \n",
       "Davenport_19                                                      MBARI   \n",
       "LaJolla_03                                                         NOAA   \n",
       "LaJolla_09                                                         NOAA   \n",
       "LaJolla_15                                                         NOAA   \n",
       "LaJolla_21                                                         NOAA   \n",
       "LaJolla_27                                                         NOAA   \n",
       "Plymouth_06                                                         NOC   \n",
       "Plymouth_13                                                         NOC   \n",
       "Plymouth_19                                                         NOC   \n",
       "Plymouth_29                                                         NOC   \n",
       "Plymouth_30                                                         NOC   \n",
       "Bedford_01                                                   UDalhousie   \n",
       "Bedford_07                                                   UDalhousie   \n",
       "Bedford_13                                                   UDalhousie   \n",
       "Bedford_19                                                   UDalhousie   \n",
       "Bedford_25                                                   UDalhousie   \n",
       "Samples taken / prepared by AWI and analyzed by...                  NaN   \n",
       "[to be added from GLOMICON partner's metadata]                      AWI   \n",
       "\n",
       "                                                   Analyzing Institute  \\\n",
       "sample_name                                                              \n",
       "Samples taken / prepared and analyzed by AWI (c...                 NaN   \n",
       "Framstrait_01                                                      AWI   \n",
       "Framstrait_01                                                      AWI   \n",
       "Framstrait_05                                                      AWI   \n",
       "Framstrait_05                                                      AWI   \n",
       "Framstrait_09                                                      AWI   \n",
       "Framstrait_09                                                      AWI   \n",
       "Framstrait_13                                                      AWI   \n",
       "Framstrait_13                                                      AWI   \n",
       "Framstrait_17                                                      AWI   \n",
       "Framstrait_17                                                      AWI   \n",
       "evenMock_01                                                        AWI   \n",
       "evenMock_02                                                        AWI   \n",
       "evenMock_03                                                        AWI   \n",
       "evenMock_04                                                        AWI   \n",
       "evenMock_05                                                        AWI   \n",
       "bloomMock_01                                                       AWI   \n",
       "bloomMock_02                                                       AWI   \n",
       "bloomMock_03                                                       AWI   \n",
       "bloomMock_04                                                       AWI   \n",
       "bloomMock_05                                                       AWI   \n",
       "blank_01                                                           AWI   \n",
       "Samples taken / prepared by GLOMICON partners a...                 NaN   \n",
       "Davenport_02                                                       AWI   \n",
       "Davenport_02                                                       AWI   \n",
       "Davenport_06                                                       AWI   \n",
       "Davenport_06                                                       AWI   \n",
       "Davenport_11                                                       AWI   \n",
       "Davenport_11                                                       AWI   \n",
       "Davenport_15                                                       AWI   \n",
       "Davenport_15                                                       AWI   \n",
       "Davenport_19                                                       AWI   \n",
       "Davenport_19                                                       AWI   \n",
       "LaJolla_03                                                         AWI   \n",
       "LaJolla_09                                                         AWI   \n",
       "LaJolla_15                                                         AWI   \n",
       "LaJolla_21                                                         AWI   \n",
       "LaJolla_27                                                         AWI   \n",
       "Plymouth_06                                                        AWI   \n",
       "Plymouth_13                                                        AWI   \n",
       "Plymouth_19                                                        AWI   \n",
       "Plymouth_29                                                        AWI   \n",
       "Plymouth_30                                                        AWI   \n",
       "Bedford_01                                                         AWI   \n",
       "Bedford_07                                                         AWI   \n",
       "Bedford_13                                                         AWI   \n",
       "Bedford_19                                                         AWI   \n",
       "Bedford_25                                                         AWI   \n",
       "Samples taken / prepared by AWI and analyzed by...                 NaN   \n",
       "[to be added from GLOMICON partner's metadata]                   MBARI   \n",
       "\n",
       "                                                       cruise_id     station  \\\n",
       "sample_name                                                                    \n",
       "Samples taken / prepared and analyzed by AWI (c...           NaN         NaN   \n",
       "Framstrait_01                                       PS114 (2018)  PS114-36-4   \n",
       "Framstrait_01                                       PS114 (2018)  PS114-36-4   \n",
       "Framstrait_05                                       PS114 (2018)  PS114-36-4   \n",
       "Framstrait_05                                       PS114 (2018)  PS114-36-4   \n",
       "Framstrait_09                                       PS114 (2018)  PS114-36-4   \n",
       "Framstrait_09                                       PS114 (2018)  PS114-36-4   \n",
       "Framstrait_13                                       PS114 (2018)  PS114-36-4   \n",
       "Framstrait_13                                       PS114 (2018)  PS114-36-4   \n",
       "Framstrait_17                                       PS114 (2018)  PS114-36-4   \n",
       "Framstrait_17                                       PS114 (2018)  PS114-36-4   \n",
       "evenMock_01                                                  NaN         NaN   \n",
       "evenMock_02                                                  NaN         NaN   \n",
       "evenMock_03                                                  NaN         NaN   \n",
       "evenMock_04                                                  NaN         NaN   \n",
       "evenMock_05                                                  NaN         NaN   \n",
       "bloomMock_01                                                 NaN         NaN   \n",
       "bloomMock_02                                                 NaN         NaN   \n",
       "bloomMock_03                                                 NaN         NaN   \n",
       "bloomMock_04                                                 NaN         NaN   \n",
       "bloomMock_05                                                 NaN         NaN   \n",
       "blank_01                                                     NaN         NaN   \n",
       "Samples taken / prepared by GLOMICON partners a...           NaN         NaN   \n",
       "Davenport_02                                                 NaN         NaN   \n",
       "Davenport_02                                                 NaN         NaN   \n",
       "Davenport_06                                                 NaN         NaN   \n",
       "Davenport_06                                                 NaN         NaN   \n",
       "Davenport_11                                                 NaN         NaN   \n",
       "Davenport_11                                                 NaN         NaN   \n",
       "Davenport_15                                                 NaN         NaN   \n",
       "Davenport_15                                                 NaN         NaN   \n",
       "Davenport_19                                                 NaN         NaN   \n",
       "Davenport_19                                                 NaN         NaN   \n",
       "LaJolla_03                                                   NaN         NaN   \n",
       "LaJolla_09                                                   NaN         NaN   \n",
       "LaJolla_15                                                   NaN         NaN   \n",
       "LaJolla_21                                                   NaN         NaN   \n",
       "LaJolla_27                                                   NaN         NaN   \n",
       "Plymouth_06                                                  NaN         NaN   \n",
       "Plymouth_13                                                  NaN         NaN   \n",
       "Plymouth_19                                                  NaN         NaN   \n",
       "Plymouth_29                                                  NaN         NaN   \n",
       "Plymouth_30                                                  NaN         NaN   \n",
       "Bedford_01                                                   NaN         NaN   \n",
       "Bedford_07                                                   NaN         NaN   \n",
       "Bedford_13                                                   NaN         NaN   \n",
       "Bedford_19                                                   NaN         NaN   \n",
       "Bedford_25                                                   NaN         NaN   \n",
       "Samples taken / prepared by AWI and analyzed by...           NaN         NaN   \n",
       "[to be added from GLOMICON partner's metadata]      PS114 (2018)  PS114-36-4   \n",
       "\n",
       "                                                   ctd_bottle_no  \\\n",
       "sample_name                                                        \n",
       "Samples taken / prepared and analyzed by AWI (c...           NaN   \n",
       "Framstrait_01                                             pooled   \n",
       "Framstrait_01                                             pooled   \n",
       "Framstrait_05                                             pooled   \n",
       "Framstrait_05                                             pooled   \n",
       "Framstrait_09                                             pooled   \n",
       "Framstrait_09                                             pooled   \n",
       "Framstrait_13                                             pooled   \n",
       "Framstrait_13                                             pooled   \n",
       "Framstrait_17                                             pooled   \n",
       "Framstrait_17                                             pooled   \n",
       "evenMock_01                                                  NaN   \n",
       "evenMock_02                                                  NaN   \n",
       "evenMock_03                                                  NaN   \n",
       "evenMock_04                                                  NaN   \n",
       "evenMock_05                                                  NaN   \n",
       "bloomMock_01                                                 NaN   \n",
       "bloomMock_02                                                 NaN   \n",
       "bloomMock_03                                                 NaN   \n",
       "bloomMock_04                                                 NaN   \n",
       "bloomMock_05                                                 NaN   \n",
       "blank_01                                                     NaN   \n",
       "Samples taken / prepared by GLOMICON partners a...           NaN   \n",
       "Davenport_02                                                 NaN   \n",
       "Davenport_02                                                 NaN   \n",
       "Davenport_06                                                 NaN   \n",
       "Davenport_06                                                 NaN   \n",
       "Davenport_11                                                 NaN   \n",
       "Davenport_11                                                 NaN   \n",
       "Davenport_15                                                 NaN   \n",
       "Davenport_15                                                 NaN   \n",
       "Davenport_19                                                 NaN   \n",
       "Davenport_19                                                 NaN   \n",
       "LaJolla_03                                                   NaN   \n",
       "LaJolla_09                                                   NaN   \n",
       "LaJolla_15                                                   NaN   \n",
       "LaJolla_21                                                   NaN   \n",
       "LaJolla_27                                                   NaN   \n",
       "Plymouth_06                                                  NaN   \n",
       "Plymouth_13                                                  NaN   \n",
       "Plymouth_19                                                  NaN   \n",
       "Plymouth_29                                                  NaN   \n",
       "Plymouth_30                                                  NaN   \n",
       "Bedford_01                                                   NaN   \n",
       "Bedford_07                                                   NaN   \n",
       "Bedford_13                                                   NaN   \n",
       "Bedford_19                                                   NaN   \n",
       "Bedford_25                                                   NaN   \n",
       "Samples taken / prepared by AWI and analyzed by...           NaN   \n",
       "[to be added from GLOMICON partner's metadata]            pooled   \n",
       "\n",
       "                                                   sample_replicate  \\\n",
       "sample_name                                                           \n",
       "Samples taken / prepared and analyzed by AWI (c...              NaN   \n",
       "Framstrait_01                                                    01   \n",
       "Framstrait_01                                                    01   \n",
       "Framstrait_05                                                    05   \n",
       "Framstrait_05                                                    05   \n",
       "Framstrait_09                                                    09   \n",
       "Framstrait_09                                                    09   \n",
       "Framstrait_13                                                    13   \n",
       "Framstrait_13                                                    13   \n",
       "Framstrait_17                                                    17   \n",
       "Framstrait_17                                                    17   \n",
       "evenMock_01                                                      01   \n",
       "evenMock_02                                                      02   \n",
       "evenMock_03                                                      03   \n",
       "evenMock_04                                                      04   \n",
       "evenMock_05                                                      05   \n",
       "bloomMock_01                                                     05   \n",
       "bloomMock_02                                                     11   \n",
       "bloomMock_03                                                     12   \n",
       "bloomMock_04                                                     19   \n",
       "bloomMock_05                                                     23   \n",
       "blank_01                                                         01   \n",
       "Samples taken / prepared by GLOMICON partners a...              NaN   \n",
       "Davenport_02                                                    NaN   \n",
       "Davenport_02                                                    NaN   \n",
       "Davenport_06                                                    NaN   \n",
       "Davenport_06                                                    NaN   \n",
       "Davenport_11                                                    NaN   \n",
       "Davenport_11                                                    NaN   \n",
       "Davenport_15                                                    NaN   \n",
       "Davenport_15                                                    NaN   \n",
       "Davenport_19                                                    NaN   \n",
       "Davenport_19                                                    NaN   \n",
       "LaJolla_03                                                      NaN   \n",
       "LaJolla_09                                                      NaN   \n",
       "LaJolla_15                                                      NaN   \n",
       "LaJolla_21                                                      NaN   \n",
       "LaJolla_27                                                      NaN   \n",
       "Plymouth_06                                                     NaN   \n",
       "Plymouth_13                                                     NaN   \n",
       "Plymouth_19                                                     NaN   \n",
       "Plymouth_29                                                     NaN   \n",
       "Plymouth_30                                                     NaN   \n",
       "Bedford_01                                                      NaN   \n",
       "Bedford_07                                                      NaN   \n",
       "Bedford_13                                                      NaN   \n",
       "Bedford_19                                                      NaN   \n",
       "Bedford_25                                                      NaN   \n",
       "Samples taken / prepared by AWI and analyzed by...              NaN   \n",
       "[to be added from GLOMICON partner's metadata]                   02   \n",
       "\n",
       "                                                                  source_material_id  \\\n",
       "sample_name                                                                            \n",
       "Samples taken / prepared and analyzed by AWI (c...                               NaN   \n",
       "Framstrait_01                                       GLOMICON_INTERCOMP_Framstrait_01   \n",
       "Framstrait_01                                       GLOMICON_INTERCOMP_Framstrait_01   \n",
       "Framstrait_05                                       GLOMICON_INTERCOMP_Framstrait_05   \n",
       "Framstrait_05                                       GLOMICON_INTERCOMP_Framstrait_05   \n",
       "Framstrait_09                                       GLOMICON_INTERCOMP_Framstrait_09   \n",
       "Framstrait_09                                       GLOMICON_INTERCOMP_Framstrait_09   \n",
       "Framstrait_13                                       GLOMICON_INTERCOMP_Framstrait_13   \n",
       "Framstrait_13                                       GLOMICON_INTERCOMP_Framstrait_13   \n",
       "Framstrait_17                                       GLOMICON_INTERCOMP_Framstrait_17   \n",
       "Framstrait_17                                       GLOMICON_INTERCOMP_Framstrait_17   \n",
       "evenMock_01                                           GLOMICON_INTERCOMP_evenMock_01   \n",
       "evenMock_02                                           GLOMICON_INTERCOMP_evenMock_02   \n",
       "evenMock_03                                           GLOMICON_INTERCOMP_evenMock_03   \n",
       "evenMock_04                                           GLOMICON_INTERCOMP_evenMock_04   \n",
       "evenMock_05                                           GLOMICON_INTERCOMP_evenMock_05   \n",
       "bloomMock_01                                         GLOMICON_INTERCOMP_bloomMock_05   \n",
       "bloomMock_02                                         GLOMICON_INTERCOMP_bloomMock_11   \n",
       "bloomMock_03                                         GLOMICON_INTERCOMP_bloomMock_12   \n",
       "bloomMock_04                                         GLOMICON_INTERCOMP_bloomMock_19   \n",
       "bloomMock_05                                         GLOMICON_INTERCOMP_bloomMock_23   \n",
       "blank_01                                                 GLOMICON_INTERCOMP_blank_01   \n",
       "Samples taken / prepared by GLOMICON partners a...                               NaN   \n",
       "Davenport_02                                                                     NaN   \n",
       "Davenport_02                                                                     NaN   \n",
       "Davenport_06                                                                     NaN   \n",
       "Davenport_06                                                                     NaN   \n",
       "Davenport_11                                                                     NaN   \n",
       "Davenport_11                                                                     NaN   \n",
       "Davenport_15                                                                     NaN   \n",
       "Davenport_15                                                                     NaN   \n",
       "Davenport_19                                                                     NaN   \n",
       "Davenport_19                                                                     NaN   \n",
       "LaJolla_03                                                                       NaN   \n",
       "LaJolla_09                                                                       NaN   \n",
       "LaJolla_15                                                                       NaN   \n",
       "LaJolla_21                                                                       NaN   \n",
       "LaJolla_27                                                                       NaN   \n",
       "Plymouth_06                                                                      NaN   \n",
       "Plymouth_13                                                                      NaN   \n",
       "Plymouth_19                                                                      NaN   \n",
       "Plymouth_29                                                                      NaN   \n",
       "Plymouth_30                                                                      NaN   \n",
       "Bedford_01                                                                       NaN   \n",
       "Bedford_07                                                                       NaN   \n",
       "Bedford_13                                                                       NaN   \n",
       "Bedford_19                                                                       NaN   \n",
       "Bedford_25                                                                       NaN   \n",
       "Samples taken / prepared by AWI and analyzed by...                               NaN   \n",
       "[to be added from GLOMICON partner's metadata]      GLOMICON_INTERCOMP_Framstrait_02   \n",
       "\n",
       "                                                                                        sample_title  \\\n",
       "sample_name                                                                                            \n",
       "Samples taken / prepared and analyzed by AWI (c...                                               NaN   \n",
       "Framstrait_01                                       Seawater sample Fram Strait (GLOMICON_INTERCOMP)   \n",
       "Framstrait_01                                       Seawater sample Fram Strait (GLOMICON_INTERCOMP)   \n",
       "Framstrait_05                                       Seawater sample Fram Strait (GLOMICON_INTERCOMP)   \n",
       "Framstrait_05                                       Seawater sample Fram Strait (GLOMICON_INTERCOMP)   \n",
       "Framstrait_09                                       Seawater sample Fram Strait (GLOMICON_INTERCOMP)   \n",
       "Framstrait_09                                       Seawater sample Fram Strait (GLOMICON_INTERCOMP)   \n",
       "Framstrait_13                                       Seawater sample Fram Strait (GLOMICON_INTERCOMP)   \n",
       "Framstrait_13                                       Seawater sample Fram Strait (GLOMICON_INTERCOMP)   \n",
       "Framstrait_17                                       Seawater sample Fram Strait (GLOMICON_INTERCOMP)   \n",
       "Framstrait_17                                       Seawater sample Fram Strait (GLOMICON_INTERCOMP)   \n",
       "evenMock_01                                                 Even mock community (GLOMICON_INTERCOMP)   \n",
       "evenMock_02                                                 Even mock community (GLOMICON_INTERCOMP)   \n",
       "evenMock_03                                                 Even mock community (GLOMICON_INTERCOMP)   \n",
       "evenMock_04                                                 Even mock community (GLOMICON_INTERCOMP)   \n",
       "evenMock_05                                                 Even mock community (GLOMICON_INTERCOMP)   \n",
       "bloomMock_01                                               Bloom mock community (GLOMICON_INTERCOMP)   \n",
       "bloomMock_02                                               Bloom mock community (GLOMICON_INTERCOMP)   \n",
       "bloomMock_03                                               Bloom mock community (GLOMICON_INTERCOMP)   \n",
       "bloomMock_04                                               Bloom mock community (GLOMICON_INTERCOMP)   \n",
       "bloomMock_05                                               Bloom mock community (GLOMICON_INTERCOMP)   \n",
       "blank_01                                                        Blank PCR water (GLOMICON_INTERCOMP)   \n",
       "Samples taken / prepared by GLOMICON partners a...                                               NaN   \n",
       "Davenport_02                                                                                     NaN   \n",
       "Davenport_02                                                                                     NaN   \n",
       "Davenport_06                                                                                     NaN   \n",
       "Davenport_06                                                                                     NaN   \n",
       "Davenport_11                                                                                     NaN   \n",
       "Davenport_11                                                                                     NaN   \n",
       "Davenport_15                                                                                     NaN   \n",
       "Davenport_15                                                                                     NaN   \n",
       "Davenport_19                                                                                     NaN   \n",
       "Davenport_19                                                                                     NaN   \n",
       "LaJolla_03                                                                                       NaN   \n",
       "LaJolla_09                                                                                       NaN   \n",
       "LaJolla_15                                                                                       NaN   \n",
       "LaJolla_21                                                                                       NaN   \n",
       "LaJolla_27                                                                                       NaN   \n",
       "Plymouth_06                                                                                      NaN   \n",
       "Plymouth_13                                                                                      NaN   \n",
       "Plymouth_19                                                                                      NaN   \n",
       "Plymouth_29                                                                                      NaN   \n",
       "Plymouth_30                                                                                      NaN   \n",
       "Bedford_01                                                                                       NaN   \n",
       "Bedford_07                                                                                       NaN   \n",
       "Bedford_13                                                                                       NaN   \n",
       "Bedford_19                                                                                       NaN   \n",
       "Bedford_25                                                                                       NaN   \n",
       "Samples taken / prepared by AWI and analyzed by...                                               NaN   \n",
       "[to be added from GLOMICON partner's metadata]      Seawater sample Fram Strait (GLOMICON_INTERCOMP)   \n",
       "\n",
       "                                                                                       notes_sampling  \\\n",
       "sample_name                                                                                             \n",
       "Samples taken / prepared and analyzed by AWI (c...                                                NaN   \n",
       "Framstrait_01                                       Surface waters from the northern part of Fram ...   \n",
       "Framstrait_01                                       Surface waters from the northern part of Fram ...   \n",
       "Framstrait_05                                       Surface waters from the northern part of Fram ...   \n",
       "Framstrait_05                                       Surface waters from the northern part of Fram ...   \n",
       "Framstrait_09                                       Surface waters from the northern part of Fram ...   \n",
       "Framstrait_09                                       Surface waters from the northern part of Fram ...   \n",
       "Framstrait_13                                       Surface waters from the northern part of Fram ...   \n",
       "Framstrait_13                                       Surface waters from the northern part of Fram ...   \n",
       "Framstrait_17                                       Surface waters from the northern part of Fram ...   \n",
       "Framstrait_17                                       Surface waters from the northern part of Fram ...   \n",
       "evenMock_01                                         mock community with even 18S rRNA gene copy nu...   \n",
       "evenMock_02                                         mock community with even 18S rRNA gene copy nu...   \n",
       "evenMock_03                                         mock community with even 18S rRNA gene copy nu...   \n",
       "evenMock_04                                         mock community with even 18S rRNA gene copy nu...   \n",
       "evenMock_05                                         mock community with even 18S rRNA gene copy nu...   \n",
       "bloomMock_01                                        mock community with uneven 18S rRNA gene copy ...   \n",
       "bloomMock_02                                        mock community with uneven 18S rRNA gene copy ...   \n",
       "bloomMock_03                                        mock community with uneven 18S rRNA gene copy ...   \n",
       "bloomMock_04                                        mock community with uneven 18S rRNA gene copy ...   \n",
       "bloomMock_05                                        mock community with uneven 18S rRNA gene copy ...   \n",
       "blank_01                                            PCR water used to dilute mock community DNA ex...   \n",
       "Samples taken / prepared by GLOMICON partners a...                                                NaN   \n",
       "Davenport_02                                                                                      NaN   \n",
       "Davenport_02                                                                                      NaN   \n",
       "Davenport_06                                                                                      NaN   \n",
       "Davenport_06                                                                                      NaN   \n",
       "Davenport_11                                                                                      NaN   \n",
       "Davenport_11                                                                                      NaN   \n",
       "Davenport_15                                                                                      NaN   \n",
       "Davenport_15                                                                                      NaN   \n",
       "Davenport_19                                                                                      NaN   \n",
       "Davenport_19                                                                                      NaN   \n",
       "LaJolla_03                                                                                        NaN   \n",
       "LaJolla_09                                                                                        NaN   \n",
       "LaJolla_15                                                                                        NaN   \n",
       "LaJolla_21                                                                                        NaN   \n",
       "LaJolla_27                                                                                        NaN   \n",
       "Plymouth_06                                                                                       NaN   \n",
       "Plymouth_13                                                                                       NaN   \n",
       "Plymouth_19                                                                                       NaN   \n",
       "Plymouth_29                                                                                       NaN   \n",
       "Plymouth_30                                                                                       NaN   \n",
       "Bedford_01                                                                                        NaN   \n",
       "Bedford_07                                                                                        NaN   \n",
       "Bedford_13                                                                                        NaN   \n",
       "Bedford_19                                                                                        NaN   \n",
       "Bedford_25                                                                                        NaN   \n",
       "Samples taken / prepared by AWI and analyzed by...                                                NaN   \n",
       "[to be added from GLOMICON partner's metadata]      Surface waters from the northern part of Fram ...   \n",
       "\n",
       "                                                            project_id  ...  \\\n",
       "sample_name                                                             ...   \n",
       "Samples taken / prepared and analyzed by AWI (c...                 NaN  ...   \n",
       "Framstrait_01                                       GLOMICON_INTERCOMP  ...   \n",
       "Framstrait_01                                       GLOMICON_INTERCOMP  ...   \n",
       "Framstrait_05                                       GLOMICON_INTERCOMP  ...   \n",
       "Framstrait_05                                       GLOMICON_INTERCOMP  ...   \n",
       "Framstrait_09                                       GLOMICON_INTERCOMP  ...   \n",
       "Framstrait_09                                       GLOMICON_INTERCOMP  ...   \n",
       "Framstrait_13                                       GLOMICON_INTERCOMP  ...   \n",
       "Framstrait_13                                       GLOMICON_INTERCOMP  ...   \n",
       "Framstrait_17                                       GLOMICON_INTERCOMP  ...   \n",
       "Framstrait_17                                       GLOMICON_INTERCOMP  ...   \n",
       "evenMock_01                                         GLOMICON_INTERCOMP  ...   \n",
       "evenMock_02                                         GLOMICON_INTERCOMP  ...   \n",
       "evenMock_03                                         GLOMICON_INTERCOMP  ...   \n",
       "evenMock_04                                         GLOMICON_INTERCOMP  ...   \n",
       "evenMock_05                                         GLOMICON_INTERCOMP  ...   \n",
       "bloomMock_01                                        GLOMICON_INTERCOMP  ...   \n",
       "bloomMock_02                                        GLOMICON_INTERCOMP  ...   \n",
       "bloomMock_03                                        GLOMICON_INTERCOMP  ...   \n",
       "bloomMock_04                                        GLOMICON_INTERCOMP  ...   \n",
       "bloomMock_05                                        GLOMICON_INTERCOMP  ...   \n",
       "blank_01                                            GLOMICON_INTERCOMP  ...   \n",
       "Samples taken / prepared by GLOMICON partners a...                 NaN  ...   \n",
       "Davenport_02                                                       NaN  ...   \n",
       "Davenport_02                                                       NaN  ...   \n",
       "Davenport_06                                                       NaN  ...   \n",
       "Davenport_06                                                       NaN  ...   \n",
       "Davenport_11                                                       NaN  ...   \n",
       "Davenport_11                                                       NaN  ...   \n",
       "Davenport_15                                                       NaN  ...   \n",
       "Davenport_15                                                       NaN  ...   \n",
       "Davenport_19                                                       NaN  ...   \n",
       "Davenport_19                                                       NaN  ...   \n",
       "LaJolla_03                                                         NaN  ...   \n",
       "LaJolla_09                                                         NaN  ...   \n",
       "LaJolla_15                                                         NaN  ...   \n",
       "LaJolla_21                                                         NaN  ...   \n",
       "LaJolla_27                                                         NaN  ...   \n",
       "Plymouth_06                                                        NaN  ...   \n",
       "Plymouth_13                                                        NaN  ...   \n",
       "Plymouth_19                                                        NaN  ...   \n",
       "Plymouth_29                                                        NaN  ...   \n",
       "Plymouth_30                                                        NaN  ...   \n",
       "Bedford_01                                                         NaN  ...   \n",
       "Bedford_07                                                         NaN  ...   \n",
       "Bedford_13                                                         NaN  ...   \n",
       "Bedford_19                                                         NaN  ...   \n",
       "Bedford_25                                                         NaN  ...   \n",
       "Samples taken / prepared by AWI and analyzed by...                 NaN  ...   \n",
       "[to be added from GLOMICON partner's metadata]      GLOMICON_INTERCOMP  ...   \n",
       "\n",
       "                                                   amplicon_sequenced  \\\n",
       "sample_name                                                             \n",
       "Samples taken / prepared and analyzed by AWI (c...                NaN   \n",
       "Framstrait_01                                                  18S V4   \n",
       "Framstrait_01                                                  18S V4   \n",
       "Framstrait_05                                                  18S V4   \n",
       "Framstrait_05                                                  18S V4   \n",
       "Framstrait_09                                                  18S V4   \n",
       "Framstrait_09                                                  18S V4   \n",
       "Framstrait_13                                                  18S V4   \n",
       "Framstrait_13                                                  18S V4   \n",
       "Framstrait_17                                                  18S V4   \n",
       "Framstrait_17                                                  18S V4   \n",
       "evenMock_01                                                    18S V4   \n",
       "evenMock_02                                                    18S V4   \n",
       "evenMock_03                                                    18S V4   \n",
       "evenMock_04                                                    18S V4   \n",
       "evenMock_05                                                    18S V4   \n",
       "bloomMock_01                                                   18S V4   \n",
       "bloomMock_02                                                   18S V4   \n",
       "bloomMock_03                                                   18S V4   \n",
       "bloomMock_04                                                   18S V4   \n",
       "bloomMock_05                                                   18S V4   \n",
       "blank_01                                                       18S V4   \n",
       "Samples taken / prepared by GLOMICON partners a...                NaN   \n",
       "Davenport_02                                                   18S V4   \n",
       "Davenport_02                                                   18S V4   \n",
       "Davenport_06                                                   18S V4   \n",
       "Davenport_06                                                   18S V4   \n",
       "Davenport_11                                                   18S V4   \n",
       "Davenport_11                                                   18S V4   \n",
       "Davenport_15                                                   18S V4   \n",
       "Davenport_15                                                   18S V4   \n",
       "Davenport_19                                                   18S V4   \n",
       "Davenport_19                                                   18S V4   \n",
       "LaJolla_03                                                        NaN   \n",
       "LaJolla_09                                                        NaN   \n",
       "LaJolla_15                                                        NaN   \n",
       "LaJolla_21                                                        NaN   \n",
       "LaJolla_27                                                        NaN   \n",
       "Plymouth_06                                                       NaN   \n",
       "Plymouth_13                                                       NaN   \n",
       "Plymouth_19                                                       NaN   \n",
       "Plymouth_29                                                       NaN   \n",
       "Plymouth_30                                                       NaN   \n",
       "Bedford_01                                                        NaN   \n",
       "Bedford_07                                                        NaN   \n",
       "Bedford_13                                                        NaN   \n",
       "Bedford_19                                                        NaN   \n",
       "Bedford_25                                                        NaN   \n",
       "Samples taken / prepared by AWI and analyzed by...                NaN   \n",
       "[to be added from GLOMICON partner's metadata]                    NaN   \n",
       "\n",
       "                                                   target_gene  \\\n",
       "sample_name                                                      \n",
       "Samples taken / prepared and analyzed by AWI (c...         NaN   \n",
       "Framstrait_01                                         18S rRNA   \n",
       "Framstrait_01                                         18S rRNA   \n",
       "Framstrait_05                                         18S rRNA   \n",
       "Framstrait_05                                         18S rRNA   \n",
       "Framstrait_09                                         18S rRNA   \n",
       "Framstrait_09                                         18S rRNA   \n",
       "Framstrait_13                                         18S rRNA   \n",
       "Framstrait_13                                         18S rRNA   \n",
       "Framstrait_17                                         18S rRNA   \n",
       "Framstrait_17                                         18S rRNA   \n",
       "evenMock_01                                           18S rRNA   \n",
       "evenMock_02                                           18S rRNA   \n",
       "evenMock_03                                           18S rRNA   \n",
       "evenMock_04                                           18S rRNA   \n",
       "evenMock_05                                           18S rRNA   \n",
       "bloomMock_01                                          18S rRNA   \n",
       "bloomMock_02                                          18S rRNA   \n",
       "bloomMock_03                                          18S rRNA   \n",
       "bloomMock_04                                          18S rRNA   \n",
       "bloomMock_05                                          18S rRNA   \n",
       "blank_01                                              18S rRNA   \n",
       "Samples taken / prepared by GLOMICON partners a...         NaN   \n",
       "Davenport_02                                          18S rRNA   \n",
       "Davenport_02                                          18S rRNA   \n",
       "Davenport_06                                          18S rRNA   \n",
       "Davenport_06                                          18S rRNA   \n",
       "Davenport_11                                          18S rRNA   \n",
       "Davenport_11                                          18S rRNA   \n",
       "Davenport_15                                          18S rRNA   \n",
       "Davenport_15                                          18S rRNA   \n",
       "Davenport_19                                          18S rRNA   \n",
       "Davenport_19                                          18S rRNA   \n",
       "LaJolla_03                                                 NaN   \n",
       "LaJolla_09                                                 NaN   \n",
       "LaJolla_15                                                 NaN   \n",
       "LaJolla_21                                                 NaN   \n",
       "LaJolla_27                                                 NaN   \n",
       "Plymouth_06                                                NaN   \n",
       "Plymouth_13                                                NaN   \n",
       "Plymouth_19                                                NaN   \n",
       "Plymouth_29                                                NaN   \n",
       "Plymouth_30                                                NaN   \n",
       "Bedford_01                                                 NaN   \n",
       "Bedford_07                                                 NaN   \n",
       "Bedford_13                                                 NaN   \n",
       "Bedford_19                                                 NaN   \n",
       "Bedford_25                                                 NaN   \n",
       "Samples taken / prepared by AWI and analyzed by...         NaN   \n",
       "[to be added from GLOMICON partner's metadata]             NaN   \n",
       "\n",
       "                                                   target_subfragment  \\\n",
       "sample_name                                                             \n",
       "Samples taken / prepared and analyzed by AWI (c...                NaN   \n",
       "Framstrait_01                                                      V4   \n",
       "Framstrait_01                                                      V4   \n",
       "Framstrait_05                                                      V4   \n",
       "Framstrait_05                                                      V4   \n",
       "Framstrait_09                                                      V4   \n",
       "Framstrait_09                                                      V4   \n",
       "Framstrait_13                                                      V4   \n",
       "Framstrait_13                                                      V4   \n",
       "Framstrait_17                                                      V4   \n",
       "Framstrait_17                                                      V4   \n",
       "evenMock_01                                                        V4   \n",
       "evenMock_02                                                        V4   \n",
       "evenMock_03                                                        V4   \n",
       "evenMock_04                                                        V4   \n",
       "evenMock_05                                                        V4   \n",
       "bloomMock_01                                                       V4   \n",
       "bloomMock_02                                                       V4   \n",
       "bloomMock_03                                                       V4   \n",
       "bloomMock_04                                                       V4   \n",
       "bloomMock_05                                                       V4   \n",
       "blank_01                                                           V4   \n",
       "Samples taken / prepared by GLOMICON partners a...                NaN   \n",
       "Davenport_02                                                       V4   \n",
       "Davenport_02                                                       V4   \n",
       "Davenport_06                                                       V4   \n",
       "Davenport_06                                                       V4   \n",
       "Davenport_11                                                       V4   \n",
       "Davenport_11                                                       V4   \n",
       "Davenport_15                                                       V4   \n",
       "Davenport_15                                                       V4   \n",
       "Davenport_19                                                       V4   \n",
       "Davenport_19                                                       V4   \n",
       "LaJolla_03                                                        NaN   \n",
       "LaJolla_09                                                        NaN   \n",
       "LaJolla_15                                                        NaN   \n",
       "LaJolla_21                                                        NaN   \n",
       "LaJolla_27                                                        NaN   \n",
       "Plymouth_06                                                       NaN   \n",
       "Plymouth_13                                                       NaN   \n",
       "Plymouth_19                                                       NaN   \n",
       "Plymouth_29                                                       NaN   \n",
       "Plymouth_30                                                       NaN   \n",
       "Bedford_01                                                        NaN   \n",
       "Bedford_07                                                        NaN   \n",
       "Bedford_13                                                        NaN   \n",
       "Bedford_19                                                        NaN   \n",
       "Bedford_25                                                        NaN   \n",
       "Samples taken / prepared by AWI and analyzed by...                NaN   \n",
       "[to be added from GLOMICON partner's metadata]                    NaN   \n",
       "\n",
       "                                                                 pcr_primer_forward  \\\n",
       "sample_name                                                                           \n",
       "Samples taken / prepared and analyzed by AWI (c...                              NaN   \n",
       "Framstrait_01                                       5'-GCG GTA ATT CCA GCT CCA A-3'   \n",
       "Framstrait_01                                       5'-GCG GTA ATT CCA GCT CCA A-3'   \n",
       "Framstrait_05                                       5'-GCG GTA ATT CCA GCT CCA A-3'   \n",
       "Framstrait_05                                       5'-GCG GTA ATT CCA GCT CCA A-3'   \n",
       "Framstrait_09                                       5'-GCG GTA ATT CCA GCT CCA A-3'   \n",
       "Framstrait_09                                       5'-GCG GTA ATT CCA GCT CCA A-3'   \n",
       "Framstrait_13                                       5'-GCG GTA ATT CCA GCT CCA A-3'   \n",
       "Framstrait_13                                       5'-GCG GTA ATT CCA GCT CCA A-3'   \n",
       "Framstrait_17                                       5'-GCG GTA ATT CCA GCT CCA A-3'   \n",
       "Framstrait_17                                       5'-GCG GTA ATT CCA GCT CCA A-3'   \n",
       "evenMock_01                                         5'-GCG GTA ATT CCA GCT CCA A-3'   \n",
       "evenMock_02                                         5'-GCG GTA ATT CCA GCT CCA A-3'   \n",
       "evenMock_03                                         5'-GCG GTA ATT CCA GCT CCA A-3'   \n",
       "evenMock_04                                         5'-GCG GTA ATT CCA GCT CCA A-3'   \n",
       "evenMock_05                                         5'-GCG GTA ATT CCA GCT CCA A-3'   \n",
       "bloomMock_01                                        5'-GCG GTA ATT CCA GCT CCA A-3'   \n",
       "bloomMock_02                                        5'-GCG GTA ATT CCA GCT CCA A-3'   \n",
       "bloomMock_03                                        5'-GCG GTA ATT CCA GCT CCA A-3'   \n",
       "bloomMock_04                                        5'-GCG GTA ATT CCA GCT CCA A-3'   \n",
       "bloomMock_05                                        5'-GCG GTA ATT CCA GCT CCA A-3'   \n",
       "blank_01                                            5'-GCG GTA ATT CCA GCT CCA A-3'   \n",
       "Samples taken / prepared by GLOMICON partners a...                              NaN   \n",
       "Davenport_02                                        5'-GCG GTA ATT CCA GCT CCA A-3'   \n",
       "Davenport_02                                        5'-GCG GTA ATT CCA GCT CCA A-3'   \n",
       "Davenport_06                                        5'-GCG GTA ATT CCA GCT CCA A-3'   \n",
       "Davenport_06                                        5'-GCG GTA ATT CCA GCT CCA A-3'   \n",
       "Davenport_11                                        5'-GCG GTA ATT CCA GCT CCA A-3'   \n",
       "Davenport_11                                        5'-GCG GTA ATT CCA GCT CCA A-3'   \n",
       "Davenport_15                                        5'-GCG GTA ATT CCA GCT CCA A-3'   \n",
       "Davenport_15                                        5'-GCG GTA ATT CCA GCT CCA A-3'   \n",
       "Davenport_19                                        5'-GCG GTA ATT CCA GCT CCA A-3'   \n",
       "Davenport_19                                        5'-GCG GTA ATT CCA GCT CCA A-3'   \n",
       "LaJolla_03                                                                      NaN   \n",
       "LaJolla_09                                                                      NaN   \n",
       "LaJolla_15                                                                      NaN   \n",
       "LaJolla_21                                                                      NaN   \n",
       "LaJolla_27                                                                      NaN   \n",
       "Plymouth_06                                                                     NaN   \n",
       "Plymouth_13                                                                     NaN   \n",
       "Plymouth_19                                                                     NaN   \n",
       "Plymouth_29                                                                     NaN   \n",
       "Plymouth_30                                                                     NaN   \n",
       "Bedford_01                                                                      NaN   \n",
       "Bedford_07                                                                      NaN   \n",
       "Bedford_13                                                                      NaN   \n",
       "Bedford_19                                                                      NaN   \n",
       "Bedford_25                                                                      NaN   \n",
       "Samples taken / prepared by AWI and analyzed by...                              NaN   \n",
       "[to be added from GLOMICON partner's metadata]                                  NaN   \n",
       "\n",
       "                                                                pcr_primer_reverse  \\\n",
       "sample_name                                                                          \n",
       "Samples taken / prepared and analyzed by AWI (c...                             NaN   \n",
       "Framstrait_01                                       5'-AC TTT CGT TCT TGA TYR R-3'   \n",
       "Framstrait_01                                       5'-AC TTT CGT TCT TGA TYR R-3'   \n",
       "Framstrait_05                                       5'-AC TTT CGT TCT TGA TYR R-3'   \n",
       "Framstrait_05                                       5'-AC TTT CGT TCT TGA TYR R-3'   \n",
       "Framstrait_09                                       5'-AC TTT CGT TCT TGA TYR R-3'   \n",
       "Framstrait_09                                       5'-AC TTT CGT TCT TGA TYR R-3'   \n",
       "Framstrait_13                                       5'-AC TTT CGT TCT TGA TYR R-3'   \n",
       "Framstrait_13                                       5'-AC TTT CGT TCT TGA TYR R-3'   \n",
       "Framstrait_17                                       5'-AC TTT CGT TCT TGA TYR R-3'   \n",
       "Framstrait_17                                       5'-AC TTT CGT TCT TGA TYR R-3'   \n",
       "evenMock_01                                         5'-AC TTT CGT TCT TGA TYR R-3'   \n",
       "evenMock_02                                         5'-AC TTT CGT TCT TGA TYR R-3'   \n",
       "evenMock_03                                         5'-AC TTT CGT TCT TGA TYR R-3'   \n",
       "evenMock_04                                         5'-AC TTT CGT TCT TGA TYR R-3'   \n",
       "evenMock_05                                         5'-AC TTT CGT TCT TGA TYR R-3'   \n",
       "bloomMock_01                                        5'-AC TTT CGT TCT TGA TYR R-3'   \n",
       "bloomMock_02                                        5'-AC TTT CGT TCT TGA TYR R-3'   \n",
       "bloomMock_03                                        5'-AC TTT CGT TCT TGA TYR R-3'   \n",
       "bloomMock_04                                        5'-AC TTT CGT TCT TGA TYR R-3'   \n",
       "bloomMock_05                                        5'-AC TTT CGT TCT TGA TYR R-3'   \n",
       "blank_01                                            5'-AC TTT CGT TCT TGA TYR R-3'   \n",
       "Samples taken / prepared by GLOMICON partners a...                             NaN   \n",
       "Davenport_02                                        5'-AC TTT CGT TCT TGA TYR R-3'   \n",
       "Davenport_02                                        5'-AC TTT CGT TCT TGA TYR R-3'   \n",
       "Davenport_06                                        5'-AC TTT CGT TCT TGA TYR R-3'   \n",
       "Davenport_06                                        5'-AC TTT CGT TCT TGA TYR R-3'   \n",
       "Davenport_11                                        5'-AC TTT CGT TCT TGA TYR R-3'   \n",
       "Davenport_11                                        5'-AC TTT CGT TCT TGA TYR R-3'   \n",
       "Davenport_15                                        5'-AC TTT CGT TCT TGA TYR R-3'   \n",
       "Davenport_15                                        5'-AC TTT CGT TCT TGA TYR R-3'   \n",
       "Davenport_19                                        5'-AC TTT CGT TCT TGA TYR R-3'   \n",
       "Davenport_19                                        5'-AC TTT CGT TCT TGA TYR R-3'   \n",
       "LaJolla_03                                                                     NaN   \n",
       "LaJolla_09                                                                     NaN   \n",
       "LaJolla_15                                                                     NaN   \n",
       "LaJolla_21                                                                     NaN   \n",
       "LaJolla_27                                                                     NaN   \n",
       "Plymouth_06                                                                    NaN   \n",
       "Plymouth_13                                                                    NaN   \n",
       "Plymouth_19                                                                    NaN   \n",
       "Plymouth_29                                                                    NaN   \n",
       "Plymouth_30                                                                    NaN   \n",
       "Bedford_01                                                                     NaN   \n",
       "Bedford_07                                                                     NaN   \n",
       "Bedford_13                                                                     NaN   \n",
       "Bedford_19                                                                     NaN   \n",
       "Bedford_25                                                                     NaN   \n",
       "Samples taken / prepared by AWI and analyzed by...                             NaN   \n",
       "[to be added from GLOMICON partner's metadata]                                 NaN   \n",
       "\n",
       "                                                   pcr_primer_name_forward  \\\n",
       "sample_name                                                                  \n",
       "Samples taken / prepared and analyzed by AWI (c...                     NaN   \n",
       "Framstrait_01                                                       528i F   \n",
       "Framstrait_01                                                       528i F   \n",
       "Framstrait_05                                                       528i F   \n",
       "Framstrait_05                                                       528i F   \n",
       "Framstrait_09                                                       528i F   \n",
       "Framstrait_09                                                       528i F   \n",
       "Framstrait_13                                                       528i F   \n",
       "Framstrait_13                                                       528i F   \n",
       "Framstrait_17                                                       528i F   \n",
       "Framstrait_17                                                       528i F   \n",
       "evenMock_01                                                         528i F   \n",
       "evenMock_02                                                         528i F   \n",
       "evenMock_03                                                         528i F   \n",
       "evenMock_04                                                         528i F   \n",
       "evenMock_05                                                         528i F   \n",
       "bloomMock_01                                                        528i F   \n",
       "bloomMock_02                                                        528i F   \n",
       "bloomMock_03                                                        528i F   \n",
       "bloomMock_04                                                        528i F   \n",
       "bloomMock_05                                                        528i F   \n",
       "blank_01                                                            528i F   \n",
       "Samples taken / prepared by GLOMICON partners a...                     NaN   \n",
       "Davenport_02                                                        528i F   \n",
       "Davenport_02                                                        528i F   \n",
       "Davenport_06                                                        528i F   \n",
       "Davenport_06                                                        528i F   \n",
       "Davenport_11                                                        528i F   \n",
       "Davenport_11                                                        528i F   \n",
       "Davenport_15                                                        528i F   \n",
       "Davenport_15                                                        528i F   \n",
       "Davenport_19                                                        528i F   \n",
       "Davenport_19                                                        528i F   \n",
       "LaJolla_03                                                             NaN   \n",
       "LaJolla_09                                                             NaN   \n",
       "LaJolla_15                                                             NaN   \n",
       "LaJolla_21                                                             NaN   \n",
       "LaJolla_27                                                             NaN   \n",
       "Plymouth_06                                                            NaN   \n",
       "Plymouth_13                                                            NaN   \n",
       "Plymouth_19                                                            NaN   \n",
       "Plymouth_29                                                            NaN   \n",
       "Plymouth_30                                                            NaN   \n",
       "Bedford_01                                                             NaN   \n",
       "Bedford_07                                                             NaN   \n",
       "Bedford_13                                                             NaN   \n",
       "Bedford_19                                                             NaN   \n",
       "Bedford_25                                                             NaN   \n",
       "Samples taken / prepared by AWI and analyzed by...                     NaN   \n",
       "[to be added from GLOMICON partner's metadata]                         NaN   \n",
       "\n",
       "                                                   pcr_primer_name_reverse  \\\n",
       "sample_name                                                                  \n",
       "Samples taken / prepared and analyzed by AWI (c...                     NaN   \n",
       "Framstrait_01                                                       964i R   \n",
       "Framstrait_01                                                       964i R   \n",
       "Framstrait_05                                                       964i R   \n",
       "Framstrait_05                                                       964i R   \n",
       "Framstrait_09                                                       964i R   \n",
       "Framstrait_09                                                       964i R   \n",
       "Framstrait_13                                                       964i R   \n",
       "Framstrait_13                                                       964i R   \n",
       "Framstrait_17                                                       964i R   \n",
       "Framstrait_17                                                       964i R   \n",
       "evenMock_01                                                         964i R   \n",
       "evenMock_02                                                         964i R   \n",
       "evenMock_03                                                         964i R   \n",
       "evenMock_04                                                         964i R   \n",
       "evenMock_05                                                         964i R   \n",
       "bloomMock_01                                                        964i R   \n",
       "bloomMock_02                                                        964i R   \n",
       "bloomMock_03                                                        964i R   \n",
       "bloomMock_04                                                        964i R   \n",
       "bloomMock_05                                                        964i R   \n",
       "blank_01                                                            964i R   \n",
       "Samples taken / prepared by GLOMICON partners a...                     NaN   \n",
       "Davenport_02                                                        964i R   \n",
       "Davenport_02                                                        964i R   \n",
       "Davenport_06                                                        964i R   \n",
       "Davenport_06                                                        964i R   \n",
       "Davenport_11                                                        964i R   \n",
       "Davenport_11                                                        964i R   \n",
       "Davenport_15                                                        964i R   \n",
       "Davenport_15                                                        964i R   \n",
       "Davenport_19                                                        964i R   \n",
       "Davenport_19                                                        964i R   \n",
       "LaJolla_03                                                             NaN   \n",
       "LaJolla_09                                                             NaN   \n",
       "LaJolla_15                                                             NaN   \n",
       "LaJolla_21                                                             NaN   \n",
       "LaJolla_27                                                             NaN   \n",
       "Plymouth_06                                                            NaN   \n",
       "Plymouth_13                                                            NaN   \n",
       "Plymouth_19                                                            NaN   \n",
       "Plymouth_29                                                            NaN   \n",
       "Plymouth_30                                                            NaN   \n",
       "Bedford_01                                                             NaN   \n",
       "Bedford_07                                                             NaN   \n",
       "Bedford_13                                                             NaN   \n",
       "Bedford_19                                                             NaN   \n",
       "Bedford_25                                                             NaN   \n",
       "Samples taken / prepared by AWI and analyzed by...                     NaN   \n",
       "[to be added from GLOMICON partner's metadata]                         NaN   \n",
       "\n",
       "                                                                                 pcr_primer_reference  \\\n",
       "sample_name                                                                                             \n",
       "Samples taken / prepared and analyzed by AWI (c...                                                NaN   \n",
       "Framstrait_01                                       528F: Elwood et al. (1985) doi:10.1093/oxfordj...   \n",
       "Framstrait_01                                       528F: Elwood et al. (1985) doi:10.1093/oxfordj...   \n",
       "Framstrait_05                                       528F: Elwood et al. (1985) doi:10.1093/oxfordj...   \n",
       "Framstrait_05                                       528F: Elwood et al. (1985) doi:10.1093/oxfordj...   \n",
       "Framstrait_09                                       528F: Elwood et al. (1985) doi:10.1093/oxfordj...   \n",
       "Framstrait_09                                       528F: Elwood et al. (1985) doi:10.1093/oxfordj...   \n",
       "Framstrait_13                                       528F: Elwood et al. (1985) doi:10.1093/oxfordj...   \n",
       "Framstrait_13                                       528F: Elwood et al. (1985) doi:10.1093/oxfordj...   \n",
       "Framstrait_17                                       528F: Elwood et al. (1985) doi:10.1093/oxfordj...   \n",
       "Framstrait_17                                       528F: Elwood et al. (1985) doi:10.1093/oxfordj...   \n",
       "evenMock_01                                         528F: Elwood et al. (1985) doi:10.1093/oxfordj...   \n",
       "evenMock_02                                         528F: Elwood et al. (1985) doi:10.1093/oxfordj...   \n",
       "evenMock_03                                         528F: Elwood et al. (1985) doi:10.1093/oxfordj...   \n",
       "evenMock_04                                         528F: Elwood et al. (1985) doi:10.1093/oxfordj...   \n",
       "evenMock_05                                         528F: Elwood et al. (1985) doi:10.1093/oxfordj...   \n",
       "bloomMock_01                                        528F: Elwood et al. (1985) doi:10.1093/oxfordj...   \n",
       "bloomMock_02                                        528F: Elwood et al. (1985) doi:10.1093/oxfordj...   \n",
       "bloomMock_03                                        528F: Elwood et al. (1985) doi:10.1093/oxfordj...   \n",
       "bloomMock_04                                        528F: Elwood et al. (1985) doi:10.1093/oxfordj...   \n",
       "bloomMock_05                                        528F: Elwood et al. (1985) doi:10.1093/oxfordj...   \n",
       "blank_01                                            528F: Elwood et al. (1985) doi:10.1093/oxfordj...   \n",
       "Samples taken / prepared by GLOMICON partners a...                                                NaN   \n",
       "Davenport_02                                        528F: Elwood et al. (1985) doi:10.1093/oxfordj...   \n",
       "Davenport_02                                        528F: Elwood et al. (1985) doi:10.1093/oxfordj...   \n",
       "Davenport_06                                        528F: Elwood et al. (1985) doi:10.1093/oxfordj...   \n",
       "Davenport_06                                        528F: Elwood et al. (1985) doi:10.1093/oxfordj...   \n",
       "Davenport_11                                        528F: Elwood et al. (1985) doi:10.1093/oxfordj...   \n",
       "Davenport_11                                        528F: Elwood et al. (1985) doi:10.1093/oxfordj...   \n",
       "Davenport_15                                        528F: Elwood et al. (1985) doi:10.1093/oxfordj...   \n",
       "Davenport_15                                        528F: Elwood et al. (1985) doi:10.1093/oxfordj...   \n",
       "Davenport_19                                        528F: Elwood et al. (1985) doi:10.1093/oxfordj...   \n",
       "Davenport_19                                        528F: Elwood et al. (1985) doi:10.1093/oxfordj...   \n",
       "LaJolla_03                                                                                        NaN   \n",
       "LaJolla_09                                                                                        NaN   \n",
       "LaJolla_15                                                                                        NaN   \n",
       "LaJolla_21                                                                                        NaN   \n",
       "LaJolla_27                                                                                        NaN   \n",
       "Plymouth_06                                                                                       NaN   \n",
       "Plymouth_13                                                                                       NaN   \n",
       "Plymouth_19                                                                                       NaN   \n",
       "Plymouth_29                                                                                       NaN   \n",
       "Plymouth_30                                                                                       NaN   \n",
       "Bedford_01                                                                                        NaN   \n",
       "Bedford_07                                                                                        NaN   \n",
       "Bedford_13                                                                                        NaN   \n",
       "Bedford_19                                                                                        NaN   \n",
       "Bedford_25                                                                                        NaN   \n",
       "Samples taken / prepared by AWI and analyzed by...                                                NaN   \n",
       "[to be added from GLOMICON partner's metadata]                                                    NaN   \n",
       "\n",
       "                                                                                             pcr_cond  \\\n",
       "sample_name                                                                                             \n",
       "Samples taken / prepared and analyzed by AWI (c...                                                NaN   \n",
       "Framstrait_01                                       \\ninitial denaturation:95degC_3min;denaturatio...   \n",
       "Framstrait_01                                       \\ninitial denaturation:95degC_3min;denaturatio...   \n",
       "Framstrait_05                                       \\ninitial denaturation:95degC_3min;denaturatio...   \n",
       "Framstrait_05                                       \\ninitial denaturation:95degC_3min;denaturatio...   \n",
       "Framstrait_09                                       \\ninitial denaturation:95degC_3min;denaturatio...   \n",
       "Framstrait_09                                       \\ninitial denaturation:95degC_3min;denaturatio...   \n",
       "Framstrait_13                                       \\ninitial denaturation:95degC_3min;denaturatio...   \n",
       "Framstrait_13                                       \\ninitial denaturation:95degC_3min;denaturatio...   \n",
       "Framstrait_17                                       \\ninitial denaturation:95degC_3min;denaturatio...   \n",
       "Framstrait_17                                       \\ninitial denaturation:95degC_3min;denaturatio...   \n",
       "evenMock_01                                         \\ninitial denaturation:95degC_3min;denaturatio...   \n",
       "evenMock_02                                         \\ninitial denaturation:95degC_3min;denaturatio...   \n",
       "evenMock_03                                         \\ninitial denaturation:95degC_3min;denaturatio...   \n",
       "evenMock_04                                         \\ninitial denaturation:95degC_3min;denaturatio...   \n",
       "evenMock_05                                         \\ninitial denaturation:95degC_3min;denaturatio...   \n",
       "bloomMock_01                                        \\ninitial denaturation:95degC_3min;denaturatio...   \n",
       "bloomMock_02                                        \\ninitial denaturation:95degC_3min;denaturatio...   \n",
       "bloomMock_03                                        \\ninitial denaturation:95degC_3min;denaturatio...   \n",
       "bloomMock_04                                        \\ninitial denaturation:95degC_3min;denaturatio...   \n",
       "bloomMock_05                                        \\ninitial denaturation:95degC_3min;denaturatio...   \n",
       "blank_01                                            \\ninitial denaturation:95degC_3min;denaturatio...   \n",
       "Samples taken / prepared by GLOMICON partners a...                                                NaN   \n",
       "Davenport_02                                        \\ninitial denaturation:95degC_3min;denaturatio...   \n",
       "Davenport_02                                        \\ninitial denaturation:95degC_3min;denaturatio...   \n",
       "Davenport_06                                        \\ninitial denaturation:95degC_3min;denaturatio...   \n",
       "Davenport_06                                        \\ninitial denaturation:95degC_3min;denaturatio...   \n",
       "Davenport_11                                        \\ninitial denaturation:95degC_3min;denaturatio...   \n",
       "Davenport_11                                        \\ninitial denaturation:95degC_3min;denaturatio...   \n",
       "Davenport_15                                        \\ninitial denaturation:95degC_3min;denaturatio...   \n",
       "Davenport_15                                        \\ninitial denaturation:95degC_3min;denaturatio...   \n",
       "Davenport_19                                        \\ninitial denaturation:95degC_3min;denaturatio...   \n",
       "Davenport_19                                        \\ninitial denaturation:95degC_3min;denaturatio...   \n",
       "LaJolla_03                                                                                        NaN   \n",
       "LaJolla_09                                                                                        NaN   \n",
       "LaJolla_15                                                                                        NaN   \n",
       "LaJolla_21                                                                                        NaN   \n",
       "LaJolla_27                                                                                        NaN   \n",
       "Plymouth_06                                                                                       NaN   \n",
       "Plymouth_13                                                                                       NaN   \n",
       "Plymouth_19                                                                                       NaN   \n",
       "Plymouth_29                                                                                       NaN   \n",
       "Plymouth_30                                                                                       NaN   \n",
       "Bedford_01                                                                                        NaN   \n",
       "Bedford_07                                                                                        NaN   \n",
       "Bedford_13                                                                                        NaN   \n",
       "Bedford_19                                                                                        NaN   \n",
       "Bedford_25                                                                                        NaN   \n",
       "Samples taken / prepared by AWI and analyzed by...                                                NaN   \n",
       "[to be added from GLOMICON partner's metadata]                                                    NaN   \n",
       "\n",
       "                                                   date_modified_amplicon_sheet  \n",
       "sample_name                                                                      \n",
       "Samples taken / prepared and analyzed by AWI (c...                          NaT  \n",
       "Framstrait_01                                                               NaT  \n",
       "Framstrait_01                                                               NaT  \n",
       "Framstrait_05                                                               NaT  \n",
       "Framstrait_05                                                               NaT  \n",
       "Framstrait_09                                                               NaT  \n",
       "Framstrait_09                                                               NaT  \n",
       "Framstrait_13                                                               NaT  \n",
       "Framstrait_13                                                               NaT  \n",
       "Framstrait_17                                                               NaT  \n",
       "Framstrait_17                                                               NaT  \n",
       "evenMock_01                                                                 NaT  \n",
       "evenMock_02                                                                 NaT  \n",
       "evenMock_03                                                                 NaT  \n",
       "evenMock_04                                                                 NaT  \n",
       "evenMock_05                                                                 NaT  \n",
       "bloomMock_01                                                                NaT  \n",
       "bloomMock_02                                                                NaT  \n",
       "bloomMock_03                                                                NaT  \n",
       "bloomMock_04                                                                NaT  \n",
       "bloomMock_05                                                                NaT  \n",
       "blank_01                                                                    NaT  \n",
       "Samples taken / prepared by GLOMICON partners a...                          NaT  \n",
       "Davenport_02                                                                NaT  \n",
       "Davenport_02                                                                NaT  \n",
       "Davenport_06                                                                NaT  \n",
       "Davenport_06                                                                NaT  \n",
       "Davenport_11                                                                NaT  \n",
       "Davenport_11                                                                NaT  \n",
       "Davenport_15                                                                NaT  \n",
       "Davenport_15                                                                NaT  \n",
       "Davenport_19                                                                NaT  \n",
       "Davenport_19                                                                NaT  \n",
       "LaJolla_03                                                                  NaT  \n",
       "LaJolla_09                                                                  NaT  \n",
       "LaJolla_15                                                                  NaT  \n",
       "LaJolla_21                                                                  NaT  \n",
       "LaJolla_27                                                                  NaT  \n",
       "Plymouth_06                                                                 NaT  \n",
       "Plymouth_13                                                                 NaT  \n",
       "Plymouth_19                                                                 NaT  \n",
       "Plymouth_29                                                                 NaT  \n",
       "Plymouth_30                                                                 NaT  \n",
       "Bedford_01                                                                  NaT  \n",
       "Bedford_07                                                                  NaT  \n",
       "Bedford_13                                                                  NaT  \n",
       "Bedford_19                                                                  NaT  \n",
       "Bedford_25                                                                  NaT  \n",
       "Samples taken / prepared by AWI and analyzed by...                          NaT  \n",
       "[to be added from GLOMICON partner's metadata]                              NaT  \n",
       "\n",
       "[50 rows x 69 columns]"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = water_amplicon_merge(dfs[1], dfs[2])\n",
    "samp_dfs.append(df)\n",
    "df.head(n=50)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "439d6d41-f363-4dd2-8bb2-f3a9b4f915f5",
   "metadata": {},
   "source": [
    "### SBR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 244,
   "id": "930273b6-f90a-4a35-8ac0-5552bad4a3bc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/kpitz/github/GLOMICON/intercomparison/metadata/SBR_NOAA_MIMARKS.survey.water.6.0_sharing.xlsx\n",
      "Sheets in excel file: ['README', 'study_data', 'water_sample_data', 'amplicon_prep_data', 'metag_prep_data', 'analysis_data', 'SRA_template', 'measurement_metadata', 'SRA Terms', 'Version log']\n",
      "study_data 5\n",
      "water_sample_data 8\n",
      "amplicon_prep_data 6\n",
      "analysis_data 6\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>amplicon_sequenced</th>\n",
       "      <th>ampliconSize</th>\n",
       "      <th>trim_method</th>\n",
       "      <th>cluster_method</th>\n",
       "      <th>pid_clustering</th>\n",
       "      <th>taxa_class_method</th>\n",
       "      <th>taxa_ref_db</th>\n",
       "      <th>code_repo</th>\n",
       "      <th>identificationReferences</th>\n",
       "      <th>controls_used</th>\n",
       "      <th>date_modified</th>\n",
       "      <th>modified_by</th>\n",
       "      <th>sheet</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>16S V4-V5</td>\n",
       "      <td>411</td>\n",
       "      <td>cutadapt</td>\n",
       "      <td>Tourmaline; qiime2-2021.2; dada2</td>\n",
       "      <td>ASV</td>\n",
       "      <td>Tourmaline; qiime2-2021.2; naive-bayes classifier</td>\n",
       "      <td>Silva SSU Ref NR 99 v138.1; 515f-926r region; ...</td>\n",
       "      <td>https://github.com/aomlomics/gomecc</td>\n",
       "      <td>10.5281/zenodo.8392695 | https://github.com/ao...</td>\n",
       "      <td>12 distilled water blanks | 2 PCR no-template ...</td>\n",
       "      <td>2024-04-11 07:12:09.354</td>\n",
       "      <td>aomlomics@gmail.com</td>\n",
       "      <td>analysis_data</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>18S V9</td>\n",
       "      <td>260</td>\n",
       "      <td>cutadapt</td>\n",
       "      <td>Tourmaline; qiime2-2021.2; dada2</td>\n",
       "      <td>ASV</td>\n",
       "      <td>Tourmaline; qiime2-2021.2; naive-bayes classifier</td>\n",
       "      <td>PR2 v5.0.1; V9 1391f-1510r region; 10.5281/zen...</td>\n",
       "      <td>https://github.com/aomlomics/gomecc</td>\n",
       "      <td>10.5281/zenodo.8392706 | https://pr2-database....</td>\n",
       "      <td>12 distilled water blanks | 2 PCR no-template ...</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>analysis_data</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  amplicon_sequenced  ampliconSize trim_method  \\\n",
       "0          16S V4-V5           411    cutadapt   \n",
       "1             18S V9           260    cutadapt   \n",
       "\n",
       "                     cluster_method pid_clustering  \\\n",
       "0  Tourmaline; qiime2-2021.2; dada2            ASV   \n",
       "1  Tourmaline; qiime2-2021.2; dada2            ASV   \n",
       "\n",
       "                                   taxa_class_method  \\\n",
       "0  Tourmaline; qiime2-2021.2; naive-bayes classifier   \n",
       "1  Tourmaline; qiime2-2021.2; naive-bayes classifier   \n",
       "\n",
       "                                         taxa_ref_db  \\\n",
       "0  Silva SSU Ref NR 99 v138.1; 515f-926r region; ...   \n",
       "1  PR2 v5.0.1; V9 1391f-1510r region; 10.5281/zen...   \n",
       "\n",
       "                             code_repo  \\\n",
       "0  https://github.com/aomlomics/gomecc   \n",
       "1  https://github.com/aomlomics/gomecc   \n",
       "\n",
       "                            identificationReferences  \\\n",
       "0  10.5281/zenodo.8392695 | https://github.com/ao...   \n",
       "1  10.5281/zenodo.8392706 | https://pr2-database....   \n",
       "\n",
       "                                       controls_used           date_modified  \\\n",
       "0  12 distilled water blanks | 2 PCR no-template ... 2024-04-11 07:12:09.354   \n",
       "1  12 distilled water blanks | 2 PCR no-template ...                     NaT   \n",
       "\n",
       "           modified_by          sheet  \n",
       "0  aomlomics@gmail.com  analysis_data  \n",
       "1                  NaN  analysis_data  "
      ]
     },
     "execution_count": 244,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# import\n",
    "#file = files[1]\n",
    "file = '/Users/kpitz/github/GLOMICON/intercomparison/metadata/SBR_NOAA_MIMARKS.survey.water.6.0_sharing.xlsx'\n",
    "print(file)\n",
    "inst_name = 'SBR'\n",
    "institutes.append(inst_name)\n",
    "xl = pd.ExcelFile(file)\n",
    "print('Sheets in excel file:',xl.sheet_names)  # see all sheet names\n",
    "#sheets = xl.sheet_names\n",
    "sheets = ['study_data', 'water_sample_data', 'amplicon_prep_data', 'analysis_data']\n",
    "dfs= []\n",
    "skiprow_num = [5,8,6,6]\n",
    "for sheet, sr in zip(sheets, skiprow_num):\n",
    "    print(sheet,sr)\n",
    "    df = xl.parse(sheet,skiprows=sr)\n",
    "    df = df.dropna(how='all', axis=1)\n",
    "    df = df.dropna(how='all', axis=0)\n",
    "    df['sheet'] = sheet\n",
    "    dfs.append(df)\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0edef3c8-c5a6-48c6-94a9-43324690f108",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "c082b48d-e7bf-466c-a7ee-f9ef2d73480f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SBR</th>\n",
       "      <th>SBR_2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>project_id</th>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>project_name</th>\n",
       "      <td>Intercomparison of filtered seawater samples a...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>project_id_external</th>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>project_contact</th>\n",
       "      <td>Kathleen Pitz, kpitz@mbari.org, https://orcid....</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>project_description</th>\n",
       "      <td>Replicate filters, originating from surface wa...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                   SBR SBR_2\n",
       "project_id                                          GLOMICON_INTERCOMP   NaN\n",
       "project_name         Intercomparison of filtered seawater samples a...   NaN\n",
       "project_id_external                                 GLOMICON_INTERCOMP   NaN\n",
       "project_contact      Kathleen Pitz, kpitz@mbari.org, https://orcid....   NaN\n",
       "project_description  Replicate filters, originating from surface wa...   NaN"
      ]
     },
     "execution_count": 89,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = study_data_merge(dfs[0], dfs[3], inst_name)\n",
    "df = df.rename(columns={1:inst_name+'_2'})\n",
    "study_dfs.append(df)\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "35f8b604-393c-4f4d-8ef6-a9d71a7a6a37",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Collecting Institute</th>\n",
       "      <th>Analyzing Institute</th>\n",
       "      <th>station</th>\n",
       "      <th>sample_replicate</th>\n",
       "      <th>source_material_id</th>\n",
       "      <th>project_id</th>\n",
       "      <th>organism</th>\n",
       "      <th>collection_date_local</th>\n",
       "      <th>collection_date</th>\n",
       "      <th>depth</th>\n",
       "      <th>...</th>\n",
       "      <th>amplicon_sequenced</th>\n",
       "      <th>target_gene</th>\n",
       "      <th>target_subfragment</th>\n",
       "      <th>pcr_primer_forward</th>\n",
       "      <th>pcr_primer_reverse</th>\n",
       "      <th>pcr_primer_name_forward</th>\n",
       "      <th>pcr_primer_name_reverse</th>\n",
       "      <th>pcr_primer_reference</th>\n",
       "      <th>pcr_cond</th>\n",
       "      <th>date_modified_amplicon_sheet</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sample_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Samples taken / prepared and analyzed by SBR (complete information)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SBR01</th>\n",
       "      <td>SBR</td>\n",
       "      <td>SBR</td>\n",
       "      <td>SOMLIT-ASTAN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>GLOMICON_INTERCOMP_SBR_01</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>seawater metagenome</td>\n",
       "      <td>2018-10-17</td>\n",
       "      <td>2018-10-17</td>\n",
       "      <td>3 m</td>\n",
       "      <td>...</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V4</td>\n",
       "      <td>5’-CCAGCASCYGCGGTAATTCC-3’</td>\n",
       "      <td>5’-ACTTTCGTTCTTGATYRA-3’</td>\n",
       "      <td>TAReuk454FWD1</td>\n",
       "      <td>TAReukREV3</td>\n",
       "      <td>TAReuk454FWD1 (5’-CCAGCASCYGCGGTAATTCC-3’) and...</td>\n",
       "      <td>initial denaturation at 98°C for 30 seconds, f...</td>\n",
       "      <td>2024-09-17 05:03:26.170</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SBR05</th>\n",
       "      <td>SBR</td>\n",
       "      <td>SBR</td>\n",
       "      <td>SOMLIT-ASTAN</td>\n",
       "      <td>5.0</td>\n",
       "      <td>GLOMICON_INTERCOMP_SBR_05</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>seawater metagenome</td>\n",
       "      <td>2018-10-17</td>\n",
       "      <td>2018-10-17</td>\n",
       "      <td>3 m</td>\n",
       "      <td>...</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V4</td>\n",
       "      <td>5’-CCAGCASCYGCGGTAATTCC-3’</td>\n",
       "      <td>5’-ACTTTCGTTCTTGATYRA-3’</td>\n",
       "      <td>TAReuk454FWD1</td>\n",
       "      <td>TAReukREV3</td>\n",
       "      <td>TAReuk454FWD1 (5’-CCAGCASCYGCGGTAATTCC-3’) and...</td>\n",
       "      <td>initial denaturation at 98°C for 30 seconds, f...</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SBR09</th>\n",
       "      <td>SBR</td>\n",
       "      <td>SBR</td>\n",
       "      <td>SOMLIT-ASTAN</td>\n",
       "      <td>9.0</td>\n",
       "      <td>GLOMICON_INTERCOMP_SBR_09</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>seawater metagenome</td>\n",
       "      <td>2018-10-17</td>\n",
       "      <td>2018-10-17</td>\n",
       "      <td>3 m</td>\n",
       "      <td>...</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V4</td>\n",
       "      <td>5’-CCAGCASCYGCGGTAATTCC-3’</td>\n",
       "      <td>5’-ACTTTCGTTCTTGATYRA-3’</td>\n",
       "      <td>TAReuk454FWD1</td>\n",
       "      <td>TAReukREV3</td>\n",
       "      <td>TAReuk454FWD1 (5’-CCAGCASCYGCGGTAATTCC-3’) and...</td>\n",
       "      <td>initial denaturation at 98°C for 30 seconds, f...</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SBR13</th>\n",
       "      <td>SBR</td>\n",
       "      <td>SBR</td>\n",
       "      <td>SOMLIT-ASTAN</td>\n",
       "      <td>13.0</td>\n",
       "      <td>GLOMICON_INTERCOMP_SBR_13</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>seawater metagenome</td>\n",
       "      <td>2018-10-17</td>\n",
       "      <td>2018-10-17</td>\n",
       "      <td>3 m</td>\n",
       "      <td>...</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V4</td>\n",
       "      <td>5’-CCAGCASCYGCGGTAATTCC-3’</td>\n",
       "      <td>5’-ACTTTCGTTCTTGATYRA-3’</td>\n",
       "      <td>TAReuk454FWD1</td>\n",
       "      <td>TAReukREV3</td>\n",
       "      <td>TAReuk454FWD1 (5’-CCAGCASCYGCGGTAATTCC-3’) and...</td>\n",
       "      <td>initial denaturation at 98°C for 30 seconds, f...</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 77 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   Collecting Institute  \\\n",
       "sample_name                                                               \n",
       "Samples taken / prepared and analyzed by SBR (c...                  NaN   \n",
       "SBR01                                                               SBR   \n",
       "SBR05                                                               SBR   \n",
       "SBR09                                                               SBR   \n",
       "SBR13                                                               SBR   \n",
       "\n",
       "                                                   Analyzing Institute  \\\n",
       "sample_name                                                              \n",
       "Samples taken / prepared and analyzed by SBR (c...                 NaN   \n",
       "SBR01                                                              SBR   \n",
       "SBR05                                                              SBR   \n",
       "SBR09                                                              SBR   \n",
       "SBR13                                                              SBR   \n",
       "\n",
       "                                                         station  \\\n",
       "sample_name                                                        \n",
       "Samples taken / prepared and analyzed by SBR (c...           NaN   \n",
       "SBR01                                               SOMLIT-ASTAN   \n",
       "SBR05                                               SOMLIT-ASTAN   \n",
       "SBR09                                               SOMLIT-ASTAN   \n",
       "SBR13                                               SOMLIT-ASTAN   \n",
       "\n",
       "                                                    sample_replicate  \\\n",
       "sample_name                                                            \n",
       "Samples taken / prepared and analyzed by SBR (c...               NaN   \n",
       "SBR01                                                            1.0   \n",
       "SBR05                                                            5.0   \n",
       "SBR09                                                            9.0   \n",
       "SBR13                                                           13.0   \n",
       "\n",
       "                                                           source_material_id  \\\n",
       "sample_name                                                                     \n",
       "Samples taken / prepared and analyzed by SBR (c...                        NaN   \n",
       "SBR01                                               GLOMICON_INTERCOMP_SBR_01   \n",
       "SBR05                                               GLOMICON_INTERCOMP_SBR_05   \n",
       "SBR09                                               GLOMICON_INTERCOMP_SBR_09   \n",
       "SBR13                                               GLOMICON_INTERCOMP_SBR_13   \n",
       "\n",
       "                                                            project_id  \\\n",
       "sample_name                                                              \n",
       "Samples taken / prepared and analyzed by SBR (c...                 NaN   \n",
       "SBR01                                               GLOMICON_INTERCOMP   \n",
       "SBR05                                               GLOMICON_INTERCOMP   \n",
       "SBR09                                               GLOMICON_INTERCOMP   \n",
       "SBR13                                               GLOMICON_INTERCOMP   \n",
       "\n",
       "                                                               organism  \\\n",
       "sample_name                                                               \n",
       "Samples taken / prepared and analyzed by SBR (c...                  NaN   \n",
       "SBR01                                               seawater metagenome   \n",
       "SBR05                                               seawater metagenome   \n",
       "SBR09                                               seawater metagenome   \n",
       "SBR13                                               seawater metagenome   \n",
       "\n",
       "                                                   collection_date_local  \\\n",
       "sample_name                                                                \n",
       "Samples taken / prepared and analyzed by SBR (c...                   NaN   \n",
       "SBR01                                                         2018-10-17   \n",
       "SBR05                                                         2018-10-17   \n",
       "SBR09                                                         2018-10-17   \n",
       "SBR13                                                         2018-10-17   \n",
       "\n",
       "                                                   collection_date depth  ...  \\\n",
       "sample_name                                                               ...   \n",
       "Samples taken / prepared and analyzed by SBR (c...             NaN   NaN  ...   \n",
       "SBR01                                                   2018-10-17   3 m  ...   \n",
       "SBR05                                                   2018-10-17   3 m  ...   \n",
       "SBR09                                                   2018-10-17   3 m  ...   \n",
       "SBR13                                                   2018-10-17   3 m  ...   \n",
       "\n",
       "                                                   amplicon_sequenced  \\\n",
       "sample_name                                                             \n",
       "Samples taken / prepared and analyzed by SBR (c...                NaN   \n",
       "SBR01                                                          18S V4   \n",
       "SBR05                                                          18S V4   \n",
       "SBR09                                                          18S V4   \n",
       "SBR13                                                          18S V4   \n",
       "\n",
       "                                                   target_gene  \\\n",
       "sample_name                                                      \n",
       "Samples taken / prepared and analyzed by SBR (c...         NaN   \n",
       "SBR01                                                 18S rRNA   \n",
       "SBR05                                                 18S rRNA   \n",
       "SBR09                                                 18S rRNA   \n",
       "SBR13                                                 18S rRNA   \n",
       "\n",
       "                                                   target_subfragment  \\\n",
       "sample_name                                                             \n",
       "Samples taken / prepared and analyzed by SBR (c...                NaN   \n",
       "SBR01                                                              V4   \n",
       "SBR05                                                              V4   \n",
       "SBR09                                                              V4   \n",
       "SBR13                                                              V4   \n",
       "\n",
       "                                                            pcr_primer_forward  \\\n",
       "sample_name                                                                      \n",
       "Samples taken / prepared and analyzed by SBR (c...                         NaN   \n",
       "SBR01                                               5’-CCAGCASCYGCGGTAATTCC-3’   \n",
       "SBR05                                               5’-CCAGCASCYGCGGTAATTCC-3’   \n",
       "SBR09                                               5’-CCAGCASCYGCGGTAATTCC-3’   \n",
       "SBR13                                               5’-CCAGCASCYGCGGTAATTCC-3’   \n",
       "\n",
       "                                                          pcr_primer_reverse  \\\n",
       "sample_name                                                                    \n",
       "Samples taken / prepared and analyzed by SBR (c...                       NaN   \n",
       "SBR01                                               5’-ACTTTCGTTCTTGATYRA-3’   \n",
       "SBR05                                               5’-ACTTTCGTTCTTGATYRA-3’   \n",
       "SBR09                                               5’-ACTTTCGTTCTTGATYRA-3’   \n",
       "SBR13                                               5’-ACTTTCGTTCTTGATYRA-3’   \n",
       "\n",
       "                                                   pcr_primer_name_forward  \\\n",
       "sample_name                                                                  \n",
       "Samples taken / prepared and analyzed by SBR (c...                     NaN   \n",
       "SBR01                                                        TAReuk454FWD1   \n",
       "SBR05                                                        TAReuk454FWD1   \n",
       "SBR09                                                        TAReuk454FWD1   \n",
       "SBR13                                                        TAReuk454FWD1   \n",
       "\n",
       "                                                   pcr_primer_name_reverse  \\\n",
       "sample_name                                                                  \n",
       "Samples taken / prepared and analyzed by SBR (c...                     NaN   \n",
       "SBR01                                                          TAReukREV3    \n",
       "SBR05                                                          TAReukREV3    \n",
       "SBR09                                                          TAReukREV3    \n",
       "SBR13                                                          TAReukREV3    \n",
       "\n",
       "                                                                                 pcr_primer_reference  \\\n",
       "sample_name                                                                                             \n",
       "Samples taken / prepared and analyzed by SBR (c...                                                NaN   \n",
       "SBR01                                               TAReuk454FWD1 (5’-CCAGCASCYGCGGTAATTCC-3’) and...   \n",
       "SBR05                                               TAReuk454FWD1 (5’-CCAGCASCYGCGGTAATTCC-3’) and...   \n",
       "SBR09                                               TAReuk454FWD1 (5’-CCAGCASCYGCGGTAATTCC-3’) and...   \n",
       "SBR13                                               TAReuk454FWD1 (5’-CCAGCASCYGCGGTAATTCC-3’) and...   \n",
       "\n",
       "                                                                                             pcr_cond  \\\n",
       "sample_name                                                                                             \n",
       "Samples taken / prepared and analyzed by SBR (c...                                                NaN   \n",
       "SBR01                                               initial denaturation at 98°C for 30 seconds, f...   \n",
       "SBR05                                               initial denaturation at 98°C for 30 seconds, f...   \n",
       "SBR09                                               initial denaturation at 98°C for 30 seconds, f...   \n",
       "SBR13                                               initial denaturation at 98°C for 30 seconds, f...   \n",
       "\n",
       "                                                   date_modified_amplicon_sheet  \n",
       "sample_name                                                                      \n",
       "Samples taken / prepared and analyzed by SBR (c...                          NaT  \n",
       "SBR01                                                   2024-09-17 05:03:26.170  \n",
       "SBR05                                                                       NaT  \n",
       "SBR09                                                                       NaT  \n",
       "SBR13                                                                       NaT  \n",
       "\n",
       "[5 rows x 77 columns]"
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = water_amplicon_merge(dfs[1], dfs[2])\n",
    "samp_dfs.append(df)\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "33951a83-3bee-4c5d-96ff-9340360e3c43",
   "metadata": {},
   "source": [
    "## Merge across institutes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "id": "2229678d-22d0-42c5-ac3f-a0e9058f38a1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MBARI</th>\n",
       "      <th>AOML</th>\n",
       "      <th>AOML_2</th>\n",
       "      <th>AWI</th>\n",
       "      <th>SBR</th>\n",
       "      <th>SBR_2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>project_id</th>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>project_name</th>\n",
       "      <td>Intercomparison of filtered seawater samples a...</td>\n",
       "      <td>Intercomparison of filtered seawater samples a...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Intercomparison of filtered seawater samples a...</td>\n",
       "      <td>Intercomparison of filtered seawater samples a...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>project_id_external</th>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>project_contact</th>\n",
       "      <td>Kathleen Pitz, kpitz@mbari.org, https://orcid....</td>\n",
       "      <td>Kathleen Pitz, kpitz@mbari.org, https://orcid....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Kathleen Pitz, kpitz@mbari.org, https://orcid....</td>\n",
       "      <td>Kathleen Pitz, kpitz@mbari.org, https://orcid....</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>project_description</th>\n",
       "      <td>Replicate filters, originating from surface wa...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Replicate filters, originating from surface wa...</td>\n",
       "      <td>Replicate filters, originating from surface wa...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>type</th>\n",
       "      <td>Occurrence</td>\n",
       "      <td>Occurrence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Occurrence</td>\n",
       "      <td>Occurrence</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>license</th>\n",
       "      <td>CC-BY 4.0</td>\n",
       "      <td>CC-BY 4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CC-BY 4.0</td>\n",
       "      <td>CC-BY 4.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>associated_parties</th>\n",
       "      <td>Francisco Chavez, chfr@mbari.org</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Felix Janssen, felix.janssen@awi.de, https://o...</td>\n",
       "      <td>Christian Jeanthon &lt;jeanthon@sb-roscoff.fr&gt;, C...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>study_area_description</th>\n",
       "      <td>eDNA from north of Monterey Bay collected duri...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Surface waters from different marine time seri...</td>\n",
       "      <td>Surface waters from different marine time seri...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>amplicon_sequenced</th>\n",
       "      <td>18S V9</td>\n",
       "      <td>16S V4-V5</td>\n",
       "      <td>18S V9</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>16S V4-V5</td>\n",
       "      <td>18S V9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>trim_method</th>\n",
       "      <td>atropos</td>\n",
       "      <td>cutadapt</td>\n",
       "      <td>cutadapt</td>\n",
       "      <td>cutadapt v2.8</td>\n",
       "      <td>cutadapt</td>\n",
       "      <td>cutadapt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cluster_method</th>\n",
       "      <td>dada2</td>\n",
       "      <td>Tourmaline; qiime2-2021.2; dada2</td>\n",
       "      <td>Tourmaline; qiime2-2021.2; dada2</td>\n",
       "      <td>dada denoising; dada2 v1.30 in R v4.3.2 (learn...</td>\n",
       "      <td>Tourmaline; qiime2-2021.2; dada2</td>\n",
       "      <td>Tourmaline; qiime2-2021.2; dada2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pid_clustering</th>\n",
       "      <td>ASV</td>\n",
       "      <td>ASV</td>\n",
       "      <td>ASV</td>\n",
       "      <td>ASV</td>\n",
       "      <td>ASV</td>\n",
       "      <td>ASV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taxa_class_method</th>\n",
       "      <td>BLAST; MEGAN LCA</td>\n",
       "      <td>Tourmaline; qiime2-2021.2; naive-bayes classifier</td>\n",
       "      <td>Tourmaline; qiime2-2021.2; naive-bayes classifier</td>\n",
       "      <td>dada2 implementation of naive-bayes classifier...</td>\n",
       "      <td>Tourmaline; qiime2-2021.2; naive-bayes classifier</td>\n",
       "      <td>Tourmaline; qiime2-2021.2; naive-bayes classifier</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taxa_ref_db</th>\n",
       "      <td>genbank NR</td>\n",
       "      <td>Silva SSU Ref NR 99 v138.1; 515f-926r region; ...</td>\n",
       "      <td>PR2 v5.0.1; V9 1391f-1510r region; 10.5281/zen...</td>\n",
       "      <td>PR2 v5.0.0; full set; https://github.com/pr2da...</td>\n",
       "      <td>Silva SSU Ref NR 99 v138.1; 515f-926r region; ...</td>\n",
       "      <td>PR2 v5.0.1; V9 1391f-1510r region; 10.5281/zen...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>code_repo</th>\n",
       "      <td>https://github.com/MBARI-BOG/BOG-Banzai-Dada2-...</td>\n",
       "      <td>https://github.com/aomlomics/gomecc</td>\n",
       "      <td>https://github.com/aomlomics/gomecc</td>\n",
       "      <td>???</td>\n",
       "      <td>https://github.com/aomlomics/gomecc</td>\n",
       "      <td>https://github.com/aomlomics/gomecc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>external_links</th>\n",
       "      <td>NaN</td>\n",
       "      <td>https://github.com/GLOMICON</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://github.com/GLOMICON</td>\n",
       "      <td>https://github.com/GLOMICON</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ampliconSize</th>\n",
       "      <td>NaN</td>\n",
       "      <td>411</td>\n",
       "      <td>260</td>\n",
       "      <td>356-394</td>\n",
       "      <td>411</td>\n",
       "      <td>260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>identificationReferences</th>\n",
       "      <td>NaN</td>\n",
       "      <td>10.5281/zenodo.8392695 | https://github.com/ao...</td>\n",
       "      <td>10.5281/zenodo.8392706 | https://pr2-database....</td>\n",
       "      <td>https://github.com/pr2database/pr2database/rel...</td>\n",
       "      <td>10.5281/zenodo.8392695 | https://github.com/ao...</td>\n",
       "      <td>10.5281/zenodo.8392706 | https://pr2-database....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>controls_used</th>\n",
       "      <td>NaN</td>\n",
       "      <td>12 distilled water blanks | 2 PCR no-template ...</td>\n",
       "      <td>12 distilled water blanks | 2 PCR no-template ...</td>\n",
       "      <td>???</td>\n",
       "      <td>12 distilled water blanks | 2 PCR no-template ...</td>\n",
       "      <td>12 distilled water blanks | 2 PCR no-template ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>modified_by</th>\n",
       "      <td>NaN</td>\n",
       "      <td>aomlomics@gmail.com</td>\n",
       "      <td>NaN</td>\n",
       "      <td>stefan.neuhaus@awi.de</td>\n",
       "      <td>aomlomics@gmail.com</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                      MBARI  \\\n",
       "project_id                                               GLOMICON_INTERCOMP   \n",
       "project_name              Intercomparison of filtered seawater samples a...   \n",
       "project_id_external                                      GLOMICON_INTERCOMP   \n",
       "project_contact           Kathleen Pitz, kpitz@mbari.org, https://orcid....   \n",
       "project_description       Replicate filters, originating from surface wa...   \n",
       "type                                                             Occurrence   \n",
       "license                                                           CC-BY 4.0   \n",
       "associated_parties                         Francisco Chavez, chfr@mbari.org   \n",
       "study_area_description    eDNA from north of Monterey Bay collected duri...   \n",
       "amplicon_sequenced                                                   18S V9   \n",
       "trim_method                                                         atropos   \n",
       "cluster_method                                                        dada2   \n",
       "pid_clustering                                                          ASV   \n",
       "taxa_class_method                                          BLAST; MEGAN LCA   \n",
       "taxa_ref_db                                                      genbank NR   \n",
       "code_repo                 https://github.com/MBARI-BOG/BOG-Banzai-Dada2-...   \n",
       "external_links                                                          NaN   \n",
       "ampliconSize                                                            NaN   \n",
       "identificationReferences                                                NaN   \n",
       "controls_used                                                           NaN   \n",
       "modified_by                                                             NaN   \n",
       "\n",
       "                                                                       AOML  \\\n",
       "project_id                                               GLOMICON_INTERCOMP   \n",
       "project_name              Intercomparison of filtered seawater samples a...   \n",
       "project_id_external                                      GLOMICON_INTERCOMP   \n",
       "project_contact           Kathleen Pitz, kpitz@mbari.org, https://orcid....   \n",
       "project_description                                                     NaN   \n",
       "type                                                             Occurrence   \n",
       "license                                                           CC-BY 4.0   \n",
       "associated_parties                                                      NaN   \n",
       "study_area_description                                                  NaN   \n",
       "amplicon_sequenced                                                16S V4-V5   \n",
       "trim_method                                                        cutadapt   \n",
       "cluster_method                             Tourmaline; qiime2-2021.2; dada2   \n",
       "pid_clustering                                                          ASV   \n",
       "taxa_class_method         Tourmaline; qiime2-2021.2; naive-bayes classifier   \n",
       "taxa_ref_db               Silva SSU Ref NR 99 v138.1; 515f-926r region; ...   \n",
       "code_repo                               https://github.com/aomlomics/gomecc   \n",
       "external_links                                  https://github.com/GLOMICON   \n",
       "ampliconSize                                                            411   \n",
       "identificationReferences  10.5281/zenodo.8392695 | https://github.com/ao...   \n",
       "controls_used             12 distilled water blanks | 2 PCR no-template ...   \n",
       "modified_by                                             aomlomics@gmail.com   \n",
       "\n",
       "                                                                     AOML_2  \\\n",
       "project_id                                                              NaN   \n",
       "project_name                                                            NaN   \n",
       "project_id_external                                                     NaN   \n",
       "project_contact                                                         NaN   \n",
       "project_description                                                     NaN   \n",
       "type                                                                    NaN   \n",
       "license                                                                 NaN   \n",
       "associated_parties                                                      NaN   \n",
       "study_area_description                                                  NaN   \n",
       "amplicon_sequenced                                                   18S V9   \n",
       "trim_method                                                        cutadapt   \n",
       "cluster_method                             Tourmaline; qiime2-2021.2; dada2   \n",
       "pid_clustering                                                          ASV   \n",
       "taxa_class_method         Tourmaline; qiime2-2021.2; naive-bayes classifier   \n",
       "taxa_ref_db               PR2 v5.0.1; V9 1391f-1510r region; 10.5281/zen...   \n",
       "code_repo                               https://github.com/aomlomics/gomecc   \n",
       "external_links                                                          NaN   \n",
       "ampliconSize                                                            260   \n",
       "identificationReferences  10.5281/zenodo.8392706 | https://pr2-database....   \n",
       "controls_used             12 distilled water blanks | 2 PCR no-template ...   \n",
       "modified_by                                                             NaN   \n",
       "\n",
       "                                                                        AWI  \\\n",
       "project_id                                               GLOMICON_INTERCOMP   \n",
       "project_name              Intercomparison of filtered seawater samples a...   \n",
       "project_id_external                                      GLOMICON_INTERCOMP   \n",
       "project_contact           Kathleen Pitz, kpitz@mbari.org, https://orcid....   \n",
       "project_description       Replicate filters, originating from surface wa...   \n",
       "type                                                             Occurrence   \n",
       "license                                                           CC-BY 4.0   \n",
       "associated_parties        Felix Janssen, felix.janssen@awi.de, https://o...   \n",
       "study_area_description    Surface waters from different marine time seri...   \n",
       "amplicon_sequenced                                                   18S V4   \n",
       "trim_method                                                   cutadapt v2.8   \n",
       "cluster_method            dada denoising; dada2 v1.30 in R v4.3.2 (learn...   \n",
       "pid_clustering                                                          ASV   \n",
       "taxa_class_method         dada2 implementation of naive-bayes classifier...   \n",
       "taxa_ref_db               PR2 v5.0.0; full set; https://github.com/pr2da...   \n",
       "code_repo                                                               ???   \n",
       "external_links                                  https://github.com/GLOMICON   \n",
       "ampliconSize                                                        356-394   \n",
       "identificationReferences  https://github.com/pr2database/pr2database/rel...   \n",
       "controls_used                                                           ???   \n",
       "modified_by                                           stefan.neuhaus@awi.de   \n",
       "\n",
       "                                                                        SBR  \\\n",
       "project_id                                               GLOMICON_INTERCOMP   \n",
       "project_name              Intercomparison of filtered seawater samples a...   \n",
       "project_id_external                                      GLOMICON_INTERCOMP   \n",
       "project_contact           Kathleen Pitz, kpitz@mbari.org, https://orcid....   \n",
       "project_description       Replicate filters, originating from surface wa...   \n",
       "type                                                             Occurrence   \n",
       "license                                                           CC-BY 4.0   \n",
       "associated_parties        Christian Jeanthon <jeanthon@sb-roscoff.fr>, C...   \n",
       "study_area_description    Surface waters from different marine time seri...   \n",
       "amplicon_sequenced                                                16S V4-V5   \n",
       "trim_method                                                        cutadapt   \n",
       "cluster_method                             Tourmaline; qiime2-2021.2; dada2   \n",
       "pid_clustering                                                          ASV   \n",
       "taxa_class_method         Tourmaline; qiime2-2021.2; naive-bayes classifier   \n",
       "taxa_ref_db               Silva SSU Ref NR 99 v138.1; 515f-926r region; ...   \n",
       "code_repo                               https://github.com/aomlomics/gomecc   \n",
       "external_links                                  https://github.com/GLOMICON   \n",
       "ampliconSize                                                            411   \n",
       "identificationReferences  10.5281/zenodo.8392695 | https://github.com/ao...   \n",
       "controls_used             12 distilled water blanks | 2 PCR no-template ...   \n",
       "modified_by                                             aomlomics@gmail.com   \n",
       "\n",
       "                                                                      SBR_2  \n",
       "project_id                                                              NaN  \n",
       "project_name                                                            NaN  \n",
       "project_id_external                                                     NaN  \n",
       "project_contact                                                         NaN  \n",
       "project_description                                                     NaN  \n",
       "type                                                                    NaN  \n",
       "license                                                                 NaN  \n",
       "associated_parties                                                      NaN  \n",
       "study_area_description                                                  NaN  \n",
       "amplicon_sequenced                                                   18S V9  \n",
       "trim_method                                                        cutadapt  \n",
       "cluster_method                             Tourmaline; qiime2-2021.2; dada2  \n",
       "pid_clustering                                                          ASV  \n",
       "taxa_class_method         Tourmaline; qiime2-2021.2; naive-bayes classifier  \n",
       "taxa_ref_db               PR2 v5.0.1; V9 1391f-1510r region; 10.5281/zen...  \n",
       "code_repo                               https://github.com/aomlomics/gomecc  \n",
       "external_links                                                          NaN  \n",
       "ampliconSize                                                            260  \n",
       "identificationReferences  10.5281/zenodo.8392706 | https://pr2-database....  \n",
       "controls_used             12 distilled water blanks | 2 PCR no-template ...  \n",
       "modified_by                                                             NaN  "
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# 04/28/25 - looks like SBR didn't alter analysis sheet, sample as AOML\n",
    "# study data\n",
    "df = pd.concat(study_dfs, axis=1)\n",
    "# export\n",
    "file = '/Users/kpitz/github/GLOMICON/intercomparison/Merged_Datasets/metadata/' + 'combined_study_data.csv'\n",
    "df.to_csv(file)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "937d581d-a72c-4973-a50f-a93ecf4add7e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>Collecting Institute</th>\n",
       "      <th>Analyzing Institute</th>\n",
       "      <th>cruise_id</th>\n",
       "      <th>station</th>\n",
       "      <th>ctd_bottle_no</th>\n",
       "      <th>sample_replicate</th>\n",
       "      <th>source_material_id</th>\n",
       "      <th>extract_number</th>\n",
       "      <th>sample_title</th>\n",
       "      <th>project_id</th>\n",
       "      <th>...</th>\n",
       "      <th>part_org_nitro</th>\n",
       "      <th>ph</th>\n",
       "      <th>phosphate</th>\n",
       "      <th>silicate</th>\n",
       "      <th>suspend_part_matter</th>\n",
       "      <th>tidal_stage</th>\n",
       "      <th>date_dna_extracted</th>\n",
       "      <th>extraction_personnel</th>\n",
       "      <th>date_pcr</th>\n",
       "      <th>pcr_personnel</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">MBARI</th>\n",
       "      <th>CN18Sc37_12_Rep_Stdy5_AO</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy5</td>\n",
       "      <td>AO9</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>GLOMICON</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN18Sc37_12_Rep_Stdy10_AO</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy10</td>\n",
       "      <td>AO10</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>GLOMICON</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN18Sc37_12_Rep_Stdy14_AO</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy14</td>\n",
       "      <td>AO11</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>GLOMICON</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN18Sc37_12_Rep_Stdy18_AO</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy18</td>\n",
       "      <td>AO12</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>GLOMICON</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN18Sc37_12_Rep_Stdy22_AO</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy22</td>\n",
       "      <td>AO13</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>GLOMICON</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">SBR</th>\n",
       "      <th>[to be added from GLOMICON partner's metadata]</th>\n",
       "      <td>SBR</td>\n",
       "      <td>UDalhousie</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SOMLIT-ASTAN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8.0</td>\n",
       "      <td>GLOMICON_INTERCOMP_SBR_08</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>19.3 µg/l</td>\n",
       "      <td>7.827</td>\n",
       "      <td>0.41 µmol/l</td>\n",
       "      <td>3.12 µmol/l</td>\n",
       "      <td>0.3 mg/l</td>\n",
       "      <td>high</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>[to be added from GLOMICON partner's metadata]</th>\n",
       "      <td>SBR</td>\n",
       "      <td>UDalhousie</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SOMLIT-ASTAN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12.0</td>\n",
       "      <td>GLOMICON_INTERCOMP_SBR_12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>19.3 µg/l</td>\n",
       "      <td>7.827</td>\n",
       "      <td>0.41 µmol/l</td>\n",
       "      <td>3.12 µmol/l</td>\n",
       "      <td>0.3 mg/l</td>\n",
       "      <td>high</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>[to be added from GLOMICON partner's metadata]</th>\n",
       "      <td>SBR</td>\n",
       "      <td>UDalhousie</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SOMLIT-ASTAN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>16.0</td>\n",
       "      <td>GLOMICON_INTERCOMP_SBR_16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>19.3 µg/l</td>\n",
       "      <td>7.827</td>\n",
       "      <td>0.41 µmol/l</td>\n",
       "      <td>3.12 µmol/l</td>\n",
       "      <td>0.3 mg/l</td>\n",
       "      <td>high</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>[to be added from GLOMICON partner's metadata]</th>\n",
       "      <td>SBR</td>\n",
       "      <td>UDalhousie</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SOMLIT-ASTAN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>20.0</td>\n",
       "      <td>GLOMICON_INTERCOMP_SBR_20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>19.3 µg/l</td>\n",
       "      <td>7.827</td>\n",
       "      <td>0.41 µmol/l</td>\n",
       "      <td>3.12 µmol/l</td>\n",
       "      <td>0.3 mg/l</td>\n",
       "      <td>high</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NaN</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>19.3 µg/l</td>\n",
       "      <td>7.827</td>\n",
       "      <td>0.41 µmol/l</td>\n",
       "      <td>3.12 µmol/l</td>\n",
       "      <td>0.3 mg/l</td>\n",
       "      <td>high</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>241 rows × 93 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                     Collecting Institute  \\\n",
       "      sample_name                                                           \n",
       "MBARI CN18Sc37_12_Rep_Stdy5_AO                                      MBARI   \n",
       "      CN18Sc37_12_Rep_Stdy10_AO                                     MBARI   \n",
       "      CN18Sc37_12_Rep_Stdy14_AO                                     MBARI   \n",
       "      CN18Sc37_12_Rep_Stdy18_AO                                     MBARI   \n",
       "      CN18Sc37_12_Rep_Stdy22_AO                                     MBARI   \n",
       "...                                                                   ...   \n",
       "SBR   [to be added from GLOMICON partner's metadata]                  SBR   \n",
       "      [to be added from GLOMICON partner's metadata]                  SBR   \n",
       "      [to be added from GLOMICON partner's metadata]                  SBR   \n",
       "      [to be added from GLOMICON partner's metadata]                  SBR   \n",
       "      NaN                                                             NaN   \n",
       "\n",
       "                                                     Analyzing Institute  \\\n",
       "      sample_name                                                          \n",
       "MBARI CN18Sc37_12_Rep_Stdy5_AO                                     MBARI   \n",
       "      CN18Sc37_12_Rep_Stdy10_AO                                    MBARI   \n",
       "      CN18Sc37_12_Rep_Stdy14_AO                                    MBARI   \n",
       "      CN18Sc37_12_Rep_Stdy18_AO                                    MBARI   \n",
       "      CN18Sc37_12_Rep_Stdy22_AO                                    MBARI   \n",
       "...                                                                  ...   \n",
       "SBR   [to be added from GLOMICON partner's metadata]          UDalhousie   \n",
       "      [to be added from GLOMICON partner's metadata]          UDalhousie   \n",
       "      [to be added from GLOMICON partner's metadata]          UDalhousie   \n",
       "      [to be added from GLOMICON partner's metadata]          UDalhousie   \n",
       "      NaN                                                            NaN   \n",
       "\n",
       "                                                     cruise_id       station  \\\n",
       "      sample_name                                                              \n",
       "MBARI CN18Sc37_12_Rep_Stdy5_AO                           CN18S            TS   \n",
       "      CN18Sc37_12_Rep_Stdy10_AO                          CN18S            TS   \n",
       "      CN18Sc37_12_Rep_Stdy14_AO                          CN18S            TS   \n",
       "      CN18Sc37_12_Rep_Stdy18_AO                          CN18S            TS   \n",
       "      CN18Sc37_12_Rep_Stdy22_AO                          CN18S            TS   \n",
       "...                                                        ...           ...   \n",
       "SBR   [to be added from GLOMICON partner's metadata]       NaN  SOMLIT-ASTAN   \n",
       "      [to be added from GLOMICON partner's metadata]       NaN  SOMLIT-ASTAN   \n",
       "      [to be added from GLOMICON partner's metadata]       NaN  SOMLIT-ASTAN   \n",
       "      [to be added from GLOMICON partner's metadata]       NaN  SOMLIT-ASTAN   \n",
       "      NaN                                                  NaN           NaN   \n",
       "\n",
       "                                                     ctd_bottle_no  \\\n",
       "      sample_name                                                    \n",
       "MBARI CN18Sc37_12_Rep_Stdy5_AO                                12.0   \n",
       "      CN18Sc37_12_Rep_Stdy10_AO                               12.0   \n",
       "      CN18Sc37_12_Rep_Stdy14_AO                               12.0   \n",
       "      CN18Sc37_12_Rep_Stdy18_AO                               12.0   \n",
       "      CN18Sc37_12_Rep_Stdy22_AO                               12.0   \n",
       "...                                                            ...   \n",
       "SBR   [to be added from GLOMICON partner's metadata]           NaN   \n",
       "      [to be added from GLOMICON partner's metadata]           NaN   \n",
       "      [to be added from GLOMICON partner's metadata]           NaN   \n",
       "      [to be added from GLOMICON partner's metadata]           NaN   \n",
       "      NaN                                                      NaN   \n",
       "\n",
       "                                                     sample_replicate  \\\n",
       "      sample_name                                                       \n",
       "MBARI CN18Sc37_12_Rep_Stdy5_AO                                    5.0   \n",
       "      CN18Sc37_12_Rep_Stdy10_AO                                  10.0   \n",
       "      CN18Sc37_12_Rep_Stdy14_AO                                  14.0   \n",
       "      CN18Sc37_12_Rep_Stdy18_AO                                  18.0   \n",
       "      CN18Sc37_12_Rep_Stdy22_AO                                  22.0   \n",
       "...                                                               ...   \n",
       "SBR   [to be added from GLOMICON partner's metadata]              8.0   \n",
       "      [to be added from GLOMICON partner's metadata]             12.0   \n",
       "      [to be added from GLOMICON partner's metadata]             16.0   \n",
       "      [to be added from GLOMICON partner's metadata]             20.0   \n",
       "      NaN                                                         NaN   \n",
       "\n",
       "                                                             source_material_id  \\\n",
       "      sample_name                                                                 \n",
       "MBARI CN18Sc37_12_Rep_Stdy5_AO                            CN18Sc37_12_Rep_Stdy5   \n",
       "      CN18Sc37_12_Rep_Stdy10_AO                          CN18Sc37_12_Rep_Stdy10   \n",
       "      CN18Sc37_12_Rep_Stdy14_AO                          CN18Sc37_12_Rep_Stdy14   \n",
       "      CN18Sc37_12_Rep_Stdy18_AO                          CN18Sc37_12_Rep_Stdy18   \n",
       "      CN18Sc37_12_Rep_Stdy22_AO                          CN18Sc37_12_Rep_Stdy22   \n",
       "...                                                                         ...   \n",
       "SBR   [to be added from GLOMICON partner's metadata]  GLOMICON_INTERCOMP_SBR_08   \n",
       "      [to be added from GLOMICON partner's metadata]  GLOMICON_INTERCOMP_SBR_12   \n",
       "      [to be added from GLOMICON partner's metadata]  GLOMICON_INTERCOMP_SBR_16   \n",
       "      [to be added from GLOMICON partner's metadata]  GLOMICON_INTERCOMP_SBR_20   \n",
       "      NaN                                                                   NaN   \n",
       "\n",
       "                                                     extract_number  \\\n",
       "      sample_name                                                     \n",
       "MBARI CN18Sc37_12_Rep_Stdy5_AO                                  AO9   \n",
       "      CN18Sc37_12_Rep_Stdy10_AO                                AO10   \n",
       "      CN18Sc37_12_Rep_Stdy14_AO                                AO11   \n",
       "      CN18Sc37_12_Rep_Stdy18_AO                                AO12   \n",
       "      CN18Sc37_12_Rep_Stdy22_AO                                AO13   \n",
       "...                                                             ...   \n",
       "SBR   [to be added from GLOMICON partner's metadata]            NaN   \n",
       "      [to be added from GLOMICON partner's metadata]            NaN   \n",
       "      [to be added from GLOMICON partner's metadata]            NaN   \n",
       "      [to be added from GLOMICON partner's metadata]            NaN   \n",
       "      NaN                                                       NaN   \n",
       "\n",
       "                                                                                 sample_title  \\\n",
       "      sample_name                                                                               \n",
       "MBARI CN18Sc37_12_Rep_Stdy5_AO                        Seawater sample Monterey Bay (GLOMICON)   \n",
       "      CN18Sc37_12_Rep_Stdy10_AO                       Seawater sample Monterey Bay (GLOMICON)   \n",
       "      CN18Sc37_12_Rep_Stdy14_AO                       Seawater sample Monterey Bay (GLOMICON)   \n",
       "      CN18Sc37_12_Rep_Stdy18_AO                       Seawater sample Monterey Bay (GLOMICON)   \n",
       "      CN18Sc37_12_Rep_Stdy22_AO                       Seawater sample Monterey Bay (GLOMICON)   \n",
       "...                                                                                       ...   \n",
       "SBR   [to be added from GLOMICON partner's metadata]                                      NaN   \n",
       "      [to be added from GLOMICON partner's metadata]                                      NaN   \n",
       "      [to be added from GLOMICON partner's metadata]                                      NaN   \n",
       "      [to be added from GLOMICON partner's metadata]                                      NaN   \n",
       "      NaN                                                                                 NaN   \n",
       "\n",
       "                                                              project_id  ...  \\\n",
       "      sample_name                                                         ...   \n",
       "MBARI CN18Sc37_12_Rep_Stdy5_AO                                  GLOMICON  ...   \n",
       "      CN18Sc37_12_Rep_Stdy10_AO                                 GLOMICON  ...   \n",
       "      CN18Sc37_12_Rep_Stdy14_AO                                 GLOMICON  ...   \n",
       "      CN18Sc37_12_Rep_Stdy18_AO                                 GLOMICON  ...   \n",
       "      CN18Sc37_12_Rep_Stdy22_AO                                 GLOMICON  ...   \n",
       "...                                                                  ...  ...   \n",
       "SBR   [to be added from GLOMICON partner's metadata]  GLOMICON_INTERCOMP  ...   \n",
       "      [to be added from GLOMICON partner's metadata]  GLOMICON_INTERCOMP  ...   \n",
       "      [to be added from GLOMICON partner's metadata]  GLOMICON_INTERCOMP  ...   \n",
       "      [to be added from GLOMICON partner's metadata]  GLOMICON_INTERCOMP  ...   \n",
       "      NaN                                             GLOMICON_INTERCOMP  ...   \n",
       "\n",
       "                                                     part_org_nitro     ph  \\\n",
       "      sample_name                                                            \n",
       "MBARI CN18Sc37_12_Rep_Stdy5_AO                                  NaN    NaN   \n",
       "      CN18Sc37_12_Rep_Stdy10_AO                                 NaN    NaN   \n",
       "      CN18Sc37_12_Rep_Stdy14_AO                                 NaN    NaN   \n",
       "      CN18Sc37_12_Rep_Stdy18_AO                                 NaN    NaN   \n",
       "      CN18Sc37_12_Rep_Stdy22_AO                                 NaN    NaN   \n",
       "...                                                             ...    ...   \n",
       "SBR   [to be added from GLOMICON partner's metadata]      19.3 µg/l  7.827   \n",
       "      [to be added from GLOMICON partner's metadata]      19.3 µg/l  7.827   \n",
       "      [to be added from GLOMICON partner's metadata]      19.3 µg/l  7.827   \n",
       "      [to be added from GLOMICON partner's metadata]      19.3 µg/l  7.827   \n",
       "      NaN                                                 19.3 µg/l  7.827   \n",
       "\n",
       "                                                        phosphate  \\\n",
       "      sample_name                                                   \n",
       "MBARI CN18Sc37_12_Rep_Stdy5_AO                                NaN   \n",
       "      CN18Sc37_12_Rep_Stdy10_AO                               NaN   \n",
       "      CN18Sc37_12_Rep_Stdy14_AO                               NaN   \n",
       "      CN18Sc37_12_Rep_Stdy18_AO                               NaN   \n",
       "      CN18Sc37_12_Rep_Stdy22_AO                               NaN   \n",
       "...                                                           ...   \n",
       "SBR   [to be added from GLOMICON partner's metadata]  0.41 µmol/l   \n",
       "      [to be added from GLOMICON partner's metadata]  0.41 µmol/l   \n",
       "      [to be added from GLOMICON partner's metadata]  0.41 µmol/l   \n",
       "      [to be added from GLOMICON partner's metadata]  0.41 µmol/l   \n",
       "      NaN                                             0.41 µmol/l   \n",
       "\n",
       "                                                         silicate  \\\n",
       "      sample_name                                                   \n",
       "MBARI CN18Sc37_12_Rep_Stdy5_AO                                NaN   \n",
       "      CN18Sc37_12_Rep_Stdy10_AO                               NaN   \n",
       "      CN18Sc37_12_Rep_Stdy14_AO                               NaN   \n",
       "      CN18Sc37_12_Rep_Stdy18_AO                               NaN   \n",
       "      CN18Sc37_12_Rep_Stdy22_AO                               NaN   \n",
       "...                                                           ...   \n",
       "SBR   [to be added from GLOMICON partner's metadata]  3.12 µmol/l   \n",
       "      [to be added from GLOMICON partner's metadata]  3.12 µmol/l   \n",
       "      [to be added from GLOMICON partner's metadata]  3.12 µmol/l   \n",
       "      [to be added from GLOMICON partner's metadata]  3.12 µmol/l   \n",
       "      NaN                                             3.12 µmol/l   \n",
       "\n",
       "                                                     suspend_part_matter  \\\n",
       "      sample_name                                                          \n",
       "MBARI CN18Sc37_12_Rep_Stdy5_AO                                       NaN   \n",
       "      CN18Sc37_12_Rep_Stdy10_AO                                      NaN   \n",
       "      CN18Sc37_12_Rep_Stdy14_AO                                      NaN   \n",
       "      CN18Sc37_12_Rep_Stdy18_AO                                      NaN   \n",
       "      CN18Sc37_12_Rep_Stdy22_AO                                      NaN   \n",
       "...                                                                  ...   \n",
       "SBR   [to be added from GLOMICON partner's metadata]            0.3 mg/l   \n",
       "      [to be added from GLOMICON partner's metadata]            0.3 mg/l   \n",
       "      [to be added from GLOMICON partner's metadata]            0.3 mg/l   \n",
       "      [to be added from GLOMICON partner's metadata]            0.3 mg/l   \n",
       "      NaN                                                       0.3 mg/l   \n",
       "\n",
       "                                                     tidal_stage  \\\n",
       "      sample_name                                                  \n",
       "MBARI CN18Sc37_12_Rep_Stdy5_AO                               NaN   \n",
       "      CN18Sc37_12_Rep_Stdy10_AO                              NaN   \n",
       "      CN18Sc37_12_Rep_Stdy14_AO                              NaN   \n",
       "      CN18Sc37_12_Rep_Stdy18_AO                              NaN   \n",
       "      CN18Sc37_12_Rep_Stdy22_AO                              NaN   \n",
       "...                                                          ...   \n",
       "SBR   [to be added from GLOMICON partner's metadata]        high   \n",
       "      [to be added from GLOMICON partner's metadata]        high   \n",
       "      [to be added from GLOMICON partner's metadata]        high   \n",
       "      [to be added from GLOMICON partner's metadata]        high   \n",
       "      NaN                                                   high   \n",
       "\n",
       "                                                     date_dna_extracted  \\\n",
       "      sample_name                                                         \n",
       "MBARI CN18Sc37_12_Rep_Stdy5_AO                                      NaT   \n",
       "      CN18Sc37_12_Rep_Stdy10_AO                                     NaT   \n",
       "      CN18Sc37_12_Rep_Stdy14_AO                                     NaT   \n",
       "      CN18Sc37_12_Rep_Stdy18_AO                                     NaT   \n",
       "      CN18Sc37_12_Rep_Stdy22_AO                                     NaT   \n",
       "...                                                                 ...   \n",
       "SBR   [to be added from GLOMICON partner's metadata]                NaT   \n",
       "      [to be added from GLOMICON partner's metadata]                NaT   \n",
       "      [to be added from GLOMICON partner's metadata]                NaT   \n",
       "      [to be added from GLOMICON partner's metadata]                NaT   \n",
       "      NaN                                                           NaT   \n",
       "\n",
       "                                                     extraction_personnel  \\\n",
       "      sample_name                                                           \n",
       "MBARI CN18Sc37_12_Rep_Stdy5_AO                                        NaN   \n",
       "      CN18Sc37_12_Rep_Stdy10_AO                                       NaN   \n",
       "      CN18Sc37_12_Rep_Stdy14_AO                                       NaN   \n",
       "      CN18Sc37_12_Rep_Stdy18_AO                                       NaN   \n",
       "      CN18Sc37_12_Rep_Stdy22_AO                                       NaN   \n",
       "...                                                                   ...   \n",
       "SBR   [to be added from GLOMICON partner's metadata]                  NaN   \n",
       "      [to be added from GLOMICON partner's metadata]                  NaN   \n",
       "      [to be added from GLOMICON partner's metadata]                  NaN   \n",
       "      [to be added from GLOMICON partner's metadata]                  NaN   \n",
       "      NaN                                                             NaN   \n",
       "\n",
       "                                                     date_pcr pcr_personnel  \n",
       "      sample_name                                                            \n",
       "MBARI CN18Sc37_12_Rep_Stdy5_AO                            NaT           NaN  \n",
       "      CN18Sc37_12_Rep_Stdy10_AO                           NaT           NaN  \n",
       "      CN18Sc37_12_Rep_Stdy14_AO                           NaT           NaN  \n",
       "      CN18Sc37_12_Rep_Stdy18_AO                           NaT           NaN  \n",
       "      CN18Sc37_12_Rep_Stdy22_AO                           NaT           NaN  \n",
       "...                                                       ...           ...  \n",
       "SBR   [to be added from GLOMICON partner's metadata]      NaT           NaN  \n",
       "      [to be added from GLOMICON partner's metadata]      NaT           NaN  \n",
       "      [to be added from GLOMICON partner's metadata]      NaT           NaN  \n",
       "      [to be added from GLOMICON partner's metadata]      NaT           NaN  \n",
       "      NaN                                                 NaT           NaN  \n",
       "\n",
       "[241 rows x 93 columns]"
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# sample data\n",
    "df = pd.concat(samp_dfs, axis=0, keys=institutes)\n",
    "# export\n",
    "file = '/Users/kpitz/github/GLOMICON/intercomparison/Merged_Datasets/metadata/' + 'combined_sample_data.csv'\n",
    "df.to_csv(file)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 273,
   "id": "794fe252-1746-4ffa-a794-3a5c5c152805",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/5v/h4s_0pz17yb6f6krsxkzr5dw000b65/T/ipykernel_4375/2190577957.py:2: FutureWarning: The behavior of pd.concat with len(keys) != len(objs) is deprecated. In a future version this will raise instead of truncating to the smaller of the two sequences\n",
      "  df = pd.concat(samp_dfs, axis=0, keys=institutes)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>Collecting Institute</th>\n",
       "      <th>Analyzing Institute</th>\n",
       "      <th>cruise_id</th>\n",
       "      <th>station</th>\n",
       "      <th>ctd_bottle_no</th>\n",
       "      <th>sample_replicate</th>\n",
       "      <th>source_material_id</th>\n",
       "      <th>extract_number</th>\n",
       "      <th>sample_title</th>\n",
       "      <th>project_id</th>\n",
       "      <th>...</th>\n",
       "      <th>basisOfRecord</th>\n",
       "      <th>chlorophyll</th>\n",
       "      <th>density</th>\n",
       "      <th>fluor</th>\n",
       "      <th>pressure</th>\n",
       "      <th>salinity</th>\n",
       "      <th>samp_store_temp</th>\n",
       "      <th>temp</th>\n",
       "      <th>date_modified</th>\n",
       "      <th>modified_by</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"20\" valign=\"top\">SBR</th>\n",
       "      <th>SOC12</th>\n",
       "      <td>NOC</td>\n",
       "      <td>SBR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SOC18</th>\n",
       "      <td>NOC</td>\n",
       "      <td>SBR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SOC24</th>\n",
       "      <td>NOC</td>\n",
       "      <td>SBR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SOC28</th>\n",
       "      <td>NOC</td>\n",
       "      <td>SBR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DAL03</th>\n",
       "      <td>UDalhousie</td>\n",
       "      <td>SBR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DAL09</th>\n",
       "      <td>UDalhousie</td>\n",
       "      <td>SBR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DAL15</th>\n",
       "      <td>UDalhousie</td>\n",
       "      <td>SBR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DAL21</th>\n",
       "      <td>UDalhousie</td>\n",
       "      <td>SBR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DAL27</th>\n",
       "      <td>UDalhousie</td>\n",
       "      <td>SBR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EMk03</th>\n",
       "      <td>AWI</td>\n",
       "      <td>SBR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EMK08</th>\n",
       "      <td>AWI</td>\n",
       "      <td>SBR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EMk09</th>\n",
       "      <td>AWI</td>\n",
       "      <td>SBR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EMk15</th>\n",
       "      <td>AWI</td>\n",
       "      <td>SBR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EMk17</th>\n",
       "      <td>AWI</td>\n",
       "      <td>SBR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BMk13</th>\n",
       "      <td>AWI</td>\n",
       "      <td>SBR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BMk15</th>\n",
       "      <td>AWI</td>\n",
       "      <td>SBR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BMk17</th>\n",
       "      <td>AWI</td>\n",
       "      <td>SBR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BMk21</th>\n",
       "      <td>AWI</td>\n",
       "      <td>SBR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BMk25</th>\n",
       "      <td>AWI</td>\n",
       "      <td>SBR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Samples taken / prepared by SBR and analyzed by GLOMICON partners (almost complete information)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>20 rows × 38 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                       Collecting Institute  \\\n",
       "    sample_name                                                               \n",
       "SBR SOC12                                                               NOC   \n",
       "    SOC18                                                               NOC   \n",
       "    SOC24                                                               NOC   \n",
       "    SOC28                                                               NOC   \n",
       "    DAL03                                                        UDalhousie   \n",
       "    DAL09                                                        UDalhousie   \n",
       "    DAL15                                                        UDalhousie   \n",
       "    DAL21                                                        UDalhousie   \n",
       "    DAL27                                                        UDalhousie   \n",
       "    EMk03                                                               AWI   \n",
       "    EMK08                                                               AWI   \n",
       "    EMk09                                                               AWI   \n",
       "    EMk15                                                               AWI   \n",
       "    EMk17                                                               AWI   \n",
       "    BMk13                                                               AWI   \n",
       "    BMk15                                                               AWI   \n",
       "    BMk17                                                               AWI   \n",
       "    BMk21                                                               AWI   \n",
       "    BMk25                                                               AWI   \n",
       "    Samples taken / prepared by SBR and analyzed by...                  NaN   \n",
       "\n",
       "                                                       Analyzing Institute  \\\n",
       "    sample_name                                                              \n",
       "SBR SOC12                                                              SBR   \n",
       "    SOC18                                                              SBR   \n",
       "    SOC24                                                              SBR   \n",
       "    SOC28                                                              SBR   \n",
       "    DAL03                                                              SBR   \n",
       "    DAL09                                                              SBR   \n",
       "    DAL15                                                              SBR   \n",
       "    DAL21                                                              SBR   \n",
       "    DAL27                                                              SBR   \n",
       "    EMk03                                                              SBR   \n",
       "    EMK08                                                              SBR   \n",
       "    EMk09                                                              SBR   \n",
       "    EMk15                                                              SBR   \n",
       "    EMk17                                                              SBR   \n",
       "    BMk13                                                              SBR   \n",
       "    BMk15                                                              SBR   \n",
       "    BMk17                                                              SBR   \n",
       "    BMk21                                                              SBR   \n",
       "    BMk25                                                              SBR   \n",
       "    Samples taken / prepared by SBR and analyzed by...                 NaN   \n",
       "\n",
       "                                                       cruise_id station  \\\n",
       "    sample_name                                                            \n",
       "SBR SOC12                                                    NaN     NaN   \n",
       "    SOC18                                                    NaN     NaN   \n",
       "    SOC24                                                    NaN     NaN   \n",
       "    SOC28                                                    NaN     NaN   \n",
       "    DAL03                                                    NaN     NaN   \n",
       "    DAL09                                                    NaN     NaN   \n",
       "    DAL15                                                    NaN     NaN   \n",
       "    DAL21                                                    NaN     NaN   \n",
       "    DAL27                                                    NaN     NaN   \n",
       "    EMk03                                                    NaN     NaN   \n",
       "    EMK08                                                    NaN     NaN   \n",
       "    EMk09                                                    NaN     NaN   \n",
       "    EMk15                                                    NaN     NaN   \n",
       "    EMk17                                                    NaN     NaN   \n",
       "    BMk13                                                    NaN     NaN   \n",
       "    BMk15                                                    NaN     NaN   \n",
       "    BMk17                                                    NaN     NaN   \n",
       "    BMk21                                                    NaN     NaN   \n",
       "    BMk25                                                    NaN     NaN   \n",
       "    Samples taken / prepared by SBR and analyzed by...       NaN     NaN   \n",
       "\n",
       "                                                       ctd_bottle_no  \\\n",
       "    sample_name                                                        \n",
       "SBR SOC12                                                        NaN   \n",
       "    SOC18                                                        NaN   \n",
       "    SOC24                                                        NaN   \n",
       "    SOC28                                                        NaN   \n",
       "    DAL03                                                        NaN   \n",
       "    DAL09                                                        NaN   \n",
       "    DAL15                                                        NaN   \n",
       "    DAL21                                                        NaN   \n",
       "    DAL27                                                        NaN   \n",
       "    EMk03                                                        NaN   \n",
       "    EMK08                                                        NaN   \n",
       "    EMk09                                                        NaN   \n",
       "    EMk15                                                        NaN   \n",
       "    EMk17                                                        NaN   \n",
       "    BMk13                                                        NaN   \n",
       "    BMk15                                                        NaN   \n",
       "    BMk17                                                        NaN   \n",
       "    BMk21                                                        NaN   \n",
       "    BMk25                                                        NaN   \n",
       "    Samples taken / prepared by SBR and analyzed by...           NaN   \n",
       "\n",
       "                                                       sample_replicate  \\\n",
       "    sample_name                                                           \n",
       "SBR SOC12                                                           NaN   \n",
       "    SOC18                                                           NaN   \n",
       "    SOC24                                                           NaN   \n",
       "    SOC28                                                           NaN   \n",
       "    DAL03                                                           NaN   \n",
       "    DAL09                                                           NaN   \n",
       "    DAL15                                                           NaN   \n",
       "    DAL21                                                           NaN   \n",
       "    DAL27                                                           NaN   \n",
       "    EMk03                                                           NaN   \n",
       "    EMK08                                                           NaN   \n",
       "    EMk09                                                           NaN   \n",
       "    EMk15                                                           NaN   \n",
       "    EMk17                                                           NaN   \n",
       "    BMk13                                                           NaN   \n",
       "    BMk15                                                           NaN   \n",
       "    BMk17                                                           NaN   \n",
       "    BMk21                                                           NaN   \n",
       "    BMk25                                                           NaN   \n",
       "    Samples taken / prepared by SBR and analyzed by...              NaN   \n",
       "\n",
       "                                                       source_material_id  \\\n",
       "    sample_name                                                             \n",
       "SBR SOC12                                                             NaN   \n",
       "    SOC18                                                             NaN   \n",
       "    SOC24                                                             NaN   \n",
       "    SOC28                                                             NaN   \n",
       "    DAL03                                                             NaN   \n",
       "    DAL09                                                             NaN   \n",
       "    DAL15                                                             NaN   \n",
       "    DAL21                                                             NaN   \n",
       "    DAL27                                                             NaN   \n",
       "    EMk03                                                             NaN   \n",
       "    EMK08                                                             NaN   \n",
       "    EMk09                                                             NaN   \n",
       "    EMk15                                                             NaN   \n",
       "    EMk17                                                             NaN   \n",
       "    BMk13                                                             NaN   \n",
       "    BMk15                                                             NaN   \n",
       "    BMk17                                                             NaN   \n",
       "    BMk21                                                             NaN   \n",
       "    BMk25                                                             NaN   \n",
       "    Samples taken / prepared by SBR and analyzed by...                NaN   \n",
       "\n",
       "                                                       extract_number  \\\n",
       "    sample_name                                                         \n",
       "SBR SOC12                                                         NaN   \n",
       "    SOC18                                                         NaN   \n",
       "    SOC24                                                         NaN   \n",
       "    SOC28                                                         NaN   \n",
       "    DAL03                                                         NaN   \n",
       "    DAL09                                                         NaN   \n",
       "    DAL15                                                         NaN   \n",
       "    DAL21                                                         NaN   \n",
       "    DAL27                                                         NaN   \n",
       "    EMk03                                                         NaN   \n",
       "    EMK08                                                         NaN   \n",
       "    EMk09                                                         NaN   \n",
       "    EMk15                                                         NaN   \n",
       "    EMk17                                                         NaN   \n",
       "    BMk13                                                         NaN   \n",
       "    BMk15                                                         NaN   \n",
       "    BMk17                                                         NaN   \n",
       "    BMk21                                                         NaN   \n",
       "    BMk25                                                         NaN   \n",
       "    Samples taken / prepared by SBR and analyzed by...            NaN   \n",
       "\n",
       "                                                       sample_title  \\\n",
       "    sample_name                                                       \n",
       "SBR SOC12                                                       NaN   \n",
       "    SOC18                                                       NaN   \n",
       "    SOC24                                                       NaN   \n",
       "    SOC28                                                       NaN   \n",
       "    DAL03                                                       NaN   \n",
       "    DAL09                                                       NaN   \n",
       "    DAL15                                                       NaN   \n",
       "    DAL21                                                       NaN   \n",
       "    DAL27                                                       NaN   \n",
       "    EMk03                                                       NaN   \n",
       "    EMK08                                                       NaN   \n",
       "    EMk09                                                       NaN   \n",
       "    EMk15                                                       NaN   \n",
       "    EMk17                                                       NaN   \n",
       "    BMk13                                                       NaN   \n",
       "    BMk15                                                       NaN   \n",
       "    BMk17                                                       NaN   \n",
       "    BMk21                                                       NaN   \n",
       "    BMk25                                                       NaN   \n",
       "    Samples taken / prepared by SBR and analyzed by...          NaN   \n",
       "\n",
       "                                                                project_id  \\\n",
       "    sample_name                                                              \n",
       "SBR SOC12                                               GLOMICON_INTERCOMP   \n",
       "    SOC18                                               GLOMICON_INTERCOMP   \n",
       "    SOC24                                               GLOMICON_INTERCOMP   \n",
       "    SOC28                                               GLOMICON_INTERCOMP   \n",
       "    DAL03                                               GLOMICON_INTERCOMP   \n",
       "    DAL09                                               GLOMICON_INTERCOMP   \n",
       "    DAL15                                               GLOMICON_INTERCOMP   \n",
       "    DAL21                                               GLOMICON_INTERCOMP   \n",
       "    DAL27                                               GLOMICON_INTERCOMP   \n",
       "    EMk03                                               GLOMICON_INTERCOMP   \n",
       "    EMK08                                               GLOMICON_INTERCOMP   \n",
       "    EMk09                                               GLOMICON_INTERCOMP   \n",
       "    EMk15                                               GLOMICON_INTERCOMP   \n",
       "    EMk17                                               GLOMICON_INTERCOMP   \n",
       "    BMk13                                               GLOMICON_INTERCOMP   \n",
       "    BMk15                                               GLOMICON_INTERCOMP   \n",
       "    BMk17                                               GLOMICON_INTERCOMP   \n",
       "    BMk21                                               GLOMICON_INTERCOMP   \n",
       "    BMk25                                               GLOMICON_INTERCOMP   \n",
       "    Samples taken / prepared by SBR and analyzed by...                 NaN   \n",
       "\n",
       "                                                        ... basisOfRecord  \\\n",
       "    sample_name                                         ...                 \n",
       "SBR SOC12                                               ...           NaN   \n",
       "    SOC18                                               ...           NaN   \n",
       "    SOC24                                               ...           NaN   \n",
       "    SOC28                                               ...           NaN   \n",
       "    DAL03                                               ...           NaN   \n",
       "    DAL09                                               ...           NaN   \n",
       "    DAL15                                               ...           NaN   \n",
       "    DAL21                                               ...           NaN   \n",
       "    DAL27                                               ...           NaN   \n",
       "    EMk03                                               ...           NaN   \n",
       "    EMK08                                               ...           NaN   \n",
       "    EMk09                                               ...           NaN   \n",
       "    EMk15                                               ...           NaN   \n",
       "    EMk17                                               ...           NaN   \n",
       "    BMk13                                               ...           NaN   \n",
       "    BMk15                                               ...           NaN   \n",
       "    BMk17                                               ...           NaN   \n",
       "    BMk21                                               ...           NaN   \n",
       "    BMk25                                               ...           NaN   \n",
       "    Samples taken / prepared by SBR and analyzed by...  ...           NaN   \n",
       "\n",
       "                                                       chlorophyll density  \\\n",
       "    sample_name                                                              \n",
       "SBR SOC12                                                      NaN     NaN   \n",
       "    SOC18                                                      NaN     NaN   \n",
       "    SOC24                                                      NaN     NaN   \n",
       "    SOC28                                                      NaN     NaN   \n",
       "    DAL03                                                      NaN     NaN   \n",
       "    DAL09                                                      NaN     NaN   \n",
       "    DAL15                                                      NaN     NaN   \n",
       "    DAL21                                                      NaN     NaN   \n",
       "    DAL27                                                      NaN     NaN   \n",
       "    EMk03                                                      NaN     NaN   \n",
       "    EMK08                                                      NaN     NaN   \n",
       "    EMk09                                                      NaN     NaN   \n",
       "    EMk15                                                      NaN     NaN   \n",
       "    EMk17                                                      NaN     NaN   \n",
       "    BMk13                                                      NaN     NaN   \n",
       "    BMk15                                                      NaN     NaN   \n",
       "    BMk17                                                      NaN     NaN   \n",
       "    BMk21                                                      NaN     NaN   \n",
       "    BMk25                                                      NaN     NaN   \n",
       "    Samples taken / prepared by SBR and analyzed by...         NaN     NaN   \n",
       "\n",
       "                                                       fluor pressure  \\\n",
       "    sample_name                                                         \n",
       "SBR SOC12                                                NaN      NaN   \n",
       "    SOC18                                                NaN      NaN   \n",
       "    SOC24                                                NaN      NaN   \n",
       "    SOC28                                                NaN      NaN   \n",
       "    DAL03                                                NaN      NaN   \n",
       "    DAL09                                                NaN      NaN   \n",
       "    DAL15                                                NaN      NaN   \n",
       "    DAL21                                                NaN      NaN   \n",
       "    DAL27                                                NaN      NaN   \n",
       "    EMk03                                                NaN      NaN   \n",
       "    EMK08                                                NaN      NaN   \n",
       "    EMk09                                                NaN      NaN   \n",
       "    EMk15                                                NaN      NaN   \n",
       "    EMk17                                                NaN      NaN   \n",
       "    BMk13                                                NaN      NaN   \n",
       "    BMk15                                                NaN      NaN   \n",
       "    BMk17                                                NaN      NaN   \n",
       "    BMk21                                                NaN      NaN   \n",
       "    BMk25                                                NaN      NaN   \n",
       "    Samples taken / prepared by SBR and analyzed by...   NaN      NaN   \n",
       "\n",
       "                                                       salinity  \\\n",
       "    sample_name                                                   \n",
       "SBR SOC12                                                   NaN   \n",
       "    SOC18                                                   NaN   \n",
       "    SOC24                                                   NaN   \n",
       "    SOC28                                                   NaN   \n",
       "    DAL03                                                   NaN   \n",
       "    DAL09                                                   NaN   \n",
       "    DAL15                                                   NaN   \n",
       "    DAL21                                                   NaN   \n",
       "    DAL27                                                   NaN   \n",
       "    EMk03                                                   NaN   \n",
       "    EMK08                                                   NaN   \n",
       "    EMk09                                                   NaN   \n",
       "    EMk15                                                   NaN   \n",
       "    EMk17                                                   NaN   \n",
       "    BMk13                                                   NaN   \n",
       "    BMk15                                                   NaN   \n",
       "    BMk17                                                   NaN   \n",
       "    BMk21                                                   NaN   \n",
       "    BMk25                                                   NaN   \n",
       "    Samples taken / prepared by SBR and analyzed by...      NaN   \n",
       "\n",
       "                                                       samp_store_temp temp  \\\n",
       "    sample_name                                                               \n",
       "SBR SOC12                                                          NaN  NaN   \n",
       "    SOC18                                                          NaN  NaN   \n",
       "    SOC24                                                          NaN  NaN   \n",
       "    SOC28                                                          NaN  NaN   \n",
       "    DAL03                                                          NaN  NaN   \n",
       "    DAL09                                                          NaN  NaN   \n",
       "    DAL15                                                          NaN  NaN   \n",
       "    DAL21                                                          NaN  NaN   \n",
       "    DAL27                                                          NaN  NaN   \n",
       "    EMk03                                                          NaN  NaN   \n",
       "    EMK08                                                          NaN  NaN   \n",
       "    EMk09                                                          NaN  NaN   \n",
       "    EMk15                                                          NaN  NaN   \n",
       "    EMk17                                                          NaN  NaN   \n",
       "    BMk13                                                          NaN  NaN   \n",
       "    BMk15                                                          NaN  NaN   \n",
       "    BMk17                                                          NaN  NaN   \n",
       "    BMk21                                                          NaN  NaN   \n",
       "    BMk25                                                          NaN  NaN   \n",
       "    Samples taken / prepared by SBR and analyzed by...             NaN  NaN   \n",
       "\n",
       "                                                       date_modified  \\\n",
       "    sample_name                                                        \n",
       "SBR SOC12                                                        NaT   \n",
       "    SOC18                                                        NaT   \n",
       "    SOC24                                                        NaT   \n",
       "    SOC28                                                        NaT   \n",
       "    DAL03                                                        NaT   \n",
       "    DAL09                                                        NaT   \n",
       "    DAL15                                                        NaT   \n",
       "    DAL21                                                        NaT   \n",
       "    DAL27                                                        NaT   \n",
       "    EMk03                                                        NaT   \n",
       "    EMK08                                                        NaT   \n",
       "    EMk09                                                        NaT   \n",
       "    EMk15                                                        NaT   \n",
       "    EMk17                                                        NaT   \n",
       "    BMk13                                                        NaT   \n",
       "    BMk15                                                        NaT   \n",
       "    BMk17                                                        NaT   \n",
       "    BMk21                                                        NaT   \n",
       "    BMk25                                                        NaT   \n",
       "    Samples taken / prepared by SBR and analyzed by...           NaT   \n",
       "\n",
       "                                                       modified_by  \n",
       "    sample_name                                                     \n",
       "SBR SOC12                                                      NaN  \n",
       "    SOC18                                                      NaN  \n",
       "    SOC24                                                      NaN  \n",
       "    SOC28                                                      NaN  \n",
       "    DAL03                                                      NaN  \n",
       "    DAL09                                                      NaN  \n",
       "    DAL15                                                      NaN  \n",
       "    DAL21                                                      NaN  \n",
       "    DAL27                                                      NaN  \n",
       "    EMk03                                                      NaN  \n",
       "    EMK08                                                      NaN  \n",
       "    EMk09                                                      NaN  \n",
       "    EMk15                                                      NaN  \n",
       "    EMk17                                                      NaN  \n",
       "    BMk13                                                      NaN  \n",
       "    BMk15                                                      NaN  \n",
       "    BMk17                                                      NaN  \n",
       "    BMk21                                                      NaN  \n",
       "    BMk25                                                      NaN  \n",
       "    Samples taken / prepared by SBR and analyzed by...         NaN  \n",
       "\n",
       "[20 rows x 38 columns]"
      ]
     },
     "execution_count": 273,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# sample data\n",
    "df = pd.concat(samp_dfs, axis=0, keys=institutes)\n",
    "#water_samp_cols.remove('sample_name')\n",
    "#water_samp_cols.remove('sheet')\n",
    "\n",
    "df = df[water_samp_cols]\n",
    "#df = df.loc[df['temp'].isna()==True]\n",
    "\n",
    "df.tail(n=20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2fa7a6d5-b2ec-40d1-8203-45c844ac169a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f1b8fa66-8e0f-41a2-9081-461511e479cf",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 261,
   "id": "840a638b-4adf-4160-a553-84c836e0f78c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['MBARI' 'NOAA' 'MOCKBLOOM' 'NOC' 'UDalhousie' 'MOCKEVEN' 'AWI' 'SBR' nan]\n",
      "short_name ['MBARI' 'NOAA' 'bloomMock' 'NOC' 'UDalhousie' 'evenMock' 'AWI' 'SBR' nan\n",
      " 'blank']\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/5v/h4s_0pz17yb6f6krsxkzr5dw000b65/T/ipykernel_4375/801486836.py:7: FutureWarning: The behavior of pd.concat with len(keys) != len(objs) is deprecated. In a future version this will raise instead of truncating to the smaller of the two sequences\n",
      "  df = pd.concat(samp_dfs, axis=0, keys=institutes)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>level_0</th>\n",
       "      <th>unified_name</th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Collecting Institute</th>\n",
       "      <th>Analyzing Institute</th>\n",
       "      <th>short_name</th>\n",
       "      <th>samp_num</th>\n",
       "      <th>source_material_id</th>\n",
       "      <th>dna_conc</th>\n",
       "      <th>size_frac</th>\n",
       "      <th>salinity</th>\n",
       "      <th>amplicon_sequenced</th>\n",
       "      <th>seq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI_5.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy5_AO</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>5.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.22um</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>18S V9</td>\n",
       "      <td>sequenced</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI_10.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy10_AO</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>10.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.22um</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>18S V9</td>\n",
       "      <td>sequenced</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI_14.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy14_AO</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>14.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy14</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.22um</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>18S V9</td>\n",
       "      <td>sequenced</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI_18.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy18_AO</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>18.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy18</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.22um</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>18S V9</td>\n",
       "      <td>sequenced</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI_22.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy22_AO</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>22.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy22</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.22um</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>18S V9</td>\n",
       "      <td>sequenced</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI_1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.22um</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI_2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>2.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.22um</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI_3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>3.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.22um</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI_4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>4.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.22um</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI_6.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>6.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.22um</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI_7.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>7.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.22um</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI_8.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>8.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.22um</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI_9.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>9.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.22um</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI_11.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>11.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.22um</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI_12.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>12.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.22um</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI_13.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>13.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.22um</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI_15.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>15.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.22um</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI_16.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>16.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.22um</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI_17.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>17.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.22um</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI_19.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>19.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.22um</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI_20.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>20.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.22um</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI_21.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>21.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy21</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.22um</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI_23.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>23.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy23</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.22um</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI_24.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>24.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy24</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.22um</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI_25.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>25.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy25</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.22um</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>AWI</td>\n",
       "      <td>MBARI_2.0</td>\n",
       "      <td>Davenport_02</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>sequenced</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>AWI</td>\n",
       "      <td>MBARI_2.0</td>\n",
       "      <td>Davenport_02</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>sequenced</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>AWI</td>\n",
       "      <td>MBARI_6.0</td>\n",
       "      <td>Davenport_06</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>6.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>sequenced</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>AWI</td>\n",
       "      <td>MBARI_6.0</td>\n",
       "      <td>Davenport_06</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>6.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>sequenced</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>AWI</td>\n",
       "      <td>MBARI_11.0</td>\n",
       "      <td>Davenport_11</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>11.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.92</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>sequenced</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101</th>\n",
       "      <td>AWI</td>\n",
       "      <td>MBARI_11.0</td>\n",
       "      <td>Davenport_11</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>11.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.92</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>sequenced</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102</th>\n",
       "      <td>AWI</td>\n",
       "      <td>MBARI_15.0</td>\n",
       "      <td>Davenport_15</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>15.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>sequenced</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>AWI</td>\n",
       "      <td>MBARI_15.0</td>\n",
       "      <td>Davenport_15</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>15.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>sequenced</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>AWI</td>\n",
       "      <td>MBARI_19.0</td>\n",
       "      <td>Davenport_19</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>19.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.02</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>sequenced</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>AWI</td>\n",
       "      <td>MBARI_19.0</td>\n",
       "      <td>Davenport_19</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>19.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.02</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>sequenced</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>194</th>\n",
       "      <td>SBR</td>\n",
       "      <td>MBARI_3.0</td>\n",
       "      <td>MBA03</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>SBR</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>80.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>sequenced</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>195</th>\n",
       "      <td>SBR</td>\n",
       "      <td>MBARI_7.0</td>\n",
       "      <td>MBA07</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>SBR</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>7.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>71.9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>sequenced</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>196</th>\n",
       "      <td>SBR</td>\n",
       "      <td>MBARI_12.0</td>\n",
       "      <td>MBA12</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>SBR</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>12.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>53.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>sequenced</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197</th>\n",
       "      <td>SBR</td>\n",
       "      <td>MBARI_16.0</td>\n",
       "      <td>MBA16</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>SBR</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>16.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>63.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>sequenced</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>198</th>\n",
       "      <td>SBR</td>\n",
       "      <td>MBARI_20.0</td>\n",
       "      <td>MBA20</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>SBR</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>20.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>80.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>18S V4</td>\n",
       "      <td>sequenced</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    level_0 unified_name                sample_name Collecting Institute  \\\n",
       "0     MBARI    MBARI_5.0   CN18Sc37_12_Rep_Stdy5_AO                MBARI   \n",
       "1     MBARI   MBARI_10.0  CN18Sc37_12_Rep_Stdy10_AO                MBARI   \n",
       "2     MBARI   MBARI_14.0  CN18Sc37_12_Rep_Stdy14_AO                MBARI   \n",
       "3     MBARI   MBARI_18.0  CN18Sc37_12_Rep_Stdy18_AO                MBARI   \n",
       "4     MBARI   MBARI_22.0  CN18Sc37_12_Rep_Stdy22_AO                MBARI   \n",
       "28    MBARI    MBARI_1.0                        NaN                MBARI   \n",
       "29    MBARI    MBARI_2.0                        NaN                MBARI   \n",
       "30    MBARI    MBARI_3.0                        NaN                MBARI   \n",
       "31    MBARI    MBARI_4.0                        NaN                MBARI   \n",
       "32    MBARI    MBARI_6.0                        NaN                MBARI   \n",
       "33    MBARI    MBARI_7.0                        NaN                MBARI   \n",
       "34    MBARI    MBARI_8.0                        NaN                MBARI   \n",
       "35    MBARI    MBARI_9.0                        NaN                MBARI   \n",
       "36    MBARI   MBARI_11.0                        NaN                MBARI   \n",
       "37    MBARI   MBARI_12.0                        NaN                MBARI   \n",
       "38    MBARI   MBARI_13.0                        NaN                MBARI   \n",
       "39    MBARI   MBARI_15.0                        NaN                MBARI   \n",
       "40    MBARI   MBARI_16.0                        NaN                MBARI   \n",
       "41    MBARI   MBARI_17.0                        NaN                MBARI   \n",
       "42    MBARI   MBARI_19.0                        NaN                MBARI   \n",
       "43    MBARI   MBARI_20.0                        NaN                MBARI   \n",
       "44    MBARI   MBARI_21.0                        NaN                MBARI   \n",
       "45    MBARI   MBARI_23.0                        NaN                MBARI   \n",
       "46    MBARI   MBARI_24.0                        NaN                MBARI   \n",
       "47    MBARI   MBARI_25.0                        NaN                MBARI   \n",
       "96      AWI    MBARI_2.0               Davenport_02                MBARI   \n",
       "97      AWI    MBARI_2.0               Davenport_02                MBARI   \n",
       "98      AWI    MBARI_6.0               Davenport_06                MBARI   \n",
       "99      AWI    MBARI_6.0               Davenport_06                MBARI   \n",
       "100     AWI   MBARI_11.0               Davenport_11                MBARI   \n",
       "101     AWI   MBARI_11.0               Davenport_11                MBARI   \n",
       "102     AWI   MBARI_15.0               Davenport_15                MBARI   \n",
       "103     AWI   MBARI_15.0               Davenport_15                MBARI   \n",
       "104     AWI   MBARI_19.0               Davenport_19                MBARI   \n",
       "105     AWI   MBARI_19.0               Davenport_19                MBARI   \n",
       "194     SBR    MBARI_3.0                      MBA03                MBARI   \n",
       "195     SBR    MBARI_7.0                      MBA07                MBARI   \n",
       "196     SBR   MBARI_12.0                      MBA12                MBARI   \n",
       "197     SBR   MBARI_16.0                      MBA16                MBARI   \n",
       "198     SBR   MBARI_20.0                      MBA20                MBARI   \n",
       "\n",
       "    Analyzing Institute short_name  samp_num      source_material_id dna_conc  \\\n",
       "0                 MBARI      MBARI       5.0   CN18Sc37_12_Rep_Stdy5      NaN   \n",
       "1                 MBARI      MBARI      10.0  CN18Sc37_12_Rep_Stdy10      NaN   \n",
       "2                 MBARI      MBARI      14.0  CN18Sc37_12_Rep_Stdy14      NaN   \n",
       "3                 MBARI      MBARI      18.0  CN18Sc37_12_Rep_Stdy18      NaN   \n",
       "4                 MBARI      MBARI      22.0  CN18Sc37_12_Rep_Stdy22      NaN   \n",
       "28                  NaN      MBARI       1.0   CN18Sc37_12_Rep_Stdy1      NaN   \n",
       "29                  NaN      MBARI       2.0   CN18Sc37_12_Rep_Stdy2      NaN   \n",
       "30                  NaN      MBARI       3.0   CN18Sc37_12_Rep_Stdy3      NaN   \n",
       "31                  NaN      MBARI       4.0   CN18Sc37_12_Rep_Stdy4      NaN   \n",
       "32                  NaN      MBARI       6.0   CN18Sc37_12_Rep_Stdy6      NaN   \n",
       "33                  NaN      MBARI       7.0   CN18Sc37_12_Rep_Stdy7      NaN   \n",
       "34                  NaN      MBARI       8.0   CN18Sc37_12_Rep_Stdy8      NaN   \n",
       "35                  NaN      MBARI       9.0   CN18Sc37_12_Rep_Stdy9      NaN   \n",
       "36                  NaN      MBARI      11.0  CN18Sc37_12_Rep_Stdy11      NaN   \n",
       "37                  NaN      MBARI      12.0  CN18Sc37_12_Rep_Stdy12      NaN   \n",
       "38                  NaN      MBARI      13.0  CN18Sc37_12_Rep_Stdy13      NaN   \n",
       "39                  NaN      MBARI      15.0  CN18Sc37_12_Rep_Stdy15      NaN   \n",
       "40                  NaN      MBARI      16.0  CN18Sc37_12_Rep_Stdy16      NaN   \n",
       "41                  NaN      MBARI      17.0  CN18Sc37_12_Rep_Stdy17      NaN   \n",
       "42                  NaN      MBARI      19.0  CN18Sc37_12_Rep_Stdy19      NaN   \n",
       "43                  NaN      MBARI      20.0  CN18Sc37_12_Rep_Stdy20      NaN   \n",
       "44                  NaN      MBARI      21.0  CN18Sc37_12_Rep_Stdy21      NaN   \n",
       "45                  NaN      MBARI      23.0  CN18Sc37_12_Rep_Stdy23      NaN   \n",
       "46                  NaN      MBARI      24.0  CN18Sc37_12_Rep_Stdy24      NaN   \n",
       "47                  NaN      MBARI      25.0  CN18Sc37_12_Rep_Stdy25      NaN   \n",
       "96                  AWI      MBARI       2.0                     NaN      6.3   \n",
       "97                  AWI      MBARI       2.0                     NaN      6.3   \n",
       "98                  AWI      MBARI       6.0                     NaN      5.6   \n",
       "99                  AWI      MBARI       6.0                     NaN      5.6   \n",
       "100                 AWI      MBARI      11.0                     NaN     2.92   \n",
       "101                 AWI      MBARI      11.0                     NaN     2.92   \n",
       "102                 AWI      MBARI      15.0                     NaN      6.6   \n",
       "103                 AWI      MBARI      15.0                     NaN      6.6   \n",
       "104                 AWI      MBARI      19.0                     NaN     4.02   \n",
       "105                 AWI      MBARI      19.0                     NaN     4.02   \n",
       "194                 SBR      MBARI       3.0                     NaN     80.3   \n",
       "195                 SBR      MBARI       7.0                     NaN     71.9   \n",
       "196                 SBR      MBARI      12.0                     NaN     53.0   \n",
       "197                 SBR      MBARI      16.0                     NaN     63.4   \n",
       "198                 SBR      MBARI      20.0                     NaN     80.1   \n",
       "\n",
       "    size_frac salinity amplicon_sequenced        seq  \n",
       "0      0.22um  33.5385             18S V9  sequenced  \n",
       "1      0.22um  33.5385             18S V9  sequenced  \n",
       "2      0.22um  33.5385             18S V9  sequenced  \n",
       "3      0.22um  33.5385             18S V9  sequenced  \n",
       "4      0.22um  33.5385             18S V9  sequenced  \n",
       "28     0.22um  33.5385                NaN             \n",
       "29     0.22um  33.5385                NaN             \n",
       "30     0.22um  33.5385                NaN             \n",
       "31     0.22um  33.5385                NaN             \n",
       "32     0.22um  33.5385                NaN             \n",
       "33     0.22um  33.5385                NaN             \n",
       "34     0.22um  33.5385                NaN             \n",
       "35     0.22um  33.5385                NaN             \n",
       "36     0.22um  33.5385                NaN             \n",
       "37     0.22um  33.5385                NaN             \n",
       "38     0.22um  33.5385                NaN             \n",
       "39     0.22um  33.5385                NaN             \n",
       "40     0.22um  33.5385                NaN             \n",
       "41     0.22um  33.5385                NaN             \n",
       "42     0.22um  33.5385                NaN             \n",
       "43     0.22um  33.5385                NaN             \n",
       "44     0.22um  33.5385                NaN             \n",
       "45     0.22um  33.5385                NaN             \n",
       "46     0.22um  33.5385                NaN             \n",
       "47     0.22um  33.5385                NaN             \n",
       "96        NaN      NaN             18S V4  sequenced  \n",
       "97        NaN      NaN             18S V4  sequenced  \n",
       "98        NaN      NaN             18S V4  sequenced  \n",
       "99        NaN      NaN             18S V4  sequenced  \n",
       "100       NaN      NaN             18S V4  sequenced  \n",
       "101       NaN      NaN             18S V4  sequenced  \n",
       "102       NaN      NaN             18S V4  sequenced  \n",
       "103       NaN      NaN             18S V4  sequenced  \n",
       "104       NaN      NaN             18S V4  sequenced  \n",
       "105       NaN      NaN             18S V4  sequenced  \n",
       "194       NaN      NaN             18S V4  sequenced  \n",
       "195       NaN      NaN             18S V4  sequenced  \n",
       "196       NaN      NaN             18S V4  sequenced  \n",
       "197       NaN      NaN             18S V4  sequenced  \n",
       "198       NaN      NaN             18S V4  sequenced  "
      ]
     },
     "execution_count": 261,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# edit sample data\n",
    "# want to merge metadata from collecting institutions with analyzing institutions with one filter per row (at the end)\n",
    "# each filter has a unique name that can be made from its collecting origin and its replicate number\n",
    "# difficulty is different naming between analyzing institutions\n",
    "\n",
    "# combine all sample data\n",
    "df = pd.concat(samp_dfs, axis=0, keys=institutes)\n",
    "df = df.reset_index()\n",
    "\n",
    "# every sample has a collecting institute and a number associated with it\n",
    "# drop columns that are all nan\n",
    "df = df.dropna(how='all', axis=1)\n",
    "\n",
    "# create new sample number column\n",
    "df['samp_num'] = np.nan\n",
    "\n",
    "# get numbers associated with names - can be either in sample_name or in source_material_id columns\n",
    "# only name with numbers is 'CN18Sc37_12_Rep_Stdy'\n",
    "df['samp_num'] = df['sample_name'].str.replace('CN18Sc37_12_Rep_Stdy', 'MBARI_').str.extract('(\\\\d+)')\n",
    "df.loc[df['samp_num'].isna(),'samp_num' ] = df['source_material_id'].str.replace('CN18Sc37_12_Rep_Stdy', 'MBARI_').str.extract('(\\\\d+)')[0]\n",
    "\n",
    "# create unified short names, corresponding with origin:\n",
    "df['short_name'] = ''\n",
    "print(df['Collecting Institute'].unique())\n",
    "# ['MBARI' 'NOAA' 'MOCKBLOOM' 'NOC' 'UDalhousie' 'MOCKEVEN' 'AWI' 'SBR' nan]\n",
    "df['short_name'] = df['Collecting Institute']\n",
    "# take care of controls:\n",
    "df.loc[df['sample_name'].str.contains('evenMock|emk|MOCKEVEN', case=False)==True, 'short_name'] = 'evenMock'\n",
    "df.loc[df['sample_name'].str.contains('bloomMock|bmk|MOCKBLOOM', case=False)==True, 'short_name'] = 'bloomMock'\n",
    "# if no sample_name ...\n",
    "df.loc[df['source_material_id'].str.contains('evenMock|emk', case=False)==True, 'short_name'] = 'evenMock'\n",
    "df.loc[df['source_material_id'].str.contains('bloomMock|bmk', case=False)==True, 'short_name'] = 'bloomMock'\n",
    "\n",
    "#blank_01 - a single blank AWI ran and included, don't want to mix it up with mock communities.\n",
    "df.loc[df['sample_name'].str.contains('blank_01', case=False)==True, 'short_name'] = 'blank'\n",
    "print('short_name',df['short_name'].unique())\n",
    "df['unified_name'] = df['short_name'] + '_' + df['samp_num'].astype(float).astype(str)\n",
    "df = df.loc[df['unified_name'].isna()==False]\n",
    "df['samp_num'] = df['samp_num'].astype(float)\n",
    "\n",
    "df['sample_name'] = df['sample_name'].str.replace(\"[to be added from GLOMICON partner's metadata]\",'')\n",
    "\n",
    "'''\n",
    "df['sample_name'] = df['sample_name'].str.replace(\"[to be added from GLOMICON partner's metadata]\",'')\n",
    "\n",
    "new_samp_dfs = []\n",
    "new_samp_names = []\n",
    "new_keys = []\n",
    "for i,group in df.groupby('level_0'):\n",
    "    group = group.drop_duplicates('unified_name')\n",
    "    #gruop = group.dropna(how='all', axis=1)\n",
    "    group.set_index(['short_name', 'samp_num','unified_name'], inplace=True)\n",
    "    new_samp_dfs.append(group)\n",
    "    new_samp_names.append(group[['sample_name', 'source_material_id']])\n",
    "    new_keys.append(i)\n",
    "    \n",
    "'''\n",
    "# want one row per sequenced sample\n",
    "cols = ['level_0', 'unified_name', 'sample_name', 'Collecting Institute', 'Analyzing Institute', 'short_name', 'samp_num', 'source_material_id', 'dna_conc', 'size_frac', 'salinity', 'amplicon_sequenced']\n",
    "df = df[cols]\n",
    "\n",
    "# for now remove mock controls\n",
    "df = df.loc[df['short_name'].isin(['bloomMock', 'evenMock'])==False]\n",
    "# subset just to make it work for one group...\n",
    "df = df.loc[df['short_name'].isin(['MBARI'])==True]\n",
    "\n",
    "\n",
    "\n",
    "df['seq'] = ''\n",
    "df.loc[df['sample_name'].isna()==False, 'seq'] = 'sequenced'\n",
    "df.loc[df['seq']=='', 'Analyzing Institute'] = np.nan\n",
    "\n",
    "#df = df.loc[df['amplicon_sequenced'].isna()==True] # get all unsequenced samples\n",
    "\n",
    "#df = df.groupby('unified_name').sum()\n",
    "\n",
    "df.head(n=50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 216,
   "id": "894e2bf9-0cc5-41df-9494-120a3579343a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr:last-of-type th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>AOML</th>\n",
       "      <th>AWI</th>\n",
       "      <th>MBARI</th>\n",
       "      <th>SBR</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>sample_name</th>\n",
       "      <th>sample_name</th>\n",
       "      <th>sample_name</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>short_name</th>\n",
       "      <th>samp_num</th>\n",
       "      <th>unified_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">AWI</th>\n",
       "      <th>1.0</th>\n",
       "      <th>AWI_1.0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Framstrait_01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2.0</th>\n",
       "      <th>AWI_2.0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>[to be added from GLOMICON partner's metadata]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3.0</th>\n",
       "      <th>AWI_3.0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>[to be added from GLOMICON partner's metadata]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4.0</th>\n",
       "      <th>AWI_4.0</th>\n",
       "      <td>GLOMICON_AWI_4</td>\n",
       "      <td>[to be added from GLOMICON partner's metadata]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5.0</th>\n",
       "      <th>AWI_5.0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Framstrait_05</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">evenMock</th>\n",
       "      <th>15.0</th>\n",
       "      <th>evenMock_15.0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>EMk15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>EMk15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17.0</th>\n",
       "      <th>evenMock_17.0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>EMk17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>EMk17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18.0</th>\n",
       "      <th>evenMock_18.0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>[to be added from GLOMICON partner's metadata]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20.0</th>\n",
       "      <th>evenMock_20.0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>[to be added from GLOMICON partner's metadata]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NaN</th>\n",
       "      <th>evenMock_nan</th>\n",
       "      <td>GLOMICON_EvenMock_A</td>\n",
       "      <td>[to be added from GLOMICON partner's metadata]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>150 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                  AOML  \\\n",
       "                                           sample_name   \n",
       "short_name samp_num unified_name                         \n",
       "AWI        1.0      AWI_1.0                        NaN   \n",
       "           2.0      AWI_2.0                        NaN   \n",
       "           3.0      AWI_3.0                        NaN   \n",
       "           4.0      AWI_4.0             GLOMICON_AWI_4   \n",
       "           5.0      AWI_5.0                        NaN   \n",
       "...                                                ...   \n",
       "evenMock   15.0     evenMock_15.0                  NaN   \n",
       "           17.0     evenMock_17.0                  NaN   \n",
       "           18.0     evenMock_18.0                  NaN   \n",
       "           20.0     evenMock_20.0                  NaN   \n",
       "           NaN      evenMock_nan   GLOMICON_EvenMock_A   \n",
       "\n",
       "                                                                              AWI  \\\n",
       "                                                                      sample_name   \n",
       "short_name samp_num unified_name                                                    \n",
       "AWI        1.0      AWI_1.0                                         Framstrait_01   \n",
       "           2.0      AWI_2.0        [to be added from GLOMICON partner's metadata]   \n",
       "           3.0      AWI_3.0        [to be added from GLOMICON partner's metadata]   \n",
       "           4.0      AWI_4.0        [to be added from GLOMICON partner's metadata]   \n",
       "           5.0      AWI_5.0                                         Framstrait_05   \n",
       "...                                                                           ...   \n",
       "evenMock   15.0     evenMock_15.0                                           EMk15   \n",
       "           17.0     evenMock_17.0                                           EMk17   \n",
       "           18.0     evenMock_18.0  [to be added from GLOMICON partner's metadata]   \n",
       "           20.0     evenMock_20.0  [to be added from GLOMICON partner's metadata]   \n",
       "           NaN      evenMock_nan   [to be added from GLOMICON partner's metadata]   \n",
       "\n",
       "                                        MBARI         SBR  \n",
       "                                  sample_name sample_name  \n",
       "short_name samp_num unified_name                           \n",
       "AWI        1.0      AWI_1.0               NaN         NaN  \n",
       "           2.0      AWI_2.0               NaN         NaN  \n",
       "           3.0      AWI_3.0               NaN         NaN  \n",
       "           4.0      AWI_4.0               NaN         NaN  \n",
       "           5.0      AWI_5.0               NaN         NaN  \n",
       "...                                       ...         ...  \n",
       "evenMock   15.0     evenMock_15.0         NaN       EMk15  \n",
       "           17.0     evenMock_17.0         NaN       EMk17  \n",
       "           18.0     evenMock_18.0         NaN         NaN  \n",
       "           20.0     evenMock_20.0         NaN         NaN  \n",
       "           NaN      evenMock_nan          NaN         NaN  \n",
       "\n",
       "[150 rows x 4 columns]"
      ]
     },
     "execution_count": 216,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.concat(new_samp_dfs, axis=1, keys = new_keys)\n",
    "#df = pd.concat(new_samp_dfs, axis=1)\n",
    "df = df.dropna(how='all', axis=1)\n",
    "df = df.sort_index()\n",
    "df.to_csv('/Users/kpitz/github/GLOMICON/intercomparison/Merged_Datasets/metadata/test.csv')\n",
    "\n",
    "df = df.loc[:, (slice(None), 'sample_name')]\n",
    "\n",
    "\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 213,
   "id": "5ab666e4-6052-4de6-9c71-64840015fb06",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>source_material_id</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>short_name</th>\n",
       "      <th>samp_num</th>\n",
       "      <th>unified_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">AOML</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">AWI</th>\n",
       "      <th>4.0</th>\n",
       "      <th>AWI_4.0</th>\n",
       "      <td>GLOMICON_AWI_4</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8.0</th>\n",
       "      <th>AWI_8.0</th>\n",
       "      <td>GLOMICON_AWI_8</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12.0</th>\n",
       "      <th>AWI_12.0</th>\n",
       "      <td>GLOMICON_AWI_12</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16.0</th>\n",
       "      <th>AWI_16.0</th>\n",
       "      <td>GLOMICON_AWI_16</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20.0</th>\n",
       "      <th>AWI_20.0</th>\n",
       "      <td>GLOMICON_AWI_20</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">SBR</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">evenMock</th>\n",
       "      <th>3.0</th>\n",
       "      <th>evenMock_3.0</th>\n",
       "      <td>EMk03</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8.0</th>\n",
       "      <th>evenMock_8.0</th>\n",
       "      <td>EMK08</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9.0</th>\n",
       "      <th>evenMock_9.0</th>\n",
       "      <td>EMk09</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15.0</th>\n",
       "      <th>evenMock_15.0</th>\n",
       "      <td>EMk15</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17.0</th>\n",
       "      <th>evenMock_17.0</th>\n",
       "      <td>EMk17</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>197 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            sample_name source_material_id\n",
       "     short_name samp_num unified_name                                     \n",
       "AOML AWI        4.0      AWI_4.0         GLOMICON_AWI_4                NaN\n",
       "                8.0      AWI_8.0         GLOMICON_AWI_8                NaN\n",
       "                12.0     AWI_12.0       GLOMICON_AWI_12                NaN\n",
       "                16.0     AWI_16.0       GLOMICON_AWI_16                NaN\n",
       "                20.0     AWI_20.0       GLOMICON_AWI_20                NaN\n",
       "...                                                 ...                ...\n",
       "SBR  evenMock   3.0      evenMock_3.0             EMk03                NaN\n",
       "                8.0      evenMock_8.0             EMK08                NaN\n",
       "                9.0      evenMock_9.0             EMk09                NaN\n",
       "                15.0     evenMock_15.0            EMk15                NaN\n",
       "                17.0     evenMock_17.0            EMk17                NaN\n",
       "\n",
       "[197 rows x 2 columns]"
      ]
     },
     "execution_count": 213,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.concat(new_samp_names, axis=0, keys = new_keys)\n",
    "#df = pd.concat(new_samp_dfs, axis=1)\n",
    "df = df.dropna(how='all', axis=1)\n",
    "df = df.sort_index()\n",
    "df.to_csv('/Users/kpitz/github/GLOMICON/intercomparison/Merged_Datasets/metadata/test.csv')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9e2cc6ac-1418-4e59-8fc9-8002273df518",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "id": "c6c20913-cfdf-49c0-b7dd-3f585bca82bf",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['MBARI' 'NOAA' 'MOCKBLOOM' 'NOC' 'UDalhousie' 'MOCKEVEN' 'AWI' 'SBR' nan]\n",
      "short_name ['MBARI' 'NOAA' 'bloomMock' 'NOC' 'UDalhousie' 'evenMock' 'AWI' 'SBR' nan\n",
      " 'blank']\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>unified_name</th>\n",
       "      <th>level_0</th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Collecting Institute</th>\n",
       "      <th>Analyzing Institute</th>\n",
       "      <th>samp_num</th>\n",
       "      <th>source_material_id</th>\n",
       "      <th>organism</th>\n",
       "      <th>short_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>AWI_4.0</td>\n",
       "      <td>AOML</td>\n",
       "      <td>GLOMICON_AWI_4</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AWI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>AWI_8.0</td>\n",
       "      <td>AOML</td>\n",
       "      <td>GLOMICON_AWI_8</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>8.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AWI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>AWI_12.0</td>\n",
       "      <td>AOML</td>\n",
       "      <td>GLOMICON_AWI_12</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>12.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AWI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>AWI_16.0</td>\n",
       "      <td>AOML</td>\n",
       "      <td>GLOMICON_AWI_16</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>16.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AWI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>AWI_20.0</td>\n",
       "      <td>AOML</td>\n",
       "      <td>GLOMICON_AWI_20</td>\n",
       "      <td>AWI</td>\n",
       "      <td>NOAA</td>\n",
       "      <td>20.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AWI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>235</th>\n",
       "      <td>SBR_4.0</td>\n",
       "      <td>SBR</td>\n",
       "      <td>[to be added from GLOMICON partner's metadata]</td>\n",
       "      <td>SBR</td>\n",
       "      <td>UDalhousie</td>\n",
       "      <td>4.0</td>\n",
       "      <td>GLOMICON_INTERCOMP_SBR_04</td>\n",
       "      <td>seawater metagenome</td>\n",
       "      <td>SBR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236</th>\n",
       "      <td>SBR_8.0</td>\n",
       "      <td>SBR</td>\n",
       "      <td>[to be added from GLOMICON partner's metadata]</td>\n",
       "      <td>SBR</td>\n",
       "      <td>UDalhousie</td>\n",
       "      <td>8.0</td>\n",
       "      <td>GLOMICON_INTERCOMP_SBR_08</td>\n",
       "      <td>seawater metagenome</td>\n",
       "      <td>SBR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>237</th>\n",
       "      <td>SBR_12.0</td>\n",
       "      <td>SBR</td>\n",
       "      <td>[to be added from GLOMICON partner's metadata]</td>\n",
       "      <td>SBR</td>\n",
       "      <td>UDalhousie</td>\n",
       "      <td>12.0</td>\n",
       "      <td>GLOMICON_INTERCOMP_SBR_12</td>\n",
       "      <td>seawater metagenome</td>\n",
       "      <td>SBR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>238</th>\n",
       "      <td>SBR_16.0</td>\n",
       "      <td>SBR</td>\n",
       "      <td>[to be added from GLOMICON partner's metadata]</td>\n",
       "      <td>SBR</td>\n",
       "      <td>UDalhousie</td>\n",
       "      <td>16.0</td>\n",
       "      <td>GLOMICON_INTERCOMP_SBR_16</td>\n",
       "      <td>seawater metagenome</td>\n",
       "      <td>SBR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>239</th>\n",
       "      <td>SBR_20.0</td>\n",
       "      <td>SBR</td>\n",
       "      <td>[to be added from GLOMICON partner's metadata]</td>\n",
       "      <td>SBR</td>\n",
       "      <td>UDalhousie</td>\n",
       "      <td>20.0</td>\n",
       "      <td>GLOMICON_INTERCOMP_SBR_20</td>\n",
       "      <td>seawater metagenome</td>\n",
       "      <td>SBR</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>208 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    unified_name level_0                                     sample_name  \\\n",
       "51       AWI_4.0    AOML                                  GLOMICON_AWI_4   \n",
       "52       AWI_8.0    AOML                                  GLOMICON_AWI_8   \n",
       "48      AWI_12.0    AOML                                 GLOMICON_AWI_12   \n",
       "49      AWI_16.0    AOML                                 GLOMICON_AWI_16   \n",
       "50      AWI_20.0    AOML                                 GLOMICON_AWI_20   \n",
       "..           ...     ...                                             ...   \n",
       "235      SBR_4.0     SBR  [to be added from GLOMICON partner's metadata]   \n",
       "236      SBR_8.0     SBR  [to be added from GLOMICON partner's metadata]   \n",
       "237     SBR_12.0     SBR  [to be added from GLOMICON partner's metadata]   \n",
       "238     SBR_16.0     SBR  [to be added from GLOMICON partner's metadata]   \n",
       "239     SBR_20.0     SBR  [to be added from GLOMICON partner's metadata]   \n",
       "\n",
       "    Collecting Institute Analyzing Institute  samp_num  \\\n",
       "51                   AWI                NOAA       4.0   \n",
       "52                   AWI                NOAA       8.0   \n",
       "48                   AWI                NOAA      12.0   \n",
       "49                   AWI                NOAA      16.0   \n",
       "50                   AWI                NOAA      20.0   \n",
       "..                   ...                 ...       ...   \n",
       "235                  SBR          UDalhousie       4.0   \n",
       "236                  SBR          UDalhousie       8.0   \n",
       "237                  SBR          UDalhousie      12.0   \n",
       "238                  SBR          UDalhousie      16.0   \n",
       "239                  SBR          UDalhousie      20.0   \n",
       "\n",
       "            source_material_id             organism short_name  \n",
       "51                         NaN                  NaN        AWI  \n",
       "52                         NaN                  NaN        AWI  \n",
       "48                         NaN                  NaN        AWI  \n",
       "49                         NaN                  NaN        AWI  \n",
       "50                         NaN                  NaN        AWI  \n",
       "..                         ...                  ...        ...  \n",
       "235  GLOMICON_INTERCOMP_SBR_04  seawater metagenome        SBR  \n",
       "236  GLOMICON_INTERCOMP_SBR_08  seawater metagenome        SBR  \n",
       "237  GLOMICON_INTERCOMP_SBR_12  seawater metagenome        SBR  \n",
       "238  GLOMICON_INTERCOMP_SBR_16  seawater metagenome        SBR  \n",
       "239  GLOMICON_INTERCOMP_SBR_20  seawater metagenome        SBR  \n",
       "\n",
       "[208 rows x 9 columns]"
      ]
     },
     "execution_count": 173,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# edit sample data\n",
    "# want to merge metadata from collecting institutions with analyzing institutions with one filter per row (at the end)\n",
    "# each filter has a unique name that can be made from its collecting origin and its replicate number\n",
    "# difficulty is different naming between analyzing institutions\n",
    "\n",
    "# combine all sample data\n",
    "df = pd.concat(samp_dfs, axis=0, keys=institutes)\n",
    "df = df.reset_index()\n",
    "\n",
    "# every sample has a collecting institute and a number associated with it\n",
    "# drop columns that are all nan\n",
    "df = df.dropna(how='all', axis=1)\n",
    "\n",
    "# create new sample number column\n",
    "df['samp_num'] = np.nan\n",
    "\n",
    "# get numbers associated with names - can be either in sample_name or in source_material_id columns\n",
    "# only name with numbers is 'CN18Sc37_12_Rep_Stdy'\n",
    "df['samp_num'] = df['sample_name'].str.replace('CN18Sc37_12_Rep_Stdy', 'MBARI_').str.extract('(\\\\d+)')\n",
    "df.loc[df['samp_num'].isna(),'samp_num' ] = df['source_material_id'].str.replace('CN18Sc37_12_Rep_Stdy', 'MBARI_').str.extract('(\\\\d+)')[0]\n",
    "\n",
    "# create unified short names, corresponding with origin:\n",
    "df['short_name'] = ''\n",
    "print(df['Collecting Institute'].unique())\n",
    "# ['MBARI' 'NOAA' 'MOCKBLOOM' 'NOC' 'UDalhousie' 'MOCKEVEN' 'AWI' 'SBR' nan]\n",
    "df['short_name'] = df['Collecting Institute']\n",
    "# take care of controls:\n",
    "df.loc[df['sample_name'].str.contains('evenMock|emk|MOCKEVEN', case=False)==True, 'short_name'] = 'evenMock'\n",
    "df.loc[df['sample_name'].str.contains('bloomMock|bmk|MOCKBLOOM', case=False)==True, 'short_name'] = 'bloomMock'\n",
    "# if no sample_name ...\n",
    "df.loc[df['source_material_id'].str.contains('evenMock|emk', case=False)==True, 'short_name'] = 'evenMock'\n",
    "df.loc[df['source_material_id'].str.contains('bloomMock|bmk', case=False)==True, 'short_name'] = 'bloomMock'\n",
    "\n",
    "#blank_01 - a single blank AWI ran and included, don't want to mix it up with mock communities.\n",
    "df.loc[df['sample_name'].str.contains('blank_01', case=False)==True, 'short_name'] = 'blank'\n",
    "\n",
    "print('short_name',df['short_name'].unique())\n",
    "\n",
    "df['unified_name'] = df['short_name'] + '_' + df['samp_num'].astype(float).astype(str)\n",
    "\n",
    "\n",
    "df = df.loc[df['unified_name'].isna()==False]\n",
    "\n",
    "cols = ['unified_name','level_0', 'sample_name', 'Collecting Institute', 'Analyzing Institute','samp_num', 'source_material_id', 'organism', 'short_name']\n",
    "#df = df.loc[df['samp_num']=='']\n",
    "#df = df.loc[df['Collecting Institute']=='NOAA']\n",
    "#print(df['Collecting Institute'].unique())\n",
    "#df = df.loc[df['Collecting Institute'].isin(['AWI', 'MOCKBLOOM', 'MOCKEVEN'])]\n",
    "#df = df.loc[df['Collecting Institute'].isin([ 'MOCKBLOOM', 'MOCKEVEN'])]\n",
    "\n",
    "#df = df.sort_values(['short_name', 'samp_num', 'unified_name','Collecting Institute', 'organism'], ascending=False)\n",
    "df['samp_num'] = df['samp_num'].astype(float)\n",
    "#df = df.sort_values(['short_name', 'samp_num', 'unified_name','level_0','Collecting Institute', 'organism'], ascending=True)\n",
    "df = df.sort_values(['level_0','Analyzing Institute','short_name', 'samp_num', 'unified_name','level_0','Collecting Institute', 'organism'], ascending=True)\n",
    "#df[cols].head(n=50)\n",
    "\n",
    "#df = df.set_index(['short_name', 'samp_num', 'unified_name' ])\n",
    "\n",
    "#df = df.pivot(index='unified_name', values='source_material_id', columns='Collecting Institute')\n",
    "df = df[cols]\n",
    "df = df.drop_duplicates()  # AWI sequenced same filters twice\n",
    "\n",
    "#df = df.pivot(index='unified_name', columns='level_0', values='sample_name')\n",
    "df[cols].to_csv('/Users/kpitz/github/GLOMICON/intercomparison/Merged_Datasets/metadata/test.csv')\n",
    "df[cols]\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "11346bc8-48be-41ad-8d2a-1f4be1bd02b0",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9307e9b6-2d42-4b2f-a247-9720d97bda34",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "10cd2311-0d54-4925-91c7-50bb6fb983a8",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "57e74e6d-f0a9-4630-a2cf-ba48f028e0e9",
   "metadata": {},
   "source": [
    "### Import Metadata files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "5c3d6deb-1613-47e6-a7f2-35865c139e92",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sheets in excel file: ['README', 'study_data', 'water_sample_data', 'amplicon_prep_data', 'metag_prep_data', 'analysis_data', 'SRA_template', 'measurement_metadata', 'SRA Terms', 'Version log']\n",
      "study_data 5\n",
      "water_sample_data 8\n",
      "amplicon_prep_data 6\n",
      "analysis_data 6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/kpitz/anaconda3/envs/scipy_test/lib/python3.12/site-packages/openpyxl/worksheet/_read_only.py:85: UserWarning: Data Validation extension is not supported and will be removed\n",
      "  for idx, row in parser.parse():\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>amplicon_sequenced</th>\n",
       "      <th>trim_method</th>\n",
       "      <th>cluster_method</th>\n",
       "      <th>pid_clustering</th>\n",
       "      <th>taxa_class_method</th>\n",
       "      <th>taxa_ref_db</th>\n",
       "      <th>code_repo</th>\n",
       "      <th>date_modified</th>\n",
       "      <th>sheet</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>18S V9</td>\n",
       "      <td>atropos</td>\n",
       "      <td>dada2</td>\n",
       "      <td>ASV</td>\n",
       "      <td>BLAST; MEGAN LCA</td>\n",
       "      <td>genbank NR</td>\n",
       "      <td>https://github.com/MBARI-BOG/BOG-Banzai-Dada2-...</td>\n",
       "      <td>2024-08-15 08:44:22.928</td>\n",
       "      <td>analysis_data</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  amplicon_sequenced trim_method cluster_method pid_clustering  \\\n",
       "0             18S V9     atropos          dada2            ASV   \n",
       "\n",
       "  taxa_class_method taxa_ref_db  \\\n",
       "0  BLAST; MEGAN LCA  genbank NR   \n",
       "\n",
       "                                           code_repo           date_modified  \\\n",
       "0  https://github.com/MBARI-BOG/BOG-Banzai-Dada2-... 2024-08-15 08:44:22.928   \n",
       "\n",
       "           sheet  \n",
       "0  analysis_data  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#look at test file\n",
    "file = '/Users/kpitz/github/GLOMICON/intercomparison/metadata/MBARI_NOAA_MIMARKS.survey.water.6.0_sharing.xlsx'\n",
    "xl = pd.ExcelFile(file)\n",
    "print('Sheets in excel file:',xl.sheet_names)  # see all sheet names\n",
    "#sheets = xl.sheet_names\n",
    "sheets = ['study_data', 'water_sample_data', 'amplicon_prep_data', 'analysis_data']\n",
    "dfs= []\n",
    "skiprow_num = [5,8,6,6]\n",
    "for sheet, sr in zip(sheets, skiprow_num):\n",
    "    print(sheet,sr)\n",
    "    df = xl.parse(sheet,skiprows=sr)\n",
    "    df = df.dropna(how='all', axis=1)\n",
    "    df = df.dropna(how='all', axis=0)\n",
    "    df['sheet'] = sheet\n",
    "    dfs.append(df)\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "8ee0e059-f832-4686-83d3-163aefc73858",
   "metadata": {},
   "outputs": [],
   "source": [
    "def water_amplicon_merge(water_df, amplicon_df):\n",
    "    df = water_df.set_index('sample_name')\n",
    "    df = df.drop(['sheet', 'date_modified','modified_by'], axis=1)\n",
    "    # add amplicon metadata to water sample metadata (some water samples don't have amplicon data if not sequenced in institute)\n",
    "    df = df.join(amplicon_df.set_index('sample_name'), rsuffix='_amplicon_sheet')\n",
    "    df = df.drop(['sheet', 'date_modified','modified_by'], axis=1)\n",
    "    return df\n",
    "def study_data_merge(study_meta, bioinf_meta, institute_name):\n",
    "    df = pd.concat([study_meta, bioinf_meta], axis=1)\n",
    "    df = df.drop(['sheet', 'date_modified'], axis=1)\n",
    "    df = df.T\n",
    "    df = df.rename(columns={0:institute_name})\n",
    "    return df\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "b75c62bc-0913-454b-8aa0-4c4828232d62",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MBARI</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>project_id</th>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>project_name</th>\n",
       "      <td>Intercomparison of filtered seawater samples a...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>project_id_external</th>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>project_contact</th>\n",
       "      <td>Kathleen Pitz, kpitz@mbari.org, https://orcid....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>project_description</th>\n",
       "      <td>Replicate filters, originating from surface wa...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                 MBARI\n",
       "project_id                                          GLOMICON_INTERCOMP\n",
       "project_name         Intercomparison of filtered seawater samples a...\n",
       "project_id_external                                 GLOMICON_INTERCOMP\n",
       "project_contact      Kathleen Pitz, kpitz@mbari.org, https://orcid....\n",
       "project_description  Replicate filters, originating from surface wa..."
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# study_dfs = []\n",
    "df = study_data_merge(dfs[0], dfs[3], 'MBARI')\n",
    "study_dfs.append(df)\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "cb80f340-a6c9-491d-bdee-8c6c368fc5c8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Collecting Institute</th>\n",
       "      <th>Analyzing Institute</th>\n",
       "      <th>cruise_id</th>\n",
       "      <th>station</th>\n",
       "      <th>ctd_bottle_no</th>\n",
       "      <th>sample_replicate</th>\n",
       "      <th>source_material_id</th>\n",
       "      <th>extract_number</th>\n",
       "      <th>sample_title</th>\n",
       "      <th>project_id</th>\n",
       "      <th>...</th>\n",
       "      <th>seq_meth</th>\n",
       "      <th>amplicon_sequenced</th>\n",
       "      <th>target_gene</th>\n",
       "      <th>target_subfragment</th>\n",
       "      <th>pcr_primer_forward</th>\n",
       "      <th>pcr_primer_reverse</th>\n",
       "      <th>pcr_primer_name_forward</th>\n",
       "      <th>pcr_primer_name_reverse</th>\n",
       "      <th>pcr_primer_reference</th>\n",
       "      <th>pcr_cond</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sample_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CN18Sc37_12_Rep_Stdy5_AO</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy5</td>\n",
       "      <td>AO9</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>GLOMICON</td>\n",
       "      <td>...</td>\n",
       "      <td>Illumina MiSeq 2x150</td>\n",
       "      <td>18S V9</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V9</td>\n",
       "      <td>GTACACACCGCCCGTC</td>\n",
       "      <td>TGATCCTTCTGCAGGTTCACCTAC</td>\n",
       "      <td>Euk_1391f</td>\n",
       "      <td>EukBr</td>\n",
       "      <td>Amaral-Zettler et al. 2009, Stoeck et al. 2010...</td>\n",
       "      <td>dx.doi.org/10.17504/protocols.io.36wgq3d8olk5/v2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN18Sc37_12_Rep_Stdy10_AO</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy10</td>\n",
       "      <td>AO10</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>GLOMICON</td>\n",
       "      <td>...</td>\n",
       "      <td>Illumina MiSeq 2x150</td>\n",
       "      <td>18S V9</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V9</td>\n",
       "      <td>GTACACACCGCCCGTC</td>\n",
       "      <td>TGATCCTTCTGCAGGTTCACCTAC</td>\n",
       "      <td>Euk_1391f</td>\n",
       "      <td>EukBr</td>\n",
       "      <td>Amaral-Zettler et al. 2009, Stoeck et al. 2010...</td>\n",
       "      <td>dx.doi.org/10.17504/protocols.io.36wgq3d8olk5/v2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN18Sc37_12_Rep_Stdy14_AO</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy14</td>\n",
       "      <td>AO11</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>GLOMICON</td>\n",
       "      <td>...</td>\n",
       "      <td>Illumina MiSeq 2x150</td>\n",
       "      <td>18S V9</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V9</td>\n",
       "      <td>GTACACACCGCCCGTC</td>\n",
       "      <td>TGATCCTTCTGCAGGTTCACCTAC</td>\n",
       "      <td>Euk_1391f</td>\n",
       "      <td>EukBr</td>\n",
       "      <td>Amaral-Zettler et al. 2009, Stoeck et al. 2010...</td>\n",
       "      <td>dx.doi.org/10.17504/protocols.io.36wgq3d8olk5/v2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN18Sc37_12_Rep_Stdy18_AO</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy18</td>\n",
       "      <td>AO12</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>GLOMICON</td>\n",
       "      <td>...</td>\n",
       "      <td>Illumina MiSeq 2x150</td>\n",
       "      <td>18S V9</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V9</td>\n",
       "      <td>GTACACACCGCCCGTC</td>\n",
       "      <td>TGATCCTTCTGCAGGTTCACCTAC</td>\n",
       "      <td>Euk_1391f</td>\n",
       "      <td>EukBr</td>\n",
       "      <td>Amaral-Zettler et al. 2009, Stoeck et al. 2010...</td>\n",
       "      <td>dx.doi.org/10.17504/protocols.io.36wgq3d8olk5/v2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN18Sc37_12_Rep_Stdy22_AO</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy22</td>\n",
       "      <td>AO13</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>GLOMICON</td>\n",
       "      <td>...</td>\n",
       "      <td>Illumina MiSeq 2x150</td>\n",
       "      <td>18S V9</td>\n",
       "      <td>18S rRNA</td>\n",
       "      <td>V9</td>\n",
       "      <td>GTACACACCGCCCGTC</td>\n",
       "      <td>TGATCCTTCTGCAGGTTCACCTAC</td>\n",
       "      <td>Euk_1391f</td>\n",
       "      <td>EukBr</td>\n",
       "      <td>Amaral-Zettler et al. 2009, Stoeck et al. 2010...</td>\n",
       "      <td>dx.doi.org/10.17504/protocols.io.36wgq3d8olk5/v2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 58 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                          Collecting Institute Analyzing Institute cruise_id  \\\n",
       "sample_name                                                                    \n",
       "CN18Sc37_12_Rep_Stdy5_AO                 MBARI               MBARI     CN18S   \n",
       "CN18Sc37_12_Rep_Stdy10_AO                MBARI               MBARI     CN18S   \n",
       "CN18Sc37_12_Rep_Stdy14_AO                MBARI               MBARI     CN18S   \n",
       "CN18Sc37_12_Rep_Stdy18_AO                MBARI               MBARI     CN18S   \n",
       "CN18Sc37_12_Rep_Stdy22_AO                MBARI               MBARI     CN18S   \n",
       "\n",
       "                          station  ctd_bottle_no  sample_replicate  \\\n",
       "sample_name                                                          \n",
       "CN18Sc37_12_Rep_Stdy5_AO       TS           12.0               5.0   \n",
       "CN18Sc37_12_Rep_Stdy10_AO      TS           12.0              10.0   \n",
       "CN18Sc37_12_Rep_Stdy14_AO      TS           12.0              14.0   \n",
       "CN18Sc37_12_Rep_Stdy18_AO      TS           12.0              18.0   \n",
       "CN18Sc37_12_Rep_Stdy22_AO      TS           12.0              22.0   \n",
       "\n",
       "                               source_material_id extract_number  \\\n",
       "sample_name                                                        \n",
       "CN18Sc37_12_Rep_Stdy5_AO    CN18Sc37_12_Rep_Stdy5            AO9   \n",
       "CN18Sc37_12_Rep_Stdy10_AO  CN18Sc37_12_Rep_Stdy10           AO10   \n",
       "CN18Sc37_12_Rep_Stdy14_AO  CN18Sc37_12_Rep_Stdy14           AO11   \n",
       "CN18Sc37_12_Rep_Stdy18_AO  CN18Sc37_12_Rep_Stdy18           AO12   \n",
       "CN18Sc37_12_Rep_Stdy22_AO  CN18Sc37_12_Rep_Stdy22           AO13   \n",
       "\n",
       "                                                      sample_title project_id  \\\n",
       "sample_name                                                                     \n",
       "CN18Sc37_12_Rep_Stdy5_AO   Seawater sample Monterey Bay (GLOMICON)   GLOMICON   \n",
       "CN18Sc37_12_Rep_Stdy10_AO  Seawater sample Monterey Bay (GLOMICON)   GLOMICON   \n",
       "CN18Sc37_12_Rep_Stdy14_AO  Seawater sample Monterey Bay (GLOMICON)   GLOMICON   \n",
       "CN18Sc37_12_Rep_Stdy18_AO  Seawater sample Monterey Bay (GLOMICON)   GLOMICON   \n",
       "CN18Sc37_12_Rep_Stdy22_AO  Seawater sample Monterey Bay (GLOMICON)   GLOMICON   \n",
       "\n",
       "                           ...              seq_meth amplicon_sequenced  \\\n",
       "sample_name                ...                                            \n",
       "CN18Sc37_12_Rep_Stdy5_AO   ...  Illumina MiSeq 2x150             18S V9   \n",
       "CN18Sc37_12_Rep_Stdy10_AO  ...  Illumina MiSeq 2x150             18S V9   \n",
       "CN18Sc37_12_Rep_Stdy14_AO  ...  Illumina MiSeq 2x150             18S V9   \n",
       "CN18Sc37_12_Rep_Stdy18_AO  ...  Illumina MiSeq 2x150             18S V9   \n",
       "CN18Sc37_12_Rep_Stdy22_AO  ...  Illumina MiSeq 2x150             18S V9   \n",
       "\n",
       "                          target_gene  target_subfragment pcr_primer_forward  \\\n",
       "sample_name                                                                    \n",
       "CN18Sc37_12_Rep_Stdy5_AO     18S rRNA                  V9   GTACACACCGCCCGTC   \n",
       "CN18Sc37_12_Rep_Stdy10_AO    18S rRNA                  V9   GTACACACCGCCCGTC   \n",
       "CN18Sc37_12_Rep_Stdy14_AO    18S rRNA                  V9   GTACACACCGCCCGTC   \n",
       "CN18Sc37_12_Rep_Stdy18_AO    18S rRNA                  V9   GTACACACCGCCCGTC   \n",
       "CN18Sc37_12_Rep_Stdy22_AO    18S rRNA                  V9   GTACACACCGCCCGTC   \n",
       "\n",
       "                                 pcr_primer_reverse pcr_primer_name_forward  \\\n",
       "sample_name                                                                   \n",
       "CN18Sc37_12_Rep_Stdy5_AO   TGATCCTTCTGCAGGTTCACCTAC               Euk_1391f   \n",
       "CN18Sc37_12_Rep_Stdy10_AO  TGATCCTTCTGCAGGTTCACCTAC               Euk_1391f   \n",
       "CN18Sc37_12_Rep_Stdy14_AO  TGATCCTTCTGCAGGTTCACCTAC               Euk_1391f   \n",
       "CN18Sc37_12_Rep_Stdy18_AO  TGATCCTTCTGCAGGTTCACCTAC               Euk_1391f   \n",
       "CN18Sc37_12_Rep_Stdy22_AO  TGATCCTTCTGCAGGTTCACCTAC               Euk_1391f   \n",
       "\n",
       "                          pcr_primer_name_reverse  \\\n",
       "sample_name                                         \n",
       "CN18Sc37_12_Rep_Stdy5_AO                    EukBr   \n",
       "CN18Sc37_12_Rep_Stdy10_AO                   EukBr   \n",
       "CN18Sc37_12_Rep_Stdy14_AO                   EukBr   \n",
       "CN18Sc37_12_Rep_Stdy18_AO                   EukBr   \n",
       "CN18Sc37_12_Rep_Stdy22_AO                   EukBr   \n",
       "\n",
       "                                                        pcr_primer_reference  \\\n",
       "sample_name                                                                    \n",
       "CN18Sc37_12_Rep_Stdy5_AO   Amaral-Zettler et al. 2009, Stoeck et al. 2010...   \n",
       "CN18Sc37_12_Rep_Stdy10_AO  Amaral-Zettler et al. 2009, Stoeck et al. 2010...   \n",
       "CN18Sc37_12_Rep_Stdy14_AO  Amaral-Zettler et al. 2009, Stoeck et al. 2010...   \n",
       "CN18Sc37_12_Rep_Stdy18_AO  Amaral-Zettler et al. 2009, Stoeck et al. 2010...   \n",
       "CN18Sc37_12_Rep_Stdy22_AO  Amaral-Zettler et al. 2009, Stoeck et al. 2010...   \n",
       "\n",
       "                                                                   pcr_cond  \n",
       "sample_name                                                                  \n",
       "CN18Sc37_12_Rep_Stdy5_AO   dx.doi.org/10.17504/protocols.io.36wgq3d8olk5/v2  \n",
       "CN18Sc37_12_Rep_Stdy10_AO  dx.doi.org/10.17504/protocols.io.36wgq3d8olk5/v2  \n",
       "CN18Sc37_12_Rep_Stdy14_AO  dx.doi.org/10.17504/protocols.io.36wgq3d8olk5/v2  \n",
       "CN18Sc37_12_Rep_Stdy18_AO  dx.doi.org/10.17504/protocols.io.36wgq3d8olk5/v2  \n",
       "CN18Sc37_12_Rep_Stdy22_AO  dx.doi.org/10.17504/protocols.io.36wgq3d8olk5/v2  \n",
       "\n",
       "[5 rows x 58 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = water_amplicon_merge(dfs[1], dfs[2])\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4aa90638-3cfc-4245-9e19-8c1a0f78360c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dad0b0ea-a7f8-4fae-88f9-6e1b66f81e73",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c7989b14-8876-4105-8b33-1dd317e1021a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dcab9957-3a5b-4124-9dae-acef006a14c5",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "098e847e-f4f7-418b-8dfa-3257a585e22f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "26077a12-c672-4982-9c70-615d42fb3cea",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Collecting Institute', 'Analyzing Institute', 'cruise_id', 'station', 'ctd_bottle_no', 'sample_replicate', 'source_material_id', 'extract_number', 'sample_title', 'project_id', 'organism', 'collection_date_local', 'collection_date', 'depth', 'env_broad_scale', 'env_local_scale', 'env_medium', 'geo_loc_name', 'waterBody', 'countryCode', 'decimalLatitude', 'decimalLongitude', 'samp_vol_we_dna_ext', 'samp_collect_device', 'samp_mat_process', 'size_frac', 'sample_type', 'collection_method', 'basisOfRecord', 'chlorophyll', 'density', 'fluor', 'pressure', 'salinity', 'samp_store_temp', 'temp', 'date_modified', 'modified_by', 'sheet', 'library_id', 'title', 'library_strategy', 'library_source', 'library_selection', 'lib_layout', 'platform', 'instrument_model', 'filetype', 'filename', 'filename2', 'seq_facility', 'seq_meth', 'amplicon_sequenced', 'target_gene', 'target_subfragment', 'pcr_primer_forward', 'pcr_primer_reverse', 'pcr_primer_name_forward', 'pcr_primer_name_reverse', 'pcr_primer_reference', 'pcr_cond', 'date_modified_amplicon_sheet', 'modified_by_amplicon_sheet', 'sheet_amplicon_sheet']\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Collecting Institute</th>\n",
       "      <th>Analyzing Institute</th>\n",
       "      <th>cruise_id</th>\n",
       "      <th>station</th>\n",
       "      <th>ctd_bottle_no</th>\n",
       "      <th>sample_replicate</th>\n",
       "      <th>source_material_id</th>\n",
       "      <th>extract_number</th>\n",
       "      <th>sample_title</th>\n",
       "      <th>project_id</th>\n",
       "      <th>...</th>\n",
       "      <th>target_subfragment</th>\n",
       "      <th>pcr_primer_forward</th>\n",
       "      <th>pcr_primer_reverse</th>\n",
       "      <th>pcr_primer_name_forward</th>\n",
       "      <th>pcr_primer_name_reverse</th>\n",
       "      <th>pcr_primer_reference</th>\n",
       "      <th>pcr_cond</th>\n",
       "      <th>date_modified_amplicon_sheet</th>\n",
       "      <th>modified_by_amplicon_sheet</th>\n",
       "      <th>sheet_amplicon_sheet</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sample_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CN18Sc37_12_Rep_Stdy5_AO</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy5</td>\n",
       "      <td>AO9</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>GLOMICON</td>\n",
       "      <td>...</td>\n",
       "      <td>V9</td>\n",
       "      <td>GTACACACCGCCCGTC</td>\n",
       "      <td>TGATCCTTCTGCAGGTTCACCTAC</td>\n",
       "      <td>Euk_1391f</td>\n",
       "      <td>EukBr</td>\n",
       "      <td>Amaral-Zettler et al. 2009, Stoeck et al. 2010...</td>\n",
       "      <td>dx.doi.org/10.17504/protocols.io.36wgq3d8olk5/v2</td>\n",
       "      <td>2024-08-15 08:42:28.469</td>\n",
       "      <td>NaN</td>\n",
       "      <td>amplicon_prep_data</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN18Sc37_12_Rep_Stdy10_AO</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy10</td>\n",
       "      <td>AO10</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>GLOMICON</td>\n",
       "      <td>...</td>\n",
       "      <td>V9</td>\n",
       "      <td>GTACACACCGCCCGTC</td>\n",
       "      <td>TGATCCTTCTGCAGGTTCACCTAC</td>\n",
       "      <td>Euk_1391f</td>\n",
       "      <td>EukBr</td>\n",
       "      <td>Amaral-Zettler et al. 2009, Stoeck et al. 2010...</td>\n",
       "      <td>dx.doi.org/10.17504/protocols.io.36wgq3d8olk5/v2</td>\n",
       "      <td>2024-06-18 19:31:14.358</td>\n",
       "      <td>kpitz@mbari.org</td>\n",
       "      <td>amplicon_prep_data</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN18Sc37_12_Rep_Stdy14_AO</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy14</td>\n",
       "      <td>AO11</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>GLOMICON</td>\n",
       "      <td>...</td>\n",
       "      <td>V9</td>\n",
       "      <td>GTACACACCGCCCGTC</td>\n",
       "      <td>TGATCCTTCTGCAGGTTCACCTAC</td>\n",
       "      <td>Euk_1391f</td>\n",
       "      <td>EukBr</td>\n",
       "      <td>Amaral-Zettler et al. 2009, Stoeck et al. 2010...</td>\n",
       "      <td>dx.doi.org/10.17504/protocols.io.36wgq3d8olk5/v2</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>amplicon_prep_data</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN18Sc37_12_Rep_Stdy18_AO</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy18</td>\n",
       "      <td>AO12</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>GLOMICON</td>\n",
       "      <td>...</td>\n",
       "      <td>V9</td>\n",
       "      <td>GTACACACCGCCCGTC</td>\n",
       "      <td>TGATCCTTCTGCAGGTTCACCTAC</td>\n",
       "      <td>Euk_1391f</td>\n",
       "      <td>EukBr</td>\n",
       "      <td>Amaral-Zettler et al. 2009, Stoeck et al. 2010...</td>\n",
       "      <td>dx.doi.org/10.17504/protocols.io.36wgq3d8olk5/v2</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>amplicon_prep_data</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN18Sc37_12_Rep_Stdy22_AO</th>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy22</td>\n",
       "      <td>AO13</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>GLOMICON</td>\n",
       "      <td>...</td>\n",
       "      <td>V9</td>\n",
       "      <td>GTACACACCGCCCGTC</td>\n",
       "      <td>TGATCCTTCTGCAGGTTCACCTAC</td>\n",
       "      <td>Euk_1391f</td>\n",
       "      <td>EukBr</td>\n",
       "      <td>Amaral-Zettler et al. 2009, Stoeck et al. 2010...</td>\n",
       "      <td>dx.doi.org/10.17504/protocols.io.36wgq3d8olk5/v2</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>amplicon_prep_data</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 64 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                          Collecting Institute Analyzing Institute cruise_id  \\\n",
       "sample_name                                                                    \n",
       "CN18Sc37_12_Rep_Stdy5_AO                 MBARI               MBARI     CN18S   \n",
       "CN18Sc37_12_Rep_Stdy10_AO                MBARI               MBARI     CN18S   \n",
       "CN18Sc37_12_Rep_Stdy14_AO                MBARI               MBARI     CN18S   \n",
       "CN18Sc37_12_Rep_Stdy18_AO                MBARI               MBARI     CN18S   \n",
       "CN18Sc37_12_Rep_Stdy22_AO                MBARI               MBARI     CN18S   \n",
       "\n",
       "                          station  ctd_bottle_no  sample_replicate  \\\n",
       "sample_name                                                          \n",
       "CN18Sc37_12_Rep_Stdy5_AO       TS           12.0               5.0   \n",
       "CN18Sc37_12_Rep_Stdy10_AO      TS           12.0              10.0   \n",
       "CN18Sc37_12_Rep_Stdy14_AO      TS           12.0              14.0   \n",
       "CN18Sc37_12_Rep_Stdy18_AO      TS           12.0              18.0   \n",
       "CN18Sc37_12_Rep_Stdy22_AO      TS           12.0              22.0   \n",
       "\n",
       "                               source_material_id extract_number  \\\n",
       "sample_name                                                        \n",
       "CN18Sc37_12_Rep_Stdy5_AO    CN18Sc37_12_Rep_Stdy5            AO9   \n",
       "CN18Sc37_12_Rep_Stdy10_AO  CN18Sc37_12_Rep_Stdy10           AO10   \n",
       "CN18Sc37_12_Rep_Stdy14_AO  CN18Sc37_12_Rep_Stdy14           AO11   \n",
       "CN18Sc37_12_Rep_Stdy18_AO  CN18Sc37_12_Rep_Stdy18           AO12   \n",
       "CN18Sc37_12_Rep_Stdy22_AO  CN18Sc37_12_Rep_Stdy22           AO13   \n",
       "\n",
       "                                                      sample_title project_id  \\\n",
       "sample_name                                                                     \n",
       "CN18Sc37_12_Rep_Stdy5_AO   Seawater sample Monterey Bay (GLOMICON)   GLOMICON   \n",
       "CN18Sc37_12_Rep_Stdy10_AO  Seawater sample Monterey Bay (GLOMICON)   GLOMICON   \n",
       "CN18Sc37_12_Rep_Stdy14_AO  Seawater sample Monterey Bay (GLOMICON)   GLOMICON   \n",
       "CN18Sc37_12_Rep_Stdy18_AO  Seawater sample Monterey Bay (GLOMICON)   GLOMICON   \n",
       "CN18Sc37_12_Rep_Stdy22_AO  Seawater sample Monterey Bay (GLOMICON)   GLOMICON   \n",
       "\n",
       "                           ... target_subfragment pcr_primer_forward  \\\n",
       "sample_name                ...                                         \n",
       "CN18Sc37_12_Rep_Stdy5_AO   ...                 V9   GTACACACCGCCCGTC   \n",
       "CN18Sc37_12_Rep_Stdy10_AO  ...                 V9   GTACACACCGCCCGTC   \n",
       "CN18Sc37_12_Rep_Stdy14_AO  ...                 V9   GTACACACCGCCCGTC   \n",
       "CN18Sc37_12_Rep_Stdy18_AO  ...                 V9   GTACACACCGCCCGTC   \n",
       "CN18Sc37_12_Rep_Stdy22_AO  ...                 V9   GTACACACCGCCCGTC   \n",
       "\n",
       "                                 pcr_primer_reverse  pcr_primer_name_forward  \\\n",
       "sample_name                                                                    \n",
       "CN18Sc37_12_Rep_Stdy5_AO   TGATCCTTCTGCAGGTTCACCTAC                Euk_1391f   \n",
       "CN18Sc37_12_Rep_Stdy10_AO  TGATCCTTCTGCAGGTTCACCTAC                Euk_1391f   \n",
       "CN18Sc37_12_Rep_Stdy14_AO  TGATCCTTCTGCAGGTTCACCTAC                Euk_1391f   \n",
       "CN18Sc37_12_Rep_Stdy18_AO  TGATCCTTCTGCAGGTTCACCTAC                Euk_1391f   \n",
       "CN18Sc37_12_Rep_Stdy22_AO  TGATCCTTCTGCAGGTTCACCTAC                Euk_1391f   \n",
       "\n",
       "                          pcr_primer_name_reverse  \\\n",
       "sample_name                                         \n",
       "CN18Sc37_12_Rep_Stdy5_AO                    EukBr   \n",
       "CN18Sc37_12_Rep_Stdy10_AO                   EukBr   \n",
       "CN18Sc37_12_Rep_Stdy14_AO                   EukBr   \n",
       "CN18Sc37_12_Rep_Stdy18_AO                   EukBr   \n",
       "CN18Sc37_12_Rep_Stdy22_AO                   EukBr   \n",
       "\n",
       "                                                        pcr_primer_reference  \\\n",
       "sample_name                                                                    \n",
       "CN18Sc37_12_Rep_Stdy5_AO   Amaral-Zettler et al. 2009, Stoeck et al. 2010...   \n",
       "CN18Sc37_12_Rep_Stdy10_AO  Amaral-Zettler et al. 2009, Stoeck et al. 2010...   \n",
       "CN18Sc37_12_Rep_Stdy14_AO  Amaral-Zettler et al. 2009, Stoeck et al. 2010...   \n",
       "CN18Sc37_12_Rep_Stdy18_AO  Amaral-Zettler et al. 2009, Stoeck et al. 2010...   \n",
       "CN18Sc37_12_Rep_Stdy22_AO  Amaral-Zettler et al. 2009, Stoeck et al. 2010...   \n",
       "\n",
       "                                                                   pcr_cond  \\\n",
       "sample_name                                                                   \n",
       "CN18Sc37_12_Rep_Stdy5_AO   dx.doi.org/10.17504/protocols.io.36wgq3d8olk5/v2   \n",
       "CN18Sc37_12_Rep_Stdy10_AO  dx.doi.org/10.17504/protocols.io.36wgq3d8olk5/v2   \n",
       "CN18Sc37_12_Rep_Stdy14_AO  dx.doi.org/10.17504/protocols.io.36wgq3d8olk5/v2   \n",
       "CN18Sc37_12_Rep_Stdy18_AO  dx.doi.org/10.17504/protocols.io.36wgq3d8olk5/v2   \n",
       "CN18Sc37_12_Rep_Stdy22_AO  dx.doi.org/10.17504/protocols.io.36wgq3d8olk5/v2   \n",
       "\n",
       "                          date_modified_amplicon_sheet  \\\n",
       "sample_name                                              \n",
       "CN18Sc37_12_Rep_Stdy5_AO       2024-08-15 08:42:28.469   \n",
       "CN18Sc37_12_Rep_Stdy10_AO      2024-06-18 19:31:14.358   \n",
       "CN18Sc37_12_Rep_Stdy14_AO                          NaT   \n",
       "CN18Sc37_12_Rep_Stdy18_AO                          NaT   \n",
       "CN18Sc37_12_Rep_Stdy22_AO                          NaT   \n",
       "\n",
       "                          modified_by_amplicon_sheet sheet_amplicon_sheet  \n",
       "sample_name                                                                \n",
       "CN18Sc37_12_Rep_Stdy5_AO                         NaN   amplicon_prep_data  \n",
       "CN18Sc37_12_Rep_Stdy10_AO            kpitz@mbari.org   amplicon_prep_data  \n",
       "CN18Sc37_12_Rep_Stdy14_AO                        NaN   amplicon_prep_data  \n",
       "CN18Sc37_12_Rep_Stdy18_AO                        NaN   amplicon_prep_data  \n",
       "CN18Sc37_12_Rep_Stdy22_AO                        NaN   amplicon_prep_data  \n",
       "\n",
       "[5 rows x 64 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = dfs[1].copy()\n",
    "df.set_index('sample_name', inplace=True)\n",
    "# add amplicon metadata to water sample metadata (some water samples don't have amplicon data if not sequenced in institute)\n",
    "df = df.join(dfs[2].set_index('sample_name'), rsuffix='_amplicon_sheet')\n",
    "print(list(df))\n",
    "\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "a2f1ca19-ca5d-4ce0-900a-0256eb101739",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['amplicon_sequenced', 'trim_method', 'cluster_method', 'pid_clustering', 'taxa_class_method', 'taxa_ref_db', 'code_repo', 'date_modified', 'sheet', 'project_id', 'project_name', 'project_id_external', 'project_contact', 'project_description', 'type', 'license', 'associated_parties', 'study_area_description', 'date_modified', 'sheet']\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MBARI</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>amplicon_sequenced</th>\n",
       "      <td>18S V9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>trim_method</th>\n",
       "      <td>atropos</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cluster_method</th>\n",
       "      <td>dada2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pid_clustering</th>\n",
       "      <td>ASV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taxa_class_method</th>\n",
       "      <td>BLAST; MEGAN LCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taxa_ref_db</th>\n",
       "      <td>genbank NR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>code_repo</th>\n",
       "      <td>https://github.com/MBARI-BOG/BOG-Banzai-Dada2-...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>project_id</th>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>project_name</th>\n",
       "      <td>Intercomparison of filtered seawater samples a...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>project_id_external</th>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>project_contact</th>\n",
       "      <td>Kathleen Pitz, kpitz@mbari.org, https://orcid....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>project_description</th>\n",
       "      <td>Replicate filters, originating from surface wa...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>type</th>\n",
       "      <td>Occurrence</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>license</th>\n",
       "      <td>CC-BY 4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>associated_parties</th>\n",
       "      <td>Francisco Chavez, chfr@mbari.org</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>study_area_description</th>\n",
       "      <td>eDNA from north of Monterey Bay collected duri...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                    MBARI\n",
       "amplicon_sequenced                                                 18S V9\n",
       "trim_method                                                       atropos\n",
       "cluster_method                                                      dada2\n",
       "pid_clustering                                                        ASV\n",
       "taxa_class_method                                        BLAST; MEGAN LCA\n",
       "taxa_ref_db                                                    genbank NR\n",
       "code_repo               https://github.com/MBARI-BOG/BOG-Banzai-Dada2-...\n",
       "project_id                                             GLOMICON_INTERCOMP\n",
       "project_name            Intercomparison of filtered seawater samples a...\n",
       "project_id_external                                    GLOMICON_INTERCOMP\n",
       "project_contact         Kathleen Pitz, kpitz@mbari.org, https://orcid....\n",
       "project_description     Replicate filters, originating from surface wa...\n",
       "type                                                           Occurrence\n",
       "license                                                         CC-BY 4.0\n",
       "associated_parties                       Francisco Chavez, chfr@mbari.org\n",
       "study_area_description  eDNA from north of Monterey Bay collected duri..."
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# study-wide metadata can organize by institution?\n",
    "df = pd.concat([dfs[3], dfs[0]], axis=1)\n",
    "print(list(df))\n",
    "df = df.drop(['sheet', 'date_modified'], axis=1)\n",
    "df = df.T\n",
    "df = df.rename(columns={0:'MBARI'})\n",
    "\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "a0200b29-bce8-46af-af15-6e95a6330094",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>project_id</th>\n",
       "      <th>project_name</th>\n",
       "      <th>project_id_external</th>\n",
       "      <th>project_contact</th>\n",
       "      <th>project_description</th>\n",
       "      <th>type</th>\n",
       "      <th>license</th>\n",
       "      <th>associated_parties</th>\n",
       "      <th>study_area_description</th>\n",
       "      <th>date_modified</th>\n",
       "      <th>sheet</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>Intercomparison of filtered seawater samples a...</td>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>Kathleen Pitz, kpitz@mbari.org, https://orcid....</td>\n",
       "      <td>Replicate filters, originating from surface wa...</td>\n",
       "      <td>Occurrence</td>\n",
       "      <td>CC-BY 4.0</td>\n",
       "      <td>Francisco Chavez, chfr@mbari.org</td>\n",
       "      <td>eDNA from north of Monterey Bay collected duri...</td>\n",
       "      <td>2024-08-22 14:14:36.023</td>\n",
       "      <td>study_data</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           project_id                                       project_name  \\\n",
       "0  GLOMICON_INTERCOMP  Intercomparison of filtered seawater samples a...   \n",
       "\n",
       "  project_id_external                                    project_contact  \\\n",
       "0  GLOMICON_INTERCOMP  Kathleen Pitz, kpitz@mbari.org, https://orcid....   \n",
       "\n",
       "                                 project_description        type    license  \\\n",
       "0  Replicate filters, originating from surface wa...  Occurrence  CC-BY 4.0   \n",
       "\n",
       "                 associated_parties  \\\n",
       "0  Francisco Chavez, chfr@mbari.org   \n",
       "\n",
       "                              study_area_description           date_modified  \\\n",
       "0  eDNA from north of Monterey Bay collected duri... 2024-08-22 14:14:36.023   \n",
       "\n",
       "        sheet  \n",
       "0  study_data  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfs[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2a87a043-315d-4fc4-87b3-3a18566f255a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "5f44165f-db25-4b59-bc97-470c7a6e99ba",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Collecting Institute</th>\n",
       "      <th>Analyzing Institute</th>\n",
       "      <th>cruise_id</th>\n",
       "      <th>station</th>\n",
       "      <th>ctd_bottle_no</th>\n",
       "      <th>sample_replicate</th>\n",
       "      <th>source_material_id</th>\n",
       "      <th>extract_number</th>\n",
       "      <th>sample_title</th>\n",
       "      <th>...</th>\n",
       "      <th>density</th>\n",
       "      <th>fluor</th>\n",
       "      <th>pressure</th>\n",
       "      <th>salinity</th>\n",
       "      <th>samp_store_temp</th>\n",
       "      <th>temp</th>\n",
       "      <th>date_modified</th>\n",
       "      <th>modified_by</th>\n",
       "      <th>sheet</th>\n",
       "      <th>dup</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CN18Sc37_12_Rep_Stdy5_AO</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy5</td>\n",
       "      <td>AO9</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>...</td>\n",
       "      <td>25.328</td>\n",
       "      <td>0.3458</td>\n",
       "      <td>1.532</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>-80.0</td>\n",
       "      <td>12.6583 °C</td>\n",
       "      <td>2024-07-31 12:16:48.569</td>\n",
       "      <td>NaN</td>\n",
       "      <td>water_sample_data</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CN18Sc37_12_Rep_Stdy10_AO</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy10</td>\n",
       "      <td>AO10</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>...</td>\n",
       "      <td>25.328</td>\n",
       "      <td>0.3458</td>\n",
       "      <td>1.532</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>-80.0</td>\n",
       "      <td>12.6583 °C</td>\n",
       "      <td>2024-06-18 19:04:45.094</td>\n",
       "      <td>kpitz@mbari.org</td>\n",
       "      <td>water_sample_data</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CN18Sc37_12_Rep_Stdy14_AO</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy14</td>\n",
       "      <td>AO11</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>...</td>\n",
       "      <td>25.328</td>\n",
       "      <td>0.3458</td>\n",
       "      <td>1.532</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>-80.0</td>\n",
       "      <td>12.6583 °C</td>\n",
       "      <td>2024-06-18 19:04:45.094</td>\n",
       "      <td>kpitz@mbari.org</td>\n",
       "      <td>water_sample_data</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CN18Sc37_12_Rep_Stdy18_AO</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy18</td>\n",
       "      <td>AO12</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>...</td>\n",
       "      <td>25.328</td>\n",
       "      <td>0.3458</td>\n",
       "      <td>1.532</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>-80.0</td>\n",
       "      <td>12.6583 °C</td>\n",
       "      <td>2024-06-18 19:05:42.928</td>\n",
       "      <td>kpitz@mbari.org</td>\n",
       "      <td>water_sample_data</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CN18Sc37_12_Rep_Stdy22_AO</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy22</td>\n",
       "      <td>AO13</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>...</td>\n",
       "      <td>25.328</td>\n",
       "      <td>0.3458</td>\n",
       "      <td>1.532</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>-80.0</td>\n",
       "      <td>12.6583 °C</td>\n",
       "      <td>2024-06-18 19:04:45.094</td>\n",
       "      <td>kpitz@mbari.org</td>\n",
       "      <td>water_sample_data</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy23</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>...</td>\n",
       "      <td>25.328</td>\n",
       "      <td>0.3458</td>\n",
       "      <td>1.532</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>-80.0</td>\n",
       "      <td>12.6583 °C</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>water_sample_data</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy21</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>...</td>\n",
       "      <td>25.328</td>\n",
       "      <td>0.3458</td>\n",
       "      <td>1.532</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>-80.0</td>\n",
       "      <td>12.6583 °C</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>water_sample_data</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>SBR</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>...</td>\n",
       "      <td>25.328</td>\n",
       "      <td>0.3458</td>\n",
       "      <td>1.532</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>-80.0</td>\n",
       "      <td>12.6583 °C</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>water_sample_data</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>...</td>\n",
       "      <td>25.328</td>\n",
       "      <td>0.3458</td>\n",
       "      <td>1.532</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>-80.0</td>\n",
       "      <td>12.6583 °C</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>water_sample_data</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>...</td>\n",
       "      <td>25.328</td>\n",
       "      <td>0.3458</td>\n",
       "      <td>1.532</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>-80.0</td>\n",
       "      <td>12.6583 °C</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>water_sample_data</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>SBR</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>...</td>\n",
       "      <td>25.328</td>\n",
       "      <td>0.3458</td>\n",
       "      <td>1.532</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>-80.0</td>\n",
       "      <td>12.6583 °C</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>water_sample_data</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>...</td>\n",
       "      <td>25.328</td>\n",
       "      <td>0.3458</td>\n",
       "      <td>1.532</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>-80.0</td>\n",
       "      <td>12.6583 °C</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>water_sample_data</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>...</td>\n",
       "      <td>25.328</td>\n",
       "      <td>0.3458</td>\n",
       "      <td>1.532</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>-80.0</td>\n",
       "      <td>12.6583 °C</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>water_sample_data</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>SBR</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>...</td>\n",
       "      <td>25.328</td>\n",
       "      <td>0.3458</td>\n",
       "      <td>1.532</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>-80.0</td>\n",
       "      <td>12.6583 °C</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>water_sample_data</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>...</td>\n",
       "      <td>25.328</td>\n",
       "      <td>0.3458</td>\n",
       "      <td>1.532</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>-80.0</td>\n",
       "      <td>12.6583 °C</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>water_sample_data</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy24</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>...</td>\n",
       "      <td>25.328</td>\n",
       "      <td>0.3458</td>\n",
       "      <td>1.532</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>-80.0</td>\n",
       "      <td>12.6583 °C</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>water_sample_data</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>...</td>\n",
       "      <td>25.328</td>\n",
       "      <td>0.3458</td>\n",
       "      <td>1.532</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>-80.0</td>\n",
       "      <td>12.6583 °C</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>water_sample_data</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>SBR</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>...</td>\n",
       "      <td>25.328</td>\n",
       "      <td>0.3458</td>\n",
       "      <td>1.532</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>-80.0</td>\n",
       "      <td>12.6583 °C</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>water_sample_data</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>...</td>\n",
       "      <td>25.328</td>\n",
       "      <td>0.3458</td>\n",
       "      <td>1.532</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>-80.0</td>\n",
       "      <td>12.6583 °C</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>water_sample_data</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>...</td>\n",
       "      <td>25.328</td>\n",
       "      <td>0.3458</td>\n",
       "      <td>1.532</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>-80.0</td>\n",
       "      <td>12.6583 °C</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>water_sample_data</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>SBR</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>...</td>\n",
       "      <td>25.328</td>\n",
       "      <td>0.3458</td>\n",
       "      <td>1.532</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>-80.0</td>\n",
       "      <td>12.6583 °C</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>water_sample_data</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>...</td>\n",
       "      <td>25.328</td>\n",
       "      <td>0.3458</td>\n",
       "      <td>1.532</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>-80.0</td>\n",
       "      <td>12.6583 °C</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>water_sample_data</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>...</td>\n",
       "      <td>25.328</td>\n",
       "      <td>0.3458</td>\n",
       "      <td>1.532</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>-80.0</td>\n",
       "      <td>12.6583 °C</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>water_sample_data</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>AWI</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>...</td>\n",
       "      <td>25.328</td>\n",
       "      <td>0.3458</td>\n",
       "      <td>1.532</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>-80.0</td>\n",
       "      <td>12.6583 °C</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>water_sample_data</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>NaN</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy25</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>...</td>\n",
       "      <td>25.328</td>\n",
       "      <td>0.3458</td>\n",
       "      <td>1.532</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>-80.0</td>\n",
       "      <td>12.6583 °C</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>water_sample_data</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>25 rows × 41 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  sample_name Collecting Institute Analyzing Institute  \\\n",
       "0    CN18Sc37_12_Rep_Stdy5_AO                MBARI               MBARI   \n",
       "1   CN18Sc37_12_Rep_Stdy10_AO                MBARI               MBARI   \n",
       "2   CN18Sc37_12_Rep_Stdy14_AO                MBARI               MBARI   \n",
       "3   CN18Sc37_12_Rep_Stdy18_AO                MBARI               MBARI   \n",
       "4   CN18Sc37_12_Rep_Stdy22_AO                MBARI               MBARI   \n",
       "45                        NaN                MBARI                 NaN   \n",
       "44                        NaN                MBARI                 NaN   \n",
       "43                        NaN                MBARI                 SBR   \n",
       "42                        NaN                MBARI                 AWI   \n",
       "41                        NaN                MBARI                 NaN   \n",
       "40                        NaN                MBARI                 SBR   \n",
       "39                        NaN                MBARI                 AWI   \n",
       "38                        NaN                MBARI                 NaN   \n",
       "37                        NaN                MBARI                 SBR   \n",
       "35                        NaN                MBARI                 NaN   \n",
       "46                        NaN                MBARI                 NaN   \n",
       "34                        NaN                MBARI                 NaN   \n",
       "33                        NaN                MBARI                 SBR   \n",
       "32                        NaN                MBARI                 AWI   \n",
       "31                        NaN                MBARI                 NaN   \n",
       "30                        NaN                MBARI                 SBR   \n",
       "29                        NaN                MBARI                 AWI   \n",
       "28                        NaN                MBARI                 NaN   \n",
       "36                        NaN                MBARI                 AWI   \n",
       "47                        NaN                MBARI                 NaN   \n",
       "\n",
       "   cruise_id station  ctd_bottle_no  sample_replicate      source_material_id  \\\n",
       "0      CN18S      TS           12.0               5.0   CN18Sc37_12_Rep_Stdy5   \n",
       "1      CN18S      TS           12.0              10.0  CN18Sc37_12_Rep_Stdy10   \n",
       "2      CN18S      TS           12.0              14.0  CN18Sc37_12_Rep_Stdy14   \n",
       "3      CN18S      TS           12.0              18.0  CN18Sc37_12_Rep_Stdy18   \n",
       "4      CN18S      TS           12.0              22.0  CN18Sc37_12_Rep_Stdy22   \n",
       "45     CN18S      TS           12.0               NaN  CN18Sc37_12_Rep_Stdy23   \n",
       "44     CN18S      TS           12.0               NaN  CN18Sc37_12_Rep_Stdy21   \n",
       "43     CN18S      TS           12.0               NaN  CN18Sc37_12_Rep_Stdy20   \n",
       "42     CN18S      TS           12.0               NaN  CN18Sc37_12_Rep_Stdy19   \n",
       "41     CN18S      TS           12.0               NaN  CN18Sc37_12_Rep_Stdy17   \n",
       "40     CN18S      TS           12.0               NaN  CN18Sc37_12_Rep_Stdy16   \n",
       "39     CN18S      TS           12.0               NaN  CN18Sc37_12_Rep_Stdy15   \n",
       "38     CN18S      TS           12.0               NaN  CN18Sc37_12_Rep_Stdy13   \n",
       "37     CN18S      TS           12.0               NaN  CN18Sc37_12_Rep_Stdy12   \n",
       "35     CN18S      TS           12.0               NaN   CN18Sc37_12_Rep_Stdy9   \n",
       "46     CN18S      TS           12.0               NaN  CN18Sc37_12_Rep_Stdy24   \n",
       "34     CN18S      TS           12.0               NaN   CN18Sc37_12_Rep_Stdy8   \n",
       "33     CN18S      TS           12.0               NaN   CN18Sc37_12_Rep_Stdy7   \n",
       "32     CN18S      TS           12.0               NaN   CN18Sc37_12_Rep_Stdy6   \n",
       "31     CN18S      TS           12.0               NaN   CN18Sc37_12_Rep_Stdy4   \n",
       "30     CN18S      TS           12.0               NaN   CN18Sc37_12_Rep_Stdy3   \n",
       "29     CN18S      TS           12.0               NaN   CN18Sc37_12_Rep_Stdy2   \n",
       "28     CN18S      TS           12.0               NaN   CN18Sc37_12_Rep_Stdy1   \n",
       "36     CN18S      TS           12.0               NaN  CN18Sc37_12_Rep_Stdy11   \n",
       "47     CN18S      TS           12.0               NaN  CN18Sc37_12_Rep_Stdy25   \n",
       "\n",
       "   extract_number                             sample_title  ... density  \\\n",
       "0             AO9  Seawater sample Monterey Bay (GLOMICON)  ...  25.328   \n",
       "1            AO10  Seawater sample Monterey Bay (GLOMICON)  ...  25.328   \n",
       "2            AO11  Seawater sample Monterey Bay (GLOMICON)  ...  25.328   \n",
       "3            AO12  Seawater sample Monterey Bay (GLOMICON)  ...  25.328   \n",
       "4            AO13  Seawater sample Monterey Bay (GLOMICON)  ...  25.328   \n",
       "45            NaN  Seawater sample Monterey Bay (GLOMICON)  ...  25.328   \n",
       "44            NaN  Seawater sample Monterey Bay (GLOMICON)  ...  25.328   \n",
       "43            NaN  Seawater sample Monterey Bay (GLOMICON)  ...  25.328   \n",
       "42            NaN  Seawater sample Monterey Bay (GLOMICON)  ...  25.328   \n",
       "41            NaN  Seawater sample Monterey Bay (GLOMICON)  ...  25.328   \n",
       "40            NaN  Seawater sample Monterey Bay (GLOMICON)  ...  25.328   \n",
       "39            NaN  Seawater sample Monterey Bay (GLOMICON)  ...  25.328   \n",
       "38            NaN  Seawater sample Monterey Bay (GLOMICON)  ...  25.328   \n",
       "37            NaN  Seawater sample Monterey Bay (GLOMICON)  ...  25.328   \n",
       "35            NaN  Seawater sample Monterey Bay (GLOMICON)  ...  25.328   \n",
       "46            NaN  Seawater sample Monterey Bay (GLOMICON)  ...  25.328   \n",
       "34            NaN  Seawater sample Monterey Bay (GLOMICON)  ...  25.328   \n",
       "33            NaN  Seawater sample Monterey Bay (GLOMICON)  ...  25.328   \n",
       "32            NaN  Seawater sample Monterey Bay (GLOMICON)  ...  25.328   \n",
       "31            NaN  Seawater sample Monterey Bay (GLOMICON)  ...  25.328   \n",
       "30            NaN  Seawater sample Monterey Bay (GLOMICON)  ...  25.328   \n",
       "29            NaN  Seawater sample Monterey Bay (GLOMICON)  ...  25.328   \n",
       "28            NaN  Seawater sample Monterey Bay (GLOMICON)  ...  25.328   \n",
       "36            NaN  Seawater sample Monterey Bay (GLOMICON)  ...  25.328   \n",
       "47            NaN  Seawater sample Monterey Bay (GLOMICON)  ...  25.328   \n",
       "\n",
       "     fluor pressure salinity  samp_store_temp        temp  \\\n",
       "0   0.3458    1.532  33.5385            -80.0  12.6583 °C   \n",
       "1   0.3458    1.532  33.5385            -80.0  12.6583 °C   \n",
       "2   0.3458    1.532  33.5385            -80.0  12.6583 °C   \n",
       "3   0.3458    1.532  33.5385            -80.0  12.6583 °C   \n",
       "4   0.3458    1.532  33.5385            -80.0  12.6583 °C   \n",
       "45  0.3458    1.532  33.5385            -80.0  12.6583 °C   \n",
       "44  0.3458    1.532  33.5385            -80.0  12.6583 °C   \n",
       "43  0.3458    1.532  33.5385            -80.0  12.6583 °C   \n",
       "42  0.3458    1.532  33.5385            -80.0  12.6583 °C   \n",
       "41  0.3458    1.532  33.5385            -80.0  12.6583 °C   \n",
       "40  0.3458    1.532  33.5385            -80.0  12.6583 °C   \n",
       "39  0.3458    1.532  33.5385            -80.0  12.6583 °C   \n",
       "38  0.3458    1.532  33.5385            -80.0  12.6583 °C   \n",
       "37  0.3458    1.532  33.5385            -80.0  12.6583 °C   \n",
       "35  0.3458    1.532  33.5385            -80.0  12.6583 °C   \n",
       "46  0.3458    1.532  33.5385            -80.0  12.6583 °C   \n",
       "34  0.3458    1.532  33.5385            -80.0  12.6583 °C   \n",
       "33  0.3458    1.532  33.5385            -80.0  12.6583 °C   \n",
       "32  0.3458    1.532  33.5385            -80.0  12.6583 °C   \n",
       "31  0.3458    1.532  33.5385            -80.0  12.6583 °C   \n",
       "30  0.3458    1.532  33.5385            -80.0  12.6583 °C   \n",
       "29  0.3458    1.532  33.5385            -80.0  12.6583 °C   \n",
       "28  0.3458    1.532  33.5385            -80.0  12.6583 °C   \n",
       "36  0.3458    1.532  33.5385            -80.0  12.6583 °C   \n",
       "47  0.3458    1.532  33.5385            -80.0  12.6583 °C   \n",
       "\n",
       "             date_modified      modified_by              sheet    dup  \n",
       "0  2024-07-31 12:16:48.569              NaN  water_sample_data  False  \n",
       "1  2024-06-18 19:04:45.094  kpitz@mbari.org  water_sample_data  False  \n",
       "2  2024-06-18 19:04:45.094  kpitz@mbari.org  water_sample_data  False  \n",
       "3  2024-06-18 19:05:42.928  kpitz@mbari.org  water_sample_data  False  \n",
       "4  2024-06-18 19:04:45.094  kpitz@mbari.org  water_sample_data  False  \n",
       "45                     NaT              NaN  water_sample_data   True  \n",
       "44                     NaT              NaN  water_sample_data   True  \n",
       "43                     NaT              NaN  water_sample_data   True  \n",
       "42                     NaT              NaN  water_sample_data   True  \n",
       "41                     NaT              NaN  water_sample_data   True  \n",
       "40                     NaT              NaN  water_sample_data   True  \n",
       "39                     NaT              NaN  water_sample_data   True  \n",
       "38                     NaT              NaN  water_sample_data   True  \n",
       "37                     NaT              NaN  water_sample_data   True  \n",
       "35                     NaT              NaN  water_sample_data   True  \n",
       "46                     NaT              NaN  water_sample_data   True  \n",
       "34                     NaT              NaN  water_sample_data   True  \n",
       "33                     NaT              NaN  water_sample_data   True  \n",
       "32                     NaT              NaN  water_sample_data   True  \n",
       "31                     NaT              NaN  water_sample_data   True  \n",
       "30                     NaT              NaN  water_sample_data   True  \n",
       "29                     NaT              NaN  water_sample_data   True  \n",
       "28                     NaT              NaN  water_sample_data   True  \n",
       "36                     NaT              NaN  water_sample_data   True  \n",
       "47                     NaT              NaN  water_sample_data   True  \n",
       "\n",
       "[25 rows x 41 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Join two sample_name dfs:\n",
    "#df = pd.concat([dfs[1].set_index('sample_name'), dfs[2].set_index('sample_name')], axis=0)\n",
    "df = dfs[1].copy()\n",
    "df['dup'] = df.duplicated('sample_name', keep=False)\n",
    "df = df.loc[df['source_material_id'].isna()==False]\n",
    "#df = df.dropna('source_material_id')\n",
    "df = df.sort_values('dup')\n",
    "\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "d3884ea2-3f91-4413-90b4-dc5e9442c620",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>library_id</th>\n",
       "      <th>title</th>\n",
       "      <th>library_strategy</th>\n",
       "      <th>library_source</th>\n",
       "      <th>library_selection</th>\n",
       "      <th>lib_layout</th>\n",
       "      <th>platform</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>filetype</th>\n",
       "      <th>...</th>\n",
       "      <th>target_subfragment</th>\n",
       "      <th>pcr_primer_forward</th>\n",
       "      <th>pcr_primer_reverse</th>\n",
       "      <th>pcr_primer_name_forward</th>\n",
       "      <th>pcr_primer_name_reverse</th>\n",
       "      <th>pcr_primer_reference</th>\n",
       "      <th>pcr_cond</th>\n",
       "      <th>date_modified</th>\n",
       "      <th>modified_by</th>\n",
       "      <th>sheet</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CN18Sc37_12_Rep_Stdy5_AO</td>\n",
       "      <td>AO9</td>\n",
       "      <td>18S amplicon metabarcoding GLOMICON comparison...</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>paired</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>fastq</td>\n",
       "      <td>...</td>\n",
       "      <td>V9</td>\n",
       "      <td>GTACACACCGCCCGTC</td>\n",
       "      <td>TGATCCTTCTGCAGGTTCACCTAC</td>\n",
       "      <td>Euk_1391f</td>\n",
       "      <td>EukBr</td>\n",
       "      <td>Amaral-Zettler et al. 2009, Stoeck et al. 2010...</td>\n",
       "      <td>dx.doi.org/10.17504/protocols.io.36wgq3d8olk5/v2</td>\n",
       "      <td>2024-08-15 08:42:28.469</td>\n",
       "      <td>NaN</td>\n",
       "      <td>amplicon_prep_data</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CN18Sc37_12_Rep_Stdy10_AO</td>\n",
       "      <td>AO10</td>\n",
       "      <td>18S amplicon metabarcoding GLOMICON comparison...</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>paired</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>fastq</td>\n",
       "      <td>...</td>\n",
       "      <td>V9</td>\n",
       "      <td>GTACACACCGCCCGTC</td>\n",
       "      <td>TGATCCTTCTGCAGGTTCACCTAC</td>\n",
       "      <td>Euk_1391f</td>\n",
       "      <td>EukBr</td>\n",
       "      <td>Amaral-Zettler et al. 2009, Stoeck et al. 2010...</td>\n",
       "      <td>dx.doi.org/10.17504/protocols.io.36wgq3d8olk5/v2</td>\n",
       "      <td>2024-06-18 19:31:14.358</td>\n",
       "      <td>kpitz@mbari.org</td>\n",
       "      <td>amplicon_prep_data</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CN18Sc37_12_Rep_Stdy14_AO</td>\n",
       "      <td>AO11</td>\n",
       "      <td>18S amplicon metabarcoding GLOMICON comparison...</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>paired</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>fastq</td>\n",
       "      <td>...</td>\n",
       "      <td>V9</td>\n",
       "      <td>GTACACACCGCCCGTC</td>\n",
       "      <td>TGATCCTTCTGCAGGTTCACCTAC</td>\n",
       "      <td>Euk_1391f</td>\n",
       "      <td>EukBr</td>\n",
       "      <td>Amaral-Zettler et al. 2009, Stoeck et al. 2010...</td>\n",
       "      <td>dx.doi.org/10.17504/protocols.io.36wgq3d8olk5/v2</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>amplicon_prep_data</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CN18Sc37_12_Rep_Stdy18_AO</td>\n",
       "      <td>AO12</td>\n",
       "      <td>18S amplicon metabarcoding GLOMICON comparison...</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>paired</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>fastq</td>\n",
       "      <td>...</td>\n",
       "      <td>V9</td>\n",
       "      <td>GTACACACCGCCCGTC</td>\n",
       "      <td>TGATCCTTCTGCAGGTTCACCTAC</td>\n",
       "      <td>Euk_1391f</td>\n",
       "      <td>EukBr</td>\n",
       "      <td>Amaral-Zettler et al. 2009, Stoeck et al. 2010...</td>\n",
       "      <td>dx.doi.org/10.17504/protocols.io.36wgq3d8olk5/v2</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>amplicon_prep_data</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CN18Sc37_12_Rep_Stdy22_AO</td>\n",
       "      <td>AO13</td>\n",
       "      <td>18S amplicon metabarcoding GLOMICON comparison...</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>paired</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>fastq</td>\n",
       "      <td>...</td>\n",
       "      <td>V9</td>\n",
       "      <td>GTACACACCGCCCGTC</td>\n",
       "      <td>TGATCCTTCTGCAGGTTCACCTAC</td>\n",
       "      <td>Euk_1391f</td>\n",
       "      <td>EukBr</td>\n",
       "      <td>Amaral-Zettler et al. 2009, Stoeck et al. 2010...</td>\n",
       "      <td>dx.doi.org/10.17504/protocols.io.36wgq3d8olk5/v2</td>\n",
       "      <td>NaT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>amplicon_prep_data</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 sample_name library_id  \\\n",
       "0   CN18Sc37_12_Rep_Stdy5_AO        AO9   \n",
       "1  CN18Sc37_12_Rep_Stdy10_AO       AO10   \n",
       "2  CN18Sc37_12_Rep_Stdy14_AO       AO11   \n",
       "3  CN18Sc37_12_Rep_Stdy18_AO       AO12   \n",
       "4  CN18Sc37_12_Rep_Stdy22_AO       AO13   \n",
       "\n",
       "                                               title library_strategy  \\\n",
       "0  18S amplicon metabarcoding GLOMICON comparison...         AMPLICON   \n",
       "1  18S amplicon metabarcoding GLOMICON comparison...         AMPLICON   \n",
       "2  18S amplicon metabarcoding GLOMICON comparison...         AMPLICON   \n",
       "3  18S amplicon metabarcoding GLOMICON comparison...         AMPLICON   \n",
       "4  18S amplicon metabarcoding GLOMICON comparison...         AMPLICON   \n",
       "\n",
       "  library_source library_selection lib_layout  platform instrument_model  \\\n",
       "0    METAGENOMIC               PCR     paired  ILLUMINA   Illumina MiSeq   \n",
       "1    METAGENOMIC               PCR     paired  ILLUMINA   Illumina MiSeq   \n",
       "2    METAGENOMIC               PCR     paired  ILLUMINA   Illumina MiSeq   \n",
       "3    METAGENOMIC               PCR     paired  ILLUMINA   Illumina MiSeq   \n",
       "4    METAGENOMIC               PCR     paired  ILLUMINA   Illumina MiSeq   \n",
       "\n",
       "  filetype  ... target_subfragment pcr_primer_forward  \\\n",
       "0    fastq  ...                 V9   GTACACACCGCCCGTC   \n",
       "1    fastq  ...                 V9   GTACACACCGCCCGTC   \n",
       "2    fastq  ...                 V9   GTACACACCGCCCGTC   \n",
       "3    fastq  ...                 V9   GTACACACCGCCCGTC   \n",
       "4    fastq  ...                 V9   GTACACACCGCCCGTC   \n",
       "\n",
       "         pcr_primer_reverse pcr_primer_name_forward pcr_primer_name_reverse  \\\n",
       "0  TGATCCTTCTGCAGGTTCACCTAC               Euk_1391f                   EukBr   \n",
       "1  TGATCCTTCTGCAGGTTCACCTAC               Euk_1391f                   EukBr   \n",
       "2  TGATCCTTCTGCAGGTTCACCTAC               Euk_1391f                   EukBr   \n",
       "3  TGATCCTTCTGCAGGTTCACCTAC               Euk_1391f                   EukBr   \n",
       "4  TGATCCTTCTGCAGGTTCACCTAC               Euk_1391f                   EukBr   \n",
       "\n",
       "                                pcr_primer_reference  \\\n",
       "0  Amaral-Zettler et al. 2009, Stoeck et al. 2010...   \n",
       "1  Amaral-Zettler et al. 2009, Stoeck et al. 2010...   \n",
       "2  Amaral-Zettler et al. 2009, Stoeck et al. 2010...   \n",
       "3  Amaral-Zettler et al. 2009, Stoeck et al. 2010...   \n",
       "4  Amaral-Zettler et al. 2009, Stoeck et al. 2010...   \n",
       "\n",
       "                                           pcr_cond           date_modified  \\\n",
       "0  dx.doi.org/10.17504/protocols.io.36wgq3d8olk5/v2 2024-08-15 08:42:28.469   \n",
       "1  dx.doi.org/10.17504/protocols.io.36wgq3d8olk5/v2 2024-06-18 19:31:14.358   \n",
       "2  dx.doi.org/10.17504/protocols.io.36wgq3d8olk5/v2                     NaT   \n",
       "3  dx.doi.org/10.17504/protocols.io.36wgq3d8olk5/v2                     NaT   \n",
       "4  dx.doi.org/10.17504/protocols.io.36wgq3d8olk5/v2                     NaT   \n",
       "\n",
       "       modified_by               sheet  \n",
       "0              NaN  amplicon_prep_data  \n",
       "1  kpitz@mbari.org  amplicon_prep_data  \n",
       "2              NaN  amplicon_prep_data  \n",
       "3              NaN  amplicon_prep_data  \n",
       "4              NaN  amplicon_prep_data  \n",
       "\n",
       "[5 rows x 26 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfs[2].head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "1a81b42b-727d-4681-997c-7abd4fad9de1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Collecting Institute</th>\n",
       "      <th>Analyzing Institute</th>\n",
       "      <th>cruise_id</th>\n",
       "      <th>station</th>\n",
       "      <th>ctd_bottle_no</th>\n",
       "      <th>sample_replicate</th>\n",
       "      <th>source_material_id</th>\n",
       "      <th>extract_number</th>\n",
       "      <th>sample_title</th>\n",
       "      <th>...</th>\n",
       "      <th>chlorophyll</th>\n",
       "      <th>density</th>\n",
       "      <th>fluor</th>\n",
       "      <th>pressure</th>\n",
       "      <th>salinity</th>\n",
       "      <th>samp_store_temp</th>\n",
       "      <th>temp</th>\n",
       "      <th>date_modified</th>\n",
       "      <th>modified_by</th>\n",
       "      <th>sheet</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CN18Sc37_12_Rep_Stdy5_AO</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy5</td>\n",
       "      <td>AO9</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>...</td>\n",
       "      <td>8.1461</td>\n",
       "      <td>25.328</td>\n",
       "      <td>0.3458</td>\n",
       "      <td>1.532</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>-80.0</td>\n",
       "      <td>12.6583 °C</td>\n",
       "      <td>2024-07-31 12:16:48.569</td>\n",
       "      <td>NaN</td>\n",
       "      <td>water_sample_data</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CN18Sc37_12_Rep_Stdy10_AO</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy10</td>\n",
       "      <td>AO10</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>...</td>\n",
       "      <td>8.1461</td>\n",
       "      <td>25.328</td>\n",
       "      <td>0.3458</td>\n",
       "      <td>1.532</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>-80.0</td>\n",
       "      <td>12.6583 °C</td>\n",
       "      <td>2024-06-18 19:04:45.094</td>\n",
       "      <td>kpitz@mbari.org</td>\n",
       "      <td>water_sample_data</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CN18Sc37_12_Rep_Stdy14_AO</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy14</td>\n",
       "      <td>AO11</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>...</td>\n",
       "      <td>8.1461</td>\n",
       "      <td>25.328</td>\n",
       "      <td>0.3458</td>\n",
       "      <td>1.532</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>-80.0</td>\n",
       "      <td>12.6583 °C</td>\n",
       "      <td>2024-06-18 19:04:45.094</td>\n",
       "      <td>kpitz@mbari.org</td>\n",
       "      <td>water_sample_data</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CN18Sc37_12_Rep_Stdy18_AO</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy18</td>\n",
       "      <td>AO12</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>...</td>\n",
       "      <td>8.1461</td>\n",
       "      <td>25.328</td>\n",
       "      <td>0.3458</td>\n",
       "      <td>1.532</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>-80.0</td>\n",
       "      <td>12.6583 °C</td>\n",
       "      <td>2024-06-18 19:05:42.928</td>\n",
       "      <td>kpitz@mbari.org</td>\n",
       "      <td>water_sample_data</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CN18Sc37_12_Rep_Stdy22_AO</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>MBARI</td>\n",
       "      <td>CN18S</td>\n",
       "      <td>TS</td>\n",
       "      <td>12.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>CN18Sc37_12_Rep_Stdy22</td>\n",
       "      <td>AO13</td>\n",
       "      <td>Seawater sample Monterey Bay (GLOMICON)</td>\n",
       "      <td>...</td>\n",
       "      <td>8.1461</td>\n",
       "      <td>25.328</td>\n",
       "      <td>0.3458</td>\n",
       "      <td>1.532</td>\n",
       "      <td>33.5385</td>\n",
       "      <td>-80.0</td>\n",
       "      <td>12.6583 °C</td>\n",
       "      <td>2024-06-18 19:04:45.094</td>\n",
       "      <td>kpitz@mbari.org</td>\n",
       "      <td>water_sample_data</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 40 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 sample_name Collecting Institute Analyzing Institute  \\\n",
       "0   CN18Sc37_12_Rep_Stdy5_AO                MBARI               MBARI   \n",
       "1  CN18Sc37_12_Rep_Stdy10_AO                MBARI               MBARI   \n",
       "2  CN18Sc37_12_Rep_Stdy14_AO                MBARI               MBARI   \n",
       "3  CN18Sc37_12_Rep_Stdy18_AO                MBARI               MBARI   \n",
       "4  CN18Sc37_12_Rep_Stdy22_AO                MBARI               MBARI   \n",
       "\n",
       "  cruise_id station  ctd_bottle_no  sample_replicate      source_material_id  \\\n",
       "0     CN18S      TS           12.0               5.0   CN18Sc37_12_Rep_Stdy5   \n",
       "1     CN18S      TS           12.0              10.0  CN18Sc37_12_Rep_Stdy10   \n",
       "2     CN18S      TS           12.0              14.0  CN18Sc37_12_Rep_Stdy14   \n",
       "3     CN18S      TS           12.0              18.0  CN18Sc37_12_Rep_Stdy18   \n",
       "4     CN18S      TS           12.0              22.0  CN18Sc37_12_Rep_Stdy22   \n",
       "\n",
       "  extract_number                             sample_title  ... chlorophyll  \\\n",
       "0            AO9  Seawater sample Monterey Bay (GLOMICON)  ...      8.1461   \n",
       "1           AO10  Seawater sample Monterey Bay (GLOMICON)  ...      8.1461   \n",
       "2           AO11  Seawater sample Monterey Bay (GLOMICON)  ...      8.1461   \n",
       "3           AO12  Seawater sample Monterey Bay (GLOMICON)  ...      8.1461   \n",
       "4           AO13  Seawater sample Monterey Bay (GLOMICON)  ...      8.1461   \n",
       "\n",
       "  density   fluor pressure  salinity samp_store_temp        temp  \\\n",
       "0  25.328  0.3458    1.532   33.5385           -80.0  12.6583 °C   \n",
       "1  25.328  0.3458    1.532   33.5385           -80.0  12.6583 °C   \n",
       "2  25.328  0.3458    1.532   33.5385           -80.0  12.6583 °C   \n",
       "3  25.328  0.3458    1.532   33.5385           -80.0  12.6583 °C   \n",
       "4  25.328  0.3458    1.532   33.5385           -80.0  12.6583 °C   \n",
       "\n",
       "            date_modified      modified_by              sheet  \n",
       "0 2024-07-31 12:16:48.569              NaN  water_sample_data  \n",
       "1 2024-06-18 19:04:45.094  kpitz@mbari.org  water_sample_data  \n",
       "2 2024-06-18 19:04:45.094  kpitz@mbari.org  water_sample_data  \n",
       "3 2024-06-18 19:05:42.928  kpitz@mbari.org  water_sample_data  \n",
       "4 2024-06-18 19:04:45.094  kpitz@mbari.org  water_sample_data  \n",
       "\n",
       "[5 rows x 40 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfs[1].head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "091ba98a-966f-4cf8-9d66-3eacc943d486",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>...</th>\n",
       "      <th>19</th>\n",
       "      <th>20</th>\n",
       "      <th>21</th>\n",
       "      <th>22</th>\n",
       "      <th>23</th>\n",
       "      <th>24</th>\n",
       "      <th>25</th>\n",
       "      <th>26</th>\n",
       "      <th>27</th>\n",
       "      <th>0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>project_id</th>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>GLOMICON</td>\n",
       "      <td>GLOMICON</td>\n",
       "      <td>GLOMICON</td>\n",
       "      <td>GLOMICON</td>\n",
       "      <td>GLOMICON</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>project_name</th>\n",
       "      <td>Intercomparison of filtered seawater samples a...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>project_id_external</th>\n",
       "      <td>GLOMICON_INTERCOMP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>project_contact</th>\n",
       "      <td>Kathleen Pitz, kpitz@mbari.org, https://orcid....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>project_description</th>\n",
       "      <td>Replicate filters, originating from surface wa...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cluster_method</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>dada2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pid_clustering</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ASV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taxa_class_method</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BLAST; MEGAN LCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taxa_ref_db</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>genbank NR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>code_repo</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://github.com/MBARI-BOG/BOG-Banzai-Dada2-...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>76 rows × 78 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                    0   \\\n",
       "project_id                                          GLOMICON_INTERCOMP   \n",
       "project_name         Intercomparison of filtered seawater samples a...   \n",
       "project_id_external                                 GLOMICON_INTERCOMP   \n",
       "project_contact      Kathleen Pitz, kpitz@mbari.org, https://orcid....   \n",
       "project_description  Replicate filters, originating from surface wa...   \n",
       "...                                                                ...   \n",
       "cluster_method                                                     NaN   \n",
       "pid_clustering                                                     NaN   \n",
       "taxa_class_method                                                  NaN   \n",
       "taxa_ref_db                                                        NaN   \n",
       "code_repo                                                          NaN   \n",
       "\n",
       "                           0         1         2         3         4    5   \\\n",
       "project_id           GLOMICON  GLOMICON  GLOMICON  GLOMICON  GLOMICON  NaN   \n",
       "project_name              NaN       NaN       NaN       NaN       NaN  NaN   \n",
       "project_id_external       NaN       NaN       NaN       NaN       NaN  NaN   \n",
       "project_contact           NaN       NaN       NaN       NaN       NaN  NaN   \n",
       "project_description       NaN       NaN       NaN       NaN       NaN  NaN   \n",
       "...                       ...       ...       ...       ...       ...  ...   \n",
       "cluster_method            NaN       NaN       NaN       NaN       NaN  NaN   \n",
       "pid_clustering            NaN       NaN       NaN       NaN       NaN  NaN   \n",
       "taxa_class_method         NaN       NaN       NaN       NaN       NaN  NaN   \n",
       "taxa_ref_db               NaN       NaN       NaN       NaN       NaN  NaN   \n",
       "code_repo                 NaN       NaN       NaN       NaN       NaN  NaN   \n",
       "\n",
       "                      6    7    8   ...   19   20   21   22   23   24   25  \\\n",
       "project_id           NaN  NaN  NaN  ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN   \n",
       "project_name         NaN  NaN  NaN  ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN   \n",
       "project_id_external  NaN  NaN  NaN  ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN   \n",
       "project_contact      NaN  NaN  NaN  ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN   \n",
       "project_description  NaN  NaN  NaN  ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN   \n",
       "...                  ...  ...  ...  ...  ...  ...  ...  ...  ...  ...  ...   \n",
       "cluster_method       NaN  NaN  NaN  ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN   \n",
       "pid_clustering       NaN  NaN  NaN  ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN   \n",
       "taxa_class_method    NaN  NaN  NaN  ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN   \n",
       "taxa_ref_db          NaN  NaN  NaN  ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN   \n",
       "code_repo            NaN  NaN  NaN  ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN   \n",
       "\n",
       "                      26   27  \\\n",
       "project_id           NaN  NaN   \n",
       "project_name         NaN  NaN   \n",
       "project_id_external  NaN  NaN   \n",
       "project_contact      NaN  NaN   \n",
       "project_description  NaN  NaN   \n",
       "...                  ...  ...   \n",
       "cluster_method       NaN  NaN   \n",
       "pid_clustering       NaN  NaN   \n",
       "taxa_class_method    NaN  NaN   \n",
       "taxa_ref_db          NaN  NaN   \n",
       "code_repo            NaN  NaN   \n",
       "\n",
       "                                                                    0   \n",
       "project_id                                                         NaN  \n",
       "project_name                                                       NaN  \n",
       "project_id_external                                                NaN  \n",
       "project_contact                                                    NaN  \n",
       "project_description                                                NaN  \n",
       "...                                                                ...  \n",
       "cluster_method                                                   dada2  \n",
       "pid_clustering                                                     ASV  \n",
       "taxa_class_method                                     BLAST; MEGAN LCA  \n",
       "taxa_ref_db                                                 genbank NR  \n",
       "code_repo            https://github.com/MBARI-BOG/BOG-Banzai-Dada2-...  \n",
       "\n",
       "[76 rows x 78 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.concat(dfs, axis=0)\n",
    "df = df.dropna(how='all', axis=1)\n",
    "df = df.dropna(how='all', axis=0)\n",
    "df = df.T\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "eee550e8-cdbd-47f6-ad43-ca1392a2f566",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python scipy_test",
   "language": "python",
   "name": "scipy_test"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
