# Intercomparison Rationale

The goal of this study is to better understand how we can combine information from multiple omics observatories to make conclusions about global biodiversity.

Institutional methods in place will have typically evolved in response to the specific needs (e.g., scientific rationale for observations, characteristics of samples and communities at observatory sites) and will hence differ for good reasons. Instead of attempting to enforce standard methods that may not be equally appropriate, with this study, we try to answer the question **at what level we can draw conclusions from data generated across observatories that all target the eukaryotic microbial community?** This boils down to the question to what degree differences are introduced by methodology (extraction, primers used, sequencing, bioinformatics) and how this influences our estimates of biodiversity. Here we compare replicate samples which all focus on pelagic protists taken from the upper water column. Apart from that, no attempt was made to harmonize the methodologies applied (e.g., in terms of filters, DNA extraction kits, primers, bioinformatics pipelines, reference data bases).

## Targets of Study

The following targets are seen as core components of the overall study

1. How much variability is there in community composition between samples taken from the same environment at the same point in time (effectively technical replicates) that are analyzed at different institutions? What are the taxonomic biases inherent to the applied standard methods?
- *Comparison of taxonomic information obtained by different partners with different lab and analytical pipelines (occurrence and relative abundance of taxa)*
2. Are the overall patterns of biodiversity and community similarity between samples consistent across methods? 
- *Even if there is variability in taxa detected, is there consistency in which samples have higher or lower overall biodiversity and which samples are most similar or different to each other? (comparison of biodiversity metrics and multivariate analyses between samples)*

Further, with the sequencing of artificial communities and replication between and within institutions, we can ask:

3. How much variability is introduced from DNA extraction forward?
- *Level of difference in mock communities (that start from technical replicates of DNA extracts) compared to communities from the environment*
4. How much internal variability occurs between replicates ?
- *Level of variability in data from replicate samples analyzed at the same institution*
6. How should metadata and protocol information be presented and shared to achieve FAIRness?
- *Provide a good example for the community*
7. (Potentially in a future study) How much variability is introduced by bioinformatics pipelines?
- *level of disagreement between communities from analyses with different in-house pipelines compared to centralized analyses of data at one institution with the same pipeline. This would involve more effort (i.e., analyzing the same raw data across multiple pipelines) and may not be possible in this first paper.*

# Summary

## Aim
To compare different procedures for microbial community observations from sampling to annotation; answering questions such as:
- How well do results with different pipelines agree?
- At which taxonomic levels are cross-observatory comparisons meaningful without common methods?
- Do we see indications that procedures are aligned to site / community characteristics?
- Can we identify procedures that work particularly well for certain environments (best represent mock composition? largest richness identified in environmental samples?)

## Focus 
18S rDNA based investigations of unicellular eukaryotes
- Analyses with pipelines in place at the different institutions

## Samples (5 replicates):
- Environmental samples (filtered surface water) from intercomparison participants’ study locations
- mock community DNA extracts with balanced (‘evenMock’) and non-balanced composition (‘bloomMock’)

## Participants:
Alfred Wegener Institute (AWI), Monterey Bay Aquarium Research Institute (MBARI), National Oceanic and Atmospheric Administration (NOAA), National Oceanography Centre (NOC), Dalhousie University, Station Biologique Roscoff (SBR)

![GLOMICON_sample_posession_test](https://github.com/GLOMICON/intercomparison/assets/7292507/40ab3e6f-561c-40d7-b97f-46eb8ca694a3)

## About this repository
In this repository, we will share the outputs of the intercomparison project and guidance for (meta)data management and sharing.

Have a suggestion or see something that should be changed? The best way to engage with this project is via its [Issue tracker](https://github.com/GLOMICON/intercomparison/issues).
