# The project

## Aim
To compare different procedures for microbial community observations from sampling to annotation; answering questions such as:
- How well do results with different pipelines agree?
- At which taxonomic levels are cross-observatory comparisons meaningful without common methods?
- Do we see indications that procedures are aligned to site / community characteristics?
- Can we identify procedures that work particularly well (best represent mock composition? largest richness identified in environmental samples?)

## Focus 
18S rDNA based investigations of unicellular eukaryotes
- Analyses with pipelines in place at the different institutions
- In additions central bioinformatics analyses at Ribocon or at one partner

## Samples (always 5 replicates):
- Environmental samples (filtered surface water) from intercomparison participants’ pet-sites’
- mock community DNA extracts with balanced (‘evenMock’) and non-balanced composition (‘bloomMock’)

## Participants:
AWI, MBARI, NOAA, NOC, UDalhousie, Station Biologique Roscoff

## About this repository
In this reporitory, we will share the outputs of the intercomparison project and guidance for (meta)data management and sharing.

The best way to engage with this project is via its [Issue tracker](https://github.com/GLOMICON/intercomparison/issues).
